{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2e110c05-d327-401b-93fc-081d6102bab1",
   "metadata": {},
   "source": [
    "notebook is sm_ont-process_reads\n",
    "\n",
    "\n",
    "envs used = nanofilt, n50, minimap2, samtools, mosdepth, samtools, mosdepth, cutesv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9cd4c83d-03e3-44f3-8e5a-5e64db850897",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from pathlib import Path\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3bb0e454-1134-4194-8c67-e685847b6669",
   "metadata": {},
   "outputs": [],
   "source": [
    "proj_dir = \"/master/nplatt/sm_nanopore\"\n",
    "ref_fas = \"{}/data/genome/GCA_000237925.5.fa\".format(proj_dir)\n",
    "Path(\"{}/results\".format(proj_dir)).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "625a0a9e-c38c-4c9a-a768-c3a22131bfb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "samples = [ \"smor\", \n",
    "            \"smle_pzq_es\",\n",
    "            \"smle_pzq_er\",\n",
    "            \"smbre\", \n",
    "            \"smeg\"        ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d62d2d02-6e9f-4c61-8028-73b97326de49",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"{}/results\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4fef4238-0533-44e4-a139-b08d6950cf3c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Filter and map raw reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13dd5d1e-db7d-4a53-a6bb-8bc790f41f71",
   "metadata": {},
   "outputs": [],
   "source": [
    "#all raw reads are provided in results/raw_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4446e16f-dd96-4417-9da8-deebe8d38001",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/nanofilt\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/nanofilt\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "c2485bf9-bb27-4391-a8ed-a0dbb15cf362",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    raw_fq = \"{}/results/raw_reads/{}.raw_pass.fastq\".format(proj_dir, sample)\n",
    "\n",
    "    !~/anaconda3/envs/sm_ont-process_reads/bin/NanoFilt --length 500 --quality 7 --headcrop 25 --tailcrop 25 {raw_fq} >{sample}.filtered.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e6cbcdc-cfe9-4087-96c3-0e0269379dcd",
   "metadata": {},
   "outputs": [],
   "source": [
    "!conda run -n n50 --cwd . --live-stream n50 --format csv $(ls *.filtered.fastq) >n50_filtered_reads.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "ab2652e7-383e-4b9a-9c95-4f9da176d9c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#path</th>\n",
       "      <th>seqs</th>\n",
       "      <th>size</th>\n",
       "      <th>N50</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>N75</th>\n",
       "      <th>N90</th>\n",
       "      <th>auN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>smbre.filtered.fastq</td>\n",
       "      <td>2156891</td>\n",
       "      <td>13733527914</td>\n",
       "      <td>16145</td>\n",
       "      <td>500</td>\n",
       "      <td>193069</td>\n",
       "      <td>7657</td>\n",
       "      <td>2728</td>\n",
       "      <td>357349.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smeg.filtered.fastq</td>\n",
       "      <td>2096508</td>\n",
       "      <td>12631573135</td>\n",
       "      <td>12491</td>\n",
       "      <td>500</td>\n",
       "      <td>236942</td>\n",
       "      <td>5659</td>\n",
       "      <td>2587</td>\n",
       "      <td>399825.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smle_pzq_er.filtered.fastq</td>\n",
       "      <td>1330051</td>\n",
       "      <td>7000372416</td>\n",
       "      <td>12071</td>\n",
       "      <td>500</td>\n",
       "      <td>447354</td>\n",
       "      <td>4953</td>\n",
       "      <td>2105</td>\n",
       "      <td>227185.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smor.filtered.fastq</td>\n",
       "      <td>3413369</td>\n",
       "      <td>14119478724</td>\n",
       "      <td>8668</td>\n",
       "      <td>500</td>\n",
       "      <td>168273</td>\n",
       "      <td>3430</td>\n",
       "      <td>1618</td>\n",
       "      <td>598308.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smle_pzq_es.filtered.fastq</td>\n",
       "      <td>3662716</td>\n",
       "      <td>23840253863</td>\n",
       "      <td>13824</td>\n",
       "      <td>500</td>\n",
       "      <td>202196</td>\n",
       "      <td>5893</td>\n",
       "      <td>2659</td>\n",
       "      <td>624239.05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        #path     seqs         size    N50  min     max   N75  \\\n",
       "0        smbre.filtered.fastq  2156891  13733527914  16145  500  193069  7657   \n",
       "1         smeg.filtered.fastq  2096508  12631573135  12491  500  236942  5659   \n",
       "2  smle_pzq_er.filtered.fastq  1330051   7000372416  12071  500  447354  4953   \n",
       "3         smor.filtered.fastq  3413369  14119478724   8668  500  168273  3430   \n",
       "4  smle_pzq_es.filtered.fastq  3662716  23840253863  13824  500  202196  5893   \n",
       "\n",
       "    N90        auN  \n",
       "0  2728  357349.74  \n",
       "1  2587  399825.88  \n",
       "2  2105  227185.13  \n",
       "3  1618  598308.11  \n",
       "4  2659  624239.05  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n50_df = pd.read_csv(\"n50_filtered_reads.csv\", sep=\",\")\n",
    "n50_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c55d5b71-923a-41e2-966b-b082b11b9f05",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/minimap2\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/minimap2\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "6db32522-3695-4aaa-9269-e8fb6b5be5c4",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smle_pzq_es ##################\n",
      "[M::mm_idx_gen::11.489*1.68] collected minimizers\n",
      "[M::mm_idx_gen::13.279*2.42] sorted minimizers\n",
      "[M::main::13.279*2.42] loaded/built the index for 12 target sequence(s)\n",
      "[M::mm_mapopt_update::14.118*2.34] mid_occ = 416\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 12\n",
      "[M::mm_idx_stat::14.681*2.28] distinct minimizers: 30119087 (65.35% are singletons); average occurrences: 2.439; average spacing: 5.327; total length: 391422238\n",
      "[M::worker_pipeline::214.298*29.98] mapped 42338 sequences\n",
      "[M::worker_pipeline::390.138*31.90] mapped 40808 sequences\n",
      "[M::worker_pipeline::588.947*32.25] mapped 43306 sequences\n",
      "[M::worker_pipeline::830.022*32.30] mapped 45455 sequences\n",
      "[M::worker_pipeline::1051.682*32.12] mapped 44703 sequences\n",
      "[M::worker_pipeline::1289.362*31.59] mapped 45693 sequences\n",
      "[M::worker_pipeline::1470.012*32.22] mapped 47806 sequences\n",
      "[M::worker_pipeline::1696.452*32.12] mapped 45932 sequences\n",
      "[M::worker_pipeline::1916.264*31.95] mapped 47284 sequences\n",
      "[M::worker_pipeline::2103.592*32.13] mapped 48028 sequences\n",
      "[M::worker_pipeline::2304.990*32.24] mapped 49325 sequences\n",
      "[M::worker_pipeline::2494.646*32.47] mapped 50218 sequences\n",
      "[M::worker_pipeline::2717.671*32.28] mapped 49571 sequences\n",
      "[M::worker_pipeline::2898.281*32.52] mapped 49790 sequences\n",
      "[M::worker_pipeline::3096.096*32.53] mapped 51413 sequences\n",
      "[M::worker_pipeline::3288.826*32.55] mapped 53354 sequences\n",
      "[M::worker_pipeline::3537.461*32.11] mapped 53951 sequences\n",
      "[M::worker_pipeline::3732.637*32.17] mapped 53719 sequences\n",
      "[M::worker_pipeline::3940.025*32.09] mapped 56807 sequences\n",
      "[M::worker_pipeline::4112.592*32.21] mapped 55685 sequences\n",
      "[M::worker_pipeline::4310.755*32.19] mapped 59269 sequences\n",
      "[M::worker_pipeline::4539.062*32.29] mapped 48605 sequences\n",
      "[M::worker_pipeline::4782.077*32.26] mapped 45151 sequences\n",
      "[M::worker_pipeline::4994.360*32.35] mapped 43831 sequences\n",
      "[M::worker_pipeline::5221.479*32.45] mapped 44343 sequences\n",
      "[M::worker_pipeline::5444.817*32.50] mapped 44640 sequences\n",
      "[M::worker_pipeline::5684.431*32.44] mapped 45538 sequences\n",
      "[M::worker_pipeline::5866.860*32.59] mapped 134943 sequences\n",
      "[M::worker_pipeline::6041.373*32.73] mapped 133880 sequences\n",
      "[M::worker_pipeline::6222.160*32.86] mapped 133272 sequences\n",
      "[M::worker_pipeline::6419.560*32.90] mapped 132816 sequences\n",
      "[M::worker_pipeline::6601.796*33.00] mapped 131336 sequences\n",
      "[M::worker_pipeline::6793.441*33.04] mapped 130610 sequences\n",
      "[M::worker_pipeline::6972.643*33.17] mapped 127668 sequences\n",
      "[M::worker_pipeline::7149.058*33.29] mapped 127555 sequences\n",
      "[M::worker_pipeline::7314.339*33.38] mapped 141249 sequences\n",
      "[M::worker_pipeline::7499.906*33.46] mapped 118698 sequences\n",
      "[M::worker_pipeline::7698.053*33.54] mapped 95874 sequences\n",
      "[M::worker_pipeline::7900.189*33.61] mapped 95068 sequences\n",
      "[M::worker_pipeline::8098.936*33.70] mapped 95263 sequences\n",
      "[M::worker_pipeline::8289.017*33.81] mapped 95789 sequences\n",
      "[M::worker_pipeline::8475.551*33.92] mapped 97350 sequences\n",
      "[M::worker_pipeline::8688.873*33.92] mapped 97320 sequences\n",
      "[M::worker_pipeline::8884.251*34.00] mapped 99976 sequences\n",
      "[M::worker_pipeline::9054.041*34.08] mapped 100132 sequences\n",
      "[M::worker_pipeline::9228.106*34.16] mapped 100400 sequences\n",
      "[M::worker_pipeline::9421.437*34.20] mapped 99759 sequences\n",
      "[M::worker_pipeline::9539.774*34.24] mapped 67195 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: /master/nplatt/anaconda3/envs/sm_ont-process_reads/bin/minimap2 -a -o /master/nplatt/sm_nanopore/results/minimap2/smle_pzq_es.sam -t 48 -x map-ont /master/nplatt/sm_nanopore/data/genome/GCA_000237925.5.fa /master/nplatt/sm_nanopore/results/nanofilt/smle_pzq_es.filtered.fastq\n",
      "[M::main] Real time: 9540.184 sec; CPU: 326642.700 sec; Peak RSS: 26.741 GB\n"
     ]
    }
   ],
   "source": [
    "# for sample in samples:\n",
    "for sample in [\"smle_pzq_es\"]:\n",
    "    sam      = \"{}/results/minimap2/{}.sam\".format(proj_dir, sample)\n",
    "    query_fq = \"{}/results/nanofilt/{}.filtered.fastq\".format(proj_dir, sample)\n",
    "\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "    !~/anaconda3/envs/sm_ont-process_reads/bin/minimap2 -a -o {sam} -t 48 -x map-ont {ref_fas} {query_fq}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "5a175957-2dae-4bac-a37e-aa063e0d1a48",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smle_pzq_es ##################\n",
      "...sort...\n",
      "[bam_sort_core] merging from 60 files and 1 in-memory blocks...\n",
      "...index...\n",
      "...coverage...\n"
     ]
    }
   ],
   "source": [
    "# for sample in samples:\n",
    "for sample in [\"smle_pzq_es\"]:\n",
    "    sam      = \"{}/results/minimap2/{}.sam\".format(proj_dir, sample)\n",
    "    query_fq = \"{}/results/nanofilt/{}.filtered.fastq\".format(proj_dir, sample)\n",
    "    \n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    #sort bam\n",
    "    print(\"...sort...\")\n",
    "    ! ~/anaconda3/envs/sm_ont-process_reads/bin/samtools sort -O bam -o {sample}.bam {sample}.sam\n",
    "    \n",
    "    #index bam\n",
    "    print(\"...index...\")\n",
    "    ! ~/anaconda3/envs/sm_ont-process_reads/bin/samtools index {sample}.bam\n",
    "    \n",
    "    #calculate coverage\n",
    "    print(\"...coverage...\")\n",
    "    ! ~/anaconda3/envs/sm_ont-process_reads/bin/mosdepth --threads 4 {sample} {sample}.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "23b040b1-f1f2-4873-a2d7-6772dfc168b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>length</th>\n",
       "      <th>bases</th>\n",
       "      <th>mean</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>smbre</td>\n",
       "      <td>391422238</td>\n",
       "      <td>12929849971</td>\n",
       "      <td>33.03</td>\n",
       "      <td>0</td>\n",
       "      <td>50586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smeg</td>\n",
       "      <td>391422238</td>\n",
       "      <td>8667326476</td>\n",
       "      <td>22.14</td>\n",
       "      <td>0</td>\n",
       "      <td>34407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smle_pzq_er</td>\n",
       "      <td>391422238</td>\n",
       "      <td>6624879196</td>\n",
       "      <td>16.93</td>\n",
       "      <td>0</td>\n",
       "      <td>10350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smle_pzq_es</td>\n",
       "      <td>391422238</td>\n",
       "      <td>21017040798</td>\n",
       "      <td>53.69</td>\n",
       "      <td>0</td>\n",
       "      <td>43602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smor</td>\n",
       "      <td>391422238</td>\n",
       "      <td>12178154968</td>\n",
       "      <td>31.11</td>\n",
       "      <td>0</td>\n",
       "      <td>80622</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample     length        bases   mean  min    max\n",
       "0        smbre  391422238  12929849971  33.03    0  50586\n",
       "1         smeg  391422238   8667326476  22.14    0  34407\n",
       "2  smle_pzq_er  391422238   6624879196  16.93    0  10350\n",
       "3  smle_pzq_es  391422238  21017040798  53.69    0  43602\n",
       "4         smor  391422238  12178154968  31.11    0  80622"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#calculate coverage for a table\n",
    "!echo \"sample,length,bases,mean,min,max\">covs.csv\n",
    "!grep \"total\" *mosdepth.summary.txt | sed 's/:/\\t/' | sed 's/\\t/,/g' | cut -f1,3-7 -d\",\" | sed 's/.mosdepth.summary.txt//' >>covs.csv\n",
    "\n",
    "cov_df=pd.read_csv(\"covs.csv\", sep=\",\")\n",
    "cov_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d4367a81-1675-42bf-a3e3-db05cac1f315",
   "metadata": {},
   "outputs": [],
   "source": [
    "#play around with igv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0694869-8ba5-4996-ae3d-e122538f69e1",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Identify structural variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "71d67d7b-66e0-434f-a420-f75d36a7870e",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/cutesv\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/cutesv\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8fd9c2fd-7b35-4641-b475-8e55df79f64d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smle_pzq_es ##################\n",
      "2023-01-18 11:31:03,178 [INFO] Running /master/nplatt/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 --genotype /master/nplatt/sm_nanopore/results/minimap2/smle_pzq_es.bam /master/nplatt/sm_nanopore/data/genome/GCA_000237925.5.fa smle_pzq_es.cutesv.vcf /master/nplatt/sm_nanopore/results/smle_pzq_es\n",
      "2023-01-18 11:31:10,070 [INFO] The total number of chromsomes: 12\n",
      "2023-01-18 11:31:38,245 [INFO] Finished OU426849.1:10000000-10680109.\n",
      "2023-01-18 11:33:46,325 [INFO] Finished HE601624.3:80000000-87984036.\n",
      "2023-01-18 11:34:13,522 [INFO] Finished HE601624.3:30000000-40000000.\n",
      "2023-01-18 11:34:14,461 [INFO] Finished HE601624.3:60000000-70000000.\n",
      "2023-01-18 11:34:14,861 [INFO] Finished HE601624.3:50000000-60000000.\n",
      "2023-01-18 11:34:15,292 [INFO] Finished HE601624.3:40000000-50000000.\n",
      "2023-01-18 11:34:15,396 [INFO] Finished HE601624.3:20000000-30000000.\n",
      "2023-01-18 11:34:16,824 [INFO] Finished HE601624.3:10000000-20000000.\n",
      "2023-01-18 11:34:17,049 [INFO] Finished HE601624.3:70000000-80000000.\n",
      "2023-01-18 11:34:17,800 [INFO] Finished OU426849.1:0-10000000.\n",
      "2023-01-18 11:34:19,942 [INFO] Finished HE601625.3:0-10000000.\n",
      "2023-01-18 11:34:34,885 [INFO] Finished HE601625.3:10000000-20000000.\n",
      "2023-01-18 11:35:10,155 [INFO] Finished HE601624.3:0-10000000.\n",
      "2023-01-18 11:36:26,000 [INFO] Finished OU426850.1:40000000-42949100.\n",
      "2023-01-18 11:39:30,490 [INFO] Finished HE601625.3:40000000-45716228.\n",
      "2023-01-18 11:41:25,195 [INFO] Finished HE601625.3:20000000-30000000.\n",
      "2023-01-18 11:41:56,222 [INFO] Finished OU426850.1:0-10000000.\n",
      "2023-01-18 11:42:09,883 [INFO] Finished OU426850.1:20000000-30000000.\n",
      "2023-01-18 11:42:20,785 [INFO] Finished OU426850.1:30000000-40000000.\n",
      "2023-01-18 11:42:31,292 [INFO] Finished OU426850.1:10000000-20000000.\n",
      "2023-01-18 11:42:41,655 [INFO] Finished HE601626.3:20000000-30000000.\n",
      "2023-01-18 11:44:43,023 [INFO] Finished HE601626.3:0-10000000.\n",
      "2023-01-18 11:44:45,384 [INFO] Finished HE601626.3:30000000-40000000.\n",
      "2023-01-18 11:44:49,712 [INFO] Finished HE601626.3:10000000-20000000.\n",
      "2023-01-18 11:46:12,421 [INFO] Finished HE601626.3:40000000-49793703.\n",
      "2023-01-18 11:47:34,679 [INFO] Finished HE601627.3:40000000-46471565.\n",
      "2023-01-18 11:47:50,871 [INFO] Finished HE601625.3:30000000-40000000.\n",
      "2023-01-18 11:48:14,295 [INFO] Finished HE601628.3:20000000-24128855.\n",
      "2023-01-18 11:48:40,115 [INFO] Finished HE601627.3:0-10000000.\n",
      "2023-01-18 11:49:23,561 [INFO] Finished HE601627.3:10000000-20000000.\n",
      "2023-01-18 11:49:32,025 [INFO] Finished HE601627.3:20000000-30000000.\n",
      "2023-01-18 11:50:03,023 [INFO] Finished HE601627.3:30000000-40000000.\n",
      "2023-01-18 11:50:11,537 [INFO] Finished HE601629.3:20000000-24696071.\n",
      "2023-01-18 11:52:31,369 [INFO] Finished HE601628.3:10000000-20000000.\n",
      "2023-01-18 11:52:33,504 [INFO] Finished HE601612.2:0-26917.\n",
      "2023-01-18 11:52:38,871 [INFO] Finished OU426848.1:30000000-33063208.\n",
      "2023-01-18 11:52:53,278 [INFO] Finished HE601628.3:0-10000000.\n",
      "2023-01-18 11:53:11,865 [INFO] Finished HE601629.3:10000000-20000000.\n",
      "2023-01-18 11:53:32,868 [INFO] Finished HE601629.3:0-10000000.\n",
      "2023-01-18 11:53:35,289 [INFO] Finished OU426848.1:0-10000000.\n",
      "2023-01-18 11:53:36,901 [INFO] Finished OU426848.1:10000000-20000000.\n",
      "2023-01-18 11:53:37,627 [INFO] Finished OU426848.1:20000000-30000000.\n",
      "2023-01-18 11:53:43,843 [INFO] Finished HE601630.3:0-10000000.\n",
      "2023-01-18 11:53:51,094 [INFO] Finished HE601630.3:10000000-19942475.\n",
      "2023-01-18 11:54:25,263 [INFO] Finished HE601631.3:0-5969971.\n",
      "2023-01-18 11:54:25,329 [INFO] Rebuilding signatures of structural variants.\n",
      "2023-01-18 11:56:15,987 [INFO] Clustering structural variants.\n",
      "2023-01-18 11:56:17,909 [INFO] Finished HE601612.2:DEL.\n",
      "2023-01-18 11:56:17,962 [INFO] Finished HE601631.3:DEL.\n",
      "2023-01-18 11:56:18,360 [INFO] Finished OU426849.1:DEL.\n",
      "2023-01-18 11:56:18,622 [INFO] Finished OU426848.1:DEL.\n",
      "2023-01-18 11:56:18,799 [INFO] Finished HE601630.3:DEL.\n",
      "2023-01-18 11:56:19,175 [INFO] Finished HE601629.3:DEL.\n",
      "2023-01-18 11:56:19,185 [INFO] Finished HE601628.3:DEL.\n",
      "2023-01-18 11:56:19,819 [INFO] Finished OU426850.1:DEL.\n",
      "2023-01-18 11:56:19,840 [INFO] Finished HE601612.2:INS.\n",
      "2023-01-18 11:56:20,084 [INFO] Finished HE601627.3:DEL.\n",
      "2023-01-18 11:56:20,261 [INFO] Finished HE601625.3:DEL.\n",
      "2023-01-18 11:56:20,445 [INFO] Finished HE601626.3:DEL.\n",
      "2023-01-18 11:56:21,791 [INFO] Finished HE601631.3:INS.\n",
      "2023-01-18 11:56:22,037 [INFO] Finished HE601624.3:DEL.\n",
      "2023-01-18 11:56:22,241 [INFO] Finished HE601628.3:INS.\n",
      "2023-01-18 11:56:22,295 [INFO] Finished HE601629.3:INS.\n",
      "2023-01-18 11:56:22,403 [INFO] Finished HE601625.3:INS.\n",
      "2023-01-18 11:56:22,448 [INFO] Finished HE601630.3:INS.\n",
      "2023-01-18 11:56:22,468 [INFO] Finished OU426849.1:INS.\n",
      "2023-01-18 11:56:22,555 [INFO] Finished OU426848.1:INS.\n",
      "2023-01-18 11:56:22,626 [INFO] Finished HE601627.3:INS.\n",
      "2023-01-18 11:56:22,841 [INFO] Finished HE601626.3:INS.\n",
      "2023-01-18 11:56:23,023 [INFO] Finished HE601612.2:INV.\n",
      "2023-01-18 11:56:23,719 [INFO] Finished HE601624.3:INS.\n",
      "2023-01-18 11:56:23,801 [INFO] Finished HE601631.3:INV.\n",
      "2023-01-18 11:56:24,047 [INFO] Finished OU426850.1:INS.\n",
      "2023-01-18 11:56:24,217 [INFO] Finished HE601630.3:INV.\n",
      "2023-01-18 11:56:24,223 [INFO] Finished HE601628.3:INV.\n",
      "2023-01-18 11:56:24,337 [INFO] Finished HE601629.3:INV.\n",
      "2023-01-18 11:56:24,422 [INFO] Finished OU426849.1:INV.\n",
      "2023-01-18 11:56:24,499 [INFO] Finished OU426848.1:INV.\n",
      "2023-01-18 11:56:24,849 [INFO] Finished HE601625.3:INV.\n",
      "2023-01-18 11:56:24,864 [INFO] Finished HE601627.3:INV.\n",
      "2023-01-18 11:56:25,029 [INFO] Finished HE601626.3:INV.\n",
      "2023-01-18 11:56:25,060 [INFO] Finished HE601612.2:DUP.\n",
      "2023-01-18 11:56:25,831 [INFO] Finished HE601624.3:INV.\n",
      "2023-01-18 11:56:26,023 [INFO] Finished HE601631.3:DUP.\n",
      "2023-01-18 11:56:26,031 [INFO] Finished OU426850.1:INV.\n",
      "2023-01-18 11:56:26,091 [INFO] Finished HE601612.2-HE601624.3:TRA/BND.\n",
      "2023-01-18 11:56:26,098 [INFO] Finished HE601612.2-HE601625.3:TRA/BND.\n",
      "2023-01-18 11:56:26,158 [INFO] Finished HE601612.2-HE601626.3:TRA/BND.\n",
      "2023-01-18 11:56:26,160 [INFO] Finished HE601628.3:DUP.\n",
      "2023-01-18 11:56:26,164 [INFO] Finished HE601612.2-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:26,224 [INFO] Finished HE601612.2-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:26,226 [INFO] Finished HE601612.2-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:26,230 [INFO] Finished HE601612.2-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:26,290 [INFO] Finished HE601612.2-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:26,291 [INFO] Finished HE601612.2-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:26,297 [INFO] Finished HE601612.2-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:26,335 [INFO] Finished HE601629.3:DUP.\n",
      "2023-01-18 11:56:26,356 [INFO] Finished HE601612.2-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:26,436 [INFO] Finished OU426849.1:DUP.\n",
      "2023-01-18 11:56:26,479 [INFO] Finished HE601630.3:DUP.\n",
      "2023-01-18 11:56:26,638 [INFO] Finished HE601625.3:DUP.\n",
      "2023-01-18 11:56:26,668 [INFO] Finished OU426848.1:DUP.\n",
      "2023-01-18 11:56:26,695 [INFO] Finished HE601627.3:DUP.\n",
      "2023-01-18 11:56:26,747 [INFO] Finished HE601624.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:26,818 [INFO] Finished HE601626.3:DUP.\n",
      "2023-01-18 11:56:26,852 [INFO] Finished HE601624.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:26,899 [INFO] Finished HE601624.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:26,960 [INFO] Finished HE601624.3-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:26,976 [INFO] Finished HE601624.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,004 [INFO] Finished HE601625.3-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:27,018 [INFO] Finished HE601624.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,073 [INFO] Finished HE601624.3-HE601625.3:TRA/BND.\n",
      "2023-01-18 11:56:27,094 [INFO] Finished HE601625.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,119 [INFO] Finished HE601624.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,123 [INFO] Finished HE601625.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,146 [INFO] Finished HE601625.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:27,152 [INFO] Finished HE601625.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,220 [INFO] Finished HE601624.3-HE601626.3:TRA/BND.\n",
      "2023-01-18 11:56:27,233 [INFO] Finished HE601625.3-HE601626.3:TRA/BND.\n",
      "2023-01-18 11:56:27,294 [INFO] Finished HE601625.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,314 [INFO] Finished HE601626.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,360 [INFO] Finished HE601626.3-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:27,385 [INFO] Finished HE601626.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:27,385 [INFO] Finished HE601625.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,399 [INFO] Finished HE601625.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,427 [INFO] Finished HE601626.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,431 [INFO] Finished HE601626.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,469 [INFO] Finished HE601626.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,489 [INFO] Finished HE601624.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,576 [INFO] Finished HE601627.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,582 [INFO] Finished HE601627.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,590 [INFO] Finished HE601627.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,658 [INFO] Finished HE601627.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:27,709 [INFO] Finished HE601626.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,712 [INFO] Finished HE601624.3:DUP.\n",
      "2023-01-18 11:56:27,727 [INFO] Finished HE601628.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,737 [INFO] Finished HE601628.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,744 [INFO] Finished HE601627.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,755 [INFO] Finished HE601628.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,766 [INFO] Finished HE601628.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,782 [INFO] Finished HE601627.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,792 [INFO] Finished HE601629.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,799 [INFO] Finished HE601628.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,806 [INFO] Finished HE601626.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,819 [INFO] Finished HE601627.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,832 [INFO] Finished HE601630.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,847 [INFO] Finished HE601630.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,856 [INFO] Finished HE601630.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,860 [INFO] Finished HE601629.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,864 [INFO] Finished HE601630.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,874 [INFO] Finished HE601631.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,896 [INFO] Finished HE601629.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,918 [INFO] Finished HE601631.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,931 [INFO] Finished HE601629.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,949 [INFO] Finished OU426848.1-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,949 [INFO] Finished HE601629.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,950 [INFO] Finished OU426849.1-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,977 [INFO] Finished HE601631.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:28,066 [INFO] Finished OU426848.1-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:28,132 [INFO] Finished OU426850.1:DUP.\n",
      "2023-01-18 11:56:28,176 [INFO] Finished HE601628.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:28,282 [INFO] Writing to your output file.\n",
      "2023-01-18 11:56:28,304 [INFO] Loading reference genome...\n",
      "2023-01-18 11:56:51,966 [INFO] Cleaning temporary files.\n",
      "2023-01-18 11:56:52,691 [INFO] Finished in 1549.51 seconds.\n"
     ]
    }
   ],
   "source": [
    "#sv call\n",
    "# for sample in samples:\n",
    "for sample in [\"smle_pzq_es\"]:\n",
    "\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    bam      = \"{}/results/minimap2/{}.bam\".format(proj_dir, sample)\n",
    "    query_fq = \"{}/results/nanofilt/{}.filtered.fastq.gz\".format(proj_dir, sample)\n",
    "    out_dir  = \"{}/{}\".format(os.getcwd(), sample)\n",
    "    \n",
    "    Path(\"{}\".format(out_dir)).mkdir(parents=True, exist_ok=True)\n",
    "    !~/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 --genotype {bam} {ref_fas} {sample}.cutesv.vcf {out_dir}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1e83da5-f68a-40fb-aa3a-29a1cf626880",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Read Correction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "172a44a7-2e93-45bf-9df2-0303ecfb389e",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/read_correction\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/read_correction\".format(proj_dir))\n",
    "\n",
    "for sample in samples:\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "    Path(\"{}/results/read_correction/{}\".format(proj_dir, sample)).mkdir(parents=True, exist_ok=True)\n",
    "    \n",
    "    os.chdir(\"{}/results/read_correction\".format(proj_dir, sample))\n",
    "    \n",
    "    !conda run -n ont_assembly --cwd . --live-stream canu -correct -genomesize=385m -useGrid=False -p {sample}  -nanopore {proj_dir}/results/raw_reads/{sample}.raw_pass.fastq\n",
    "    \n",
    "    os.chdir(\"{}/results/read_correction\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39d8bb33-5379-4a91-b4bc-6c0f510761db",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash\n",
    "\n",
    "for SAMPLE in 'smor' 'smle_pzq_es' 'smle_pzq_er' 'smbre' 'smeg'; do\n",
    "\n",
    "    if [ -d ${SAMPLE} ]; then\n",
    "        rm -rf ${SAMPLE}\n",
    "    fi\n",
    "    \n",
    "    mkdir ${SAMPLE}\n",
    "    cd ${SAMPLE}\n",
    "    \n",
    "    LOG=$(pwd)/${SAMPLE}.log \n",
    "    \n",
    "    QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 192 -N ${SAMPLE} -o ${LOG}\"\n",
    "    CMD=\"conda run -n ont_assembly --cwd . --live-stream canu -correct -genomesize=385m -useGrid=False -p ${SAMPLE}  -nanopore /master/nplatt/sm_nanopore/results/raw_reads/${SAMPLE}.raw_pass.fastq\"\n",
    "    \n",
    "    echo $CMD | $QSUB\n",
    "    cd ..\n",
    "done\n",
    "\n",
    "for SAMPLE in 'smor' 'smle_pzq_es' 'smbre'; do\n",
    "    rm -rf $SAMPLE*\n",
    "    QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 192 -N ${SAMPLE} -o ${SAMPLE}.log\"\n",
    "    CMD=\"conda run -n ont_assembly --cwd . --live-stream canu -correct -genomesize=385m -useGrid=False -p ${SAMPLE}/${SAMPLE}  -nanopore /master/nplatt/sm_nanopore/results/raw_reads/${SAMPLE}.raw_pass.fastq\"\n",
    "    \n",
    "    echo $CMD | $QSUB\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "218b0bb6-ecc0-4350-9c66-cbcb0666e51f",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Genome assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "2622026b-669a-41a4-b722-2ff5ccb79193",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/assembly\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/assembly\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13c30768-4f95-4904-a110-4d27b38ff59b",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "    !flye --nano-cor {proj_dir}/results/read_correction/{sample}.correctedReads.fasta.gz -g 385m -t 128 -i 3 -o {sample}_rnd0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2a56d32-e431-4195-a5f8-ea56d1accc5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "for SAMPLE in 'smor' 'smle_pzq_es' 'smle_pzq_er' 'smbre' 'smeg'; do\n",
    "\n",
    "    if [ -d ${SAMPLE} ]; then\n",
    "        rm -rf ${SAMPLE}\n",
    "    fi\n",
    "    \n",
    "    mkdir ${SAMPLE}\n",
    "    cd ${SAMPLE}\n",
    "    \n",
    "    LOG=$(pwd)/${SAMPLE}.log \n",
    "    READS=\"/master/nplatt/sm_nanopore/results/read_correction/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz\"\n",
    "    OUT=\"/master/nplatt/sm_nanopore/results/assembly/${SAMPLE}/${SAMPLE}_rnd0\"\n",
    "    \n",
    "    QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 128 -N ${SAMPLE} -o ${LOG}\"\n",
    "    CMD=\"conda run -n sm_ont-process_reads --cwd . --live-stream flye --nano-cor ${READS} -g 385m -t 128 -i 3 -o ${OUT}\"\n",
    "    \n",
    "    echo $CMD | $QSUB\n",
    "    cd ..\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "47970cec-d20a-4f7a-b46f-a6a210585b9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>total_length</th>\n",
       "      <th>fragments</th>\n",
       "      <th>fragment_n50</th>\n",
       "      <th>largest_fragment</th>\n",
       "      <th>scaffolds</th>\n",
       "      <th>mean_coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>smor</td>\n",
       "      <td>386656054</td>\n",
       "      <td>1674</td>\n",
       "      <td>2150179</td>\n",
       "      <td>8090473</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smle_pzq_es</td>\n",
       "      <td>403425661</td>\n",
       "      <td>1122</td>\n",
       "      <td>3481572</td>\n",
       "      <td>14813648</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smle_pzq_er</td>\n",
       "      <td>382213358</td>\n",
       "      <td>1968</td>\n",
       "      <td>1273691</td>\n",
       "      <td>5768858</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smbre</td>\n",
       "      <td>384951218</td>\n",
       "      <td>1020</td>\n",
       "      <td>2298594</td>\n",
       "      <td>14522024</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smeg</td>\n",
       "      <td>370379461</td>\n",
       "      <td>1963</td>\n",
       "      <td>871323</td>\n",
       "      <td>6130221</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample total_length fragments fragment_n50 largest_fragment scaffolds  \\\n",
       "0         smor    386656054      1674      2150179          8090473         0   \n",
       "1  smle_pzq_es    403425661      1122      3481572         14813648         0   \n",
       "2  smle_pzq_er    382213358      1968      1273691          5768858         0   \n",
       "3        smbre    384951218      1020      2298594         14522024         0   \n",
       "4         smeg    370379461      1963       871323          6130221         0   \n",
       "\n",
       "  mean_coverage  \n",
       "0            23  \n",
       "1            33  \n",
       "2            15  \n",
       "3            24  \n",
       "4            14  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#get df of assembly stats\n",
    "total_lengths=[]\n",
    "fragments=[]\n",
    "fragment_n50s=[]\n",
    "largest_frags=[]\n",
    "scaffolds=[]\n",
    "mean_coverages=[]\n",
    "    \n",
    "for sample in samples:\n",
    "    info = !tail -n 8 {sample}/{sample}.log | head -n 6\n",
    "    total_lengths.append(info[0].split(\"\\t\")[-1])\n",
    "    fragments.append(info[1].split(\"\\t\")[-1])\n",
    "    fragment_n50s.append(info[2].split(\"\\t\")[-1])\n",
    "    largest_frags.append(info[3].split(\"\\t\")[-1])\n",
    "    scaffolds.append(info[4].split(\"\\t\")[-1])\n",
    "    mean_coverages.append(info[5].split(\"\\t\")[-1])\n",
    "    \n",
    "assembly_df=pd.DataFrame(data = [samples, total_lengths, fragments, fragment_n50s, largest_frags, scaffolds, mean_coverages]).T\n",
    "assembly_df.columns=[\"sample\", \"total_length\", \"fragments\", \"fragment_n50\", \"largest_fragment\", \"scaffolds\", \"mean_coverage\"]\n",
    "assembly_df.to_csv(\"assembly_info.csv\", index=False)\n",
    "assembly_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "c87716bd-431b-4e06-ab21-e8afc04f2a24",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "smor\n",
      "smle_pzq_es\n",
      "smle_pzq_er\n",
      "smbre\n",
      "smeg\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    print(sample)\n",
    "    ! cp /master/nplatt/sm_nanopore/results/assembly/{sample}/{sample}_rnd0/assembly.fasta {sample}_rnd0.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12de9d93-c60c-41f4-8556-370f4e265f84",
   "metadata": {},
   "source": [
    "## polish assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "c1ea2ab8-cc08-4c9e-8819-9803b66f777c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/polish_assembly\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/polish_assembly\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "e71a2c33-555b-4c0a-aa42-7af40c7e2947",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "smor\n",
      "smle_pzq_es\n",
      "smle_pzq_er\n",
      "smbre\n",
      "smeg\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    print(sample)\n",
    "    !ln -s /master/nplatt/sm_nanopore/results/assembly/{sample}_rnd0.fasta {sample}.assembly_rnd0.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "f2841bc5-98dc-490b-b1a6-c1968fdbf9a6",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smor ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::7.984*1.68] collected minimizers\n",
      "[M::mm_idx_gen::9.287*2.93] sorted minimizers\n",
      "[M::main::9.288*2.93] loaded/built the index for 1674 target sequence(s)\n",
      "[M::mm_mapopt_update::10.191*2.76] mid_occ = 384\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1674\n",
      "[M::mm_idx_stat::10.857*2.65] distinct minimizers: 30374815 (65.19% are singletons); average occurrences: 2.391; average spacing: 5.323; total length: 386656054\n",
      "[M::worker_pipeline::152.226*39.59] mapped 79894 sequences\n",
      "[M::worker_pipeline::307.705*40.33] mapped 77039 sequences\n",
      "[M::worker_pipeline::476.213*40.86] mapped 74651 sequences\n",
      "[M::worker_pipeline::629.258*41.74] mapped 76580 sequences\n",
      "[M::worker_pipeline::794.384*41.56] mapped 73726 sequences\n",
      "[M::worker_pipeline::965.772*41.19] mapped 76750 sequences\n",
      "[M::worker_pipeline::1144.195*40.83] mapped 76077 sequences\n",
      "[M::worker_pipeline::1289.656*41.18] mapped 77343 sequences\n",
      "[M::worker_pipeline::1453.221*41.22] mapped 75661 sequences\n",
      "[M::worker_pipeline::1617.462*40.85] mapped 78790 sequences\n",
      "[M::worker_pipeline::1764.942*41.05] mapped 78853 sequences\n",
      "[M::worker_pipeline::1933.526*41.06] mapped 76202 sequences\n",
      "[M::worker_pipeline::2095.499*41.02] mapped 73981 sequences\n",
      "[M::worker_pipeline::2273.516*41.00] mapped 70832 sequences\n",
      "[M::worker_pipeline::2436.701*41.12] mapped 70994 sequences\n",
      "[M::worker_pipeline::2600.532*41.11] mapped 72574 sequences\n",
      "[M::worker_pipeline::2774.577*40.99] mapped 71286 sequences\n",
      "[M::worker_pipeline::2945.959*40.93] mapped 72852 sequences\n",
      "[M::worker_pipeline::3124.810*40.87] mapped 73865 sequences\n",
      "[M::worker_pipeline::3281.401*40.90] mapped 75586 sequences\n",
      "[M::worker_pipeline::3432.424*40.51] mapped 60781 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smor.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smor/smor.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3432.825 sec; CPU: 139062.435 sec; Peak RSS: 22.556 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.139771 s\n",
      "[racon::Polisher::initialize] loaded sequences 101.550652 s\n",
      "[racon::Polisher::initialize] loaded overlaps 69.268296 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 19.301929 s\n",
      "[racon::Polisher::initialize] transformed data into windows 18.209538 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 195.514138 s\n",
      "[racon::Polisher::] total = 410.996559 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::7.581*1.74] collected minimizers\n",
      "[M::mm_idx_gen::8.810*2.82] sorted minimizers\n",
      "[M::main::8.811*2.82] loaded/built the index for 1399 target sequence(s)\n",
      "[M::mm_mapopt_update::9.549*2.68] mid_occ = 381\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1399\n",
      "[M::mm_idx_stat::10.053*2.59] distinct minimizers: 30397504 (65.24% are singletons); average occurrences: 2.386; average spacing: 5.322; total length: 385948710\n",
      "[M::worker_pipeline::147.574*39.59] mapped 79894 sequences\n",
      "[M::worker_pipeline::306.812*39.36] mapped 77039 sequences\n",
      "[M::worker_pipeline::475.756*40.33] mapped 74651 sequences\n",
      "[M::worker_pipeline::626.544*41.34] mapped 76580 sequences\n",
      "[M::worker_pipeline::782.457*41.45] mapped 73726 sequences\n",
      "[M::worker_pipeline::953.096*40.96] mapped 76750 sequences\n",
      "[M::worker_pipeline::1131.399*40.50] mapped 76077 sequences\n",
      "[M::worker_pipeline::1274.639*40.97] mapped 77343 sequences\n",
      "[M::worker_pipeline::1429.701*41.17] mapped 75661 sequences\n",
      "[M::worker_pipeline::1598.395*40.68] mapped 78790 sequences\n",
      "[M::worker_pipeline::1744.064*40.88] mapped 78853 sequences\n",
      "[M::worker_pipeline::1908.716*40.89] mapped 76202 sequences\n",
      "[M::worker_pipeline::2065.932*40.87] mapped 73981 sequences\n",
      "[M::worker_pipeline::2236.008*40.94] mapped 70832 sequences\n",
      "[M::worker_pipeline::2390.563*41.10] mapped 70994 sequences\n",
      "[M::worker_pipeline::2546.478*41.15] mapped 72574 sequences\n",
      "[M::worker_pipeline::2715.094*41.06] mapped 71286 sequences\n",
      "[M::worker_pipeline::2882.604*41.01] mapped 72852 sequences\n",
      "[M::worker_pipeline::3054.230*40.98] mapped 73865 sequences\n",
      "[M::worker_pipeline::3206.423*41.03] mapped 75586 sequences\n",
      "[M::worker_pipeline::3349.963*40.66] mapped 60781 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smor.assembly_rnd1.fasta /master/nplatt/sm_nanopore/results/read_correction/smor/smor.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3350.317 sec; CPU: 136217.193 sec; Peak RSS: 23.181 GB\n",
      "R2: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.144430 s\n",
      "[racon::Polisher::initialize] loaded sequences 102.767956 s\n",
      "[racon::Polisher::initialize] loaded overlaps 26.603356 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 14.940295 s\n",
      "[racon::Polisher::initialize] transformed data into windows 19.916187 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 185.418094 s\n",
      "[racon::Polisher::] total = 356.933251 s\n",
      "#\n",
      "R3: map\n",
      "[M::mm_idx_gen::7.605*1.71] collected minimizers\n",
      "[M::mm_idx_gen::8.852*2.83] sorted minimizers\n",
      "[M::main::8.852*2.83] loaded/built the index for 1369 target sequence(s)\n",
      "[M::mm_mapopt_update::9.541*2.69] mid_occ = 380\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1369\n",
      "[M::mm_idx_stat::10.048*2.61] distinct minimizers: 30396595 (65.25% are singletons); average occurrences: 2.385; average spacing: 5.322; total length: 385763855\n",
      "[M::worker_pipeline::146.197*39.51] mapped 79894 sequences\n",
      "[M::worker_pipeline::302.011*39.61] mapped 77039 sequences\n",
      "[M::worker_pipeline::468.536*40.38] mapped 74651 sequences\n",
      "[M::worker_pipeline::617.183*41.40] mapped 76580 sequences\n",
      "[M::worker_pipeline::773.340*41.48] mapped 73726 sequences\n",
      "[M::worker_pipeline::942.110*41.02] mapped 76750 sequences\n",
      "[M::worker_pipeline::1116.024*40.56] mapped 76077 sequences\n",
      "[M::worker_pipeline::1255.488*41.05] mapped 77343 sequences\n",
      "[M::worker_pipeline::1407.854*41.27] mapped 75661 sequences\n",
      "[M::worker_pipeline::1573.150*40.80] mapped 78790 sequences\n",
      "[M::worker_pipeline::1716.360*40.98] mapped 78853 sequences\n",
      "[M::worker_pipeline::1882.881*40.93] mapped 76202 sequences\n",
      "[M::worker_pipeline::2039.833*40.90] mapped 73981 sequences\n",
      "[M::worker_pipeline::2211.866*40.93] mapped 70832 sequences\n",
      "[M::worker_pipeline::2363.239*41.12] mapped 70994 sequences\n",
      "[M::worker_pipeline::2526.774*41.04] mapped 72574 sequences\n",
      "[M::worker_pipeline::2688.484*41.03] mapped 71286 sequences\n",
      "[M::worker_pipeline::2854.296*40.93] mapped 72852 sequences\n",
      "[M::worker_pipeline::3021.929*40.90] mapped 73865 sequences\n",
      "[M::worker_pipeline::3172.232*40.99] mapped 75586 sequences\n",
      "[M::worker_pipeline::3315.903*40.62] mapped 60781 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smor.assembly_rnd2.fasta /master/nplatt/sm_nanopore/results/read_correction/smor/smor.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3316.345 sec; CPU: 134699.331 sec; Peak RSS: 23.595 GB\n",
      "R3: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.149253 s\n",
      "[racon::Polisher::initialize] loaded sequences 101.847246 s\n",
      "[racon::Polisher::initialize] loaded overlaps 30.492327 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 17.914960 s\n",
      "[racon::Polisher::initialize] transformed data into windows 19.132240 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 188.459564 s\n",
      "[racon::Polisher::] total = 364.686051 s\n",
      "#\n",
      "###################### smle_pzq_es ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::7.896*1.71] collected minimizers\n",
      "[M::mm_idx_gen::9.202*2.90] sorted minimizers\n",
      "[M::main::9.202*2.90] loaded/built the index for 1122 target sequence(s)\n",
      "[M::mm_mapopt_update::9.895*2.77] mid_occ = 407\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1122\n",
      "[M::mm_idx_stat::10.381*2.69] distinct minimizers: 30553214 (64.52% are singletons); average occurrences: 2.479; average spacing: 5.325; total length: 403425661\n",
      "[M::worker_pipeline::172.686*39.54] mapped 19853 sequences\n",
      "[M::worker_pipeline::386.693*37.10] mapped 19771 sequences\n",
      "[M::worker_pipeline::603.000*37.46] mapped 20643 sequences\n",
      "[M::worker_pipeline::811.530*37.68] mapped 20836 sequences\n",
      "[M::worker_pipeline::1011.733*37.64] mapped 21265 sequences\n",
      "[M::worker_pipeline::1197.420*38.36] mapped 21689 sequences\n",
      "[M::worker_pipeline::1399.452*38.45] mapped 21616 sequences\n",
      "[M::worker_pipeline::1618.992*37.78] mapped 21757 sequences\n",
      "[M::worker_pipeline::1838.404*37.67] mapped 22496 sequences\n",
      "[M::worker_pipeline::2029.941*37.99] mapped 22806 sequences\n",
      "[M::worker_pipeline::2225.667*38.09] mapped 22165 sequences\n",
      "[M::worker_pipeline::2440.700*37.72] mapped 22957 sequences\n",
      "[M::worker_pipeline::2630.425*37.87] mapped 23544 sequences\n",
      "[M::worker_pipeline::2796.484*38.21] mapped 23371 sequences\n",
      "[M::worker_pipeline::2984.129*38.37] mapped 24068 sequences\n",
      "[M::worker_pipeline::3170.378*38.56] mapped 24289 sequences\n",
      "[M::worker_pipeline::3426.888*38.10] mapped 19452 sequences\n",
      "[M::worker_pipeline::3632.253*38.14] mapped 20038 sequences\n",
      "[M::worker_pipeline::3866.863*38.10] mapped 20058 sequences\n",
      "[M::worker_pipeline::4075.356*38.07] mapped 20416 sequences\n",
      "[M::worker_pipeline::4231.939*38.23] mapped 34327 sequences\n",
      "[M::worker_pipeline::4367.387*38.40] mapped 41954 sequences\n",
      "[M::worker_pipeline::4494.029*38.60] mapped 41664 sequences\n",
      "[M::worker_pipeline::4626.822*38.78] mapped 41677 sequences\n",
      "[M::worker_pipeline::4780.803*38.85] mapped 41139 sequences\n",
      "[M::worker_pipeline::4940.716*39.09] mapped 38763 sequences\n",
      "[M::worker_pipeline::5111.640*39.18] mapped 39457 sequences\n",
      "[M::worker_pipeline::5289.543*39.24] mapped 39956 sequences\n",
      "[M::worker_pipeline::5445.584*39.43] mapped 40366 sequences\n",
      "[M::worker_pipeline::5601.922*39.59] mapped 38901 sequences\n",
      "[M::worker_pipeline::5639.609*39.60] mapped 8242 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_es.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_es/smle_pzq_es.correctedReads.fasta.gz\n",
      "[M::main] Real time: 5640.001 sec; CPU: 223305.887 sec; Peak RSS: 23.044 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.158625 s\n",
      "[racon::Polisher::initialize] loaded sequences 146.908151 s\n",
      "[racon::Polisher::initialize] loaded overlaps 42.327313 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 15.036950 s\n",
      "[racon::Polisher::initialize] transformed data into windows 21.089318 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 254.333967 s\n",
      "[racon::Polisher::] total = 484.502313 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::8.046*1.69] collected minimizers\n",
      "[M::mm_idx_gen::9.265*2.75] sorted minimizers\n",
      "[M::main::9.265*2.75] loaded/built the index for 841 target sequence(s)\n",
      "[M::mm_mapopt_update::9.942*2.63] mid_occ = 407\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 841\n",
      "[M::mm_idx_stat::10.428*2.56] distinct minimizers: 30580531 (64.58% are singletons); average occurrences: 2.471; average spacing: 5.324; total length: 402322301\n",
      "[M::worker_pipeline::168.352*40.18] mapped 19853 sequences\n",
      "[M::worker_pipeline::373.321*37.82] mapped 19771 sequences\n",
      "[M::worker_pipeline::590.770*37.80] mapped 20643 sequences\n",
      "[M::worker_pipeline::783.351*38.57] mapped 20836 sequences\n",
      "[M::worker_pipeline::983.700*38.27] mapped 21265 sequences\n",
      "[M::worker_pipeline::1170.170*38.76] mapped 21689 sequences\n",
      "[M::worker_pipeline::1364.587*38.98] mapped 21616 sequences\n",
      "[M::worker_pipeline::1580.043*38.22] mapped 21757 sequences\n",
      "[M::worker_pipeline::1793.657*38.14] mapped 22496 sequences\n",
      "[M::worker_pipeline::1985.303*38.38] mapped 22806 sequences\n",
      "[M::worker_pipeline::2168.712*38.64] mapped 22165 sequences\n",
      "[M::worker_pipeline::2383.598*38.23] mapped 22957 sequences\n",
      "[M::worker_pipeline::2569.444*38.34] mapped 23544 sequences\n",
      "[M::worker_pipeline::2729.772*38.70] mapped 23371 sequences\n",
      "[M::worker_pipeline::2910.914*38.89] mapped 24068 sequences\n",
      "[M::worker_pipeline::3096.942*39.04] mapped 24289 sequences\n",
      "[M::worker_pipeline::3352.871*38.55] mapped 19452 sequences\n",
      "[M::worker_pipeline::3553.176*38.58] mapped 20038 sequences\n",
      "[M::worker_pipeline::3786.482*38.50] mapped 20058 sequences\n",
      "[M::worker_pipeline::3987.821*38.53] mapped 20416 sequences\n",
      "[M::worker_pipeline::4133.279*38.73] mapped 34327 sequences\n",
      "[M::worker_pipeline::4264.550*38.92] mapped 41954 sequences\n",
      "[M::worker_pipeline::4391.924*39.11] mapped 41664 sequences\n",
      "[M::worker_pipeline::4523.552*39.28] mapped 41677 sequences\n",
      "[M::worker_pipeline::4675.127*39.36] mapped 41139 sequences\n",
      "[M::worker_pipeline::4834.138*39.59] mapped 38763 sequences\n",
      "[M::worker_pipeline::5000.242*39.70] mapped 39457 sequences\n",
      "[M::worker_pipeline::5174.190*39.76] mapped 39956 sequences\n",
      "[M::worker_pipeline::5327.673*39.93] mapped 40366 sequences\n",
      "[M::worker_pipeline::5480.221*40.07] mapped 38901 sequences\n",
      "[M::worker_pipeline::5517.473*40.07] mapped 8242 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_es.assembly_rnd1.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_es/smle_pzq_es.correctedReads.fasta.gz\n",
      "[M::main] Real time: 5517.820 sec; CPU: 221092.124 sec; Peak RSS: 22.918 GB\n",
      "R2: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.256752 s\n",
      "[racon::Polisher::initialize] loaded sequences 147.591536 s\n",
      "[racon::Polisher::initialize] loaded overlaps 40.335285 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 17.719516 s\n",
      "[racon::Polisher::initialize] transformed data into windows 20.023588 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 253.454532 s\n",
      "[racon::Polisher::] total = 484.181523 s\n",
      "#\n",
      "R3: map\n",
      "[M::mm_idx_gen::9.634*1.66] collected minimizers\n",
      "[M::mm_idx_gen::10.842*2.57] sorted minimizers\n",
      "[M::main::10.842*2.57] loaded/built the index for 789 target sequence(s)\n",
      "[M::mm_mapopt_update::11.581*2.47] mid_occ = 407\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 789\n",
      "[M::mm_idx_stat::12.072*2.41] distinct minimizers: 30573180 (64.62% are singletons); average occurrences: 2.470; average spacing: 5.324; total length: 401969069\n",
      "[M::worker_pipeline::171.402*39.34] mapped 19853 sequences\n",
      "[M::worker_pipeline::380.799*36.89] mapped 19771 sequences\n",
      "[M::worker_pipeline::594.338*37.59] mapped 20643 sequences\n",
      "[M::worker_pipeline::783.218*38.59] mapped 20836 sequences\n",
      "[M::worker_pipeline::978.978*38.38] mapped 21265 sequences\n",
      "[M::worker_pipeline::1162.084*39.02] mapped 21689 sequences\n",
      "[M::worker_pipeline::1362.541*38.99] mapped 21616 sequences\n",
      "[M::worker_pipeline::1576.207*38.31] mapped 21757 sequences\n",
      "[M::worker_pipeline::1788.328*38.24] mapped 22496 sequences\n",
      "[M::worker_pipeline::1977.860*38.59] mapped 22806 sequences\n",
      "[M::worker_pipeline::2170.103*38.70] mapped 22165 sequences\n",
      "[M::worker_pipeline::2381.712*38.33] mapped 22957 sequences\n",
      "[M::worker_pipeline::2566.540*38.50] mapped 23544 sequences\n",
      "[M::worker_pipeline::2727.783*38.87] mapped 23371 sequences\n",
      "[M::worker_pipeline::2906.828*39.04] mapped 24068 sequences\n",
      "[M::worker_pipeline::3093.745*39.18] mapped 24289 sequences\n",
      "[M::worker_pipeline::3343.000*38.73] mapped 19452 sequences\n",
      "[M::worker_pipeline::3544.310*38.78] mapped 20038 sequences\n",
      "[M::worker_pipeline::3779.800*38.68] mapped 20058 sequences\n",
      "[M::worker_pipeline::3984.200*38.67] mapped 20416 sequences\n",
      "[M::worker_pipeline::4138.044*38.83] mapped 34327 sequences\n",
      "[M::worker_pipeline::4273.452*38.97] mapped 41954 sequences\n",
      "[M::worker_pipeline::4401.064*39.15] mapped 41664 sequences\n",
      "[M::worker_pipeline::4532.924*39.32] mapped 41677 sequences\n",
      "[M::worker_pipeline::4684.672*39.39] mapped 41139 sequences\n",
      "[M::worker_pipeline::4843.334*39.62] mapped 38763 sequences\n",
      "[M::worker_pipeline::5007.122*39.75] mapped 39457 sequences\n",
      "[M::worker_pipeline::5182.820*39.81] mapped 39956 sequences\n",
      "[M::worker_pipeline::5334.561*40.00] mapped 40366 sequences\n",
      "[M::worker_pipeline::5487.920*40.14] mapped 38901 sequences\n",
      "[M::worker_pipeline::5525.582*40.14] mapped 8242 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_es.assembly_rnd2.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_es/smle_pzq_es.correctedReads.fasta.gz\n",
      "[M::main] Real time: 5525.873 sec; CPU: 221794.069 sec; Peak RSS: 23.020 GB\n",
      "R3: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.346187 s\n",
      "[racon::Polisher::initialize] loaded sequences 148.885734 s\n",
      "[racon::Polisher::initialize] loaded overlaps 41.326534 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 14.228298 s\n",
      "[racon::Polisher::initialize] transformed data into windows 18.844883 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 252.791732 s\n",
      "[racon::Polisher::] total = 481.391174 s\n",
      "#\n",
      "###################### smle_pzq_er ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::8.547*1.66] collected minimizers\n",
      "[M::mm_idx_gen::9.878*2.76] sorted minimizers\n",
      "[M::main::9.878*2.76] loaded/built the index for 1968 target sequence(s)\n",
      "[M::mm_mapopt_update::10.615*2.64] mid_occ = 373\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1968\n",
      "[M::mm_idx_stat::11.119*2.57] distinct minimizers: 30209362 (65.19% are singletons); average occurrences: 2.377; average spacing: 5.323; total length: 382213358\n",
      "[M::worker_pipeline::101.212*37.43] mapped 76770 sequences\n",
      "[M::worker_pipeline::197.376*39.82] mapped 71320 sequences\n",
      "[M::worker_pipeline::292.345*40.90] mapped 73006 sequences\n",
      "[M::worker_pipeline::395.987*41.15] mapped 69078 sequences\n",
      "[M::worker_pipeline::507.789*39.95] mapped 68270 sequences\n",
      "[M::worker_pipeline::605.266*39.60] mapped 76406 sequences\n",
      "[M::worker_pipeline::690.203*40.15] mapped 75581 sequences\n",
      "[M::worker_pipeline::780.265*40.53] mapped 73411 sequences\n",
      "[M::worker_pipeline::874.280*40.80] mapped 73667 sequences\n",
      "[M::worker_pipeline::968.959*40.55] mapped 74686 sequences\n",
      "[M::worker_pipeline::1064.246*40.48] mapped 77941 sequences\n",
      "[M::worker_pipeline::1150.558*40.63] mapped 78752 sequences\n",
      "[M::worker_pipeline::1239.019*40.81] mapped 76620 sequences\n",
      "[M::worker_pipeline::1294.453*40.67] mapped 43491 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_er.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_er/smle_pzq_er.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1294.799 sec; CPU: 52650.649 sec; Peak RSS: 22.097 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.077128 s\n",
      "[racon::Polisher::initialize] loaded sequences 65.019206 s\n",
      "[racon::Polisher::initialize] loaded overlaps 24.796660 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 11.954963 s\n",
      "[racon::Polisher::initialize] transformed data into windows 7.371992 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 143.387897 s4.845812 s\n",
      "[racon::Polisher::] total = 257.503844 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::7.541*1.69] collected minimizers\n",
      "[M::mm_idx_gen::8.788*2.77] sorted minimizers\n",
      "[M::main::8.789*2.77] loaded/built the index for 1611 target sequence(s)\n",
      "[M::mm_mapopt_update::9.488*2.64] mid_occ = 364\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1611\n",
      "[M::mm_idx_stat::9.973*2.56] distinct minimizers: 30322343 (65.28% are singletons); average occurrences: 2.365; average spacing: 5.321; total length: 381614371\n",
      "[M::worker_pipeline::100.131*37.83] mapped 76770 sequences\n",
      "[M::worker_pipeline::195.433*39.64] mapped 71320 sequences\n",
      "[M::worker_pipeline::286.855*40.64] mapped 73006 sequences\n",
      "[M::worker_pipeline::389.978*40.81] mapped 69078 sequences\n",
      "[M::worker_pipeline::498.317*39.88] mapped 68270 sequences\n",
      "[M::worker_pipeline::592.994*39.57] mapped 76406 sequences\n",
      "[M::worker_pipeline::673.276*40.20] mapped 75581 sequences\n",
      "[M::worker_pipeline::761.828*40.59] mapped 73411 sequences\n",
      "[M::worker_pipeline::855.219*40.80] mapped 73667 sequences\n",
      "[M::worker_pipeline::944.630*40.70] mapped 74686 sequences\n",
      "[M::worker_pipeline::1034.611*40.65] mapped 77941 sequences\n",
      "[M::worker_pipeline::1125.942*40.59] mapped 78752 sequences\n",
      "[M::worker_pipeline::1209.989*40.83] mapped 76620 sequences\n",
      "[M::worker_pipeline::1262.212*40.77] mapped 43491 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_er.assembly_rnd1.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_er/smle_pzq_er.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1262.587 sec; CPU: 51454.778 sec; Peak RSS: 21.712 GB\n",
      "R2: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.109804 s\n",
      "[racon::Polisher::initialize] loaded sequences 65.880837 s\n",
      "[racon::Polisher::initialize] loaded overlaps 26.117030 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 12.140027 s\n",
      "[racon::Polisher::initialize] transformed data into windows 7.580020 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 140.047301 s\n",
      "[racon::Polisher::] total = 256.554459 s\n",
      "#\n",
      "R3: map\n",
      "[M::mm_idx_gen::7.511*1.70] collected minimizers\n",
      "[M::mm_idx_gen::8.833*2.85] sorted minimizers\n",
      "[M::main::8.834*2.85] loaded/built the index for 1582 target sequence(s)\n",
      "[M::mm_mapopt_update::9.492*2.72] mid_occ = 364\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1582\n",
      "[M::mm_idx_stat::9.977*2.64] distinct minimizers: 30331368 (65.29% are singletons); average occurrences: 2.365; average spacing: 5.321; total length: 381704002\n",
      "[M::worker_pipeline::100.601*38.07] mapped 76770 sequences\n",
      "[M::worker_pipeline::193.603*40.14] mapped 71320 sequences\n",
      "[M::worker_pipeline::286.347*40.77] mapped 73006 sequences\n",
      "[M::worker_pipeline::387.207*41.10] mapped 69078 sequences\n",
      "[M::worker_pipeline::491.405*40.24] mapped 68270 sequences\n",
      "[M::worker_pipeline::584.664*40.02] mapped 76406 sequences\n",
      "[M::worker_pipeline::664.283*40.59] mapped 75581 sequences\n",
      "[M::worker_pipeline::751.438*41.05] mapped 73411 sequences\n",
      "[M::worker_pipeline::844.293*41.29] mapped 73667 sequences\n",
      "[M::worker_pipeline::935.497*41.07] mapped 74686 sequences\n",
      "[M::worker_pipeline::1024.562*41.09] mapped 77941 sequences\n",
      "[M::worker_pipeline::1109.975*41.20] mapped 78752 sequences\n",
      "[M::worker_pipeline::1194.609*41.44] mapped 76620 sequences\n",
      "[M::worker_pipeline::1245.823*41.34] mapped 43491 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_er.assembly_rnd2.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_er/smle_pzq_er.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1246.252 sec; CPU: 51501.835 sec; Peak RSS: 21.574 GB\n",
      "R3: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.172103 s\n",
      "[racon::Polisher::initialize] loaded sequences 66.222945 s\n",
      "[racon::Polisher::initialize] loaded overlaps 26.870027 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 12.611046 s\n",
      "[racon::Polisher::initialize] transformed data into windows 10.259038 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 146.096231 s\n",
      "[racon::Polisher::] total = 267.307171 s\n",
      "#\n",
      "###################### smbre ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::7.514*1.69] collected minimizers\n",
      "[M::mm_idx_gen::8.769*2.80] sorted minimizers\n",
      "[M::main::8.770*2.80] loaded/built the index for 1020 target sequence(s)\n",
      "[M::mm_mapopt_update::9.478*2.66] mid_occ = 394\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1020\n",
      "[M::mm_idx_stat::9.978*2.58] distinct minimizers: 30091754 (65.42% are singletons); average occurrences: 2.403; average spacing: 5.324; total length: 384951218\n",
      "[M::worker_pipeline::193.150*34.78] mapped 53283 sequences\n",
      "[M::worker_pipeline::345.736*38.72] mapped 52871 sequences\n",
      "[M::worker_pipeline::512.004*39.90] mapped 48725 sequences\n",
      "[M::worker_pipeline::677.763*40.43] mapped 46760 sequences\n",
      "[M::worker_pipeline::843.969*40.72] mapped 43965 sequences\n",
      "[M::worker_pipeline::1001.822*41.18] mapped 43922 sequences\n",
      "[M::worker_pipeline::1163.905*41.65] mapped 42570 sequences\n",
      "[M::worker_pipeline::1336.817*41.55] mapped 40977 sequences\n",
      "[M::worker_pipeline::1513.143*41.51] mapped 39946 sequences\n",
      "[M::worker_pipeline::1676.543*41.50] mapped 46552 sequences\n",
      "[M::worker_pipeline::1856.737*41.03] mapped 47237 sequences\n",
      "[M::worker_pipeline::2018.012*41.15] mapped 46567 sequences\n",
      "[M::worker_pipeline::2178.742*41.25] mapped 46788 sequences\n",
      "[M::worker_pipeline::2330.101*41.50] mapped 45522 sequences\n",
      "[M::worker_pipeline::2488.746*41.61] mapped 45538 sequences\n",
      "[M::worker_pipeline::2634.157*41.77] mapped 45607 sequences\n",
      "[M::worker_pipeline::2790.556*41.71] mapped 45182 sequences\n",
      "[M::worker_pipeline::2942.096*41.77] mapped 45409 sequences\n",
      "[M::worker_pipeline::3073.416*41.84] mapped 44380 sequences\n",
      "[M::worker_pipeline::3211.393*41.90] mapped 44938 sequences\n",
      "[M::worker_pipeline::3357.080*41.72] mapped 45630 sequences\n",
      "[M::worker_pipeline::3498.487*41.47] mapped 40156 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smbre.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smbre/smbre.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3498.906 sec; CPU: 145074.942 sec; Peak RSS: 24.759 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.078661 s\n",
      "[racon::Polisher::initialize] loaded sequences 107.383010 s\n",
      "[racon::Polisher::initialize] loaded overlaps 36.186241 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 16.406332 s\n",
      "[racon::Polisher::initialize] transformed data into windows 15.943155 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 195.406518 s\n",
      "[racon::Polisher::] total = 376.592673 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::7.578*1.70] collected minimizers\n",
      "[M::mm_idx_gen::8.792*2.78] sorted minimizers\n",
      "[M::main::8.793*2.78] loaded/built the index for 863 target sequence(s)\n",
      "[M::mm_mapopt_update::9.495*2.64] mid_occ = 392\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 863\n",
      "[M::mm_idx_stat::9.976*2.57] distinct minimizers: 30104826 (65.45% are singletons); average occurrences: 2.399; average spacing: 5.324; total length: 384537830\n",
      "[M::worker_pipeline::196.025*35.18] mapped 53283 sequences\n",
      "[M::worker_pipeline::348.845*38.96] mapped 52871 sequences\n",
      "[M::worker_pipeline::528.009*39.62] mapped 48725 sequences\n",
      "[M::worker_pipeline::700.069*39.96] mapped 46760 sequences\n",
      "[M::worker_pipeline::874.344*40.26] mapped 43965 sequences\n",
      "[M::worker_pipeline::1036.449*40.81] mapped 43922 sequences\n",
      "[M::worker_pipeline::1206.986*41.14] mapped 42570 sequences\n",
      "[M::worker_pipeline::1380.384*41.19] mapped 40977 sequences\n",
      "[M::worker_pipeline::1561.838*41.20] mapped 39946 sequences\n",
      "[M::worker_pipeline::1730.506*41.10] mapped 46552 sequences\n",
      "[M::worker_pipeline::1911.670*40.70] mapped 47237 sequences\n",
      "[M::worker_pipeline::2066.317*41.02] mapped 46567 sequences\n",
      "[M::worker_pipeline::2233.712*41.04] mapped 46788 sequences\n",
      "[M::worker_pipeline::2394.546*41.20] mapped 45522 sequences\n",
      "[M::worker_pipeline::2558.987*41.24] mapped 45538 sequences\n",
      "[M::worker_pipeline::2715.758*41.36] mapped 45607 sequences\n",
      "[M::worker_pipeline::2871.871*41.39] mapped 45182 sequences\n",
      "[M::worker_pipeline::3039.750*41.31] mapped 45409 sequences\n",
      "[M::worker_pipeline::3175.911*41.36] mapped 44380 sequences\n",
      "[M::worker_pipeline::3327.163*41.34] mapped 44938 sequences\n",
      "[M::worker_pipeline::3478.397*41.17] mapped 45630 sequences\n",
      "[M::worker_pipeline::3613.450*41.07] mapped 40156 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smbre.assembly_rnd1.fasta /master/nplatt/sm_nanopore/results/read_correction/smbre/smbre.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3613.843 sec; CPU: 148414.394 sec; Peak RSS: 23.428 GB\n",
      "R2: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.138405 s\n",
      "[racon::Polisher::initialize] loaded sequences 102.947766 s\n",
      "[racon::Polisher::initialize] loaded overlaps 25.953495 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 12.424720 s\n",
      "[racon::Polisher::initialize] transformed data into windows 14.520990 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 193.301941 s\n",
      "[racon::Polisher::] total = 354.562270 s\n",
      "#\n",
      "R3: map\n",
      "[M::mm_idx_gen::8.107*1.69] collected minimizers\n",
      "[M::mm_idx_gen::9.529*2.98] sorted minimizers\n",
      "[M::main::9.530*2.98] loaded/built the index for 831 target sequence(s)\n",
      "[M::mm_mapopt_update::10.381*2.81] mid_occ = 391\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 831\n",
      "[M::mm_idx_stat::10.960*2.72] distinct minimizers: 30104374 (65.45% are singletons); average occurrences: 2.399; average spacing: 5.324; total length: 384487880\n",
      "[M::worker_pipeline::191.993*35.65] mapped 53283 sequences\n",
      "[M::worker_pipeline::344.222*39.09] mapped 52871 sequences\n",
      "[M::worker_pipeline::522.196*39.74] mapped 48725 sequences\n",
      "[M::worker_pipeline::690.916*40.17] mapped 46760 sequences\n",
      "[M::worker_pipeline::864.704*40.33] mapped 43965 sequences\n",
      "[M::worker_pipeline::1021.268*40.83] mapped 43922 sequences\n",
      "[M::worker_pipeline::1192.620*41.12] mapped 42570 sequences\n",
      "[M::worker_pipeline::1362.209*41.19] mapped 40977 sequences\n",
      "[M::worker_pipeline::1537.998*41.34] mapped 39946 sequences\n",
      "[M::worker_pipeline::1704.273*41.23] mapped 46552 sequences\n",
      "[M::worker_pipeline::1889.321*40.75] mapped 47237 sequences\n",
      "[M::worker_pipeline::2048.109*40.89] mapped 46567 sequences\n",
      "[M::worker_pipeline::2213.962*40.91] mapped 46788 sequences\n",
      "[M::worker_pipeline::2369.793*41.09] mapped 45522 sequences\n",
      "[M::worker_pipeline::2533.122*41.16] mapped 45538 sequences\n",
      "[M::worker_pipeline::2685.127*41.29] mapped 45607 sequences\n",
      "[M::worker_pipeline::2839.448*41.33] mapped 45182 sequences\n",
      "[M::worker_pipeline::3000.577*41.33] mapped 45409 sequences\n",
      "[M::worker_pipeline::3134.637*41.40] mapped 44380 sequences\n",
      "[M::worker_pipeline::3279.134*41.38] mapped 44938 sequences\n",
      "[M::worker_pipeline::3422.723*41.27] mapped 45630 sequences\n",
      "[M::worker_pipeline::3563.041*41.09] mapped 40156 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smbre.assembly_rnd2.fasta /master/nplatt/sm_nanopore/results/read_correction/smbre/smbre.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3563.511 sec; CPU: 146416.730 sec; Peak RSS: 23.257 GB\n",
      "R3: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.310798 s\n",
      "[racon::Polisher::initialize] loaded sequences 107.066274 s\n",
      "[racon::Polisher::initialize] loaded overlaps 26.866338 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 14.815388 s\n",
      "[racon::Polisher::initialize] transformed data into windows 16.890523 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 188.803997 s\n",
      "[racon::Polisher::] total = 360.313372 s\n",
      "#\n",
      "###################### smeg ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::7.068*1.68] collected minimizers\n",
      "[M::mm_idx_gen::8.334*2.85] sorted minimizers\n",
      "[M::main::8.334*2.85] loaded/built the index for 1963 target sequence(s)\n",
      "[M::mm_mapopt_update::9.029*2.71] mid_occ = 374\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1963\n",
      "[M::mm_idx_stat::9.508*2.62] distinct minimizers: 29693941 (65.87% are singletons); average occurrences: 2.343; average spacing: 5.325; total length: 370379461\n",
      "[M::worker_pipeline::150.602*36.83] mapped 46550 sequences\n",
      "[M::worker_pipeline::269.723*39.19] mapped 48696 sequences\n",
      "[M::worker_pipeline::410.898*39.44] mapped 48598 sequences\n",
      "[M::worker_pipeline::548.265*39.24] mapped 46410 sequences\n",
      "[M::worker_pipeline::688.176*38.50] mapped 49713 sequences\n",
      "[M::worker_pipeline::843.005*38.13] mapped 49778 sequences\n",
      "[M::worker_pipeline::979.189*38.40] mapped 48029 sequences\n",
      "[M::worker_pipeline::1119.284*38.46] mapped 47397 sequences\n",
      "[M::worker_pipeline::1236.508*38.74] mapped 63067 sequences\n",
      "[M::worker_pipeline::1343.908*39.18] mapped 69436 sequences\n",
      "[M::worker_pipeline::1466.845*39.07] mapped 67208 sequences\n",
      "[M::worker_pipeline::1591.167*39.05] mapped 65917 sequences\n",
      "[M::worker_pipeline::1608.341*38.89] mapped 7204 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smeg.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smeg/smeg.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1608.539 sec; CPU: 62555.827 sec; Peak RSS: 21.167 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.004812 s\n",
      "[racon::Polisher::initialize] loaded sequences 57.975932 s\n",
      "[racon::Polisher::initialize] loaded overlaps 18.154140 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 7.779614 s\n",
      "[racon::Polisher::initialize] transformed data into windows 6.775523 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 106.537008 s\n",
      "[racon::Polisher::] total = 201.228917 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::7.289*1.65] collected minimizers\n",
      "[M::mm_idx_gen::8.646*2.85] sorted minimizers\n",
      "[M::main::8.646*2.85] loaded/built the index for 1704 target sequence(s)\n",
      "[M::mm_mapopt_update::9.344*2.71] mid_occ = 369\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1704\n",
      "[M::mm_idx_stat::9.822*2.63] distinct minimizers: 29717886 (65.87% are singletons); average occurrences: 2.337; average spacing: 5.324; total length: 369728076\n",
      "[M::worker_pipeline::148.591*36.65] mapped 46550 sequences\n",
      "[M::worker_pipeline::266.160*39.31] mapped 48696 sequences\n",
      "[M::worker_pipeline::398.206*40.43] mapped 48598 sequences\n",
      "[M::worker_pipeline::528.110*40.40] mapped 46410 sequences\n",
      "[M::worker_pipeline::666.205*39.49] mapped 49713 sequences\n",
      "[M::worker_pipeline::818.088*38.86] mapped 49778 sequences\n",
      "[M::worker_pipeline::952.898*39.00] mapped 48029 sequences\n",
      "[M::worker_pipeline::1090.429*39.05] mapped 47397 sequences\n",
      "[M::worker_pipeline::1203.600*39.29] mapped 63067 sequences\n",
      "[M::worker_pipeline::1305.587*39.72] mapped 69436 sequences\n",
      "[M::worker_pipeline::1426.536*39.66] mapped 67208 sequences\n",
      "[M::worker_pipeline::1542.291*39.80] mapped 65917 sequences\n",
      "[M::worker_pipeline::1559.713*39.65] mapped 7204 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smeg.assembly_rnd1.fasta /master/nplatt/sm_nanopore/results/read_correction/smeg/smeg.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1560.060 sec; CPU: 61840.668 sec; Peak RSS: 22.912 GB\n",
      "R2: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.715573 s\n",
      "[racon::Polisher::initialize] loaded sequences 57.372261 s\n",
      "[racon::Polisher::initialize] loaded overlaps 16.191489 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 8.778752 s\n",
      "[racon::Polisher::initialize] transformed data into windows 7.200109 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 109.529461 s\n",
      "[racon::Polisher::] total = 203.975579 s\n",
      "#\n",
      "R3: map\n",
      "[M::mm_idx_gen::7.185*1.67] collected minimizers\n",
      "[M::mm_idx_gen::8.396*2.79] sorted minimizers\n",
      "[M::main::8.397*2.79] loaded/built the index for 1669 target sequence(s)\n",
      "[M::mm_mapopt_update::9.086*2.65] mid_occ = 368\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1669\n",
      "[M::mm_idx_stat::9.561*2.57] distinct minimizers: 29717896 (65.88% are singletons); average occurrences: 2.336; average spacing: 5.324; total length: 369543290\n",
      "[M::worker_pipeline::149.730*38.03] mapped 46550 sequences\n",
      "[M::worker_pipeline::273.541*39.97] mapped 48696 sequences\n",
      "[M::worker_pipeline::413.320*40.10] mapped 48598 sequences\n",
      "[M::worker_pipeline::544.038*40.43] mapped 46410 sequences\n",
      "[M::worker_pipeline::683.224*39.58] mapped 49713 sequences\n",
      "[M::worker_pipeline::832.897*39.42] mapped 49778 sequences\n",
      "[M::worker_pipeline::967.910*39.63] mapped 48029 sequences\n",
      "[M::worker_pipeline::1108.349*39.56] mapped 47397 sequences\n",
      "[M::worker_pipeline::1226.402*39.75] mapped 63067 sequences\n",
      "[M::worker_pipeline::1329.992*40.16] mapped 69436 sequences\n",
      "[M::worker_pipeline::1454.251*40.01] mapped 67208 sequences\n",
      "[M::worker_pipeline::1574.498*40.08] mapped 65917 sequences\n",
      "[M::worker_pipeline::1591.432*39.93] mapped 7204 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smeg.assembly_rnd2.fasta /master/nplatt/sm_nanopore/results/read_correction/smeg/smeg.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1591.774 sec; CPU: 63541.456 sec; Peak RSS: 21.523 GB\n",
      "R3: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.071946 s\n",
      "[racon::Polisher::initialize] loaded sequences 58.422955 s\n",
      "[racon::Polisher::initialize] loaded overlaps 22.068410 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 6.597411 s\n",
      "[racon::Polisher::initialize] transformed data into windows 6.582953 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 108.250975 s\n",
      "[racon::Polisher::] total = 206.464460 s\n",
      "#\n"
     ]
    }
   ],
   "source": [
    "#set the initial round\n",
    "rnd=1\n",
    "for sample in samples:\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    for rnd in [1, 2, 3]:\n",
    "        #define major i/o files\n",
    "        current_assembly = \"{}.assembly_rnd{}.fasta\".format(sample, str(rnd-1))\n",
    "        corrected_reads = \"{}/results/read_correction/{}/{}.correctedReads.fasta.gz\".format(proj_dir, sample, sample)\n",
    "        polished_assembly = \"{}.assembly_rnd{}.fasta\".format(sample, str(rnd))\n",
    "        n_threads=48\n",
    "        \n",
    "        #map corrected reads to the assembly\n",
    "        print(\"R{}: map\".format(rnd))\n",
    "        minimap_cmd = \"conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "            minimap2 \\\n",
    "                -a \\\n",
    "                -map-ont \\\n",
    "                -t {} \\\n",
    "                {} \\\n",
    "                {}  \\\n",
    "                >{}.racon_rnd{}.sam\".format(n_threads, current_assembly, corrected_reads, sample, rnd)\n",
    "        minimap_cmd=' '.join(minimap_cmd.split())\n",
    "        ! {minimap_cmd}\n",
    "\n",
    "        #polish the assembly\n",
    "        print(\"R{}: polish\".format(rnd))\n",
    "        racon_cmd = \"conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "            racon \\\n",
    "                -t {} \\\n",
    "                {} \\\n",
    "                {}.racon_rnd{}.sam \\\n",
    "                {} \\\n",
    "                >{}\".format(n_threads, corrected_reads, sample, rnd, current_assembly, polished_assembly)\n",
    "        racon_cmd=' '.join(racon_cmd.split())\n",
    "        ! {racon_cmd}\n",
    "        print(\"#\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "769878fa-14fb-4009-bcda-6bfb0ebfb0cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Genome contamination "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe7b6d67-4fe8-4ebe-9c99-e0b2e496768f",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p ${PROJ_DIR}/data/remove_contamination && cd ${PROJ_DIR}/data/remove_contamination"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9f52c18-0713-4d8c-9f23-54cf4f0e2db9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#https://github.com/timkahlke/BASTA/wiki/3.-BASTA-Usage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d36c2fd-7da4-4c2c-81fe-c7235e4646b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "conda activate basta_py3\n",
    "\n",
    "mkdir -p ${PROJ_DIR}/data/remove_contamination && cd ${PROJ_DIR}/data/remove_contamination\n",
    "\n",
    "genome_updater.sh \\\n",
    "    -g \"\" \\\n",
    "    -d \"refseq\" \\\n",
    "    -c \"representative genome\" \\\n",
    "    -l \"complete genome,chromosome\" \\\n",
    "    -f \"genomic.fna.gz,assembly_report.txt\" \\\n",
    "    -o \"refseq_216_fas\" \\\n",
    "    -b \"v1\" \\\n",
    "    -a \\\n",
    "    -A species:1 \\\n",
    "    -m \\\n",
    "    -u \\\n",
    "    -r \\\n",
    "    -p \\\n",
    "    -t 48\n",
    "\n",
    "#uncompress for kraken2\n",
    "cd refseq_216_fas/v1/files\n",
    "    \n",
    "ls *.gz | parallel -j 48 gunzip {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4cb02c95-be25-4c46-a008-df182698c5eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create blast database\n",
    "cat *.fas >refseq_216.fas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d04f719-c4f9-4018-8f8f-9835b0d29d4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "makeblastdb -dbtype nucle -in "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73defe93-b772-4160-9b68-eaf749526c4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "blastm -outfmt 6 -in <assembly> -out <assembly>.blastn --threads 48"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3de6022-1acc-405b-95e8-f20e510c835e",
   "metadata": {},
   "outputs": [],
   "source": [
    "./bin/basta taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66a39e42-ef28-4e3c-95b3-43a947b9ce38",
   "metadata": {},
   "outputs": [],
   "source": [
    "./bin/basta download gb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55039b2c-0aa4-4dc9-a570-0281d131bbcd",
   "metadata": {},
   "outputs": [],
   "source": [
    "./bin/basta sequence INPUT_FILE OUTPUT_FILE MAPPING_FILE_TYPE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "097dcc29-a9bc-4ac1-82b3-0bee37219dc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "./scripts/filter_basta_fasta.py [options] FASTA_FILE FILTERED_OUTPUT_FILE NAME_OF_TAXON BASTA_FILE"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c43a6f3f-029f-45af-935a-50ea0e2da649",
   "metadata": {},
   "source": [
    "# "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf5d162e-af14-49b6-9d97-ed0ce9101400",
   "metadata": {},
   "source": [
    "## genome quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b64bd824-e1a6-439e-9b07-2d4addd4dd33",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/busco\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/busco\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8434c2ca-cfee-4a25-9d97-750a6b90928c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-02-07 16:58:55 INFO:\t***** Start a BUSCO v5.4.3 analysis, current time: 02/07/2023 16:58:55 *****\n",
      "2023-02-07 16:58:55 INFO:\tConfiguring BUSCO with local environment\n",
      "2023-02-07 16:58:55 INFO:\tMode is genome\n",
      "2023-02-07 16:58:55 INFO:\tDownloading information on latest versions of BUSCO data...\n",
      "2023-02-07 16:58:57 INFO:\tInput file is /master/nplatt/sm_nanopore/results/polish_assembly/smor.assembly_rnd3.fasta\n",
      "2023-02-07 16:58:57 INFO:\tNo lineage specified. Running lineage auto selector.\n",
      "\n",
      "2023-02-07 16:58:57 INFO:\t***** Starting Auto Select Lineage *****\n",
      "\tThis process runs BUSCO on the generic lineage datasets for the domains archaea, bacteria and eukaryota. Once the optimal domain is selected, BUSCO automatically attempts to find the most appropriate BUSCO dataset to use based on phylogenetic placement.\n",
      "\t--auto-lineage-euk and --auto-lineage-prok are also available if you know your input assembly is, or is not, an eukaryote. See the user guide for more information.\n",
      "\tA reminder: Busco evaluations are valid when an appropriate dataset is used, i.e., the dataset belongs to the lineage of the species to test. Because of overlapping markers/spurious matches among domains, busco matches in another domain do not necessarily mean that your genome/proteome contains sequences from this domain. However, a high busco score in multiple domains might help you identify possible contaminations.\n",
      "2023-02-07 16:58:57 INFO:\tDownloading file 'https://busco-data.ezlab.org/v5/data/lineages/eukaryota_odb10.2020-09-10.tar.gz'\n",
      "2023-02-07 16:59:10 INFO:\tDecompressing file '/master/nplatt/sm_nanopore/results/busco/busco_downloads/lineages/eukaryota_odb10.tar.gz'\n",
      "2023-02-07 16:59:16 INFO:\tRunning BUSCO using lineage dataset eukaryota_odb10 (eukaryota, 2020-09-10)\n",
      "2023-02-07 16:59:16 INFO:\tRunning 1 job(s) on bbtools, starting at 02/07/2023 16:59:16\n",
      "2023-02-07 16:59:22 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-02-07 16:59:25 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 16:59:25\n",
      "2023-02-07 17:06:26 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:06:39 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 17:06:39 INFO:\tRunning 255 job(s) on hmmsearch, starting at 02/07/2023 17:06:39\n",
      "2023-02-07 17:06:45 INFO:\t[hmmsearch]\t26 of 255 task(s) completed\n",
      "2023-02-07 17:06:45 INFO:\t[hmmsearch]\t51 of 255 task(s) completed\n",
      "2023-02-07 17:06:45 INFO:\t[hmmsearch]\t77 of 255 task(s) completed\n",
      "2023-02-07 17:06:45 INFO:\t[hmmsearch]\t102 of 255 task(s) completed\n",
      "2023-02-07 17:06:45 INFO:\t[hmmsearch]\t128 of 255 task(s) completed\n",
      "2023-02-07 17:06:45 INFO:\t[hmmsearch]\t153 of 255 task(s) completed\n",
      "2023-02-07 17:06:45 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-02-07 17:06:45 INFO:\t[hmmsearch]\t204 of 255 task(s) completed\n",
      "2023-02-07 17:06:46 INFO:\t[hmmsearch]\t230 of 255 task(s) completed\n",
      "2023-02-07 17:06:46 INFO:\t[hmmsearch]\t255 of 255 task(s) completed\n",
      "2023-02-07 17:06:47 INFO:\tResults:\tC:82.0%[S:81.2%,D:0.8%],F:10.2%,M:7.8%,n:255\t   \n",
      "\n",
      "2023-02-07 17:06:47 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-02-07 17:06:51 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 17:06:51\n",
      "2023-02-07 17:17:48 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:17:50 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 17:17:50 INFO:\tRunning 46 job(s) on hmmsearch, starting at 02/07/2023 17:17:50\n",
      "2023-02-07 17:17:55 INFO:\t[hmmsearch]\t5 of 46 task(s) completed\n",
      "2023-02-07 17:17:56 INFO:\t[hmmsearch]\t10 of 46 task(s) completed\n",
      "2023-02-07 17:17:56 INFO:\t[hmmsearch]\t14 of 46 task(s) completed\n",
      "2023-02-07 17:17:56 INFO:\t[hmmsearch]\t19 of 46 task(s) completed\n",
      "2023-02-07 17:17:56 INFO:\t[hmmsearch]\t23 of 46 task(s) completed\n",
      "2023-02-07 17:17:56 INFO:\t[hmmsearch]\t28 of 46 task(s) completed\n",
      "2023-02-07 17:17:56 INFO:\t[hmmsearch]\t33 of 46 task(s) completed\n",
      "2023-02-07 17:17:56 INFO:\t[hmmsearch]\t37 of 46 task(s) completed\n",
      "2023-02-07 17:17:56 INFO:\t[hmmsearch]\t42 of 46 task(s) completed\n",
      "2023-02-07 17:17:56 INFO:\t[hmmsearch]\t46 of 46 task(s) completed\n",
      "2023-02-07 17:17:58 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-02-07 17:18:00 INFO:\tResults:\tC:83.6%[S:82.4%,D:1.2%],F:9.0%,M:7.4%,n:255\t   \n",
      "\n",
      "2023-02-07 17:18:13 INFO:\teukaryota_odb10 selected\n",
      "\n",
      "2023-02-07 17:18:13 INFO:\t***** Searching tree for chosen lineage to find best taxonomic match *****\n",
      "\n",
      "2023-02-07 17:18:16 INFO:\tExtract markers...\n",
      "2023-02-07 17:18:16 INFO:\tDownloading file 'https://busco-data.ezlab.org/v5/data/placement_files/list_of_reference_markers.eukaryota_odb10.2019-12-16.txt.tar.gz'\n",
      "2023-02-07 17:18:17 INFO:\tDecompressing file '/master/nplatt/sm_nanopore/results/busco/busco_downloads/placement_files/list_of_reference_markers.eukaryota_odb10.2019-12-16.txt.tar.gz'\n",
      "2023-02-07 17:18:17 INFO:\tDownloading file 'https://busco-data.ezlab.org/v5/data/placement_files/tree.eukaryota_odb10.2019-12-16.nwk.tar.gz'\n",
      "2023-02-07 17:18:18 INFO:\tDecompressing file '/master/nplatt/sm_nanopore/results/busco/busco_downloads/placement_files/tree.eukaryota_odb10.2019-12-16.nwk.tar.gz'\n",
      "2023-02-07 17:18:18 INFO:\tDownloading file 'https://busco-data.ezlab.org/v5/data/placement_files/tree_metadata.eukaryota_odb10.2019-12-16.txt.tar.gz'\n",
      "2023-02-07 17:18:19 INFO:\tDecompressing file '/master/nplatt/sm_nanopore/results/busco/busco_downloads/placement_files/tree_metadata.eukaryota_odb10.2019-12-16.txt.tar.gz'\n",
      "2023-02-07 17:18:19 INFO:\tDownloading file 'https://busco-data.ezlab.org/v5/data/placement_files/supermatrix.aln.eukaryota_odb10.2019-12-16.faa.tar.gz'\n",
      "2023-02-07 17:18:26 INFO:\tDecompressing file '/master/nplatt/sm_nanopore/results/busco/busco_downloads/placement_files/supermatrix.aln.eukaryota_odb10.2019-12-16.faa.tar.gz'\n",
      "2023-02-07 17:18:27 INFO:\tDownloading file 'https://busco-data.ezlab.org/v5/data/placement_files/mapping_taxids-busco_dataset_name.eukaryota_odb10.2019-12-16.txt.tar.gz'\n",
      "2023-02-07 17:18:27 INFO:\tDecompressing file '/master/nplatt/sm_nanopore/results/busco/busco_downloads/placement_files/mapping_taxids-busco_dataset_name.eukaryota_odb10.2019-12-16.txt.tar.gz'\n",
      "2023-02-07 17:18:27 INFO:\tDownloading file 'https://busco-data.ezlab.org/v5/data/placement_files/mapping_taxid-lineage.eukaryota_odb10.2019-12-16.txt.tar.gz'\n",
      "2023-02-07 17:18:29 INFO:\tDecompressing file '/master/nplatt/sm_nanopore/results/busco/busco_downloads/placement_files/mapping_taxid-lineage.eukaryota_odb10.2019-12-16.txt.tar.gz'\n",
      "2023-02-07 17:18:29 INFO:\tPlace the markers on the reference tree...\n",
      "2023-02-07 17:18:29 INFO:\tRunning 1 job(s) on sepp, starting at 02/07/2023 17:18:29\n",
      "2023-02-07 17:19:32 INFO:\t[sepp]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:19:35 INFO:\tLineage metazoa is selected, supported by 48 markers out of 91\n",
      "2023-02-07 17:19:35 INFO:\tDownloading file 'https://busco-data.ezlab.org/v5/data/lineages/metazoa_odb10.2021-02-24.tar.gz'\n",
      "2023-02-07 17:19:48 INFO:\tDecompressing file '/master/nplatt/sm_nanopore/results/busco/busco_downloads/lineages/metazoa_odb10.tar.gz'\n",
      "2023-02-07 17:20:01 INFO:\tRunning BUSCO using lineage dataset metazoa_odb10 (eukaryota, 2021-02-17)\n",
      "2023-02-07 17:20:01 INFO:\tRunning 1 job(s) on bbtools, starting at 02/07/2023 17:20:01\n",
      "2023-02-07 17:20:08 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:20:11 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 17:20:11\n",
      "2023-02-07 17:26:52 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:27:09 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 17:27:09 INFO:\tRunning 954 job(s) on hmmsearch, starting at 02/07/2023 17:27:09\n",
      "2023-02-07 17:27:15 INFO:\t[hmmsearch]\t96 of 954 task(s) completed\n",
      "2023-02-07 17:27:15 INFO:\t[hmmsearch]\t191 of 954 task(s) completed\n",
      "2023-02-07 17:27:15 INFO:\t[hmmsearch]\t287 of 954 task(s) completed\n",
      "2023-02-07 17:27:15 INFO:\t[hmmsearch]\t287 of 954 task(s) completed\n",
      "2023-02-07 17:27:16 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-02-07 17:27:16 INFO:\t[hmmsearch]\t478 of 954 task(s) completed\n",
      "2023-02-07 17:27:17 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-02-07 17:27:17 INFO:\t[hmmsearch]\t859 of 954 task(s) completed\n",
      "2023-02-07 17:27:18 INFO:\t[hmmsearch]\t954 of 954 task(s) completed\n",
      "2023-02-07 17:27:21 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-02-07 17:27:37 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 17:27:37\n",
      "2023-02-07 17:39:06 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:39:14 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 17:39:14 INFO:\tRunning 327 job(s) on hmmsearch, starting at 02/07/2023 17:39:14\n",
      "2023-02-07 17:39:19 INFO:\t[hmmsearch]\t33 of 327 task(s) completed\n",
      "2023-02-07 17:39:20 INFO:\t[hmmsearch]\t66 of 327 task(s) completed\n",
      "2023-02-07 17:39:20 INFO:\t[hmmsearch]\t99 of 327 task(s) completed\n",
      "2023-02-07 17:39:20 INFO:\t[hmmsearch]\t131 of 327 task(s) completed\n",
      "2023-02-07 17:39:21 INFO:\t[hmmsearch]\t164 of 327 task(s) completed\n",
      "2023-02-07 17:39:21 INFO:\t[hmmsearch]\t197 of 327 task(s) completed\n",
      "2023-02-07 17:39:21 INFO:\t[hmmsearch]\t229 of 327 task(s) completed\n",
      "2023-02-07 17:39:21 INFO:\t[hmmsearch]\t262 of 327 task(s) completed\n",
      "2023-02-07 17:39:22 INFO:\t[hmmsearch]\t295 of 327 task(s) completed\n",
      "2023-02-07 17:39:22 INFO:\t[hmmsearch]\t327 of 327 task(s) completed\n",
      "2023-02-07 17:39:31 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-02-07 17:39:36 INFO:\tResults:\tC:68.1%[S:66.8%,D:1.3%],F:7.1%,M:24.8%,n:954\t   \n",
      "\n",
      "2023-02-07 17:40:05 INFO:\t\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from generic domain eukaryota_odb10      |\n",
      "\t--------------------------------------------------\n",
      "\t|C:83.6%[S:82.4%,D:1.2%],F:9.0%,M:7.4%,n:255      |\n",
      "\t|213\tComplete BUSCOs (C)                       |\n",
      "\t|210\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|3\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|23\tFragmented BUSCOs (F)                     |\n",
      "\t|19\tMissing BUSCOs (M)                        |\n",
      "\t|255\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from dataset metazoa_odb10               |\n",
      "\t--------------------------------------------------\n",
      "\t|C:68.1%[S:66.8%,D:1.3%],F:7.1%,M:24.8%,n:954     |\n",
      "\t|649\tComplete BUSCOs (C)                       |\n",
      "\t|637\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|12\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|68\tFragmented BUSCOs (F)                     |\n",
      "\t|237\tMissing BUSCOs (M)                        |\n",
      "\t|954\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "2023-02-07 17:40:05 INFO:\tBUSCO analysis done. Total running time: 2468 seconds\n",
      "2023-02-07 17:40:05 INFO:\tResults written in /master/nplatt/sm_nanopore/results/busco/smor\n",
      "2023-02-07 17:40:05 INFO:\tFor assistance with interpreting the results, please consult the userguide: https://busco.ezlab.org/busco_userguide.html\n",
      "\n",
      "2023-02-07 17:40:05 INFO:\tVisit this page https://gitlab.com/ezlab/busco#how-to-cite-busco to see how to cite BUSCO\n",
      "2023-02-07 17:40:08 INFO:\t***** Start a BUSCO v5.4.3 analysis, current time: 02/07/2023 17:40:08 *****\n",
      "2023-02-07 17:40:08 INFO:\tConfiguring BUSCO with local environment\n",
      "2023-02-07 17:40:08 INFO:\tMode is genome\n",
      "2023-02-07 17:40:08 INFO:\tDownloading information on latest versions of BUSCO data...\n",
      "2023-02-07 17:40:11 INFO:\tInput file is /master/nplatt/sm_nanopore/results/polish_assembly/smle_pzq_es.assembly_rnd3.fasta\n",
      "2023-02-07 17:40:11 INFO:\tNo lineage specified. Running lineage auto selector.\n",
      "\n",
      "2023-02-07 17:40:11 INFO:\t***** Starting Auto Select Lineage *****\n",
      "\tThis process runs BUSCO on the generic lineage datasets for the domains archaea, bacteria and eukaryota. Once the optimal domain is selected, BUSCO automatically attempts to find the most appropriate BUSCO dataset to use based on phylogenetic placement.\n",
      "\t--auto-lineage-euk and --auto-lineage-prok are also available if you know your input assembly is, or is not, an eukaryote. See the user guide for more information.\n",
      "\tA reminder: Busco evaluations are valid when an appropriate dataset is used, i.e., the dataset belongs to the lineage of the species to test. Because of overlapping markers/spurious matches among domains, busco matches in another domain do not necessarily mean that your genome/proteome contains sequences from this domain. However, a high busco score in multiple domains might help you identify possible contaminations.\n",
      "2023-02-07 17:40:14 INFO:\tRunning BUSCO using lineage dataset eukaryota_odb10 (eukaryota, 2020-09-10)\n",
      "2023-02-07 17:40:14 INFO:\tRunning 1 job(s) on bbtools, starting at 02/07/2023 17:40:14\n",
      "2023-02-07 17:40:17 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:40:17 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 17:40:17\n",
      "2023-02-07 17:47:21 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:47:32 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 17:47:32 INFO:\tRunning 255 job(s) on hmmsearch, starting at 02/07/2023 17:47:32\n",
      "2023-02-07 17:47:37 INFO:\t[hmmsearch]\t51 of 255 task(s) completed\n",
      "2023-02-07 17:47:37 INFO:\t[hmmsearch]\t77 of 255 task(s) completed\n",
      "2023-02-07 17:47:38 INFO:\t[hmmsearch]\t102 of 255 task(s) completed\n",
      "2023-02-07 17:47:38 INFO:\t[hmmsearch]\t128 of 255 task(s) completed\n",
      "2023-02-07 17:47:38 INFO:\t[hmmsearch]\t153 of 255 task(s) completed\n",
      "2023-02-07 17:47:38 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-02-07 17:47:38 INFO:\t[hmmsearch]\t204 of 255 task(s) completed\n",
      "2023-02-07 17:47:38 INFO:\t[hmmsearch]\t230 of 255 task(s) completed\n",
      "2023-02-07 17:47:38 INFO:\t[hmmsearch]\t255 of 255 task(s) completed\n",
      "2023-02-07 17:47:39 INFO:\tResults:\tC:87.5%[S:85.9%,D:1.6%],F:6.7%,M:5.8%,n:255\t   \n",
      "\n",
      "2023-02-07 17:47:39 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-02-07 17:47:43 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 17:47:43\n",
      "2023-02-07 17:57:41 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:57:43 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 17:57:43 INFO:\tRunning 32 job(s) on hmmsearch, starting at 02/07/2023 17:57:43\n",
      "2023-02-07 17:57:49 INFO:\t[hmmsearch]\t4 of 32 task(s) completed\n",
      "2023-02-07 17:57:49 INFO:\t[hmmsearch]\t7 of 32 task(s) completed\n",
      "2023-02-07 17:57:49 INFO:\t[hmmsearch]\t10 of 32 task(s) completed\n",
      "2023-02-07 17:57:49 INFO:\t[hmmsearch]\t13 of 32 task(s) completed\n",
      "2023-02-07 17:57:49 INFO:\t[hmmsearch]\t17 of 32 task(s) completed\n",
      "2023-02-07 17:57:49 INFO:\t[hmmsearch]\t23 of 32 task(s) completed\n",
      "2023-02-07 17:57:49 INFO:\t[hmmsearch]\t26 of 32 task(s) completed\n",
      "2023-02-07 17:57:49 INFO:\t[hmmsearch]\t29 of 32 task(s) completed\n",
      "2023-02-07 17:57:49 INFO:\t[hmmsearch]\t32 of 32 task(s) completed\n",
      "2023-02-07 17:57:51 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-02-07 17:57:53 INFO:\tResults:\tC:88.3%[S:86.7%,D:1.6%],F:6.3%,M:5.4%,n:255\t   \n",
      "\n",
      "2023-02-07 17:58:05 INFO:\teukaryota_odb10 selected\n",
      "\n",
      "2023-02-07 17:58:05 INFO:\t***** Searching tree for chosen lineage to find best taxonomic match *****\n",
      "\n",
      "2023-02-07 17:58:06 INFO:\tExtract markers...\n",
      "2023-02-07 17:58:06 INFO:\tPlace the markers on the reference tree...\n",
      "2023-02-07 17:58:06 INFO:\tRunning 1 job(s) on sepp, starting at 02/07/2023 17:58:06\n",
      "2023-02-07 17:59:09 INFO:\t[sepp]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:59:11 INFO:\tLineage metazoa is selected, supported by 52 markers out of 92\n",
      "2023-02-07 17:59:11 WARNING:\tThe file or folder /master/nplatt/sm_nanopore/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "2023-02-07 17:59:12 INFO:\tRunning BUSCO using lineage dataset metazoa_odb10 (eukaryota, 2021-02-17)\n",
      "2023-02-07 17:59:12 INFO:\tRunning 1 job(s) on bbtools, starting at 02/07/2023 17:59:12\n",
      "2023-02-07 17:59:18 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-02-07 17:59:18 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 17:59:18\n",
      "2023-02-07 18:06:04 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 18:06:18 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 18:06:18 INFO:\tRunning 954 job(s) on hmmsearch, starting at 02/07/2023 18:06:18\n",
      "2023-02-07 18:06:25 INFO:\t[hmmsearch]\t96 of 954 task(s) completed\n",
      "2023-02-07 18:06:26 INFO:\t[hmmsearch]\t191 of 954 task(s) completed\n",
      "2023-02-07 18:06:26 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-02-07 18:06:26 INFO:\t[hmmsearch]\t478 of 954 task(s) completed\n",
      "2023-02-07 18:06:27 INFO:\t[hmmsearch]\t573 of 954 task(s) completed\n",
      "2023-02-07 18:06:27 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-02-07 18:06:27 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-02-07 18:06:27 INFO:\t[hmmsearch]\t859 of 954 task(s) completed\n",
      "2023-02-07 18:06:29 INFO:\t[hmmsearch]\t954 of 954 task(s) completed\n",
      "2023-02-07 18:06:32 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-02-07 18:06:46 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 18:06:46\n",
      "2023-02-07 18:18:18 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 18:18:24 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 18:18:24 INFO:\tRunning 302 job(s) on hmmsearch, starting at 02/07/2023 18:18:24\n",
      "2023-02-07 18:18:29 INFO:\t[hmmsearch]\t31 of 302 task(s) completed\n",
      "2023-02-07 18:18:29 INFO:\t[hmmsearch]\t61 of 302 task(s) completed\n",
      "2023-02-07 18:18:29 INFO:\t[hmmsearch]\t91 of 302 task(s) completed\n",
      "2023-02-07 18:18:29 INFO:\t[hmmsearch]\t121 of 302 task(s) completed\n",
      "2023-02-07 18:18:30 INFO:\t[hmmsearch]\t151 of 302 task(s) completed\n",
      "2023-02-07 18:18:30 INFO:\t[hmmsearch]\t182 of 302 task(s) completed\n",
      "2023-02-07 18:18:30 INFO:\t[hmmsearch]\t182 of 302 task(s) completed\n",
      "2023-02-07 18:18:30 INFO:\t[hmmsearch]\t212 of 302 task(s) completed\n",
      "2023-02-07 18:18:30 INFO:\t[hmmsearch]\t242 of 302 task(s) completed\n",
      "2023-02-07 18:18:30 INFO:\t[hmmsearch]\t272 of 302 task(s) completed\n",
      "2023-02-07 18:18:30 INFO:\t[hmmsearch]\t302 of 302 task(s) completed\n",
      "2023-02-07 18:18:40 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-02-07 18:18:45 INFO:\tResults:\tC:69.6%[S:68.0%,D:1.6%],F:6.6%,M:23.8%,n:954\t   \n",
      "\n",
      "2023-02-07 18:19:04 INFO:\t\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from generic domain eukaryota_odb10      |\n",
      "\t--------------------------------------------------\n",
      "\t|C:88.3%[S:86.7%,D:1.6%],F:6.3%,M:5.4%,n:255      |\n",
      "\t|225\tComplete BUSCOs (C)                       |\n",
      "\t|221\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|4\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|16\tFragmented BUSCOs (F)                     |\n",
      "\t|14\tMissing BUSCOs (M)                        |\n",
      "\t|255\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from dataset metazoa_odb10               |\n",
      "\t--------------------------------------------------\n",
      "\t|C:69.6%[S:68.0%,D:1.6%],F:6.6%,M:23.8%,n:954     |\n",
      "\t|664\tComplete BUSCOs (C)                       |\n",
      "\t|649\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|15\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|63\tFragmented BUSCOs (F)                     |\n",
      "\t|227\tMissing BUSCOs (M)                        |\n",
      "\t|954\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "2023-02-07 18:19:04 INFO:\tBUSCO analysis done with WARNING(s). Total running time: 2333 seconds\n",
      "\n",
      "***** Summary of warnings: *****\n",
      "2023-02-07 17:59:11 WARNING:busco.BuscoDownloadManager\tThe file or folder /master/nplatt/sm_nanopore/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "\n",
      "2023-02-07 18:19:04 INFO:\tResults written in /master/nplatt/sm_nanopore/results/busco/smle_pzq_es\n",
      "2023-02-07 18:19:04 INFO:\tFor assistance with interpreting the results, please consult the userguide: https://busco.ezlab.org/busco_userguide.html\n",
      "\n",
      "2023-02-07 18:19:04 INFO:\tVisit this page https://gitlab.com/ezlab/busco#how-to-cite-busco to see how to cite BUSCO\n",
      "2023-02-07 18:19:07 INFO:\t***** Start a BUSCO v5.4.3 analysis, current time: 02/07/2023 18:19:07 *****\n",
      "2023-02-07 18:19:07 INFO:\tConfiguring BUSCO with local environment\n",
      "2023-02-07 18:19:07 INFO:\tMode is genome\n",
      "2023-02-07 18:19:07 INFO:\tDownloading information on latest versions of BUSCO data...\n",
      "2023-02-07 18:19:10 INFO:\tInput file is /master/nplatt/sm_nanopore/results/polish_assembly/smle_pzq_er.assembly_rnd3.fasta\n",
      "2023-02-07 18:19:10 INFO:\tNo lineage specified. Running lineage auto selector.\n",
      "\n",
      "2023-02-07 18:19:10 INFO:\t***** Starting Auto Select Lineage *****\n",
      "\tThis process runs BUSCO on the generic lineage datasets for the domains archaea, bacteria and eukaryota. Once the optimal domain is selected, BUSCO automatically attempts to find the most appropriate BUSCO dataset to use based on phylogenetic placement.\n",
      "\t--auto-lineage-euk and --auto-lineage-prok are also available if you know your input assembly is, or is not, an eukaryote. See the user guide for more information.\n",
      "\tA reminder: Busco evaluations are valid when an appropriate dataset is used, i.e., the dataset belongs to the lineage of the species to test. Because of overlapping markers/spurious matches among domains, busco matches in another domain do not necessarily mean that your genome/proteome contains sequences from this domain. However, a high busco score in multiple domains might help you identify possible contaminations.\n",
      "2023-02-07 18:19:12 INFO:\tRunning BUSCO using lineage dataset eukaryota_odb10 (eukaryota, 2020-09-10)\n",
      "2023-02-07 18:19:12 INFO:\tRunning 1 job(s) on bbtools, starting at 02/07/2023 18:19:12\n",
      "2023-02-07 18:19:17 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-02-07 18:19:17 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 18:19:17\n",
      "2023-02-07 18:26:13 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 18:26:28 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 18:26:28 INFO:\tRunning 255 job(s) on hmmsearch, starting at 02/07/2023 18:26:28\n",
      "2023-02-07 18:26:33 INFO:\t[hmmsearch]\t26 of 255 task(s) completed\n",
      "2023-02-07 18:26:34 INFO:\t[hmmsearch]\t51 of 255 task(s) completed\n",
      "2023-02-07 18:26:34 INFO:\t[hmmsearch]\t77 of 255 task(s) completed\n",
      "2023-02-07 18:26:34 INFO:\t[hmmsearch]\t102 of 255 task(s) completed\n",
      "2023-02-07 18:26:34 INFO:\t[hmmsearch]\t128 of 255 task(s) completed\n",
      "2023-02-07 18:26:34 INFO:\t[hmmsearch]\t153 of 255 task(s) completed\n",
      "2023-02-07 18:26:34 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-02-07 18:26:34 INFO:\t[hmmsearch]\t230 of 255 task(s) completed\n",
      "2023-02-07 18:26:34 INFO:\t[hmmsearch]\t255 of 255 task(s) completed\n",
      "2023-02-07 18:26:35 INFO:\tResults:\tC:79.3%[S:77.3%,D:2.0%],F:11.0%,M:9.7%,n:255\t   \n",
      "\n",
      "2023-02-07 18:26:35 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-02-07 18:26:40 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 18:26:40\n",
      "2023-02-07 18:38:04 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 18:38:10 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 18:38:10 INFO:\tRunning 53 job(s) on hmmsearch, starting at 02/07/2023 18:38:10\n",
      "2023-02-07 18:38:15 INFO:\t[hmmsearch]\t6 of 53 task(s) completed\n",
      "2023-02-07 18:38:15 INFO:\t[hmmsearch]\t11 of 53 task(s) completed\n",
      "2023-02-07 18:38:15 INFO:\t[hmmsearch]\t16 of 53 task(s) completed\n",
      "2023-02-07 18:38:15 INFO:\t[hmmsearch]\t22 of 53 task(s) completed\n",
      "2023-02-07 18:38:15 INFO:\t[hmmsearch]\t27 of 53 task(s) completed\n",
      "2023-02-07 18:38:15 INFO:\t[hmmsearch]\t32 of 53 task(s) completed\n",
      "2023-02-07 18:38:15 INFO:\t[hmmsearch]\t38 of 53 task(s) completed\n",
      "2023-02-07 18:38:16 INFO:\t[hmmsearch]\t43 of 53 task(s) completed\n",
      "2023-02-07 18:38:16 INFO:\t[hmmsearch]\t48 of 53 task(s) completed\n",
      "2023-02-07 18:38:16 INFO:\t[hmmsearch]\t53 of 53 task(s) completed\n",
      "2023-02-07 18:38:18 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-02-07 18:38:20 INFO:\tResults:\tC:81.6%[S:79.6%,D:2.0%],F:9.0%,M:9.4%,n:255\t   \n",
      "\n",
      "2023-02-07 18:38:35 INFO:\teukaryota_odb10 selected\n",
      "\n",
      "2023-02-07 18:38:35 INFO:\t***** Searching tree for chosen lineage to find best taxonomic match *****\n",
      "\n",
      "2023-02-07 18:38:35 INFO:\tExtract markers...\n",
      "2023-02-07 18:38:35 INFO:\tPlace the markers on the reference tree...\n",
      "2023-02-07 18:38:35 INFO:\tRunning 1 job(s) on sepp, starting at 02/07/2023 18:38:35\n",
      "2023-02-07 18:39:36 INFO:\t[sepp]\t1 of 1 task(s) completed\n",
      "2023-02-07 18:39:39 INFO:\tLineage metazoa is selected, supported by 45 markers out of 88\n",
      "2023-02-07 18:39:39 WARNING:\tThe file or folder /master/nplatt/sm_nanopore/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "2023-02-07 18:39:39 INFO:\tRunning BUSCO using lineage dataset metazoa_odb10 (eukaryota, 2021-02-17)\n",
      "2023-02-07 18:39:39 INFO:\tRunning 1 job(s) on bbtools, starting at 02/07/2023 18:39:39\n",
      "2023-02-07 18:39:46 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-02-07 18:39:46 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 18:39:46\n",
      "2023-02-07 18:46:25 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 18:46:44 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 18:46:44 INFO:\tRunning 954 job(s) on hmmsearch, starting at 02/07/2023 18:46:44\n",
      "2023-02-07 18:46:50 INFO:\t[hmmsearch]\t96 of 954 task(s) completed\n",
      "2023-02-07 18:46:50 INFO:\t[hmmsearch]\t191 of 954 task(s) completed\n",
      "2023-02-07 18:46:51 INFO:\t[hmmsearch]\t287 of 954 task(s) completed\n",
      "2023-02-07 18:46:51 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-02-07 18:46:51 INFO:\t[hmmsearch]\t478 of 954 task(s) completed\n",
      "2023-02-07 18:46:51 INFO:\t[hmmsearch]\t573 of 954 task(s) completed\n",
      "2023-02-07 18:46:52 INFO:\t[hmmsearch]\t668 of 954 task(s) completed\n",
      "2023-02-07 18:46:52 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-02-07 18:46:52 INFO:\t[hmmsearch]\t859 of 954 task(s) completed\n",
      "2023-02-07 18:46:53 INFO:\t[hmmsearch]\t954 of 954 task(s) completed\n",
      "2023-02-07 18:46:56 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-02-07 18:47:14 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 18:47:14\n",
      "2023-02-07 18:59:10 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 18:59:20 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 18:59:20 INFO:\tRunning 375 job(s) on hmmsearch, starting at 02/07/2023 18:59:20\n",
      "2023-02-07 18:59:26 INFO:\t[hmmsearch]\t38 of 375 task(s) completed\n",
      "2023-02-07 18:59:26 INFO:\t[hmmsearch]\t75 of 375 task(s) completed\n",
      "2023-02-07 18:59:27 INFO:\t[hmmsearch]\t113 of 375 task(s) completed\n",
      "2023-02-07 18:59:27 INFO:\t[hmmsearch]\t150 of 375 task(s) completed\n",
      "2023-02-07 18:59:28 INFO:\t[hmmsearch]\t188 of 375 task(s) completed\n",
      "2023-02-07 18:59:28 INFO:\t[hmmsearch]\t225 of 375 task(s) completed\n",
      "2023-02-07 18:59:29 INFO:\t[hmmsearch]\t263 of 375 task(s) completed\n",
      "2023-02-07 18:59:29 INFO:\t[hmmsearch]\t300 of 375 task(s) completed\n",
      "2023-02-07 18:59:29 INFO:\t[hmmsearch]\t338 of 375 task(s) completed\n",
      "2023-02-07 18:59:30 INFO:\t[hmmsearch]\t375 of 375 task(s) completed\n",
      "2023-02-07 18:59:37 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-02-07 18:59:42 INFO:\tResults:\tC:63.8%[S:62.5%,D:1.3%],F:9.0%,M:27.2%,n:954\t   \n",
      "\n",
      "2023-02-07 19:00:02 INFO:\t\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from generic domain eukaryota_odb10      |\n",
      "\t--------------------------------------------------\n",
      "\t|C:81.6%[S:79.6%,D:2.0%],F:9.0%,M:9.4%,n:255      |\n",
      "\t|208\tComplete BUSCOs (C)                       |\n",
      "\t|203\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|5\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|23\tFragmented BUSCOs (F)                     |\n",
      "\t|24\tMissing BUSCOs (M)                        |\n",
      "\t|255\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from dataset metazoa_odb10               |\n",
      "\t--------------------------------------------------\n",
      "\t|C:63.8%[S:62.5%,D:1.3%],F:9.0%,M:27.2%,n:954     |\n",
      "\t|608\tComplete BUSCOs (C)                       |\n",
      "\t|596\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|12\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|86\tFragmented BUSCOs (F)                     |\n",
      "\t|260\tMissing BUSCOs (M)                        |\n",
      "\t|954\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "2023-02-07 19:00:02 INFO:\tBUSCO analysis done with WARNING(s). Total running time: 2452 seconds\n",
      "\n",
      "***** Summary of warnings: *****\n",
      "2023-02-07 18:39:39 WARNING:busco.BuscoDownloadManager\tThe file or folder /master/nplatt/sm_nanopore/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "\n",
      "2023-02-07 19:00:02 INFO:\tResults written in /master/nplatt/sm_nanopore/results/busco/smle_pzq_er\n",
      "2023-02-07 19:00:02 INFO:\tFor assistance with interpreting the results, please consult the userguide: https://busco.ezlab.org/busco_userguide.html\n",
      "\n",
      "2023-02-07 19:00:02 INFO:\tVisit this page https://gitlab.com/ezlab/busco#how-to-cite-busco to see how to cite BUSCO\n",
      "2023-02-07 19:00:05 INFO:\t***** Start a BUSCO v5.4.3 analysis, current time: 02/07/2023 19:00:05 *****\n",
      "2023-02-07 19:00:05 INFO:\tConfiguring BUSCO with local environment\n",
      "2023-02-07 19:00:05 INFO:\tMode is genome\n",
      "2023-02-07 19:00:05 INFO:\tDownloading information on latest versions of BUSCO data...\n",
      "2023-02-07 19:00:08 INFO:\tInput file is /master/nplatt/sm_nanopore/results/polish_assembly/smbre.assembly_rnd3.fasta\n",
      "2023-02-07 19:00:08 INFO:\tNo lineage specified. Running lineage auto selector.\n",
      "\n",
      "2023-02-07 19:00:08 INFO:\t***** Starting Auto Select Lineage *****\n",
      "\tThis process runs BUSCO on the generic lineage datasets for the domains archaea, bacteria and eukaryota. Once the optimal domain is selected, BUSCO automatically attempts to find the most appropriate BUSCO dataset to use based on phylogenetic placement.\n",
      "\t--auto-lineage-euk and --auto-lineage-prok are also available if you know your input assembly is, or is not, an eukaryote. See the user guide for more information.\n",
      "\tA reminder: Busco evaluations are valid when an appropriate dataset is used, i.e., the dataset belongs to the lineage of the species to test. Because of overlapping markers/spurious matches among domains, busco matches in another domain do not necessarily mean that your genome/proteome contains sequences from this domain. However, a high busco score in multiple domains might help you identify possible contaminations.\n",
      "2023-02-07 19:00:08 INFO:\tRunning BUSCO using lineage dataset eukaryota_odb10 (eukaryota, 2020-09-10)\n",
      "2023-02-07 19:00:08 INFO:\tRunning 1 job(s) on bbtools, starting at 02/07/2023 19:00:08\n",
      "2023-02-07 19:00:12 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:00:13 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 19:00:13\n",
      "2023-02-07 19:07:15 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:07:28 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 19:07:28 INFO:\tRunning 255 job(s) on hmmsearch, starting at 02/07/2023 19:07:28\n",
      "2023-02-07 19:07:33 INFO:\t[hmmsearch]\t51 of 255 task(s) completed\n",
      "2023-02-07 19:07:34 INFO:\t[hmmsearch]\t102 of 255 task(s) completed\n",
      "2023-02-07 19:07:34 INFO:\t[hmmsearch]\t128 of 255 task(s) completed\n",
      "2023-02-07 19:07:34 INFO:\t[hmmsearch]\t153 of 255 task(s) completed\n",
      "2023-02-07 19:07:34 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-02-07 19:07:34 INFO:\t[hmmsearch]\t204 of 255 task(s) completed\n",
      "2023-02-07 19:07:34 INFO:\t[hmmsearch]\t230 of 255 task(s) completed\n",
      "2023-02-07 19:07:34 INFO:\t[hmmsearch]\t255 of 255 task(s) completed\n",
      "2023-02-07 19:07:35 INFO:\tResults:\tC:82.8%[S:81.6%,D:1.2%],F:9.8%,M:7.4%,n:255\t   \n",
      "\n",
      "2023-02-07 19:07:35 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-02-07 19:07:39 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 19:07:39\n",
      "2023-02-07 19:17:24 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:17:27 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 19:17:27 INFO:\tRunning 44 job(s) on hmmsearch, starting at 02/07/2023 19:17:27\n",
      "2023-02-07 19:17:33 INFO:\t[hmmsearch]\t5 of 44 task(s) completed\n",
      "2023-02-07 19:17:33 INFO:\t[hmmsearch]\t9 of 44 task(s) completed\n",
      "2023-02-07 19:17:33 INFO:\t[hmmsearch]\t14 of 44 task(s) completed\n",
      "2023-02-07 19:17:33 INFO:\t[hmmsearch]\t18 of 44 task(s) completed\n",
      "2023-02-07 19:17:33 INFO:\t[hmmsearch]\t23 of 44 task(s) completed\n",
      "2023-02-07 19:17:33 INFO:\t[hmmsearch]\t27 of 44 task(s) completed\n",
      "2023-02-07 19:17:33 INFO:\t[hmmsearch]\t31 of 44 task(s) completed\n",
      "2023-02-07 19:17:33 INFO:\t[hmmsearch]\t36 of 44 task(s) completed\n",
      "2023-02-07 19:17:33 INFO:\t[hmmsearch]\t40 of 44 task(s) completed\n",
      "2023-02-07 19:17:33 INFO:\t[hmmsearch]\t44 of 44 task(s) completed\n",
      "2023-02-07 19:17:35 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-02-07 19:17:37 INFO:\tResults:\tC:83.2%[S:82.0%,D:1.2%],F:9.8%,M:7.0%,n:255\t   \n",
      "\n",
      "2023-02-07 19:17:50 INFO:\teukaryota_odb10 selected\n",
      "\n",
      "2023-02-07 19:17:50 INFO:\t***** Searching tree for chosen lineage to find best taxonomic match *****\n",
      "\n",
      "2023-02-07 19:17:51 INFO:\tExtract markers...\n",
      "2023-02-07 19:17:51 INFO:\tPlace the markers on the reference tree...\n",
      "2023-02-07 19:17:51 INFO:\tRunning 1 job(s) on sepp, starting at 02/07/2023 19:17:51\n",
      "2023-02-07 19:18:54 INFO:\t[sepp]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:18:56 INFO:\tLineage metazoa is selected, supported by 48 markers out of 89\n",
      "2023-02-07 19:18:56 WARNING:\tThe file or folder /master/nplatt/sm_nanopore/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "2023-02-07 19:18:56 INFO:\tRunning BUSCO using lineage dataset metazoa_odb10 (eukaryota, 2021-02-17)\n",
      "2023-02-07 19:18:56 INFO:\tRunning 1 job(s) on bbtools, starting at 02/07/2023 19:18:56\n",
      "2023-02-07 19:19:03 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:19:03 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 19:19:03\n",
      "2023-02-07 19:25:48 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:26:03 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 19:26:03 INFO:\tRunning 954 job(s) on hmmsearch, starting at 02/07/2023 19:26:03\n",
      "2023-02-07 19:26:09 INFO:\t[hmmsearch]\t96 of 954 task(s) completed\n",
      "2023-02-07 19:26:09 INFO:\t[hmmsearch]\t191 of 954 task(s) completed\n",
      "2023-02-07 19:26:09 INFO:\t[hmmsearch]\t287 of 954 task(s) completed\n",
      "2023-02-07 19:26:10 INFO:\t[hmmsearch]\t478 of 954 task(s) completed\n",
      "2023-02-07 19:26:10 INFO:\t[hmmsearch]\t573 of 954 task(s) completed\n",
      "2023-02-07 19:26:10 INFO:\t[hmmsearch]\t668 of 954 task(s) completed\n",
      "2023-02-07 19:26:11 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-02-07 19:26:11 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-02-07 19:26:11 INFO:\t[hmmsearch]\t859 of 954 task(s) completed\n",
      "2023-02-07 19:26:12 INFO:\t[hmmsearch]\t954 of 954 task(s) completed\n",
      "2023-02-07 19:26:15 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-02-07 19:26:31 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 19:26:31\n",
      "2023-02-07 19:38:05 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:38:10 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 19:38:10 INFO:\tRunning 335 job(s) on hmmsearch, starting at 02/07/2023 19:38:10\n",
      "2023-02-07 19:38:16 INFO:\t[hmmsearch]\t34 of 335 task(s) completed\n",
      "2023-02-07 19:38:16 INFO:\t[hmmsearch]\t67 of 335 task(s) completed\n",
      "2023-02-07 19:38:16 INFO:\t[hmmsearch]\t101 of 335 task(s) completed\n",
      "2023-02-07 19:38:17 INFO:\t[hmmsearch]\t134 of 335 task(s) completed\n",
      "2023-02-07 19:38:17 INFO:\t[hmmsearch]\t168 of 335 task(s) completed\n",
      "2023-02-07 19:38:17 INFO:\t[hmmsearch]\t201 of 335 task(s) completed\n",
      "2023-02-07 19:38:17 INFO:\t[hmmsearch]\t235 of 335 task(s) completed\n",
      "2023-02-07 19:38:18 INFO:\t[hmmsearch]\t268 of 335 task(s) completed\n",
      "2023-02-07 19:38:18 INFO:\t[hmmsearch]\t302 of 335 task(s) completed\n",
      "2023-02-07 19:38:18 INFO:\t[hmmsearch]\t335 of 335 task(s) completed\n",
      "2023-02-07 19:38:27 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-02-07 19:38:32 INFO:\tResults:\tC:66.6%[S:65.3%,D:1.3%],F:7.4%,M:26.0%,n:954\t   \n",
      "\n",
      "2023-02-07 19:38:51 INFO:\t\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from generic domain eukaryota_odb10      |\n",
      "\t--------------------------------------------------\n",
      "\t|C:83.2%[S:82.0%,D:1.2%],F:9.8%,M:7.0%,n:255      |\n",
      "\t|212\tComplete BUSCOs (C)                       |\n",
      "\t|209\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|3\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|25\tFragmented BUSCOs (F)                     |\n",
      "\t|18\tMissing BUSCOs (M)                        |\n",
      "\t|255\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from dataset metazoa_odb10               |\n",
      "\t--------------------------------------------------\n",
      "\t|C:66.6%[S:65.3%,D:1.3%],F:7.4%,M:26.0%,n:954     |\n",
      "\t|635\tComplete BUSCOs (C)                       |\n",
      "\t|623\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|12\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|71\tFragmented BUSCOs (F)                     |\n",
      "\t|248\tMissing BUSCOs (M)                        |\n",
      "\t|954\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "2023-02-07 19:38:51 INFO:\tBUSCO analysis done with WARNING(s). Total running time: 2323 seconds\n",
      "\n",
      "***** Summary of warnings: *****\n",
      "2023-02-07 19:18:56 WARNING:busco.BuscoDownloadManager\tThe file or folder /master/nplatt/sm_nanopore/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "\n",
      "2023-02-07 19:38:51 INFO:\tResults written in /master/nplatt/sm_nanopore/results/busco/smbre\n",
      "2023-02-07 19:38:51 INFO:\tFor assistance with interpreting the results, please consult the userguide: https://busco.ezlab.org/busco_userguide.html\n",
      "\n",
      "2023-02-07 19:38:51 INFO:\tVisit this page https://gitlab.com/ezlab/busco#how-to-cite-busco to see how to cite BUSCO\n",
      "2023-02-07 19:38:54 INFO:\t***** Start a BUSCO v5.4.3 analysis, current time: 02/07/2023 19:38:54 *****\n",
      "2023-02-07 19:38:54 INFO:\tConfiguring BUSCO with local environment\n",
      "2023-02-07 19:38:54 INFO:\tMode is genome\n",
      "2023-02-07 19:38:54 INFO:\tDownloading information on latest versions of BUSCO data...\n",
      "2023-02-07 19:38:57 INFO:\tInput file is /master/nplatt/sm_nanopore/results/polish_assembly/smeg.assembly_rnd3.fasta\n",
      "2023-02-07 19:38:57 INFO:\tNo lineage specified. Running lineage auto selector.\n",
      "\n",
      "2023-02-07 19:38:57 INFO:\t***** Starting Auto Select Lineage *****\n",
      "\tThis process runs BUSCO on the generic lineage datasets for the domains archaea, bacteria and eukaryota. Once the optimal domain is selected, BUSCO automatically attempts to find the most appropriate BUSCO dataset to use based on phylogenetic placement.\n",
      "\t--auto-lineage-euk and --auto-lineage-prok are also available if you know your input assembly is, or is not, an eukaryote. See the user guide for more information.\n",
      "\tA reminder: Busco evaluations are valid when an appropriate dataset is used, i.e., the dataset belongs to the lineage of the species to test. Because of overlapping markers/spurious matches among domains, busco matches in another domain do not necessarily mean that your genome/proteome contains sequences from this domain. However, a high busco score in multiple domains might help you identify possible contaminations.\n",
      "2023-02-07 19:38:59 INFO:\tRunning BUSCO using lineage dataset eukaryota_odb10 (eukaryota, 2020-09-10)\n",
      "2023-02-07 19:38:59 INFO:\tRunning 1 job(s) on bbtools, starting at 02/07/2023 19:38:59\n",
      "2023-02-07 19:39:04 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:39:04 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 19:39:04\n",
      "2023-02-07 19:45:51 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:46:09 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 19:46:09 INFO:\tRunning 255 job(s) on hmmsearch, starting at 02/07/2023 19:46:09\n",
      "2023-02-07 19:46:14 INFO:\t[hmmsearch]\t26 of 255 task(s) completed\n",
      "2023-02-07 19:46:14 INFO:\t[hmmsearch]\t51 of 255 task(s) completed\n",
      "2023-02-07 19:46:14 INFO:\t[hmmsearch]\t77 of 255 task(s) completed\n",
      "2023-02-07 19:46:15 INFO:\t[hmmsearch]\t102 of 255 task(s) completed\n",
      "2023-02-07 19:46:15 INFO:\t[hmmsearch]\t128 of 255 task(s) completed\n",
      "2023-02-07 19:46:15 INFO:\t[hmmsearch]\t153 of 255 task(s) completed\n",
      "2023-02-07 19:46:15 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-02-07 19:46:15 INFO:\t[hmmsearch]\t204 of 255 task(s) completed\n",
      "2023-02-07 19:46:15 INFO:\t[hmmsearch]\t230 of 255 task(s) completed\n",
      "2023-02-07 19:46:16 INFO:\t[hmmsearch]\t255 of 255 task(s) completed\n",
      "2023-02-07 19:46:16 INFO:\tResults:\tC:80.8%[S:79.2%,D:1.6%],F:10.2%,M:9.0%,n:255\t   \n",
      "\n",
      "2023-02-07 19:46:16 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-02-07 19:46:21 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 19:46:21\n",
      "2023-02-07 19:57:13 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:57:16 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 19:57:16 INFO:\tRunning 49 job(s) on hmmsearch, starting at 02/07/2023 19:57:16\n",
      "2023-02-07 19:57:21 INFO:\t[hmmsearch]\t5 of 49 task(s) completed\n",
      "2023-02-07 19:57:21 INFO:\t[hmmsearch]\t10 of 49 task(s) completed\n",
      "2023-02-07 19:57:22 INFO:\t[hmmsearch]\t15 of 49 task(s) completed\n",
      "2023-02-07 19:57:22 INFO:\t[hmmsearch]\t20 of 49 task(s) completed\n",
      "2023-02-07 19:57:22 INFO:\t[hmmsearch]\t25 of 49 task(s) completed\n",
      "2023-02-07 19:57:22 INFO:\t[hmmsearch]\t30 of 49 task(s) completed\n",
      "2023-02-07 19:57:22 INFO:\t[hmmsearch]\t35 of 49 task(s) completed\n",
      "2023-02-07 19:57:22 INFO:\t[hmmsearch]\t40 of 49 task(s) completed\n",
      "2023-02-07 19:57:22 INFO:\t[hmmsearch]\t45 of 49 task(s) completed\n",
      "2023-02-07 19:57:22 INFO:\t[hmmsearch]\t49 of 49 task(s) completed\n",
      "2023-02-07 19:57:24 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-02-07 19:57:26 INFO:\tResults:\tC:82.0%[S:80.4%,D:1.6%],F:10.2%,M:7.8%,n:255\t   \n",
      "\n",
      "2023-02-07 19:57:41 INFO:\teukaryota_odb10 selected\n",
      "\n",
      "2023-02-07 19:57:41 INFO:\t***** Searching tree for chosen lineage to find best taxonomic match *****\n",
      "\n",
      "2023-02-07 19:57:41 INFO:\tExtract markers...\n",
      "2023-02-07 19:57:41 INFO:\tPlace the markers on the reference tree...\n",
      "2023-02-07 19:57:41 INFO:\tRunning 1 job(s) on sepp, starting at 02/07/2023 19:57:41\n",
      "2023-02-07 19:58:43 INFO:\t[sepp]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:58:45 INFO:\tLineage metazoa is selected, supported by 43 markers out of 85\n",
      "2023-02-07 19:58:45 WARNING:\tThe file or folder /master/nplatt/sm_nanopore/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "2023-02-07 19:58:46 INFO:\tRunning BUSCO using lineage dataset metazoa_odb10 (eukaryota, 2021-02-17)\n",
      "2023-02-07 19:58:46 INFO:\tRunning 1 job(s) on bbtools, starting at 02/07/2023 19:58:46\n",
      "2023-02-07 19:58:52 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-02-07 19:58:52 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 19:58:52\n",
      "2023-02-07 20:05:01 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 20:05:22 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 20:05:22 INFO:\tRunning 954 job(s) on hmmsearch, starting at 02/07/2023 20:05:22\n",
      "2023-02-07 20:05:28 INFO:\t[hmmsearch]\t96 of 954 task(s) completed\n",
      "2023-02-07 20:05:28 INFO:\t[hmmsearch]\t287 of 954 task(s) completed\n",
      "2023-02-07 20:05:29 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-02-07 20:05:29 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-02-07 20:05:29 INFO:\t[hmmsearch]\t478 of 954 task(s) completed\n",
      "2023-02-07 20:05:29 INFO:\t[hmmsearch]\t573 of 954 task(s) completed\n",
      "2023-02-07 20:05:29 INFO:\t[hmmsearch]\t668 of 954 task(s) completed\n",
      "2023-02-07 20:05:30 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-02-07 20:05:30 INFO:\t[hmmsearch]\t859 of 954 task(s) completed\n",
      "2023-02-07 20:05:31 INFO:\t[hmmsearch]\t954 of 954 task(s) completed\n",
      "2023-02-07 20:05:35 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-02-07 20:05:50 INFO:\tRunning 1 job(s) on metaeuk, starting at 02/07/2023 20:05:50\n",
      "2023-02-07 20:17:10 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-02-07 20:17:22 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-02-07 20:17:22 INFO:\tRunning 331 job(s) on hmmsearch, starting at 02/07/2023 20:17:22\n",
      "2023-02-07 20:17:28 INFO:\t[hmmsearch]\t34 of 331 task(s) completed\n",
      "2023-02-07 20:17:28 INFO:\t[hmmsearch]\t67 of 331 task(s) completed\n",
      "2023-02-07 20:17:29 INFO:\t[hmmsearch]\t100 of 331 task(s) completed\n",
      "2023-02-07 20:17:29 INFO:\t[hmmsearch]\t133 of 331 task(s) completed\n",
      "2023-02-07 20:17:29 INFO:\t[hmmsearch]\t166 of 331 task(s) completed\n",
      "2023-02-07 20:17:30 INFO:\t[hmmsearch]\t199 of 331 task(s) completed\n",
      "2023-02-07 20:17:30 INFO:\t[hmmsearch]\t232 of 331 task(s) completed\n",
      "2023-02-07 20:17:30 INFO:\t[hmmsearch]\t265 of 331 task(s) completed\n",
      "2023-02-07 20:17:31 INFO:\t[hmmsearch]\t298 of 331 task(s) completed\n",
      "2023-02-07 20:17:31 INFO:\t[hmmsearch]\t331 of 331 task(s) completed\n",
      "2023-02-07 20:17:39 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-02-07 20:17:44 INFO:\tResults:\tC:67.4%[S:66.4%,D:1.0%],F:7.4%,M:25.2%,n:954\t   \n",
      "\n",
      "2023-02-07 20:18:05 INFO:\t\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from generic domain eukaryota_odb10      |\n",
      "\t--------------------------------------------------\n",
      "\t|C:82.0%[S:80.4%,D:1.6%],F:10.2%,M:7.8%,n:255     |\n",
      "\t|209\tComplete BUSCOs (C)                       |\n",
      "\t|205\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|4\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|26\tFragmented BUSCOs (F)                     |\n",
      "\t|20\tMissing BUSCOs (M)                        |\n",
      "\t|255\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from dataset metazoa_odb10               |\n",
      "\t--------------------------------------------------\n",
      "\t|C:67.4%[S:66.4%,D:1.0%],F:7.4%,M:25.2%,n:954     |\n",
      "\t|643\tComplete BUSCOs (C)                       |\n",
      "\t|633\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|10\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|71\tFragmented BUSCOs (F)                     |\n",
      "\t|240\tMissing BUSCOs (M)                        |\n",
      "\t|954\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "2023-02-07 20:18:05 INFO:\tBUSCO analysis done with WARNING(s). Total running time: 2347 seconds\n",
      "\n",
      "***** Summary of warnings: *****\n",
      "2023-02-07 19:58:45 WARNING:busco.BuscoDownloadManager\tThe file or folder /master/nplatt/sm_nanopore/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "\n",
      "2023-02-07 20:18:05 INFO:\tResults written in /master/nplatt/sm_nanopore/results/busco/smeg\n",
      "2023-02-07 20:18:05 INFO:\tFor assistance with interpreting the results, please consult the userguide: https://busco.ezlab.org/busco_userguide.html\n",
      "\n",
      "2023-02-07 20:18:05 INFO:\tVisit this page https://gitlab.com/ezlab/busco#how-to-cite-busco to see how to cite BUSCO\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    assembly = \"{}/results/polish_assembly/{}.assembly_rnd3.fasta\".format(proj_dir, sample)\n",
    "    !conda run -n busco --cwd . --live-stream busco -i {assembly} --auto-lineage-euk -c 48 -o {sample} -m genome "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9ffd5276-7d80-4904-8fcb-2e8d90c0ca33",
   "metadata": {},
   "source": [
    "## genome quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "faae6500-e42f-45ab-b73c-127fa4a4aecf",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    assembly = \"{}/polish_assembly/{}.assembly_rnd3.fasta\".format(proj_dir, sample)\n",
    "    ./JupiterPlot/jupiter t=48 name={sample} ng=85 ref={ref_fas} fa={assembly}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "818182ca-701c-4c0c-86be-8094904e6032",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## Grave Yard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "316d74b7-320a-4a5c-9201-bba9709eb023",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate sm_ont-process_reads\n",
    "\n",
    "cd /master/nplatt/sm_nanopore\n",
    "\n",
    "cd sm_nanopore/results/nanofilt\n",
    "3\n",
    " ~/anaconda3/envs/sm_ont-process_reads/bin/flye --nano-cor ../read_correction_canu/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz -g 385m -t 96  -i 3 -o ${SAMPLE} >${SAMPLE}.canu.log 2>&1 &\n",
    "\n",
    "conda activate repeatmasker\n",
    "RepeatMasker \\\n",
    "    -pa 12 \\\n",
    "    -lib ../repeatmodeler/Smansoni_v7-families.fa \\\n",
    "    -gff \\\n",
    "    -s \\\n",
    "    -dir $(pwd) \\\n",
    "    -small \\\n",
    "    -x \\\n",
    "    $REF\n",
    "\n",
    "\n",
    "\n",
    "SmOR_Female_6_02_22_vSM_V9_sorted.vcf\n",
    "schisto_pairs_04132022_vSM_V9_sorted.vcf\n",
    "SmEG_Female_5_25_22_vSM_V9_sorted.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500) && AF>0.75' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pzq-es_filtered.vcf\n",
    "\n",
    "bedtools intersect \\\n",
    "    -c \\\n",
    "    -b smle-pzq-es_filtered.vcf \\\n",
    "    -a ../lift/v9.bed \\\n",
    "    >smle-pzq-es_intersect.bed\n",
    "\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-200 || SVLEN>200)' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout\n",
    "\n",
    "#############################################################################################\n",
    "\n",
    "#genrate reads\n",
    "art_illumina \\\n",
    "    --len 150 \\\n",
    "    --fcov 100 \\\n",
    "    --errfree \\\n",
    "    -na \\\n",
    "    --in SM_V9.fa \\\n",
    "    --out sm_v9_illumina \\\n",
    "    >sm_v9_illumina.log 2>&1 &\n",
    "\n",
    "nanosim-h \\\n",
    "    --perfect \\\n",
    "    -o sm_v9_ont \\\n",
    "    -n 5000000 \\\n",
    "    SM_V9.fa \\\n",
    "    >sm_v9_ont.log 2>&1 &\n",
    "\n",
    "#convert to fasta\n",
    "sed -n '1~4s/^@/>/p;2~4p' sm_v9_illumina.fq > sm_v9_illumina.fa\n",
    "\n",
    "#make sure amount of data is similar\n",
    "echo sm_v9_ont.fa $(cat sm_v9_ont.fa | paste  - - | cut -f2 | wc -c) >>counts &\n",
    "echo sm_v9_illumina.fa $(cat sm_v9_illumina.fa | paste  - - | cut -f2 | wc -c) >>counts &\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#map reads (perfectly and unambiguosly) back to the reference\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "    bbmap.sh \\\n",
    "        ref=SM_V9.fa \\\n",
    "        perfectmode=t \\\n",
    "        threads=48 \\\n",
    "        ambiguous=toss \\\n",
    "        in=${SAMPLE}.fa \\\n",
    "        overwrite=t \\\n",
    "        out=${SAMPLE}.sam \\\n",
    "        mappedonly=t \\\n",
    "        >${SAMPLE}.bbmap.log 2>&1 &\n",
    "done\n",
    "\n",
    "\n",
    "#calculate coverage\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "\n",
    "    echo $SAMPLE\n",
    "    echo \"...sort\"\n",
    "    samtools view -b -F 4 ${SAMPLE}.sam | samtools sort >${SAMPLE}.bam\n",
    "    echo \"...idx\"\n",
    "    samtools index ${SAMPLE}.bam\n",
    "    echo \"...cov\"\n",
    "    mosdepth \\\n",
    "        --threads 4 \\\n",
    "        ${SAMPLE} \\\n",
    "        ${SAMPLE}.bam\n",
    "done\n",
    "\n",
    "#how much is inaccessible per read type\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "\n",
    "    MB=$(zcat ${SAMPLE}.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}')\n",
    "    echo -e $SAMPLE\"\\t\"$MB\n",
    "\n",
    "done\n",
    "\n",
    "art_illumina \\\n",
    "--paired \\\n",
    "--len 150 \\\n",
    "--fcov 10 \\\n",
    "--mflen 300 \\\n",
    "--sdev 80 \\\n",
    "-na \\\n",
    "--in SM_V9.fa \\\n",
    "--out sm_v9_wes\n",
    "\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_wes1.fq \\\n",
    "    in2=sm_v9_wes2.fq \\\n",
    "    samestrandpairs=t \\\n",
    "    killbadpairs=t \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_wes.sam \\\n",
    "    mappedonly=t \\\n",
    "    ignorebadquality=t\n",
    "\n",
    "samtools view -b -F 4 sm_v9_wes.sam | samtools sort >sm_v9_wes.bam\n",
    "samtools index sm_v9_wes.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_wes \\\n",
    "    sm_v9_wes.bam\n",
    "\n",
    "\n",
    "\n",
    "zcat sm_v9_wes.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_wes.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "sed -i 's/SM_V9_1/HE601624.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_2/HE601625.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_3/HE601626.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_4/HE601627.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_5/HE601628.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_6/HE601629.3/' sm_v9.gff3\n",
    "\n",
    "\n",
    "\n",
    "bedtools intersect \\\n",
    "    -wb \\\n",
    "    -a smle-pzq-es_filtered.vcf \\\n",
    "    -b sm_v9.gff3\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500) && AF>0.75' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pzq-es_filtered.vcf\n",
    "\n",
    "bedtools intersect \\\n",
    "    -c \\\n",
    "    -b smle-pzq-es_filtered.vcf \\\n",
    "    -a ../lift/v9.bed\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "cp ../cutesv/sm_v9.gff3 .\n",
    "awk '{if ($3==\"exon\") print $0}' sm_v9.gff3 >sm_v9_exons.gff3\n",
    "cut -f1 -d\";\" sm_v9_exons.gff3 | cut -f1,4,5,9 | sed 's/ID=exon://' >sm_v9_exons.bed\n",
    "bedtools intersect -a sm_v9_exons.bed  -b qtls_v9.bed >qtl_exons.bed\n",
    "\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smeg_filtered.vcf  >smeg_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smeg_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smeg_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smeg_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smeg_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smor_filtered.vcf  >smor_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smor_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smor_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smor_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smor_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smle-pzq-es_filtered.vcf  >smle-pzq-es_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smle-pzq-es_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smle-pzq-es_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smle-pzq-es_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smle-pzq-es_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "nanosim-h --perfect -o sm_v9_ont_reads -n 500000 SM_V9.fa\n",
    "\n",
    "\n",
    "    minimap2 \\\n",
    "        -a \\\n",
    "        -o sm_v9_ont.sam \\\n",
    "        -t 48 \\\n",
    "        -x map-ont \\\n",
    "        SM_V9.fa \\\n",
    "        sm_v9_ont_reads.fa\n",
    "\n",
    "\n",
    "\n",
    "samtools view -b -F 4 sm_v9_ont.sam | samtools sort >sm_v9_ont.bam\n",
    "\n",
    "samtools index sm_v9_ont.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_ont \\\n",
    "    sm_v9_ont.bam\n",
    "\n",
    "zcat sm_v9_ont.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_ont.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_ont_reads.fa \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_ont_bbmap.sam \\\n",
    "    mappedonly=t \\\n",
    "    ignorebadquality=t\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    perfectmode=t \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_ont_reads.fa \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_ont_bbmap.sam \\\n",
    "    mappedonly=t\n",
    "\n",
    "\n",
    "samtools view -b -F 4 sm_v9_ont_bbmap.sam | samtools sort >sm_v9_ont_bbmap.bam\n",
    "samtools index sm_v9_ont_bbmap.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_ont_bbmap \\\n",
    "    sm_v9_ont_bbmap.bam\n",
    "\n",
    "zcat sm_v9_ont_bbmap.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_ont_bbmap.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "RepeatMasker -qq -pa 48 -nolow -norna -dir . -gff -lib sm_v7_tes.fas SM_V9.fa\n",
    "\n",
    "SmEG_Female_5_25_22\n",
    "schisto_pairs_04132022  ../cutesv/SmOR_Female_6_02_22_vSM_V9_sorted.vcf\n",
    "\n",
    "for VCF in $(ls ../cutesv/*_vSM_V9_sorted.vcf); do \n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pze-es_bigsvs.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/SmEG_Female_5_25_22_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smeg_bigsvs.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/SmOR_Female_6_02_22_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smor_bigsvs.vcf\n",
    "\n",
    "   396,000,000\n",
    "39,777,694,771\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "sm_ont-process_reads",
   "language": "python",
   "name": "sm_ont-process_reads"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
