{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2e110c05-d327-401b-93fc-081d6102bab1",
   "metadata": {},
   "source": [
    "notebook is sm_ont-process_reads\n",
    "\n",
    "\n",
    "envs used = nanofilt, n50, minimap2, samtools, mosdepth, samtools, mosdepth, cutesv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9cd4c83d-03e3-44f3-8e5a-5e64db850897",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from pathlib import Path\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3bb0e454-1134-4194-8c67-e685847b6669",
   "metadata": {},
   "outputs": [],
   "source": [
    "proj_dir = \"/master/nplatt/sm_nanopore\"\n",
    "ref_fas = \"{}/data/genome/GCA_000237925.5.fa\".format(proj_dir)\n",
    "Path(\"{}/results\".format(proj_dir)).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "625a0a9e-c38c-4c9a-a768-c3a22131bfb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "samples = [ \"smor\", \n",
    "            \"smle_pzq_es\",\n",
    "            \"smle_pzq_er\",\n",
    "            \"smbre\", \n",
    "            \"smeg\"        ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d62d2d02-6e9f-4c61-8028-73b97326de49",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"{}/results\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4fef4238-0533-44e4-a139-b08d6950cf3c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Filter and map raw reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13dd5d1e-db7d-4a53-a6bb-8bc790f41f71",
   "metadata": {},
   "outputs": [],
   "source": [
    "#all raw reads are provided in results/raw_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4446e16f-dd96-4417-9da8-deebe8d38001",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/nanofilt\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/nanofilt\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "c2485bf9-bb27-4391-a8ed-a0dbb15cf362",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    raw_fq = \"{}/results/raw_reads/{}.raw_pass.fastq\".format(proj_dir, sample)\n",
    "\n",
    "    !~/anaconda3/envs/sm_ont-process_reads/bin/NanoFilt --length 500 --quality 7 --headcrop 25 --tailcrop 25 {raw_fq} >{sample}.filtered.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e6cbcdc-cfe9-4087-96c3-0e0269379dcd",
   "metadata": {},
   "outputs": [],
   "source": [
    "!conda run -n n50 --cwd . --live-stream n50 --format csv $(ls *.filtered.fastq) >n50_filtered_reads.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "ab2652e7-383e-4b9a-9c95-4f9da176d9c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#path</th>\n",
       "      <th>seqs</th>\n",
       "      <th>size</th>\n",
       "      <th>N50</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>N75</th>\n",
       "      <th>N90</th>\n",
       "      <th>auN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>smbre.filtered.fastq</td>\n",
       "      <td>2156891</td>\n",
       "      <td>13733527914</td>\n",
       "      <td>16145</td>\n",
       "      <td>500</td>\n",
       "      <td>193069</td>\n",
       "      <td>7657</td>\n",
       "      <td>2728</td>\n",
       "      <td>357349.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smeg.filtered.fastq</td>\n",
       "      <td>2096508</td>\n",
       "      <td>12631573135</td>\n",
       "      <td>12491</td>\n",
       "      <td>500</td>\n",
       "      <td>236942</td>\n",
       "      <td>5659</td>\n",
       "      <td>2587</td>\n",
       "      <td>399825.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smle_pzq_er.filtered.fastq</td>\n",
       "      <td>1330051</td>\n",
       "      <td>7000372416</td>\n",
       "      <td>12071</td>\n",
       "      <td>500</td>\n",
       "      <td>447354</td>\n",
       "      <td>4953</td>\n",
       "      <td>2105</td>\n",
       "      <td>227185.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smor.filtered.fastq</td>\n",
       "      <td>3413369</td>\n",
       "      <td>14119478724</td>\n",
       "      <td>8668</td>\n",
       "      <td>500</td>\n",
       "      <td>168273</td>\n",
       "      <td>3430</td>\n",
       "      <td>1618</td>\n",
       "      <td>598308.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smle_pzq_es.filtered.fastq</td>\n",
       "      <td>3662716</td>\n",
       "      <td>23840253863</td>\n",
       "      <td>13824</td>\n",
       "      <td>500</td>\n",
       "      <td>202196</td>\n",
       "      <td>5893</td>\n",
       "      <td>2659</td>\n",
       "      <td>624239.05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        #path     seqs         size    N50  min     max   N75  \\\n",
       "0        smbre.filtered.fastq  2156891  13733527914  16145  500  193069  7657   \n",
       "1         smeg.filtered.fastq  2096508  12631573135  12491  500  236942  5659   \n",
       "2  smle_pzq_er.filtered.fastq  1330051   7000372416  12071  500  447354  4953   \n",
       "3         smor.filtered.fastq  3413369  14119478724   8668  500  168273  3430   \n",
       "4  smle_pzq_es.filtered.fastq  3662716  23840253863  13824  500  202196  5893   \n",
       "\n",
       "    N90        auN  \n",
       "0  2728  357349.74  \n",
       "1  2587  399825.88  \n",
       "2  2105  227185.13  \n",
       "3  1618  598308.11  \n",
       "4  2659  624239.05  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n50_df = pd.read_csv(\"n50_filtered_reads.csv\", sep=\",\")\n",
    "n50_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c55d5b71-923a-41e2-966b-b082b11b9f05",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/minimap2\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/minimap2\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "6db32522-3695-4aaa-9269-e8fb6b5be5c4",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smle_pzq_es ##################\n",
      "[M::mm_idx_gen::11.489*1.68] collected minimizers\n",
      "[M::mm_idx_gen::13.279*2.42] sorted minimizers\n",
      "[M::main::13.279*2.42] loaded/built the index for 12 target sequence(s)\n",
      "[M::mm_mapopt_update::14.118*2.34] mid_occ = 416\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 12\n",
      "[M::mm_idx_stat::14.681*2.28] distinct minimizers: 30119087 (65.35% are singletons); average occurrences: 2.439; average spacing: 5.327; total length: 391422238\n",
      "[M::worker_pipeline::214.298*29.98] mapped 42338 sequences\n",
      "[M::worker_pipeline::390.138*31.90] mapped 40808 sequences\n",
      "[M::worker_pipeline::588.947*32.25] mapped 43306 sequences\n",
      "[M::worker_pipeline::830.022*32.30] mapped 45455 sequences\n",
      "[M::worker_pipeline::1051.682*32.12] mapped 44703 sequences\n",
      "[M::worker_pipeline::1289.362*31.59] mapped 45693 sequences\n",
      "[M::worker_pipeline::1470.012*32.22] mapped 47806 sequences\n",
      "[M::worker_pipeline::1696.452*32.12] mapped 45932 sequences\n",
      "[M::worker_pipeline::1916.264*31.95] mapped 47284 sequences\n",
      "[M::worker_pipeline::2103.592*32.13] mapped 48028 sequences\n",
      "[M::worker_pipeline::2304.990*32.24] mapped 49325 sequences\n",
      "[M::worker_pipeline::2494.646*32.47] mapped 50218 sequences\n",
      "[M::worker_pipeline::2717.671*32.28] mapped 49571 sequences\n",
      "[M::worker_pipeline::2898.281*32.52] mapped 49790 sequences\n",
      "[M::worker_pipeline::3096.096*32.53] mapped 51413 sequences\n",
      "[M::worker_pipeline::3288.826*32.55] mapped 53354 sequences\n",
      "[M::worker_pipeline::3537.461*32.11] mapped 53951 sequences\n",
      "[M::worker_pipeline::3732.637*32.17] mapped 53719 sequences\n",
      "[M::worker_pipeline::3940.025*32.09] mapped 56807 sequences\n",
      "[M::worker_pipeline::4112.592*32.21] mapped 55685 sequences\n",
      "[M::worker_pipeline::4310.755*32.19] mapped 59269 sequences\n",
      "[M::worker_pipeline::4539.062*32.29] mapped 48605 sequences\n",
      "[M::worker_pipeline::4782.077*32.26] mapped 45151 sequences\n",
      "[M::worker_pipeline::4994.360*32.35] mapped 43831 sequences\n",
      "[M::worker_pipeline::5221.479*32.45] mapped 44343 sequences\n",
      "[M::worker_pipeline::5444.817*32.50] mapped 44640 sequences\n",
      "[M::worker_pipeline::5684.431*32.44] mapped 45538 sequences\n",
      "[M::worker_pipeline::5866.860*32.59] mapped 134943 sequences\n",
      "[M::worker_pipeline::6041.373*32.73] mapped 133880 sequences\n",
      "[M::worker_pipeline::6222.160*32.86] mapped 133272 sequences\n",
      "[M::worker_pipeline::6419.560*32.90] mapped 132816 sequences\n",
      "[M::worker_pipeline::6601.796*33.00] mapped 131336 sequences\n",
      "[M::worker_pipeline::6793.441*33.04] mapped 130610 sequences\n",
      "[M::worker_pipeline::6972.643*33.17] mapped 127668 sequences\n",
      "[M::worker_pipeline::7149.058*33.29] mapped 127555 sequences\n",
      "[M::worker_pipeline::7314.339*33.38] mapped 141249 sequences\n",
      "[M::worker_pipeline::7499.906*33.46] mapped 118698 sequences\n",
      "[M::worker_pipeline::7698.053*33.54] mapped 95874 sequences\n",
      "[M::worker_pipeline::7900.189*33.61] mapped 95068 sequences\n",
      "[M::worker_pipeline::8098.936*33.70] mapped 95263 sequences\n",
      "[M::worker_pipeline::8289.017*33.81] mapped 95789 sequences\n",
      "[M::worker_pipeline::8475.551*33.92] mapped 97350 sequences\n",
      "[M::worker_pipeline::8688.873*33.92] mapped 97320 sequences\n",
      "[M::worker_pipeline::8884.251*34.00] mapped 99976 sequences\n",
      "[M::worker_pipeline::9054.041*34.08] mapped 100132 sequences\n",
      "[M::worker_pipeline::9228.106*34.16] mapped 100400 sequences\n",
      "[M::worker_pipeline::9421.437*34.20] mapped 99759 sequences\n",
      "[M::worker_pipeline::9539.774*34.24] mapped 67195 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: /master/nplatt/anaconda3/envs/sm_ont-process_reads/bin/minimap2 -a -o /master/nplatt/sm_nanopore/results/minimap2/smle_pzq_es.sam -t 48 -x map-ont /master/nplatt/sm_nanopore/data/genome/GCA_000237925.5.fa /master/nplatt/sm_nanopore/results/nanofilt/smle_pzq_es.filtered.fastq\n",
      "[M::main] Real time: 9540.184 sec; CPU: 326642.700 sec; Peak RSS: 26.741 GB\n"
     ]
    }
   ],
   "source": [
    "# for sample in samples:\n",
    "for sample in [\"smle_pzq_es\"]:\n",
    "    sam      = \"{}/results/minimap2/{}.sam\".format(proj_dir, sample)\n",
    "    query_fq = \"{}/results/nanofilt/{}.filtered.fastq\".format(proj_dir, sample)\n",
    "\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "    !~/anaconda3/envs/sm_ont-process_reads/bin/minimap2 -a -o {sam} -t 48 -x map-ont {ref_fas} {query_fq}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "5a175957-2dae-4bac-a37e-aa063e0d1a48",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smle_pzq_es ##################\n",
      "...sort...\n",
      "[bam_sort_core] merging from 60 files and 1 in-memory blocks...\n",
      "...index...\n",
      "...coverage...\n"
     ]
    }
   ],
   "source": [
    "# for sample in samples:\n",
    "for sample in [\"smle_pzq_es\"]:\n",
    "    sam      = \"{}/results/minimap2/{}.sam\".format(proj_dir, sample)\n",
    "    query_fq = \"{}/results/nanofilt/{}.filtered.fastq\".format(proj_dir, sample)\n",
    "    \n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    #sort bam\n",
    "    print(\"...sort...\")\n",
    "    ! ~/anaconda3/envs/sm_ont-process_reads/bin/samtools sort -O bam -o {sample}.bam {sample}.sam\n",
    "    \n",
    "    #index bam\n",
    "    print(\"...index...\")\n",
    "    ! ~/anaconda3/envs/sm_ont-process_reads/bin/samtools index {sample}.bam\n",
    "    \n",
    "    #calculate coverage\n",
    "    print(\"...coverage...\")\n",
    "    ! ~/anaconda3/envs/sm_ont-process_reads/bin/mosdepth --threads 4 {sample} {sample}.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "23b040b1-f1f2-4873-a2d7-6772dfc168b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>length</th>\n",
       "      <th>bases</th>\n",
       "      <th>mean</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>smbre</td>\n",
       "      <td>391422238</td>\n",
       "      <td>12929849971</td>\n",
       "      <td>33.03</td>\n",
       "      <td>0</td>\n",
       "      <td>50586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smeg</td>\n",
       "      <td>391422238</td>\n",
       "      <td>8667326476</td>\n",
       "      <td>22.14</td>\n",
       "      <td>0</td>\n",
       "      <td>34407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smle_pzq_er</td>\n",
       "      <td>391422238</td>\n",
       "      <td>6624879196</td>\n",
       "      <td>16.93</td>\n",
       "      <td>0</td>\n",
       "      <td>10350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smle_pzq_es</td>\n",
       "      <td>391422238</td>\n",
       "      <td>21017040798</td>\n",
       "      <td>53.69</td>\n",
       "      <td>0</td>\n",
       "      <td>43602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smor</td>\n",
       "      <td>391422238</td>\n",
       "      <td>12178154968</td>\n",
       "      <td>31.11</td>\n",
       "      <td>0</td>\n",
       "      <td>80622</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample     length        bases   mean  min    max\n",
       "0        smbre  391422238  12929849971  33.03    0  50586\n",
       "1         smeg  391422238   8667326476  22.14    0  34407\n",
       "2  smle_pzq_er  391422238   6624879196  16.93    0  10350\n",
       "3  smle_pzq_es  391422238  21017040798  53.69    0  43602\n",
       "4         smor  391422238  12178154968  31.11    0  80622"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#calculate coverage for a table\n",
    "!echo \"sample,length,bases,mean,min,max\">covs.csv\n",
    "!grep \"total\" *mosdepth.summary.txt | sed 's/:/\\t/' | sed 's/\\t/,/g' | cut -f1,3-7 -d\",\" | sed 's/.mosdepth.summary.txt//' >>covs.csv\n",
    "\n",
    "cov_df=pd.read_csv(\"covs.csv\", sep=\",\")\n",
    "cov_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d4367a81-1675-42bf-a3e3-db05cac1f315",
   "metadata": {},
   "outputs": [],
   "source": [
    "#play around with igv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0694869-8ba5-4996-ae3d-e122538f69e1",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Identify structural variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "71d67d7b-66e0-434f-a420-f75d36a7870e",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/cutesv\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/cutesv\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8fd9c2fd-7b35-4641-b475-8e55df79f64d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smle_pzq_es ##################\n",
      "2023-01-18 11:31:03,178 [INFO] Running /master/nplatt/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 --genotype /master/nplatt/sm_nanopore/results/minimap2/smle_pzq_es.bam /master/nplatt/sm_nanopore/data/genome/GCA_000237925.5.fa smle_pzq_es.cutesv.vcf /master/nplatt/sm_nanopore/results/smle_pzq_es\n",
      "2023-01-18 11:31:10,070 [INFO] The total number of chromsomes: 12\n",
      "2023-01-18 11:31:38,245 [INFO] Finished OU426849.1:10000000-10680109.\n",
      "2023-01-18 11:33:46,325 [INFO] Finished HE601624.3:80000000-87984036.\n",
      "2023-01-18 11:34:13,522 [INFO] Finished HE601624.3:30000000-40000000.\n",
      "2023-01-18 11:34:14,461 [INFO] Finished HE601624.3:60000000-70000000.\n",
      "2023-01-18 11:34:14,861 [INFO] Finished HE601624.3:50000000-60000000.\n",
      "2023-01-18 11:34:15,292 [INFO] Finished HE601624.3:40000000-50000000.\n",
      "2023-01-18 11:34:15,396 [INFO] Finished HE601624.3:20000000-30000000.\n",
      "2023-01-18 11:34:16,824 [INFO] Finished HE601624.3:10000000-20000000.\n",
      "2023-01-18 11:34:17,049 [INFO] Finished HE601624.3:70000000-80000000.\n",
      "2023-01-18 11:34:17,800 [INFO] Finished OU426849.1:0-10000000.\n",
      "2023-01-18 11:34:19,942 [INFO] Finished HE601625.3:0-10000000.\n",
      "2023-01-18 11:34:34,885 [INFO] Finished HE601625.3:10000000-20000000.\n",
      "2023-01-18 11:35:10,155 [INFO] Finished HE601624.3:0-10000000.\n",
      "2023-01-18 11:36:26,000 [INFO] Finished OU426850.1:40000000-42949100.\n",
      "2023-01-18 11:39:30,490 [INFO] Finished HE601625.3:40000000-45716228.\n",
      "2023-01-18 11:41:25,195 [INFO] Finished HE601625.3:20000000-30000000.\n",
      "2023-01-18 11:41:56,222 [INFO] Finished OU426850.1:0-10000000.\n",
      "2023-01-18 11:42:09,883 [INFO] Finished OU426850.1:20000000-30000000.\n",
      "2023-01-18 11:42:20,785 [INFO] Finished OU426850.1:30000000-40000000.\n",
      "2023-01-18 11:42:31,292 [INFO] Finished OU426850.1:10000000-20000000.\n",
      "2023-01-18 11:42:41,655 [INFO] Finished HE601626.3:20000000-30000000.\n",
      "2023-01-18 11:44:43,023 [INFO] Finished HE601626.3:0-10000000.\n",
      "2023-01-18 11:44:45,384 [INFO] Finished HE601626.3:30000000-40000000.\n",
      "2023-01-18 11:44:49,712 [INFO] Finished HE601626.3:10000000-20000000.\n",
      "2023-01-18 11:46:12,421 [INFO] Finished HE601626.3:40000000-49793703.\n",
      "2023-01-18 11:47:34,679 [INFO] Finished HE601627.3:40000000-46471565.\n",
      "2023-01-18 11:47:50,871 [INFO] Finished HE601625.3:30000000-40000000.\n",
      "2023-01-18 11:48:14,295 [INFO] Finished HE601628.3:20000000-24128855.\n",
      "2023-01-18 11:48:40,115 [INFO] Finished HE601627.3:0-10000000.\n",
      "2023-01-18 11:49:23,561 [INFO] Finished HE601627.3:10000000-20000000.\n",
      "2023-01-18 11:49:32,025 [INFO] Finished HE601627.3:20000000-30000000.\n",
      "2023-01-18 11:50:03,023 [INFO] Finished HE601627.3:30000000-40000000.\n",
      "2023-01-18 11:50:11,537 [INFO] Finished HE601629.3:20000000-24696071.\n",
      "2023-01-18 11:52:31,369 [INFO] Finished HE601628.3:10000000-20000000.\n",
      "2023-01-18 11:52:33,504 [INFO] Finished HE601612.2:0-26917.\n",
      "2023-01-18 11:52:38,871 [INFO] Finished OU426848.1:30000000-33063208.\n",
      "2023-01-18 11:52:53,278 [INFO] Finished HE601628.3:0-10000000.\n",
      "2023-01-18 11:53:11,865 [INFO] Finished HE601629.3:10000000-20000000.\n",
      "2023-01-18 11:53:32,868 [INFO] Finished HE601629.3:0-10000000.\n",
      "2023-01-18 11:53:35,289 [INFO] Finished OU426848.1:0-10000000.\n",
      "2023-01-18 11:53:36,901 [INFO] Finished OU426848.1:10000000-20000000.\n",
      "2023-01-18 11:53:37,627 [INFO] Finished OU426848.1:20000000-30000000.\n",
      "2023-01-18 11:53:43,843 [INFO] Finished HE601630.3:0-10000000.\n",
      "2023-01-18 11:53:51,094 [INFO] Finished HE601630.3:10000000-19942475.\n",
      "2023-01-18 11:54:25,263 [INFO] Finished HE601631.3:0-5969971.\n",
      "2023-01-18 11:54:25,329 [INFO] Rebuilding signatures of structural variants.\n",
      "2023-01-18 11:56:15,987 [INFO] Clustering structural variants.\n",
      "2023-01-18 11:56:17,909 [INFO] Finished HE601612.2:DEL.\n",
      "2023-01-18 11:56:17,962 [INFO] Finished HE601631.3:DEL.\n",
      "2023-01-18 11:56:18,360 [INFO] Finished OU426849.1:DEL.\n",
      "2023-01-18 11:56:18,622 [INFO] Finished OU426848.1:DEL.\n",
      "2023-01-18 11:56:18,799 [INFO] Finished HE601630.3:DEL.\n",
      "2023-01-18 11:56:19,175 [INFO] Finished HE601629.3:DEL.\n",
      "2023-01-18 11:56:19,185 [INFO] Finished HE601628.3:DEL.\n",
      "2023-01-18 11:56:19,819 [INFO] Finished OU426850.1:DEL.\n",
      "2023-01-18 11:56:19,840 [INFO] Finished HE601612.2:INS.\n",
      "2023-01-18 11:56:20,084 [INFO] Finished HE601627.3:DEL.\n",
      "2023-01-18 11:56:20,261 [INFO] Finished HE601625.3:DEL.\n",
      "2023-01-18 11:56:20,445 [INFO] Finished HE601626.3:DEL.\n",
      "2023-01-18 11:56:21,791 [INFO] Finished HE601631.3:INS.\n",
      "2023-01-18 11:56:22,037 [INFO] Finished HE601624.3:DEL.\n",
      "2023-01-18 11:56:22,241 [INFO] Finished HE601628.3:INS.\n",
      "2023-01-18 11:56:22,295 [INFO] Finished HE601629.3:INS.\n",
      "2023-01-18 11:56:22,403 [INFO] Finished HE601625.3:INS.\n",
      "2023-01-18 11:56:22,448 [INFO] Finished HE601630.3:INS.\n",
      "2023-01-18 11:56:22,468 [INFO] Finished OU426849.1:INS.\n",
      "2023-01-18 11:56:22,555 [INFO] Finished OU426848.1:INS.\n",
      "2023-01-18 11:56:22,626 [INFO] Finished HE601627.3:INS.\n",
      "2023-01-18 11:56:22,841 [INFO] Finished HE601626.3:INS.\n",
      "2023-01-18 11:56:23,023 [INFO] Finished HE601612.2:INV.\n",
      "2023-01-18 11:56:23,719 [INFO] Finished HE601624.3:INS.\n",
      "2023-01-18 11:56:23,801 [INFO] Finished HE601631.3:INV.\n",
      "2023-01-18 11:56:24,047 [INFO] Finished OU426850.1:INS.\n",
      "2023-01-18 11:56:24,217 [INFO] Finished HE601630.3:INV.\n",
      "2023-01-18 11:56:24,223 [INFO] Finished HE601628.3:INV.\n",
      "2023-01-18 11:56:24,337 [INFO] Finished HE601629.3:INV.\n",
      "2023-01-18 11:56:24,422 [INFO] Finished OU426849.1:INV.\n",
      "2023-01-18 11:56:24,499 [INFO] Finished OU426848.1:INV.\n",
      "2023-01-18 11:56:24,849 [INFO] Finished HE601625.3:INV.\n",
      "2023-01-18 11:56:24,864 [INFO] Finished HE601627.3:INV.\n",
      "2023-01-18 11:56:25,029 [INFO] Finished HE601626.3:INV.\n",
      "2023-01-18 11:56:25,060 [INFO] Finished HE601612.2:DUP.\n",
      "2023-01-18 11:56:25,831 [INFO] Finished HE601624.3:INV.\n",
      "2023-01-18 11:56:26,023 [INFO] Finished HE601631.3:DUP.\n",
      "2023-01-18 11:56:26,031 [INFO] Finished OU426850.1:INV.\n",
      "2023-01-18 11:56:26,091 [INFO] Finished HE601612.2-HE601624.3:TRA/BND.\n",
      "2023-01-18 11:56:26,098 [INFO] Finished HE601612.2-HE601625.3:TRA/BND.\n",
      "2023-01-18 11:56:26,158 [INFO] Finished HE601612.2-HE601626.3:TRA/BND.\n",
      "2023-01-18 11:56:26,160 [INFO] Finished HE601628.3:DUP.\n",
      "2023-01-18 11:56:26,164 [INFO] Finished HE601612.2-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:26,224 [INFO] Finished HE601612.2-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:26,226 [INFO] Finished HE601612.2-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:26,230 [INFO] Finished HE601612.2-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:26,290 [INFO] Finished HE601612.2-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:26,291 [INFO] Finished HE601612.2-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:26,297 [INFO] Finished HE601612.2-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:26,335 [INFO] Finished HE601629.3:DUP.\n",
      "2023-01-18 11:56:26,356 [INFO] Finished HE601612.2-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:26,436 [INFO] Finished OU426849.1:DUP.\n",
      "2023-01-18 11:56:26,479 [INFO] Finished HE601630.3:DUP.\n",
      "2023-01-18 11:56:26,638 [INFO] Finished HE601625.3:DUP.\n",
      "2023-01-18 11:56:26,668 [INFO] Finished OU426848.1:DUP.\n",
      "2023-01-18 11:56:26,695 [INFO] Finished HE601627.3:DUP.\n",
      "2023-01-18 11:56:26,747 [INFO] Finished HE601624.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:26,818 [INFO] Finished HE601626.3:DUP.\n",
      "2023-01-18 11:56:26,852 [INFO] Finished HE601624.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:26,899 [INFO] Finished HE601624.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:26,960 [INFO] Finished HE601624.3-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:26,976 [INFO] Finished HE601624.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,004 [INFO] Finished HE601625.3-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:27,018 [INFO] Finished HE601624.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,073 [INFO] Finished HE601624.3-HE601625.3:TRA/BND.\n",
      "2023-01-18 11:56:27,094 [INFO] Finished HE601625.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,119 [INFO] Finished HE601624.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,123 [INFO] Finished HE601625.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,146 [INFO] Finished HE601625.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:27,152 [INFO] Finished HE601625.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,220 [INFO] Finished HE601624.3-HE601626.3:TRA/BND.\n",
      "2023-01-18 11:56:27,233 [INFO] Finished HE601625.3-HE601626.3:TRA/BND.\n",
      "2023-01-18 11:56:27,294 [INFO] Finished HE601625.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,314 [INFO] Finished HE601626.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,360 [INFO] Finished HE601626.3-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:27,385 [INFO] Finished HE601626.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:27,385 [INFO] Finished HE601625.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,399 [INFO] Finished HE601625.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,427 [INFO] Finished HE601626.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,431 [INFO] Finished HE601626.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,469 [INFO] Finished HE601626.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,489 [INFO] Finished HE601624.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,576 [INFO] Finished HE601627.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,582 [INFO] Finished HE601627.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,590 [INFO] Finished HE601627.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,658 [INFO] Finished HE601627.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:27,709 [INFO] Finished HE601626.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,712 [INFO] Finished HE601624.3:DUP.\n",
      "2023-01-18 11:56:27,727 [INFO] Finished HE601628.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,737 [INFO] Finished HE601628.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,744 [INFO] Finished HE601627.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,755 [INFO] Finished HE601628.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,766 [INFO] Finished HE601628.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,782 [INFO] Finished HE601627.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,792 [INFO] Finished HE601629.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,799 [INFO] Finished HE601628.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,806 [INFO] Finished HE601626.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,819 [INFO] Finished HE601627.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,832 [INFO] Finished HE601630.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,847 [INFO] Finished HE601630.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,856 [INFO] Finished HE601630.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,860 [INFO] Finished HE601629.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,864 [INFO] Finished HE601630.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,874 [INFO] Finished HE601631.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,896 [INFO] Finished HE601629.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,918 [INFO] Finished HE601631.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,931 [INFO] Finished HE601629.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,949 [INFO] Finished OU426848.1-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,949 [INFO] Finished HE601629.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,950 [INFO] Finished OU426849.1-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,977 [INFO] Finished HE601631.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:28,066 [INFO] Finished OU426848.1-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:28,132 [INFO] Finished OU426850.1:DUP.\n",
      "2023-01-18 11:56:28,176 [INFO] Finished HE601628.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:28,282 [INFO] Writing to your output file.\n",
      "2023-01-18 11:56:28,304 [INFO] Loading reference genome...\n",
      "2023-01-18 11:56:51,966 [INFO] Cleaning temporary files.\n",
      "2023-01-18 11:56:52,691 [INFO] Finished in 1549.51 seconds.\n"
     ]
    }
   ],
   "source": [
    "#sv call\n",
    "# for sample in samples:\n",
    "for sample in [\"smle_pzq_es\"]:\n",
    "\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    bam      = \"{}/results/minimap2/{}.bam\".format(proj_dir, sample)\n",
    "    query_fq = \"{}/results/nanofilt/{}.filtered.fastq.gz\".format(proj_dir, sample)\n",
    "    out_dir  = \"{}/{}\".format(os.getcwd(), sample)\n",
    "    \n",
    "    Path(\"{}\".format(out_dir)).mkdir(parents=True, exist_ok=True)\n",
    "    !~/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 --genotype {bam} {ref_fas} {sample}.cutesv.vcf {out_dir}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1e83da5-f68a-40fb-aa3a-29a1cf626880",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Read Correction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "172a44a7-2e93-45bf-9df2-0303ecfb389e",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/read_correction\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/read_correction\".format(proj_dir))\n",
    "\n",
    "for sample in samples:\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "    Path(\"{}/results/read_correction/{}\".format(proj_dir, sample)).mkdir(parents=True, exist_ok=True)\n",
    "    \n",
    "    os.chdir(\"{}/results/read_correction\".format(proj_dir, sample))\n",
    "    \n",
    "    !conda run -n ont_assembly --cwd . --live-stream canu -correct -genomesize=385m -useGrid=False -p {sample}  -nanopore {proj_dir}/results/raw_reads/{sample}.raw_pass.fastq\n",
    "    \n",
    "    os.chdir(\"{}/results/read_correction\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39d8bb33-5379-4a91-b4bc-6c0f510761db",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash\n",
    "\n",
    "for SAMPLE in 'smor' 'smle_pzq_es' 'smle_pzq_er' 'smbre' 'smeg'; do\n",
    "\n",
    "    if [ -d ${SAMPLE} ]; then\n",
    "        rm -rf ${SAMPLE}\n",
    "    fi\n",
    "    \n",
    "    mkdir ${SAMPLE}\n",
    "    cd ${SAMPLE}\n",
    "    \n",
    "    LOG=$(pwd)/${SAMPLE}.log \n",
    "    \n",
    "    QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 192 -N ${SAMPLE} -o ${LOG}\"\n",
    "    CMD=\"conda run -n ont_assembly --cwd . --live-stream canu -correct -genomesize=385m -useGrid=False -p ${SAMPLE}  -nanopore /master/nplatt/sm_nanopore/results/raw_reads/${SAMPLE}.raw_pass.fastq\"\n",
    "    \n",
    "    echo $CMD | $QSUB\n",
    "    cd ..\n",
    "done\n",
    "\n",
    "for SAMPLE in 'smor' 'smle_pzq_es' 'smbre'; do\n",
    "    rm -rf $SAMPLE*\n",
    "    QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 192 -N ${SAMPLE} -o ${SAMPLE}.log\"\n",
    "    CMD=\"conda run -n ont_assembly --cwd . --live-stream canu -correct -genomesize=385m -useGrid=False -p ${SAMPLE}/${SAMPLE}  -nanopore /master/nplatt/sm_nanopore/results/raw_reads/${SAMPLE}.raw_pass.fastq\"\n",
    "    \n",
    "    echo $CMD | $QSUB\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "218b0bb6-ecc0-4350-9c66-cbcb0666e51f",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Genome assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "2622026b-669a-41a4-b722-2ff5ccb79193",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/assembly\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/assembly\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13c30768-4f95-4904-a110-4d27b38ff59b",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "    !flye --nano-cor {proj_dir}/results/read_correction/{sample}.correctedReads.fasta.gz -g 385m -t 128 -i 3 -o {sample}_rnd0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2a56d32-e431-4195-a5f8-ea56d1accc5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "for SAMPLE in 'smor' 'smle_pzq_es' 'smle_pzq_er' 'smbre' 'smeg'; do\n",
    "\n",
    "    if [ -d ${SAMPLE} ]; then\n",
    "        rm -rf ${SAMPLE}\n",
    "    fi\n",
    "    \n",
    "    mkdir ${SAMPLE}\n",
    "    cd ${SAMPLE}\n",
    "    \n",
    "    LOG=$(pwd)/${SAMPLE}.log \n",
    "    READS=\"/master/nplatt/sm_nanopore/results/read_correction/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz\"\n",
    "    OUT=\"/master/nplatt/sm_nanopore/results/assembly/${SAMPLE}/${SAMPLE}_rnd0\"\n",
    "    \n",
    "    QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 128 -N ${SAMPLE} -o ${LOG}\"\n",
    "    CMD=\"conda run -n sm_ont-process_reads --cwd . --live-stream flye --nano-cor ${READS} -g 385m -t 128 -i 3 -o ${OUT}\"\n",
    "    \n",
    "    echo $CMD | $QSUB\n",
    "    cd ..\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "47970cec-d20a-4f7a-b46f-a6a210585b9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>total_length</th>\n",
       "      <th>fragments</th>\n",
       "      <th>fragment_n50</th>\n",
       "      <th>largest_fragment</th>\n",
       "      <th>scaffolds</th>\n",
       "      <th>mean_coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>smor</td>\n",
       "      <td>386656054</td>\n",
       "      <td>1674</td>\n",
       "      <td>2150179</td>\n",
       "      <td>8090473</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smle_pzq_es</td>\n",
       "      <td>403425661</td>\n",
       "      <td>1122</td>\n",
       "      <td>3481572</td>\n",
       "      <td>14813648</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smle_pzq_er</td>\n",
       "      <td>382213358</td>\n",
       "      <td>1968</td>\n",
       "      <td>1273691</td>\n",
       "      <td>5768858</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smbre</td>\n",
       "      <td>384951218</td>\n",
       "      <td>1020</td>\n",
       "      <td>2298594</td>\n",
       "      <td>14522024</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smeg</td>\n",
       "      <td>370379461</td>\n",
       "      <td>1963</td>\n",
       "      <td>871323</td>\n",
       "      <td>6130221</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample total_length fragments fragment_n50 largest_fragment scaffolds  \\\n",
       "0         smor    386656054      1674      2150179          8090473         0   \n",
       "1  smle_pzq_es    403425661      1122      3481572         14813648         0   \n",
       "2  smle_pzq_er    382213358      1968      1273691          5768858         0   \n",
       "3        smbre    384951218      1020      2298594         14522024         0   \n",
       "4         smeg    370379461      1963       871323          6130221         0   \n",
       "\n",
       "  mean_coverage  \n",
       "0            23  \n",
       "1            33  \n",
       "2            15  \n",
       "3            24  \n",
       "4            14  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#get df of assembly stats\n",
    "total_lengths=[]\n",
    "fragments=[]\n",
    "fragment_n50s=[]\n",
    "largest_frags=[]\n",
    "scaffolds=[]\n",
    "mean_coverages=[]\n",
    "    \n",
    "for sample in samples:\n",
    "    info = !tail -n 8 {sample}/{sample}.log | head -n 6\n",
    "    total_lengths.append(info[0].split(\"\\t\")[-1])\n",
    "    fragments.append(info[1].split(\"\\t\")[-1])\n",
    "    fragment_n50s.append(info[2].split(\"\\t\")[-1])\n",
    "    largest_frags.append(info[3].split(\"\\t\")[-1])\n",
    "    scaffolds.append(info[4].split(\"\\t\")[-1])\n",
    "    mean_coverages.append(info[5].split(\"\\t\")[-1])\n",
    "    \n",
    "assembly_df=pd.DataFrame(data = [samples, total_lengths, fragments, fragment_n50s, largest_frags, scaffolds, mean_coverages]).T\n",
    "assembly_df.columns=[\"sample\", \"total_length\", \"fragments\", \"fragment_n50\", \"largest_fragment\", \"scaffolds\", \"mean_coverage\"]\n",
    "assembly_df.to_csv(\"assembly_info.csv\", index=False)\n",
    "assembly_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "c87716bd-431b-4e06-ab21-e8afc04f2a24",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "smor\n",
      "smle_pzq_es\n",
      "smle_pzq_er\n",
      "smbre\n",
      "smeg\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    print(sample)\n",
    "    ! cp /master/nplatt/sm_nanopore/results/assembly/{sample}/{sample}_rnd0/assembly.fasta {sample}_rnd0.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12de9d93-c60c-41f4-8556-370f4e265f84",
   "metadata": {},
   "source": [
    "## polish assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "c1ea2ab8-cc08-4c9e-8819-9803b66f777c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/polish_assembly\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/polish_assembly\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "e71a2c33-555b-4c0a-aa42-7af40c7e2947",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "smor\n",
      "smle_pzq_es\n",
      "smle_pzq_er\n",
      "smbre\n",
      "smeg\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    print(sample)\n",
    "    !ln -s /master/nplatt/sm_nanopore/results/assembly/{sample}_rnd0.fasta {sample}.assembly_rnd0.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2841bc5-98dc-490b-b1a6-c1968fdbf9a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smor ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::7.984*1.68] collected minimizers\n",
      "[M::mm_idx_gen::9.287*2.93] sorted minimizers\n",
      "[M::main::9.288*2.93] loaded/built the index for 1674 target sequence(s)\n",
      "[M::mm_mapopt_update::10.191*2.76] mid_occ = 384\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1674\n",
      "[M::mm_idx_stat::10.857*2.65] distinct minimizers: 30374815 (65.19% are singletons); average occurrences: 2.391; average spacing: 5.323; total length: 386656054\n",
      "[M::worker_pipeline::152.226*39.59] mapped 79894 sequences\n",
      "[M::worker_pipeline::307.705*40.33] mapped 77039 sequences\n",
      "[M::worker_pipeline::476.213*40.86] mapped 74651 sequences\n",
      "[M::worker_pipeline::629.258*41.74] mapped 76580 sequences\n",
      "[M::worker_pipeline::794.384*41.56] mapped 73726 sequences\n",
      "[M::worker_pipeline::965.772*41.19] mapped 76750 sequences\n",
      "[M::worker_pipeline::1144.195*40.83] mapped 76077 sequences\n",
      "[M::worker_pipeline::1289.656*41.18] mapped 77343 sequences\n",
      "[M::worker_pipeline::1453.221*41.22] mapped 75661 sequences\n",
      "[M::worker_pipeline::1617.462*40.85] mapped 78790 sequences\n",
      "[M::worker_pipeline::1764.942*41.05] mapped 78853 sequences\n",
      "[M::worker_pipeline::1933.526*41.06] mapped 76202 sequences\n",
      "[M::worker_pipeline::2095.499*41.02] mapped 73981 sequences\n",
      "[M::worker_pipeline::2273.516*41.00] mapped 70832 sequences\n",
      "[M::worker_pipeline::2436.701*41.12] mapped 70994 sequences\n",
      "[M::worker_pipeline::2600.532*41.11] mapped 72574 sequences\n",
      "[M::worker_pipeline::2774.577*40.99] mapped 71286 sequences\n",
      "[M::worker_pipeline::2945.959*40.93] mapped 72852 sequences\n",
      "[M::worker_pipeline::3124.810*40.87] mapped 73865 sequences\n",
      "[M::worker_pipeline::3281.401*40.90] mapped 75586 sequences\n",
      "[M::worker_pipeline::3432.424*40.51] mapped 60781 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smor.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smor/smor.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3432.825 sec; CPU: 139062.435 sec; Peak RSS: 22.556 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.139771 s\n",
      "[racon::Polisher::initialize] loaded sequences 101.550652 s\n",
      "[racon::Polisher::initialize] loaded overlaps 69.268296 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 19.301929 s\n",
      "[racon::Polisher::initialize] transformed data into windows 18.209538 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 195.514138 s\n",
      "[racon::Polisher::] total = 410.996559 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::7.581*1.74] collected minimizers\n",
      "[M::mm_idx_gen::8.810*2.82] sorted minimizers\n",
      "[M::main::8.811*2.82] loaded/built the index for 1399 target sequence(s)\n",
      "[M::mm_mapopt_update::9.549*2.68] mid_occ = 381\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1399\n",
      "[M::mm_idx_stat::10.053*2.59] distinct minimizers: 30397504 (65.24% are singletons); average occurrences: 2.386; average spacing: 5.322; total length: 385948710\n",
      "[M::worker_pipeline::147.574*39.59] mapped 79894 sequences\n",
      "[M::worker_pipeline::306.812*39.36] mapped 77039 sequences\n",
      "[M::worker_pipeline::475.756*40.33] mapped 74651 sequences\n"
     ]
    }
   ],
   "source": [
    "#set the initial round\n",
    "rnd=1\n",
    "for sample in samples:\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    for rnd in [1, 2, 3]:\n",
    "        #define major i/o files\n",
    "        current_assembly = \"{}.assembly_rnd{}.fasta\".format(sample, str(rnd-1))\n",
    "        corrected_reads = \"{}/results/read_correction/{}/{}.correctedReads.fasta.gz\".format(proj_dir, sample, sample)\n",
    "        polished_assembly = \"{}.assembly_rnd{}.fasta\".format(sample, str(rnd))\n",
    "        n_threads=48\n",
    "        \n",
    "        #map corrected reads to the assembly\n",
    "        print(\"R{}: map\".format(rnd))\n",
    "        minimap_cmd = \"conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "            minimap2 \\\n",
    "                -a \\\n",
    "                -map-ont \\\n",
    "                -t {} \\\n",
    "                {} \\\n",
    "                {}  \\\n",
    "                >{}.racon_rnd{}.sam\".format(n_threads, current_assembly, corrected_reads, sample, rnd)\n",
    "        minimap_cmd=' '.join(minimap_cmd.split())\n",
    "        ! {minimap_cmd}\n",
    "\n",
    "        #polish the assembly\n",
    "        print(\"R{}: polish\".format(rnd))\n",
    "        racon_cmd = \"conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "            racon \\\n",
    "                -t {} \\\n",
    "                {} \\\n",
    "                {}.racon_rnd{}.sam \\\n",
    "                {} \\\n",
    "                >{}\".format(n_threads, corrected_reads, sample, rnd, current_assembly, polished_assembly)\n",
    "        racon_cmd=' '.join(racon_cmd.split())\n",
    "        ! {racon_cmd}\n",
    "        print(\"#\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "944cb7fd-823c-42ba-93ea-466eef913cbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    assembly = \"{}/polish_assembly/{}.assembly_rnd3.fasta\".format(proj_dir, sample)\n",
    "    busco -i {assembly} --auto-lineage-euk -c 48 -o {sample}.busco. -m genome \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "faae6500-e42f-45ab-b73c-127fa4a4aecf",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    assembly = \"{}/polish_assembly/{}.assembly_rnd3.fasta\".format(proj_dir, sample)\n",
    "    ./JupiterPlot/jupiter t=48 name={sample} ng=85 ref={ref_fas} fa={assembly}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "818182ca-701c-4c0c-86be-8094904e6032",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## Grave Yard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "316d74b7-320a-4a5c-9201-bba9709eb023",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate sm_ont-process_reads\n",
    "\n",
    "cd /master/nplatt/sm_nanopore\n",
    "\n",
    "cd sm_nanopore/results/nanofilt\n",
    "3\n",
    " ~/anaconda3/envs/sm_ont-process_reads/bin/flye --nano-cor ../read_correction_canu/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz -g 385m -t 96  -i 3 -o ${SAMPLE} >${SAMPLE}.canu.log 2>&1 &\n",
    "\n",
    "conda activate repeatmasker\n",
    "RepeatMasker \\\n",
    "    -pa 12 \\\n",
    "    -lib ../repeatmodeler/Smansoni_v7-families.fa \\\n",
    "    -gff \\\n",
    "    -s \\\n",
    "    -dir $(pwd) \\\n",
    "    -small \\\n",
    "    -x \\\n",
    "    $REF\n",
    "\n",
    "\n",
    "\n",
    "SmOR_Female_6_02_22_vSM_V9_sorted.vcf\n",
    "schisto_pairs_04132022_vSM_V9_sorted.vcf\n",
    "SmEG_Female_5_25_22_vSM_V9_sorted.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500) && AF>0.75' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pzq-es_filtered.vcf\n",
    "\n",
    "bedtools intersect \\\n",
    "    -c \\\n",
    "    -b smle-pzq-es_filtered.vcf \\\n",
    "    -a ../lift/v9.bed \\\n",
    "    >smle-pzq-es_intersect.bed\n",
    "\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-200 || SVLEN>200)' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout\n",
    "\n",
    "#############################################################################################\n",
    "\n",
    "#genrate reads\n",
    "art_illumina \\\n",
    "    --len 150 \\\n",
    "    --fcov 100 \\\n",
    "    --errfree \\\n",
    "    -na \\\n",
    "    --in SM_V9.fa \\\n",
    "    --out sm_v9_illumina \\\n",
    "    >sm_v9_illumina.log 2>&1 &\n",
    "\n",
    "nanosim-h \\\n",
    "    --perfect \\\n",
    "    -o sm_v9_ont \\\n",
    "    -n 5000000 \\\n",
    "    SM_V9.fa \\\n",
    "    >sm_v9_ont.log 2>&1 &\n",
    "\n",
    "#convert to fasta\n",
    "sed -n '1~4s/^@/>/p;2~4p' sm_v9_illumina.fq > sm_v9_illumina.fa\n",
    "\n",
    "#make sure amount of data is similar\n",
    "echo sm_v9_ont.fa $(cat sm_v9_ont.fa | paste  - - | cut -f2 | wc -c) >>counts &\n",
    "echo sm_v9_illumina.fa $(cat sm_v9_illumina.fa | paste  - - | cut -f2 | wc -c) >>counts &\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#map reads (perfectly and unambiguosly) back to the reference\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "    bbmap.sh \\\n",
    "        ref=SM_V9.fa \\\n",
    "        perfectmode=t \\\n",
    "        threads=48 \\\n",
    "        ambiguous=toss \\\n",
    "        in=${SAMPLE}.fa \\\n",
    "        overwrite=t \\\n",
    "        out=${SAMPLE}.sam \\\n",
    "        mappedonly=t \\\n",
    "        >${SAMPLE}.bbmap.log 2>&1 &\n",
    "done\n",
    "\n",
    "\n",
    "#calculate coverage\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "\n",
    "    echo $SAMPLE\n",
    "    echo \"...sort\"\n",
    "    samtools view -b -F 4 ${SAMPLE}.sam | samtools sort >${SAMPLE}.bam\n",
    "    echo \"...idx\"\n",
    "    samtools index ${SAMPLE}.bam\n",
    "    echo \"...cov\"\n",
    "    mosdepth \\\n",
    "        --threads 4 \\\n",
    "        ${SAMPLE} \\\n",
    "        ${SAMPLE}.bam\n",
    "done\n",
    "\n",
    "#how much is inaccessible per read type\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "\n",
    "    MB=$(zcat ${SAMPLE}.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}')\n",
    "    echo -e $SAMPLE\"\\t\"$MB\n",
    "\n",
    "done\n",
    "\n",
    "art_illumina \\\n",
    "--paired \\\n",
    "--len 150 \\\n",
    "--fcov 10 \\\n",
    "--mflen 300 \\\n",
    "--sdev 80 \\\n",
    "-na \\\n",
    "--in SM_V9.fa \\\n",
    "--out sm_v9_wes\n",
    "\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_wes1.fq \\\n",
    "    in2=sm_v9_wes2.fq \\\n",
    "    samestrandpairs=t \\\n",
    "    killbadpairs=t \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_wes.sam \\\n",
    "    mappedonly=t \\\n",
    "    ignorebadquality=t\n",
    "\n",
    "samtools view -b -F 4 sm_v9_wes.sam | samtools sort >sm_v9_wes.bam\n",
    "samtools index sm_v9_wes.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_wes \\\n",
    "    sm_v9_wes.bam\n",
    "\n",
    "\n",
    "\n",
    "zcat sm_v9_wes.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_wes.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "sed -i 's/SM_V9_1/HE601624.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_2/HE601625.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_3/HE601626.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_4/HE601627.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_5/HE601628.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_6/HE601629.3/' sm_v9.gff3\n",
    "\n",
    "\n",
    "\n",
    "bedtools intersect \\\n",
    "    -wb \\\n",
    "    -a smle-pzq-es_filtered.vcf \\\n",
    "    -b sm_v9.gff3\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500) && AF>0.75' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pzq-es_filtered.vcf\n",
    "\n",
    "bedtools intersect \\\n",
    "    -c \\\n",
    "    -b smle-pzq-es_filtered.vcf \\\n",
    "    -a ../lift/v9.bed\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "cp ../cutesv/sm_v9.gff3 .\n",
    "awk '{if ($3==\"exon\") print $0}' sm_v9.gff3 >sm_v9_exons.gff3\n",
    "cut -f1 -d\";\" sm_v9_exons.gff3 | cut -f1,4,5,9 | sed 's/ID=exon://' >sm_v9_exons.bed\n",
    "bedtools intersect -a sm_v9_exons.bed  -b qtls_v9.bed >qtl_exons.bed\n",
    "\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smeg_filtered.vcf  >smeg_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smeg_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smeg_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smeg_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smeg_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smor_filtered.vcf  >smor_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smor_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smor_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smor_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smor_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smle-pzq-es_filtered.vcf  >smle-pzq-es_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smle-pzq-es_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smle-pzq-es_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smle-pzq-es_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smle-pzq-es_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "nanosim-h --perfect -o sm_v9_ont_reads -n 500000 SM_V9.fa\n",
    "\n",
    "\n",
    "    minimap2 \\\n",
    "        -a \\\n",
    "        -o sm_v9_ont.sam \\\n",
    "        -t 48 \\\n",
    "        -x map-ont \\\n",
    "        SM_V9.fa \\\n",
    "        sm_v9_ont_reads.fa\n",
    "\n",
    "\n",
    "\n",
    "samtools view -b -F 4 sm_v9_ont.sam | samtools sort >sm_v9_ont.bam\n",
    "\n",
    "samtools index sm_v9_ont.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_ont \\\n",
    "    sm_v9_ont.bam\n",
    "\n",
    "zcat sm_v9_ont.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_ont.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_ont_reads.fa \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_ont_bbmap.sam \\\n",
    "    mappedonly=t \\\n",
    "    ignorebadquality=t\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    perfectmode=t \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_ont_reads.fa \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_ont_bbmap.sam \\\n",
    "    mappedonly=t\n",
    "\n",
    "\n",
    "samtools view -b -F 4 sm_v9_ont_bbmap.sam | samtools sort >sm_v9_ont_bbmap.bam\n",
    "samtools index sm_v9_ont_bbmap.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_ont_bbmap \\\n",
    "    sm_v9_ont_bbmap.bam\n",
    "\n",
    "zcat sm_v9_ont_bbmap.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_ont_bbmap.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "RepeatMasker -qq -pa 48 -nolow -norna -dir . -gff -lib sm_v7_tes.fas SM_V9.fa\n",
    "\n",
    "SmEG_Female_5_25_22\n",
    "schisto_pairs_04132022  ../cutesv/SmOR_Female_6_02_22_vSM_V9_sorted.vcf\n",
    "\n",
    "for VCF in $(ls ../cutesv/*_vSM_V9_sorted.vcf); do \n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pze-es_bigsvs.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/SmEG_Female_5_25_22_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smeg_bigsvs.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/SmOR_Female_6_02_22_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smor_bigsvs.vcf\n",
    "\n",
    "   396,000,000\n",
    "39,777,694,771\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "sm_ont-process_reads",
   "language": "python",
   "name": "sm_ont-process_reads"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
