{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "797f812c-0cd1-4332-bfd6-b6dda1d9d2ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "#use conda env sm_ont-process_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "68681984-0034-4a1b-8835-ba6cf659f329",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from pathlib import Path\n",
    "#import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d6fba8af-fc68-4a7c-8743-bf9c352102c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "proj_dir = \"/master/nplatt/sm_nanopore\"\n",
    "ref_fas = \"{}/data/genome/Smansoni_v7.fa\".format(proj_dir)\n",
    "\n",
    "Path(\"{}/results\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "samples=[\"smor\", \n",
    "         \"smle_pzq_es\", \n",
    "         \"smbre\", \n",
    "         \"smeg\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0ed1a605-2d69-4de2-882c-f4d1eb3e4904",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"{}/results\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "106579da-cfaa-4729-a313-8f2ad2b23619",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/read_correction_canu\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/read_correction_canu\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8a245df-25b8-4acd-a76f-c436485e5a88",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    in_fq = \"{}/results/nanofilt/{}.filtered.fastq.gz\".format(proj_dir, sample)\n",
    "\n",
    "    !conda run -n sm_ont-process_reads canu -correct -minThreads=48 -genomesize=385m -useGrid=False -d {sample} -p {sample} -nanopore {in_fq}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "70598b10-c258-4b81-b334-965929fee9ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ~/anaconda3/envs/sm_ont-process_reads/bin/canu -correct -minThreads=48 -minThreads=96 -genomesize=385m -useGrid=False -d smbre/ -p smbre -nanopore ../nanofilt/smbre.filtered.fastq.gz >smbre.canu.log 2>&1\n",
    "\n",
    "# ~/anaconda3/envs/sm_ont-process_reads/bin/canu -correct -minThreads=48 -minThreads=96 -genomesize=385m -useGrid=False -d smor/ -p smor -nanopore ../nanofilt/smor.filtered.fastq.gz >smor.canu.log 2>&1\n",
    "\n",
    "# ~/anaconda3/envs/sm_ont-process_reads/bin/canu -correct -minThreads=48 -minThreads=96 -genomesize=385m -useGrid=False -d smeg/ -p smeg -nanopore ../nanofilt/smeg.filtered.fastq.gz >smeg.canu.log 2>&1\n",
    "\n",
    "# ~/anaconda3/envs/sm_ont-process_reads/bin/canu -correct -minThreads=48 -minThreads=96 -genomesize=385m -useGrid=False -d smle_pzq_es/ -p smle_pzq_es -nanopore ../nanofilt/smle_pzq_es.filtered.fastq.gz >smle_pzq_es.canu.log 2>&1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb4482ba-4f57-4537-bb56-36ad5a5c2d00",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    in_fas = \"{}/results/read_correction_canu/{}/{}.correctedReads.fasta.gz\".format(proj_dir, sample, sample)\n",
    "\n",
    "    !conda run -n sm_ont-process_reads flye --nano-cor {in_fas} -g 385m -t 128 -i 3 -o {sample}\n",
    "\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93f7bcd3-cf61-4730-9514-0ad29e28f326",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir /master/nplatt/sm_nanopore/results/polish_r1\n",
    "cd /master/nplatt/sm_nanopore/results/polish_r1\n",
    "\n",
    "for SAMPLE in smle_pzq_es smbre smeg smor; do\n",
    "    minimap2 \\\n",
    "        -a \\\n",
    "        -map-ont \\\n",
    "        -t 192 \\\n",
    "        /master/nplatt/sm_nanopore/results/assembly/${SAMPLE}/assembly.fasta \\\n",
    "        /master/nplatt/sm_nanopore/results/read_correction_canu/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz \\\n",
    "        >${SAMPLE}_r1.sam\n",
    "\n",
    "    racon \\\n",
    "        -t 192 \\\n",
    "        /master/nplatt/sm_nanopore/results/read_correction_canu/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz \\\n",
    "        ${SAMPLE}_r1.sam \\\n",
    "        /master/nplatt/sm_nanopore/results/assembly/${SAMPLE}/assembly.fasta \\\n",
    "        >${SAMPLE}_r1_racon.fasta\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b158ab6-9cce-4ffa-93ea-c10c91e99675",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir /master/nplatt/sm_nanopore/results/polish_r2\n",
    "cd /master/nplatt/sm_nanopore/results/polish_r2\n",
    "\n",
    "for SAMPLE in smle_pzq_es smbre smeg smor; do\n",
    "    minimap2 \\\n",
    "       -a \\\n",
    "       -map-ont \\\n",
    "       -t 48 \\\n",
    "       /master/nplatt/sm_nanopore/results/polish_r1/${SAMPLE}_r1_racon.fas \\\n",
    "       /master/nplatt/sm_nanopore/results/read_correction_canu/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz \\\n",
    "       >${SAMPLE}_r2.sam 2>${SAMPLE}_r2_minimap.log &\n",
    "\n",
    "    racon \\\n",
    "        -t 48 \\\n",
    "        /master/nplatt/sm_nanopore/results/read_correction_canu/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz \\\n",
    "        ${SAMPLE}_r2.sam \\\n",
    "        /master/nplatt/sm_nanopore/results/polish_r1/${SAMPLE}_r1_racon.fasta \\\n",
    "        >${SAMPLE}_r2_racon.fas 2>${SAMPLE}_r2_racon.log &\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1a2a1fa-3cba-4e5e-85ab-0d23618d06d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir /master/nplatt/sm_nanopore/results/polish_r3\n",
    "cd /master/nplatt/sm_nanopore/results/polish_r3\n",
    "\n",
    "for SAMPLE in smle_pzq_es smbre smeg smor; do\n",
    "    minimap2 \\\n",
    "        -a \\\n",
    "        -map-ont \\\n",
    "        -t 64 \\\n",
    "        /master/nplatt/sm_nanopore/results/polish_r2/${SAMPLE}_r2_racon.fasta \\\n",
    "        /master/nplatt/sm_nanopore/results/read_correction_canu/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz \\\n",
    "        >${SAMPLE}_r3.sam 2>${SAMPLE}_r3_minimap.log &\n",
    "\n",
    "done\n",
    "\n",
    "wait \n",
    "\n",
    "for SAMPLE in smle_pzq_es smbre smeg smor; do\n",
    "    racon \\\n",
    "        -t 192 \\\n",
    "        /master/nplatt/sm_nanopore/results/read_correction_canu/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz \\\n",
    "        ${SAMPLE}_r3.sam \\\n",
    "        /master/nplatt/sm_nanopore/results/polish_r2/${SAMPLE}_r2_racon.fasta \\\n",
    "        >${SAMPLE}_r3_racon.fasta 2>${SAMPLE}_r3_racon.log &\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5affd219-6bab-4560-a7bc-5c1e9c7f3e80",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir /master/nplatt/sm_nanopore/results/busco\n",
    "cd /master/nplatt/sm_nanopore/results/busco\n",
    "\n",
    "conda activate sm_ont-busco\n",
    "\n",
    "for SAMPLE in smle_pzq_es smbre smeg smor; do\n",
    "    busco \\\n",
    "        -i /master/nplatt/sm_nanopore/results/polish_r3/${SAMPLE}_r3_racon.fasta \\\n",
    "        --auto-lineage-euk \\\n",
    "        -c 48 \\\n",
    "        -o ${SAMPLE} \\\n",
    "        -m genome \\\n",
    "        >${SAMPLE}.busco.log\n",
    "done\n",
    "\n",
    "busco \\\n",
    "    -i /master/nplatt/sm_nanopore/data/genome/SM_V9.fa \\\n",
    "    --auto-lineage-euk \\\n",
    "    -c 48 \\\n",
    "    -o SM_V9 \\\n",
    "    -m genome \\\n",
    "    >SM_V9.busco.log\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ef5ab68-a4c2-4e93-aa95-a9155d692d59",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd sm_nanopore/results/synteny/\n",
    "conda activate sm_ont-sibeliaz\n",
    "export SATSUMA2_PATH=/master/nplatt/anaconda3/envs/sm_ont-sibeliaz/bin\n",
    "\n",
    "#need to run repeat modeler and then repeatmasker to HARD MASK the repeats.\n",
    "SatsumaSynteny2 -q ../polish_r3/smle_pzq_es_r3_racon.fasta  -t ../../data/genome/SM_V9.fa  -o smle_pzq_es -slaves 12 -threads 12 -sl_mem 64\n",
    "SatsumaSynteny2 -q ../polish_r3/smor_r3_racon.fasta  -t ../../data/genome/SM_V9.fa  -o smor -slaves 12 -threads 12 -sl_mem 64\n",
    "SatsumaSynteny2 -q ../polish_r3/smle_r3_racon.fasta  -t ../../data/genome/SM_V9.fa  -o smle -slaves 12 -threads 12 -sl_mem 64\n",
    "SatsumaSynteny2 -q ../polish_r3/smbre_r3_racon.fasta  -t ../../data/genome/SM_V9.fa  -o smbre -slaves 12 -threads 12 -sl_mem 64"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "98786cae-4daf-4dae-9ebd-bf4c0d817b13",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2895930 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "2895930 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "2340800 + 0 mapped (80.83% : N/A)\n",
      "2340800 + 0 primary mapped (80.83% : N/A)\n",
      "2895930 + 0 paired in sequencing\n",
      "1447965 + 0 read1\n",
      "1447965 + 0 read2\n",
      "2285142 + 0 properly paired (78.91% : N/A)\n",
      "2300692 + 0 with itself and mate mapped\n",
      "40108 + 0 singletons (1.38% : N/A)\n",
      "2564 + 0 with mate mapped to a different chr\n",
      "2564 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cmd=\"samtools flagstat /master/nplatt/sch_hae_scan/results/mapped_reads/sh_tzpem_SRR8284792_pe.sam\"\n",
    "!conda run -n samtools {cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "efd6ad07-8de7-4fbf-99e2-06920b98be19",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6810ffa4-abe6-4eab-8b2e-ca14946d877e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bb91217f-579b-4588-9864-bb0d311a11dd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51f75516-634d-412b-bf12-2a01068e30c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    " ~/anaconda3/envs/sm_ont-process_reads/bin/flye --nano-cor ../read_correction_canu/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz -g 385m -t 96  -i 3 -o ${SAMPLE} >${SAMPLE}.canu.log 2>&1 &\n",
    "\n",
    "conda activate repeatmasker\n",
    "RepeatMasker \\\n",
    "    -pa 12 \\\n",
    "    -lib ../repeatmodeler/Smansoni_v7-families.fa \\\n",
    "    -gff \\\n",
    "    -s \\\n",
    "    -dir $(pwd) \\\n",
    "    -small \\\n",
    "    -x \\\n",
    "    $REF\n",
    "\n",
    "\n",
    "\n",
    "SmOR_Female_6_02_22_vSM_V9_sorted.vcf\n",
    "schisto_pairs_04132022_vSM_V9_sorted.vcf\n",
    "SmEG_Female_5_25_22_vSM_V9_sorted.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500) && AF>0.75' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pzq-es_filtered.vcf\n",
    "\n",
    "bedtools intersect \\\n",
    "    -c \\\n",
    "    -b smle-pzq-es_filtered.vcf \\\n",
    "    -a ../lift/v9.bed \\\n",
    "    >smle-pzq-es_intersect.bed\n",
    "\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-200 || SVLEN>200)' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout\n",
    "\n",
    "#############################################################################################\n",
    "\n",
    "#genrate reads\n",
    "art_illumina \\\n",
    "    --len 150 \\\n",
    "    --fcov 100 \\\n",
    "    --errfree \\\n",
    "    -na \\\n",
    "    --in SM_V9.fa \\\n",
    "    --out sm_v9_illumina \\\n",
    "    >sm_v9_illumina.log 2>&1 &\n",
    "\n",
    "nanosim-h \\\n",
    "    --perfect \\\n",
    "    -o sm_v9_ont \\\n",
    "    -n 5000000 \\\n",
    "    SM_V9.fa \\\n",
    "    >sm_v9_ont.log 2>&1 &\n",
    "\n",
    "#convert to fasta\n",
    "sed -n '1~4s/^@/>/p;2~4p' sm_v9_illumina.fq > sm_v9_illumina.fa\n",
    "\n",
    "#make sure amount of data is similar\n",
    "echo sm_v9_ont.fa $(cat sm_v9_ont.fa | paste  - - | cut -f2 | wc -c) >>counts &\n",
    "echo sm_v9_illumina.fa $(cat sm_v9_illumina.fa | paste  - - | cut -f2 | wc -c) >>counts &\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#map reads (perfectly and unambiguosly) back to the reference\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "    bbmap.sh \\\n",
    "        ref=SM_V9.fa \\\n",
    "        perfectmode=t \\\n",
    "        threads=48 \\\n",
    "        ambiguous=toss \\\n",
    "        in=${SAMPLE}.fa \\\n",
    "        overwrite=t \\\n",
    "        out=${SAMPLE}.sam \\\n",
    "        mappedonly=t \\\n",
    "        >${SAMPLE}.bbmap.log 2>&1 &\n",
    "done\n",
    "\n",
    "\n",
    "#calculate coverage\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "\n",
    "    echo $SAMPLE\n",
    "    echo \"...sort\"\n",
    "    samtools view -b -F 4 ${SAMPLE}.sam | samtools sort >${SAMPLE}.bam\n",
    "    echo \"...idx\"\n",
    "    samtools index ${SAMPLE}.bam\n",
    "    echo \"...cov\"\n",
    "    mosdepth \\\n",
    "        --threads 4 \\\n",
    "        ${SAMPLE} \\\n",
    "        ${SAMPLE}.bam\n",
    "done\n",
    "\n",
    "#how much is inaccessible per read type\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "\n",
    "    MB=$(zcat ${SAMPLE}.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}')\n",
    "    echo -e $SAMPLE\"\\t\"$MB\n",
    "\n",
    "done\n",
    "\n",
    "art_illumina \\\n",
    "--paired \\\n",
    "--len 150 \\\n",
    "--fcov 10 \\\n",
    "--mflen 300 \\\n",
    "--sdev 80 \\\n",
    "-na \\\n",
    "--in SM_V9.fa \\\n",
    "--out sm_v9_wes\n",
    "\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_wes1.fq \\\n",
    "    in2=sm_v9_wes2.fq \\\n",
    "    samestrandpairs=t \\\n",
    "    killbadpairs=t \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_wes.sam \\\n",
    "    mappedonly=t \\\n",
    "    ignorebadquality=t\n",
    "\n",
    "samtools view -b -F 4 sm_v9_wes.sam | samtools sort >sm_v9_wes.bam\n",
    "samtools index sm_v9_wes.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_wes \\\n",
    "    sm_v9_wes.bam\n",
    "\n",
    "\n",
    "\n",
    "zcat sm_v9_wes.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_wes.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "sed -i 's/SM_V9_1/HE601624.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_2/HE601625.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_3/HE601626.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_4/HE601627.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_5/HE601628.3/' sm_v9.gff3\n",
    "sed -i 's/SM_V9_6/HE601629.3/' sm_v9.gff3\n",
    "\n",
    "\n",
    "\n",
    "bedtools intersect \\\n",
    "    -wb \\\n",
    "    -a smle-pzq-es_filtered.vcf \\\n",
    "    -b sm_v9.gff3\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500) && AF>0.75' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pzq-es_filtered.vcf\n",
    "\n",
    "bedtools intersect \\\n",
    "    -c \\\n",
    "    -b smle-pzq-es_filtered.vcf \\\n",
    "    -a ../lift/v9.bed\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "cp ../cutesv/sm_v9.gff3 .\n",
    "awk '{if ($3==\"exon\") print $0}' sm_v9.gff3 >sm_v9_exons.gff3\n",
    "cut -f1 -d\";\" sm_v9_exons.gff3 | cut -f1,4,5,9 | sed 's/ID=exon://' >sm_v9_exons.bed\n",
    "bedtools intersect -a sm_v9_exons.bed  -b qtls_v9.bed >qtl_exons.bed\n",
    "\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smeg_filtered.vcf  >smeg_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smeg_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smeg_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smeg_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smeg_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smor_filtered.vcf  >smor_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smor_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smor_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smor_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smor_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smle-pzq-es_filtered.vcf  >smle-pzq-es_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smle-pzq-es_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smle-pzq-es_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smle-pzq-es_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smle-pzq-es_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "nanosim-h --perfect -o sm_v9_ont_reads -n 500000 SM_V9.fa\n",
    "\n",
    "\n",
    "    minimap2 \\\n",
    "        -a \\\n",
    "        -o sm_v9_ont.sam \\\n",
    "        -t 48 \\\n",
    "        -x map-ont \\\n",
    "        SM_V9.fa \\\n",
    "        sm_v9_ont_reads.fa\n",
    "\n",
    "\n",
    "\n",
    "samtools view -b -F 4 sm_v9_ont.sam | samtools sort >sm_v9_ont.bam\n",
    "\n",
    "samtools index sm_v9_ont.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_ont \\\n",
    "    sm_v9_ont.bam\n",
    "\n",
    "zcat sm_v9_ont.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_ont.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_ont_reads.fa \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_ont_bbmap.sam \\\n",
    "    mappedonly=t \\\n",
    "    ignorebadquality=t\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    perfectmode=t \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_ont_reads.fa \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_ont_bbmap.sam \\\n",
    "    mappedonly=t\n",
    "\n",
    "\n",
    "samtools view -b -F 4 sm_v9_ont_bbmap.sam | samtools sort >sm_v9_ont_bbmap.bam\n",
    "samtools index sm_v9_ont_bbmap.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_ont_bbmap \\\n",
    "    sm_v9_ont_bbmap.bam\n",
    "\n",
    "zcat sm_v9_ont_bbmap.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_ont_bbmap.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "RepeatMasker -qq -pa 48 -nolow -norna -dir . -gff -lib sm_v7_tes.fas SM_V9.fa\n",
    "\n",
    "SmEG_Female_5_25_22\n",
    "schisto_pairs_04132022  ../cutesv/SmOR_Female_6_02_22_vSM_V9_sorted.vcf\n",
    "\n",
    "for VCF in $(ls ../cutesv/*_vSM_V9_sorted.vcf); do \n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pze-es_bigsvs.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/SmEG_Female_5_25_22_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smeg_bigsvs.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/SmOR_Female_6_02_22_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smor_bigsvs.vcf\n",
    "\n",
    "   396,000,000\n",
    "39,777,694,771\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5f31b5bd-e0e8-4c69-8597-cba8ab26622f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "here\n"
     ]
    }
   ],
   "source": [
    "print(\"here\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "sm_ont-process_reads",
   "language": "python",
   "name": "sm_ont-process_reads"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
