{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5fa1b64e-a799-4ff8-896c-36534c559df6",
   "metadata": {},
   "source": [
    "# Nanopore long-read sequencing uncovers hundreds of structural variants in the Schistosome genome\n",
    "*Shalini Nair, Elisha Enabulele, Xue Li, Tim Anderson, Neal Platt*"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af9f12f9-924c-4ced-9a24-70fe777660e7",
   "metadata": {
    "tags": []
   },
   "source": [
    "__ABSTRACT__: Nanopore sequencing generates extremely long sequence reads allowing detailed characterization of genomic features (genome rearrangements/structural variants) that are difficult to accurately assay using Illumina short-read sequencing approaches. Technological improvements in Nanopore long-read sequence have resulted in reduced prices, as well as increased accuracy and output. This method was declared “Method of the year” in Nature in 2022. Our laboratory is exploring the use of nanopore for characterizing pathogen genomes. Here, we detail application of nanopore sequencing of schistosome parasites. These parasites have a 380 Mb genome that is riddled with repetitive elements and structural variants. We have developed protocols that allow generation of reads up to 200kb, with 50% of reads > 24kb, and generation of 13.5 Gb of high-quality data from a single R10 flow cell, providing 27x coverage of the parasite genome.  Using this new data, we have identified multiple structural variants segregating in a lab strain of S. mansoni that would not have been visible using short read sequencing alone.  For example, we have found 468 large structural variants (>500bp) that were completely fixed in our laboratory population compared to the reference genome/strain. We anticipate that characterization of large structural variants may help to identify causative features underlying important schistosome phenotypes. Nanopore also has many possible uses for other researchers at Texas biomed, including sequencing entire viral genomes or host MHC, and rapid monitoring of infectious diseases in global populations."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77d21088-87ad-4785-97c9-9e142e6208c5",
   "metadata": {
    "tags": []
   },
   "source": [
    "**[08/04/2023] Nanopore analysis ideas**\n",
    "\n",
    "- Are the genomes of lab populations syntenic or have large changes in gene structure/order been introduced?\n",
    "  - __This would be extremely interesting__\n",
    "  - Aside – one possible explanation for the division between Sh populations might be rearrangements/inversions\n",
    "- What impact do large SVs have on the coding regions of the genome?\n",
    "  - Comment – define impact here. It would certainly be interesting to know whether SVs in coding sequences seem likely to reduce/eliminate gene expression\n",
    "- Is there evidence of population specific SVs (including CNVs/gene family expansions/contractions)?\n",
    "- Can we recover and examine variability in complex regions like SmPoMucs and how is this variability distributed among and within lab populations?\n",
    "  - __Yes – v important.__\n",
    "  - Simply identifying the genome location of these genes will be very important – we would like to be able to definitively state that smPoMucs do/do not colocalize with the host specificity QTLs that Fred has mapped.\n",
    "- Are there large SVs in important regions associated with parasite phenotypes?\n",
    "  - Important question but difficult to do. I think we can certainly examine how many SVs are segregating in QTL regions, to highlight possible causative SVs\n",
    "- (Some interesting question about the tandem repeat region in the mitochondrial genome...not sure what yet)\n",
    "  - There is some old literature about these regions, suggesting v rapid change and extensive polymorphism within individuals. This might be difficult with the population data, but would be great to do when we have data from individual genotypes. Perhaps worth waiting until we have that data.\n",
    "- __Is it possible to identify SVs that are being selected (for/against)?__\n",
    "  - I think this is particularly interesting. We expect most SVs to be selected against.\n",
    "  - Comparing allele frequencies within populations would be informative – we expect most to be at low frequency. High frequency SVs are good candidates for positive selection\n",
    "  - __Comparing SV burden in coding vs non-coding regions would be informative to understand strength of selection against SVs__\n",
    "  - __Likewise, comparing SVs in low and high recombination rate regions. We predict higher frequency/larger SVs in high recombination regions__"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db002030-e5c3-4e4b-bf17-4ab6628a7b44",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Prep analyses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9cd4c83d-03e3-44f3-8e5a-5e64db850897",
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'vcf'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "Input \u001b[0;32mIn [8]\u001b[0m, in \u001b[0;36m<cell line: 8>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mBio\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m SeqIO\n\u001b[1;32m      7\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mBio\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mSeqRecord\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m SeqRecord\n\u001b[0;32m----> 8\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mvcf\u001b[39;00m\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'vcf'"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from pathlib import Path\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import pysam\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "import vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3bb0e454-1134-4194-8c67-e685847b6669",
   "metadata": {},
   "outputs": [],
   "source": [
    "proj_dir = \"/master/nplatt/sch_man_ont\"\n",
    "ref_fas = \"{}/data/genome/GCA_000237925.5.fa\".format(proj_dir)\n",
    "Path(\"{}/results\".format(proj_dir)).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "625a0a9e-c38c-4c9a-a768-c3a22131bfb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "samples = [ \"smor\", \n",
    "            \"smle_pzq_es\",\n",
    "            \"smle_pzq_er\",\n",
    "            \"smbre\", \n",
    "            \"smeg\"        ]\n",
    "\n",
    "pops=samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d62d2d02-6e9f-4c61-8028-73b97326de49",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"{}/results\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4fef4238-0533-44e4-a139-b08d6950cf3c",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## Filter reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13dd5d1e-db7d-4a53-a6bb-8bc790f41f71",
   "metadata": {},
   "outputs": [],
   "source": [
    "#all raw reads are provided in results/raw_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4446e16f-dd96-4417-9da8-deebe8d38001",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/nanofilt\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/nanofilt\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "c2485bf9-bb27-4391-a8ed-a0dbb15cf362",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    raw_fq = \"{}/results/raw_reads/{}.raw_pass.fastq\".format(proj_dir, sample)\n",
    "\n",
    "    !~/anaconda3/envs/sm_ont-process_reads/bin/NanoFilt --length 500 --quality 7 --headcrop 25 --tailcrop 25 {raw_fq} >{sample}.filtered.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e6cbcdc-cfe9-4087-96c3-0e0269379dcd",
   "metadata": {},
   "outputs": [],
   "source": [
    "!conda run -n n50 --cwd . --live-stream n50 --format csv $(ls *.filtered.fastq) >n50_filtered_reads.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "ab2652e7-383e-4b9a-9c95-4f9da176d9c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#path</th>\n",
       "      <th>seqs</th>\n",
       "      <th>size</th>\n",
       "      <th>N50</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>N75</th>\n",
       "      <th>N90</th>\n",
       "      <th>auN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>smbre.filtered.fastq</td>\n",
       "      <td>2156891</td>\n",
       "      <td>13733527914</td>\n",
       "      <td>16145</td>\n",
       "      <td>500</td>\n",
       "      <td>193069</td>\n",
       "      <td>7657</td>\n",
       "      <td>2728</td>\n",
       "      <td>357349.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smeg.filtered.fastq</td>\n",
       "      <td>2096508</td>\n",
       "      <td>12631573135</td>\n",
       "      <td>12491</td>\n",
       "      <td>500</td>\n",
       "      <td>236942</td>\n",
       "      <td>5659</td>\n",
       "      <td>2587</td>\n",
       "      <td>399825.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smle_pzq_er.filtered.fastq</td>\n",
       "      <td>1330051</td>\n",
       "      <td>7000372416</td>\n",
       "      <td>12071</td>\n",
       "      <td>500</td>\n",
       "      <td>447354</td>\n",
       "      <td>4953</td>\n",
       "      <td>2105</td>\n",
       "      <td>227185.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smor.filtered.fastq</td>\n",
       "      <td>3413369</td>\n",
       "      <td>14119478724</td>\n",
       "      <td>8668</td>\n",
       "      <td>500</td>\n",
       "      <td>168273</td>\n",
       "      <td>3430</td>\n",
       "      <td>1618</td>\n",
       "      <td>598308.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smle_pzq_es.filtered.fastq</td>\n",
       "      <td>3662716</td>\n",
       "      <td>23840253863</td>\n",
       "      <td>13824</td>\n",
       "      <td>500</td>\n",
       "      <td>202196</td>\n",
       "      <td>5893</td>\n",
       "      <td>2659</td>\n",
       "      <td>624239.05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        #path     seqs         size    N50  min     max   N75  \\\n",
       "0        smbre.filtered.fastq  2156891  13733527914  16145  500  193069  7657   \n",
       "1         smeg.filtered.fastq  2096508  12631573135  12491  500  236942  5659   \n",
       "2  smle_pzq_er.filtered.fastq  1330051   7000372416  12071  500  447354  4953   \n",
       "3         smor.filtered.fastq  3413369  14119478724   8668  500  168273  3430   \n",
       "4  smle_pzq_es.filtered.fastq  3662716  23840253863  13824  500  202196  5893   \n",
       "\n",
       "    N90        auN  \n",
       "0  2728  357349.74  \n",
       "1  2587  399825.88  \n",
       "2  2105  227185.13  \n",
       "3  1618  598308.11  \n",
       "4  2659  624239.05  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n50_df = pd.read_csv(\"n50_filtered_reads.csv\", sep=\",\")\n",
    "n50_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80d45e6b-b4c7-4959-8f79-1a3dbd4305c3",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Map reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c55d5b71-923a-41e2-966b-b082b11b9f05",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/minimap2\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/minimap2\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "6db32522-3695-4aaa-9269-e8fb6b5be5c4",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smle_pzq_es ##################\n",
      "[M::mm_idx_gen::11.489*1.68] collected minimizers\n",
      "[M::mm_idx_gen::13.279*2.42] sorted minimizers\n",
      "[M::main::13.279*2.42] loaded/built the index for 12 target sequence(s)\n",
      "[M::mm_mapopt_update::14.118*2.34] mid_occ = 416\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 12\n",
      "[M::mm_idx_stat::14.681*2.28] distinct minimizers: 30119087 (65.35% are singletons); average occurrences: 2.439; average spacing: 5.327; total length: 391422238\n",
      "[M::worker_pipeline::214.298*29.98] mapped 42338 sequences\n",
      "[M::worker_pipeline::390.138*31.90] mapped 40808 sequences\n",
      "[M::worker_pipeline::588.947*32.25] mapped 43306 sequences\n",
      "[M::worker_pipeline::830.022*32.30] mapped 45455 sequences\n",
      "[M::worker_pipeline::1051.682*32.12] mapped 44703 sequences\n",
      "[M::worker_pipeline::1289.362*31.59] mapped 45693 sequences\n",
      "[M::worker_pipeline::1470.012*32.22] mapped 47806 sequences\n",
      "[M::worker_pipeline::1696.452*32.12] mapped 45932 sequences\n",
      "[M::worker_pipeline::1916.264*31.95] mapped 47284 sequences\n",
      "[M::worker_pipeline::2103.592*32.13] mapped 48028 sequences\n",
      "[M::worker_pipeline::2304.990*32.24] mapped 49325 sequences\n",
      "[M::worker_pipeline::2494.646*32.47] mapped 50218 sequences\n",
      "[M::worker_pipeline::2717.671*32.28] mapped 49571 sequences\n",
      "[M::worker_pipeline::2898.281*32.52] mapped 49790 sequences\n",
      "[M::worker_pipeline::3096.096*32.53] mapped 51413 sequences\n",
      "[M::worker_pipeline::3288.826*32.55] mapped 53354 sequences\n",
      "[M::worker_pipeline::3537.461*32.11] mapped 53951 sequences\n",
      "[M::worker_pipeline::3732.637*32.17] mapped 53719 sequences\n",
      "[M::worker_pipeline::3940.025*32.09] mapped 56807 sequences\n",
      "[M::worker_pipeline::4112.592*32.21] mapped 55685 sequences\n",
      "[M::worker_pipeline::4310.755*32.19] mapped 59269 sequences\n",
      "[M::worker_pipeline::4539.062*32.29] mapped 48605 sequences\n",
      "[M::worker_pipeline::4782.077*32.26] mapped 45151 sequences\n",
      "[M::worker_pipeline::4994.360*32.35] mapped 43831 sequences\n",
      "[M::worker_pipeline::5221.479*32.45] mapped 44343 sequences\n",
      "[M::worker_pipeline::5444.817*32.50] mapped 44640 sequences\n",
      "[M::worker_pipeline::5684.431*32.44] mapped 45538 sequences\n",
      "[M::worker_pipeline::5866.860*32.59] mapped 134943 sequences\n",
      "[M::worker_pipeline::6041.373*32.73] mapped 133880 sequences\n",
      "[M::worker_pipeline::6222.160*32.86] mapped 133272 sequences\n",
      "[M::worker_pipeline::6419.560*32.90] mapped 132816 sequences\n",
      "[M::worker_pipeline::6601.796*33.00] mapped 131336 sequences\n",
      "[M::worker_pipeline::6793.441*33.04] mapped 130610 sequences\n",
      "[M::worker_pipeline::6972.643*33.17] mapped 127668 sequences\n",
      "[M::worker_pipeline::7149.058*33.29] mapped 127555 sequences\n",
      "[M::worker_pipeline::7314.339*33.38] mapped 141249 sequences\n",
      "[M::worker_pipeline::7499.906*33.46] mapped 118698 sequences\n",
      "[M::worker_pipeline::7698.053*33.54] mapped 95874 sequences\n",
      "[M::worker_pipeline::7900.189*33.61] mapped 95068 sequences\n",
      "[M::worker_pipeline::8098.936*33.70] mapped 95263 sequences\n",
      "[M::worker_pipeline::8289.017*33.81] mapped 95789 sequences\n",
      "[M::worker_pipeline::8475.551*33.92] mapped 97350 sequences\n",
      "[M::worker_pipeline::8688.873*33.92] mapped 97320 sequences\n",
      "[M::worker_pipeline::8884.251*34.00] mapped 99976 sequences\n",
      "[M::worker_pipeline::9054.041*34.08] mapped 100132 sequences\n",
      "[M::worker_pipeline::9228.106*34.16] mapped 100400 sequences\n",
      "[M::worker_pipeline::9421.437*34.20] mapped 99759 sequences\n",
      "[M::worker_pipeline::9539.774*34.24] mapped 67195 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: /master/nplatt/anaconda3/envs/sm_ont-process_reads/bin/minimap2 -a -o /master/nplatt/sm_nanopore/results/minimap2/smle_pzq_es.sam -t 48 -x map-ont /master/nplatt/sm_nanopore/data/genome/GCA_000237925.5.fa /master/nplatt/sm_nanopore/results/nanofilt/smle_pzq_es.filtered.fastq\n",
      "[M::main] Real time: 9540.184 sec; CPU: 326642.700 sec; Peak RSS: 26.741 GB\n"
     ]
    }
   ],
   "source": [
    "# for sample in samples:\n",
    "for sample in [\"smle_pzq_es\"]:\n",
    "    sam      = \"{}/results/minimap2/{}.sam\".format(proj_dir, sample)\n",
    "    query_fq = \"{}/results/nanofilt/{}.filtered.fastq\".format(proj_dir, sample)\n",
    "\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "    !~/anaconda3/envs/sm_ont-process_reads/bin/minimap2 -a -o {sam} -t 48 -x map-ont {ref_fas} {query_fq}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "5a175957-2dae-4bac-a37e-aa063e0d1a48",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smle_pzq_es ##################\n",
      "...sort...\n",
      "[bam_sort_core] merging from 60 files and 1 in-memory blocks...\n",
      "...index...\n",
      "...coverage...\n"
     ]
    }
   ],
   "source": [
    "# for sample in samples:\n",
    "for sample in [\"smle_pzq_es\"]:\n",
    "    sam      = \"{}/results/minimap2/{}.sam\".format(proj_dir, sample)\n",
    "    query_fq = \"{}/results/nanofilt/{}.filtered.fastq\".format(proj_dir, sample)\n",
    "    \n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    #sort bam\n",
    "    print(\"...sort...\")\n",
    "    ! ~/anaconda3/envs/sm_ont-process_reads/bin/samtools sort -O bam -o {sample}.bam {sample}.sam\n",
    "    \n",
    "    #index bam\n",
    "    print(\"...index...\")\n",
    "    ! ~/anaconda3/envs/sm_ont-process_reads/bin/samtools index {sample}.bam\n",
    "    \n",
    "    #calculate coverage\n",
    "    print(\"...coverage...\")\n",
    "    ! ~/anaconda3/envs/sm_ont-process_reads/bin/mosdepth --threads 4 {sample} {sample}.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "23b040b1-f1f2-4873-a2d7-6772dfc168b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>length</th>\n",
       "      <th>bases</th>\n",
       "      <th>mean</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>smbre</td>\n",
       "      <td>391422238</td>\n",
       "      <td>12929849971</td>\n",
       "      <td>33.03</td>\n",
       "      <td>0</td>\n",
       "      <td>50586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smeg</td>\n",
       "      <td>391422238</td>\n",
       "      <td>8667326476</td>\n",
       "      <td>22.14</td>\n",
       "      <td>0</td>\n",
       "      <td>34407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smle_pzq_er</td>\n",
       "      <td>391422238</td>\n",
       "      <td>6624879196</td>\n",
       "      <td>16.93</td>\n",
       "      <td>0</td>\n",
       "      <td>10350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smle_pzq_es</td>\n",
       "      <td>391422238</td>\n",
       "      <td>21017040798</td>\n",
       "      <td>53.69</td>\n",
       "      <td>0</td>\n",
       "      <td>43602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smor</td>\n",
       "      <td>391422238</td>\n",
       "      <td>12178154968</td>\n",
       "      <td>31.11</td>\n",
       "      <td>0</td>\n",
       "      <td>80622</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample     length        bases   mean  min    max\n",
       "0        smbre  391422238  12929849971  33.03    0  50586\n",
       "1         smeg  391422238   8667326476  22.14    0  34407\n",
       "2  smle_pzq_er  391422238   6624879196  16.93    0  10350\n",
       "3  smle_pzq_es  391422238  21017040798  53.69    0  43602\n",
       "4         smor  391422238  12178154968  31.11    0  80622"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#calculate coverage for a table\n",
    "!echo \"sample,length,bases,mean,min,max\">covs.csv\n",
    "!grep \"total\" *mosdepth.summary.txt | sed 's/:/\\t/' | sed 's/\\t/,/g' | cut -f1,3-7 -d\",\" | sed 's/.mosdepth.summary.txt//' >>covs.csv\n",
    "\n",
    "cov_df=pd.read_csv(\"covs.csv\", sep=\",\")\n",
    "cov_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d4367a81-1675-42bf-a3e3-db05cac1f315",
   "metadata": {},
   "outputs": [],
   "source": [
    "#play around with igv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0694869-8ba5-4996-ae3d-e122538f69e1",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Identify structural variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71d67d7b-66e0-434f-a420-f75d36a7870e",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/cutesv\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/cutesv\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8fd9c2fd-7b35-4641-b475-8e55df79f64d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smle_pzq_es ##################\n",
      "2023-01-18 11:31:03,178 [INFO] Running /master/nplatt/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 --genotype /master/nplatt/sm_nanopore/results/minimap2/smle_pzq_es.bam /master/nplatt/sm_nanopore/data/genome/GCA_000237925.5.fa smle_pzq_es.cutesv.vcf /master/nplatt/sm_nanopore/results/smle_pzq_es\n",
      "2023-01-18 11:31:10,070 [INFO] The total number of chromsomes: 12\n",
      "2023-01-18 11:31:38,245 [INFO] Finished OU426849.1:10000000-10680109.\n",
      "2023-01-18 11:33:46,325 [INFO] Finished HE601624.3:80000000-87984036.\n",
      "2023-01-18 11:34:13,522 [INFO] Finished HE601624.3:30000000-40000000.\n",
      "2023-01-18 11:34:14,461 [INFO] Finished HE601624.3:60000000-70000000.\n",
      "2023-01-18 11:34:14,861 [INFO] Finished HE601624.3:50000000-60000000.\n",
      "2023-01-18 11:34:15,292 [INFO] Finished HE601624.3:40000000-50000000.\n",
      "2023-01-18 11:34:15,396 [INFO] Finished HE601624.3:20000000-30000000.\n",
      "2023-01-18 11:34:16,824 [INFO] Finished HE601624.3:10000000-20000000.\n",
      "2023-01-18 11:34:17,049 [INFO] Finished HE601624.3:70000000-80000000.\n",
      "2023-01-18 11:34:17,800 [INFO] Finished OU426849.1:0-10000000.\n",
      "2023-01-18 11:34:19,942 [INFO] Finished HE601625.3:0-10000000.\n",
      "2023-01-18 11:34:34,885 [INFO] Finished HE601625.3:10000000-20000000.\n",
      "2023-01-18 11:35:10,155 [INFO] Finished HE601624.3:0-10000000.\n",
      "2023-01-18 11:36:26,000 [INFO] Finished OU426850.1:40000000-42949100.\n",
      "2023-01-18 11:39:30,490 [INFO] Finished HE601625.3:40000000-45716228.\n",
      "2023-01-18 11:41:25,195 [INFO] Finished HE601625.3:20000000-30000000.\n",
      "2023-01-18 11:41:56,222 [INFO] Finished OU426850.1:0-10000000.\n",
      "2023-01-18 11:42:09,883 [INFO] Finished OU426850.1:20000000-30000000.\n",
      "2023-01-18 11:42:20,785 [INFO] Finished OU426850.1:30000000-40000000.\n",
      "2023-01-18 11:42:31,292 [INFO] Finished OU426850.1:10000000-20000000.\n",
      "2023-01-18 11:42:41,655 [INFO] Finished HE601626.3:20000000-30000000.\n",
      "2023-01-18 11:44:43,023 [INFO] Finished HE601626.3:0-10000000.\n",
      "2023-01-18 11:44:45,384 [INFO] Finished HE601626.3:30000000-40000000.\n",
      "2023-01-18 11:44:49,712 [INFO] Finished HE601626.3:10000000-20000000.\n",
      "2023-01-18 11:46:12,421 [INFO] Finished HE601626.3:40000000-49793703.\n",
      "2023-01-18 11:47:34,679 [INFO] Finished HE601627.3:40000000-46471565.\n",
      "2023-01-18 11:47:50,871 [INFO] Finished HE601625.3:30000000-40000000.\n",
      "2023-01-18 11:48:14,295 [INFO] Finished HE601628.3:20000000-24128855.\n",
      "2023-01-18 11:48:40,115 [INFO] Finished HE601627.3:0-10000000.\n",
      "2023-01-18 11:49:23,561 [INFO] Finished HE601627.3:10000000-20000000.\n",
      "2023-01-18 11:49:32,025 [INFO] Finished HE601627.3:20000000-30000000.\n",
      "2023-01-18 11:50:03,023 [INFO] Finished HE601627.3:30000000-40000000.\n",
      "2023-01-18 11:50:11,537 [INFO] Finished HE601629.3:20000000-24696071.\n",
      "2023-01-18 11:52:31,369 [INFO] Finished HE601628.3:10000000-20000000.\n",
      "2023-01-18 11:52:33,504 [INFO] Finished HE601612.2:0-26917.\n",
      "2023-01-18 11:52:38,871 [INFO] Finished OU426848.1:30000000-33063208.\n",
      "2023-01-18 11:52:53,278 [INFO] Finished HE601628.3:0-10000000.\n",
      "2023-01-18 11:53:11,865 [INFO] Finished HE601629.3:10000000-20000000.\n",
      "2023-01-18 11:53:32,868 [INFO] Finished HE601629.3:0-10000000.\n",
      "2023-01-18 11:53:35,289 [INFO] Finished OU426848.1:0-10000000.\n",
      "2023-01-18 11:53:36,901 [INFO] Finished OU426848.1:10000000-20000000.\n",
      "2023-01-18 11:53:37,627 [INFO] Finished OU426848.1:20000000-30000000.\n",
      "2023-01-18 11:53:43,843 [INFO] Finished HE601630.3:0-10000000.\n",
      "2023-01-18 11:53:51,094 [INFO] Finished HE601630.3:10000000-19942475.\n",
      "2023-01-18 11:54:25,263 [INFO] Finished HE601631.3:0-5969971.\n",
      "2023-01-18 11:54:25,329 [INFO] Rebuilding signatures of structural variants.\n",
      "2023-01-18 11:56:15,987 [INFO] Clustering structural variants.\n",
      "2023-01-18 11:56:17,909 [INFO] Finished HE601612.2:DEL.\n",
      "2023-01-18 11:56:17,962 [INFO] Finished HE601631.3:DEL.\n",
      "2023-01-18 11:56:18,360 [INFO] Finished OU426849.1:DEL.\n",
      "2023-01-18 11:56:18,622 [INFO] Finished OU426848.1:DEL.\n",
      "2023-01-18 11:56:18,799 [INFO] Finished HE601630.3:DEL.\n",
      "2023-01-18 11:56:19,175 [INFO] Finished HE601629.3:DEL.\n",
      "2023-01-18 11:56:19,185 [INFO] Finished HE601628.3:DEL.\n",
      "2023-01-18 11:56:19,819 [INFO] Finished OU426850.1:DEL.\n",
      "2023-01-18 11:56:19,840 [INFO] Finished HE601612.2:INS.\n",
      "2023-01-18 11:56:20,084 [INFO] Finished HE601627.3:DEL.\n",
      "2023-01-18 11:56:20,261 [INFO] Finished HE601625.3:DEL.\n",
      "2023-01-18 11:56:20,445 [INFO] Finished HE601626.3:DEL.\n",
      "2023-01-18 11:56:21,791 [INFO] Finished HE601631.3:INS.\n",
      "2023-01-18 11:56:22,037 [INFO] Finished HE601624.3:DEL.\n",
      "2023-01-18 11:56:22,241 [INFO] Finished HE601628.3:INS.\n",
      "2023-01-18 11:56:22,295 [INFO] Finished HE601629.3:INS.\n",
      "2023-01-18 11:56:22,403 [INFO] Finished HE601625.3:INS.\n",
      "2023-01-18 11:56:22,448 [INFO] Finished HE601630.3:INS.\n",
      "2023-01-18 11:56:22,468 [INFO] Finished OU426849.1:INS.\n",
      "2023-01-18 11:56:22,555 [INFO] Finished OU426848.1:INS.\n",
      "2023-01-18 11:56:22,626 [INFO] Finished HE601627.3:INS.\n",
      "2023-01-18 11:56:22,841 [INFO] Finished HE601626.3:INS.\n",
      "2023-01-18 11:56:23,023 [INFO] Finished HE601612.2:INV.\n",
      "2023-01-18 11:56:23,719 [INFO] Finished HE601624.3:INS.\n",
      "2023-01-18 11:56:23,801 [INFO] Finished HE601631.3:INV.\n",
      "2023-01-18 11:56:24,047 [INFO] Finished OU426850.1:INS.\n",
      "2023-01-18 11:56:24,217 [INFO] Finished HE601630.3:INV.\n",
      "2023-01-18 11:56:24,223 [INFO] Finished HE601628.3:INV.\n",
      "2023-01-18 11:56:24,337 [INFO] Finished HE601629.3:INV.\n",
      "2023-01-18 11:56:24,422 [INFO] Finished OU426849.1:INV.\n",
      "2023-01-18 11:56:24,499 [INFO] Finished OU426848.1:INV.\n",
      "2023-01-18 11:56:24,849 [INFO] Finished HE601625.3:INV.\n",
      "2023-01-18 11:56:24,864 [INFO] Finished HE601627.3:INV.\n",
      "2023-01-18 11:56:25,029 [INFO] Finished HE601626.3:INV.\n",
      "2023-01-18 11:56:25,060 [INFO] Finished HE601612.2:DUP.\n",
      "2023-01-18 11:56:25,831 [INFO] Finished HE601624.3:INV.\n",
      "2023-01-18 11:56:26,023 [INFO] Finished HE601631.3:DUP.\n",
      "2023-01-18 11:56:26,031 [INFO] Finished OU426850.1:INV.\n",
      "2023-01-18 11:56:26,091 [INFO] Finished HE601612.2-HE601624.3:TRA/BND.\n",
      "2023-01-18 11:56:26,098 [INFO] Finished HE601612.2-HE601625.3:TRA/BND.\n",
      "2023-01-18 11:56:26,158 [INFO] Finished HE601612.2-HE601626.3:TRA/BND.\n",
      "2023-01-18 11:56:26,160 [INFO] Finished HE601628.3:DUP.\n",
      "2023-01-18 11:56:26,164 [INFO] Finished HE601612.2-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:26,224 [INFO] Finished HE601612.2-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:26,226 [INFO] Finished HE601612.2-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:26,230 [INFO] Finished HE601612.2-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:26,290 [INFO] Finished HE601612.2-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:26,291 [INFO] Finished HE601612.2-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:26,297 [INFO] Finished HE601612.2-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:26,335 [INFO] Finished HE601629.3:DUP.\n",
      "2023-01-18 11:56:26,356 [INFO] Finished HE601612.2-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:26,436 [INFO] Finished OU426849.1:DUP.\n",
      "2023-01-18 11:56:26,479 [INFO] Finished HE601630.3:DUP.\n",
      "2023-01-18 11:56:26,638 [INFO] Finished HE601625.3:DUP.\n",
      "2023-01-18 11:56:26,668 [INFO] Finished OU426848.1:DUP.\n",
      "2023-01-18 11:56:26,695 [INFO] Finished HE601627.3:DUP.\n",
      "2023-01-18 11:56:26,747 [INFO] Finished HE601624.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:26,818 [INFO] Finished HE601626.3:DUP.\n",
      "2023-01-18 11:56:26,852 [INFO] Finished HE601624.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:26,899 [INFO] Finished HE601624.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:26,960 [INFO] Finished HE601624.3-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:26,976 [INFO] Finished HE601624.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,004 [INFO] Finished HE601625.3-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:27,018 [INFO] Finished HE601624.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,073 [INFO] Finished HE601624.3-HE601625.3:TRA/BND.\n",
      "2023-01-18 11:56:27,094 [INFO] Finished HE601625.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,119 [INFO] Finished HE601624.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,123 [INFO] Finished HE601625.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,146 [INFO] Finished HE601625.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:27,152 [INFO] Finished HE601625.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,220 [INFO] Finished HE601624.3-HE601626.3:TRA/BND.\n",
      "2023-01-18 11:56:27,233 [INFO] Finished HE601625.3-HE601626.3:TRA/BND.\n",
      "2023-01-18 11:56:27,294 [INFO] Finished HE601625.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,314 [INFO] Finished HE601626.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,360 [INFO] Finished HE601626.3-HE601627.3:TRA/BND.\n",
      "2023-01-18 11:56:27,385 [INFO] Finished HE601626.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:27,385 [INFO] Finished HE601625.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,399 [INFO] Finished HE601625.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,427 [INFO] Finished HE601626.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,431 [INFO] Finished HE601626.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,469 [INFO] Finished HE601626.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,489 [INFO] Finished HE601624.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,576 [INFO] Finished HE601627.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,582 [INFO] Finished HE601627.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,590 [INFO] Finished HE601627.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,658 [INFO] Finished HE601627.3-HE601628.3:TRA/BND.\n",
      "2023-01-18 11:56:27,709 [INFO] Finished HE601626.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,712 [INFO] Finished HE601624.3:DUP.\n",
      "2023-01-18 11:56:27,727 [INFO] Finished HE601628.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,737 [INFO] Finished HE601628.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,744 [INFO] Finished HE601627.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,755 [INFO] Finished HE601628.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,766 [INFO] Finished HE601628.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,782 [INFO] Finished HE601627.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,792 [INFO] Finished HE601629.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,799 [INFO] Finished HE601628.3-HE601629.3:TRA/BND.\n",
      "2023-01-18 11:56:27,806 [INFO] Finished HE601626.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,819 [INFO] Finished HE601627.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,832 [INFO] Finished HE601630.3-HE601631.3:TRA/BND.\n",
      "2023-01-18 11:56:27,847 [INFO] Finished HE601630.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,856 [INFO] Finished HE601630.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,860 [INFO] Finished HE601629.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,864 [INFO] Finished HE601630.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,874 [INFO] Finished HE601631.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,896 [INFO] Finished HE601629.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,918 [INFO] Finished HE601631.3-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,931 [INFO] Finished HE601629.3-HE601630.3:TRA/BND.\n",
      "2023-01-18 11:56:27,949 [INFO] Finished OU426848.1-OU426849.1:TRA/BND.\n",
      "2023-01-18 11:56:27,949 [INFO] Finished HE601629.3-OU426848.1:TRA/BND.\n",
      "2023-01-18 11:56:27,950 [INFO] Finished OU426849.1-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:27,977 [INFO] Finished HE601631.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:28,066 [INFO] Finished OU426848.1-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:28,132 [INFO] Finished OU426850.1:DUP.\n",
      "2023-01-18 11:56:28,176 [INFO] Finished HE601628.3-OU426850.1:TRA/BND.\n",
      "2023-01-18 11:56:28,282 [INFO] Writing to your output file.\n",
      "2023-01-18 11:56:28,304 [INFO] Loading reference genome...\n",
      "2023-01-18 11:56:51,966 [INFO] Cleaning temporary files.\n",
      "2023-01-18 11:56:52,691 [INFO] Finished in 1549.51 seconds.\n"
     ]
    }
   ],
   "source": [
    "#sv call\n",
    "# for sample in samples:\n",
    "for sample in [\"smle_pzq_es\"]:\n",
    "\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    bam      = \"{}/results/minimap2/{}.bam\".format(proj_dir, sample)\n",
    "    query_fq = \"{}/results/nanofilt/{}.filtered.fastq.gz\".format(proj_dir, sample)\n",
    "    out_dir  = \"{}/{}\".format(os.getcwd(), sample)\n",
    "    \n",
    "    Path(\"{}\".format(out_dir)).mkdir(parents=True, exist_ok=True)\n",
    "    !~/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --report_readid --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 --genotype {bam} {ref_fas} {sample}.cutesv.vcf {out_dir}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9f093361-5581-4b6f-88c4-4b89f448e1bc",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smor ##################\n",
      "2023-08-02 16:33:33,389 [INFO] Running /master/nplatt/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --report_readid --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 /master/nplatt/sch_man_ont/results/minimap2/smor.bam /master/nplatt/sch_man_ont/data/genome/GCA_000237925.5.fa smor.cutesv.vcf /master/nplatt/sch_man_ont/results/cutesv/smor\n",
      "2023-08-02 16:33:33,588 [INFO] The total number of chromsomes: 12\n",
      "2023-08-02 16:33:35,747 [INFO] Finished OU426849.1:10000000-10680109.\n",
      "2023-08-02 16:33:46,133 [INFO] Finished HE601624.3:80000000-87984036.\n",
      "2023-08-02 16:33:47,986 [INFO] Finished HE601624.3:50000000-60000000.\n",
      "2023-08-02 16:33:48,219 [INFO] Finished HE601624.3:30000000-40000000.\n",
      "2023-08-02 16:33:48,219 [INFO] Finished OU426849.1:0-10000000.\n",
      "2023-08-02 16:33:48,372 [INFO] Finished HE601624.3:60000000-70000000.\n",
      "2023-08-02 16:33:48,374 [INFO] Finished HE601624.3:10000000-20000000.\n",
      "2023-08-02 16:33:48,487 [INFO] Finished HE601624.3:40000000-50000000.\n",
      "2023-08-02 16:33:48,544 [INFO] Finished HE601624.3:20000000-30000000.\n",
      "2023-08-02 16:33:48,605 [INFO] Finished HE601624.3:70000000-80000000.\n",
      "2023-08-02 16:33:49,155 [INFO] Finished HE601625.3:0-10000000.\n",
      "2023-08-02 16:33:49,155 [INFO] Finished HE601625.3:10000000-20000000.\n",
      "2023-08-02 16:33:50,250 [INFO] Finished HE601624.3:0-10000000.\n",
      "2023-08-02 16:33:55,137 [INFO] Finished OU426850.1:40000000-42949100.\n",
      "2023-08-02 16:33:58,993 [INFO] Finished HE601625.3:40000000-45716228.\n",
      "2023-08-02 16:34:01,686 [INFO] Finished HE601625.3:20000000-30000000.\n",
      "2023-08-02 16:34:02,607 [INFO] Finished OU426850.1:0-10000000.\n",
      "2023-08-02 16:34:02,714 [INFO] Finished OU426850.1:20000000-30000000.\n",
      "2023-08-02 16:34:03,125 [INFO] Finished OU426850.1:30000000-40000000.\n",
      "2023-08-02 16:34:03,287 [INFO] Finished OU426850.1:10000000-20000000.\n",
      "2023-08-02 16:34:03,424 [INFO] Finished HE601626.3:20000000-30000000.\n",
      "2023-08-02 16:34:04,519 [INFO] Finished HE601626.3:30000000-40000000.\n",
      "2023-08-02 16:34:05,921 [INFO] Finished HE601626.3:0-10000000.\n",
      "2023-08-02 16:34:08,221 [INFO] Finished HE601626.3:10000000-20000000.\n",
      "2023-08-02 16:34:08,243 [INFO] Finished HE601625.3:30000000-40000000.\n",
      "2023-08-02 16:34:08,898 [INFO] Finished HE601626.3:40000000-49793703.\n",
      "2023-08-02 16:34:11,485 [INFO] Finished HE601628.3:20000000-24128855.\n",
      "2023-08-02 16:34:12,445 [INFO] Finished HE601627.3:0-10000000.\n",
      "2023-08-02 16:34:13,367 [INFO] Finished HE601627.3:40000000-46471565.\n",
      "2023-08-02 16:34:14,559 [INFO] Finished HE601629.3:20000000-24696071.\n",
      "2023-08-02 16:34:14,791 [INFO] Finished HE601627.3:10000000-20000000.\n",
      "2023-08-02 16:34:16,373 [INFO] Finished HE601627.3:20000000-30000000.\n",
      "2023-08-02 16:34:16,818 [INFO] Finished HE601627.3:30000000-40000000.\n",
      "2023-08-02 16:34:17,677 [INFO] Finished HE601628.3:10000000-20000000.\n",
      "2023-08-02 16:34:17,817 [INFO] Finished HE601628.3:0-10000000.\n",
      "2023-08-02 16:34:18,476 [INFO] Finished OU426848.1:30000000-33063208.\n",
      "2023-08-02 16:34:19,659 [INFO] Finished HE601629.3:0-10000000.\n",
      "2023-08-02 16:34:19,859 [INFO] Finished OU426848.1:0-10000000.\n",
      "2023-08-02 16:34:20,267 [INFO] Finished HE601612.2:0-26917.\n",
      "2023-08-02 16:34:20,727 [INFO] Finished OU426848.1:10000000-20000000.\n",
      "2023-08-02 16:34:20,809 [INFO] Finished HE601630.3:0-10000000.\n",
      "2023-08-02 16:34:20,990 [INFO] Finished OU426848.1:20000000-30000000.\n",
      "2023-08-02 16:34:21,047 [INFO] Finished HE601629.3:10000000-20000000.\n",
      "2023-08-02 16:34:23,212 [INFO] Finished HE601630.3:10000000-19942475.\n",
      "2023-08-02 16:34:23,558 [INFO] Finished HE601631.3:0-5969971.\n",
      "2023-08-02 16:34:23,680 [INFO] Rebuilding signatures of structural variants.\n",
      "2023-08-02 16:34:43,810 [INFO] Clustering structural variants.\n",
      "2023-08-02 16:34:44,153 [INFO] Finished HE601612.2:DEL.\n",
      "2023-08-02 16:34:44,172 [INFO] Finished HE601631.3:DEL.\n",
      "2023-08-02 16:34:44,176 [INFO] Finished OU426848.1:DEL.\n",
      "2023-08-02 16:34:44,192 [INFO] Finished OU426849.1:DEL.\n",
      "2023-08-02 16:34:44,255 [INFO] Finished HE601630.3:DEL.\n",
      "2023-08-02 16:34:44,326 [INFO] Finished HE601629.3:DEL.\n",
      "2023-08-02 16:34:44,357 [INFO] Finished OU426850.1:DEL.\n",
      "2023-08-02 16:34:44,388 [INFO] Finished HE601628.3:DEL.\n",
      "2023-08-02 16:34:44,426 [INFO] Finished HE601627.3:DEL.\n",
      "2023-08-02 16:34:44,469 [INFO] Finished HE601625.3:DEL.\n",
      "2023-08-02 16:34:44,507 [INFO] Finished HE601626.3:DEL.\n",
      "2023-08-02 16:34:44,655 [INFO] Finished HE601624.3:DEL.\n",
      "2023-08-02 16:34:44,666 [INFO] Finished HE601612.2:INS.\n",
      "2023-08-02 16:34:44,685 [INFO] Finished HE601612.2:INV.\n",
      "2023-08-02 16:34:44,709 [INFO] Finished HE601624.3:INV.\n",
      "2023-08-02 16:34:44,730 [INFO] Finished HE601625.3:INV.\n",
      "2023-08-02 16:34:44,752 [INFO] Finished HE601626.3:INV.\n",
      "2023-08-02 16:34:44,772 [INFO] Finished HE601627.3:INV.\n",
      "2023-08-02 16:34:44,791 [INFO] Finished HE601628.3:INV.\n",
      "2023-08-02 16:34:44,810 [INFO] Finished HE601629.3:INV.\n",
      "2023-08-02 16:34:44,828 [INFO] Finished HE601630.3:INV.\n",
      "2023-08-02 16:34:44,850 [INFO] Finished HE601631.3:INV.\n",
      "2023-08-02 16:34:44,868 [INFO] Finished OU426848.1:INV.\n",
      "2023-08-02 16:34:44,884 [INFO] Finished OU426849.1:INV.\n",
      "2023-08-02 16:34:44,903 [INFO] Finished OU426850.1:INV.\n",
      "2023-08-02 16:34:44,915 [INFO] Finished HE601612.2:DUP.\n",
      "2023-08-02 16:34:44,927 [INFO] Finished HE601624.3:DUP.\n",
      "2023-08-02 16:34:44,935 [INFO] Finished HE601631.3:INS.\n",
      "2023-08-02 16:34:44,939 [INFO] Finished HE601625.3:DUP.\n",
      "2023-08-02 16:34:44,950 [INFO] Finished HE601627.3:DUP.\n",
      "2023-08-02 16:34:44,958 [INFO] Finished HE601626.3:DUP.\n",
      "2023-08-02 16:34:44,961 [INFO] Finished HE601628.3:DUP.\n",
      "2023-08-02 16:34:44,970 [INFO] Finished HE601629.3:DUP.\n",
      "2023-08-02 16:34:44,972 [INFO] Finished HE601630.3:DUP.\n",
      "2023-08-02 16:34:44,982 [INFO] Finished HE601631.3:DUP.\n",
      "2023-08-02 16:34:44,983 [INFO] Finished OU426848.1:DUP.\n",
      "2023-08-02 16:34:44,987 [INFO] Finished HE601630.3:INS.\n",
      "2023-08-02 16:34:44,993 [INFO] Finished OU426849.1:DUP.\n",
      "2023-08-02 16:34:44,994 [INFO] Finished OU426850.1:DUP.\n",
      "2023-08-02 16:34:45,010 [INFO] Finished OU426848.1:INS.\n",
      "2023-08-02 16:34:45,029 [INFO] Finished HE601629.3:INS.\n",
      "2023-08-02 16:34:45,030 [INFO] Finished HE601625.3:INS.\n",
      "2023-08-02 16:34:45,037 [INFO] Finished HE601612.2-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:34:45,037 [INFO] Finished HE601612.2-HE601624.3:TRA/BND.\n",
      "2023-08-02 16:34:45,039 [INFO] Finished HE601627.3:INS.\n",
      "2023-08-02 16:34:45,040 [INFO] Finished HE601612.2-HE601625.3:TRA/BND.\n",
      "2023-08-02 16:34:45,048 [INFO] Finished OU426849.1:INS.\n",
      "2023-08-02 16:34:45,058 [INFO] Finished HE601612.2-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:34:45,061 [INFO] Finished HE601628.3:INS.\n",
      "2023-08-02 16:34:45,079 [INFO] Finished HE601612.2-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:34:45,079 [INFO] Finished HE601612.2-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:34:45,082 [INFO] Finished HE601612.2-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:34:45,084 [INFO] Finished HE601612.2-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:34:45,085 [INFO] Finished HE601612.2-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:34:45,086 [INFO] Finished HE601626.3:INS.\n",
      "2023-08-02 16:34:45,089 [INFO] Finished HE601612.2-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:34:45,094 [INFO] Finished HE601612.2-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,108 [INFO] Finished HE601624.3-HE601625.3:TRA/BND.\n",
      "2023-08-02 16:34:45,120 [INFO] Finished HE601624.3-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:34:45,125 [INFO] Finished HE601624.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:34:45,128 [INFO] Finished HE601624.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:34:45,131 [INFO] Finished HE601624.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:34:45,131 [INFO] Finished HE601624.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:34:45,133 [INFO] Finished HE601624.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:34:45,137 [INFO] Finished HE601624.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:34:45,139 [INFO] Finished HE601624.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:34:45,143 [INFO] Finished HE601624.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,157 [INFO] Finished HE601625.3-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:34:45,167 [INFO] Finished HE601625.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:34:45,169 [INFO] Finished HE601625.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:34:45,175 [INFO] Finished HE601625.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:34:45,178 [INFO] Finished HE601625.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:34:45,179 [INFO] Finished HE601625.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:34:45,179 [INFO] Finished HE601625.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:34:45,184 [INFO] Finished HE601625.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:34:45,187 [INFO] Finished HE601625.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,197 [INFO] Finished HE601626.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:34:45,202 [INFO] Finished HE601626.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:34:45,213 [INFO] Finished HE601626.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:34:45,215 [INFO] Finished HE601626.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:34:45,226 [INFO] Finished HE601626.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:34:45,226 [INFO] Finished HE601626.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:34:45,227 [INFO] Finished HE601626.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:34:45,229 [INFO] Finished HE601626.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,230 [INFO] Finished HE601627.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:34:45,233 [INFO] Finished HE601627.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:34:45,242 [INFO] Finished HE601627.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:34:45,245 [INFO] Finished HE601627.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:34:45,255 [INFO] Finished HE601627.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:34:45,258 [INFO] Finished HE601624.3:INS.\n",
      "2023-08-02 16:34:45,261 [INFO] Finished HE601627.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:34:45,272 [INFO] Finished HE601628.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:34:45,273 [INFO] Finished HE601627.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,274 [INFO] Finished HE601628.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:34:45,275 [INFO] Finished HE601628.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:34:45,276 [INFO] Finished HE601628.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:34:45,279 [INFO] Finished HE601628.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:34:45,288 [INFO] Finished HE601628.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,290 [INFO] Finished HE601629.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:34:45,300 [INFO] Finished HE601629.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:34:45,306 [INFO] Finished HE601629.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:34:45,313 [INFO] Finished HE601629.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:34:45,318 [INFO] Finished HE601629.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,320 [INFO] Finished HE601630.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:34:45,320 [INFO] Finished HE601630.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:34:45,321 [INFO] Finished HE601630.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:34:45,322 [INFO] Finished HE601630.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,326 [INFO] Finished HE601631.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:34:45,333 [INFO] Finished HE601631.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:34:45,337 [INFO] Finished HE601631.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,345 [INFO] Finished OU426848.1-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:34:45,351 [INFO] Finished OU426848.1-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,358 [INFO] Finished OU426849.1-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:34:45,379 [INFO] Finished OU426850.1:INS.\n",
      "2023-08-02 16:34:45,441 [INFO] Writing to your output file.\n",
      "2023-08-02 16:34:45,454 [INFO] Loading reference genome...\n",
      "2023-08-02 16:34:49,246 [INFO] Cleaning temporary files.\n",
      "2023-08-02 16:34:49,779 [INFO] Finished in 76.39 seconds.\n",
      "###################### smle_pzq_es ##################\n",
      "2023-08-02 16:34:50,519 [INFO] Running /master/nplatt/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --report_readid --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 /master/nplatt/sch_man_ont/results/minimap2/smle_pzq_es.bam /master/nplatt/sch_man_ont/data/genome/GCA_000237925.5.fa smle_pzq_es.cutesv.vcf /master/nplatt/sch_man_ont/results/cutesv/smle_pzq_es\n",
      "2023-08-02 16:34:50,758 [INFO] The total number of chromsomes: 12\n",
      "2023-08-02 16:34:54,058 [INFO] Finished OU426849.1:10000000-10680109.\n",
      "2023-08-02 16:35:13,591 [INFO] Finished HE601624.3:80000000-87984036.\n",
      "2023-08-02 16:35:17,732 [INFO] Finished HE601624.3:30000000-40000000.\n",
      "2023-08-02 16:35:17,821 [INFO] Finished HE601624.3:40000000-50000000.\n",
      "2023-08-02 16:35:17,964 [INFO] Finished HE601624.3:50000000-60000000.\n",
      "2023-08-02 16:35:18,245 [INFO] Finished HE601624.3:20000000-30000000.\n",
      "2023-08-02 16:35:18,269 [INFO] Finished HE601624.3:60000000-70000000.\n",
      "2023-08-02 16:35:18,320 [INFO] Finished HE601624.3:10000000-20000000.\n",
      "2023-08-02 16:35:18,807 [INFO] Finished HE601625.3:0-10000000.\n",
      "2023-08-02 16:35:18,967 [INFO] Finished HE601624.3:70000000-80000000.\n",
      "2023-08-02 16:35:19,019 [INFO] Finished OU426849.1:0-10000000.\n",
      "2023-08-02 16:35:20,716 [INFO] Finished HE601624.3:0-10000000.\n",
      "2023-08-02 16:35:21,040 [INFO] Finished HE601625.3:10000000-20000000.\n",
      "2023-08-02 16:35:28,685 [INFO] Finished OU426850.1:40000000-42949100.\n",
      "2023-08-02 16:35:34,876 [INFO] Finished HE601625.3:40000000-45716228.\n",
      "2023-08-02 16:35:40,872 [INFO] Finished HE601625.3:20000000-30000000.\n",
      "2023-08-02 16:35:43,805 [INFO] Finished OU426850.1:0-10000000.\n",
      "2023-08-02 16:35:44,063 [INFO] Finished OU426850.1:30000000-40000000.\n",
      "2023-08-02 16:35:44,385 [INFO] Finished OU426850.1:20000000-30000000.\n",
      "2023-08-02 16:35:44,439 [INFO] Finished OU426850.1:10000000-20000000.\n",
      "2023-08-02 16:35:45,349 [INFO] Finished HE601626.3:20000000-30000000.\n",
      "2023-08-02 16:35:46,331 [INFO] Finished HE601626.3:0-10000000.\n",
      "2023-08-02 16:35:46,541 [INFO] Finished HE601626.3:30000000-40000000.\n",
      "2023-08-02 16:35:49,161 [INFO] Finished HE601626.3:10000000-20000000.\n",
      "2023-08-02 16:35:49,747 [INFO] Finished HE601625.3:30000000-40000000.\n",
      "2023-08-02 16:35:52,440 [INFO] Finished HE601626.3:40000000-49793703.\n",
      "2023-08-02 16:35:56,527 [INFO] Finished HE601628.3:20000000-24128855.\n",
      "2023-08-02 16:35:58,487 [INFO] Finished HE601627.3:0-10000000.\n",
      "2023-08-02 16:36:00,204 [INFO] Finished HE601627.3:40000000-46471565.\n",
      "2023-08-02 16:36:00,428 [INFO] Finished HE601629.3:20000000-24696071.\n",
      "2023-08-02 16:36:04,762 [INFO] Finished HE601627.3:10000000-20000000.\n",
      "2023-08-02 16:36:07,360 [INFO] Finished HE601627.3:20000000-30000000.\n",
      "2023-08-02 16:36:08,043 [INFO] Finished HE601627.3:30000000-40000000.\n",
      "2023-08-02 16:36:09,274 [INFO] Finished HE601628.3:0-10000000.\n",
      "2023-08-02 16:36:10,613 [INFO] Finished HE601628.3:10000000-20000000.\n",
      "2023-08-02 16:36:11,373 [INFO] Finished HE601612.2:0-26917.\n",
      "2023-08-02 16:36:12,420 [INFO] Finished OU426848.1:30000000-33063208.\n",
      "2023-08-02 16:36:12,686 [INFO] Finished HE601629.3:0-10000000.\n",
      "2023-08-02 16:36:13,625 [INFO] Finished HE601629.3:10000000-20000000.\n",
      "2023-08-02 16:36:14,823 [INFO] Finished HE601631.3:0-5969971.\n",
      "2023-08-02 16:36:16,198 [INFO] Finished HE601630.3:0-10000000.\n",
      "2023-08-02 16:36:17,282 [INFO] Finished OU426848.1:10000000-20000000.\n",
      "2023-08-02 16:36:18,047 [INFO] Finished OU426848.1:0-10000000.\n",
      "2023-08-02 16:36:21,163 [INFO] Finished HE601630.3:10000000-19942475.\n",
      "2023-08-02 16:36:21,189 [INFO] Finished OU426848.1:20000000-30000000.\n",
      "2023-08-02 16:36:21,281 [INFO] Rebuilding signatures of structural variants.\n",
      "2023-08-02 16:37:07,494 [INFO] Clustering structural variants.\n",
      "2023-08-02 16:37:08,202 [INFO] Finished HE601612.2:DEL.\n",
      "2023-08-02 16:37:08,256 [INFO] Finished HE601631.3:DEL.\n",
      "2023-08-02 16:37:08,323 [INFO] Finished OU426849.1:DEL.\n",
      "2023-08-02 16:37:08,350 [INFO] Finished OU426848.1:DEL.\n",
      "2023-08-02 16:37:08,447 [INFO] Finished HE601630.3:DEL.\n",
      "2023-08-02 16:37:08,594 [INFO] Finished HE601629.3:DEL.\n",
      "2023-08-02 16:37:08,625 [INFO] Finished HE601628.3:DEL.\n",
      "2023-08-02 16:37:08,654 [INFO] Finished OU426850.1:DEL.\n",
      "2023-08-02 16:37:08,722 [INFO] Finished HE601627.3:DEL.\n",
      "2023-08-02 16:37:08,822 [INFO] Finished HE601625.3:DEL.\n",
      "2023-08-02 16:37:08,949 [INFO] Finished HE601626.3:DEL.\n",
      "2023-08-02 16:37:08,979 [INFO] Finished HE601612.2:INS.\n",
      "2023-08-02 16:37:09,148 [INFO] Finished HE601624.3:DEL.\n",
      "2023-08-02 16:37:09,190 [INFO] Finished HE601612.2:INV.\n",
      "2023-08-02 16:37:09,226 [INFO] Finished HE601624.3:INV.\n",
      "2023-08-02 16:37:09,260 [INFO] Finished HE601625.3:INV.\n",
      "2023-08-02 16:37:09,292 [INFO] Finished HE601626.3:INV.\n",
      "2023-08-02 16:37:09,323 [INFO] Finished HE601627.3:INV.\n",
      "2023-08-02 16:37:09,353 [INFO] Finished HE601628.3:INV.\n",
      "2023-08-02 16:37:09,383 [INFO] Finished HE601629.3:INV.\n",
      "2023-08-02 16:37:09,412 [INFO] Finished HE601630.3:INV.\n",
      "2023-08-02 16:37:09,440 [INFO] Finished HE601631.3:INV.\n",
      "2023-08-02 16:37:09,469 [INFO] Finished OU426848.1:INV.\n",
      "2023-08-02 16:37:09,488 [INFO] Finished HE601631.3:INS.\n",
      "2023-08-02 16:37:09,497 [INFO] Finished OU426849.1:INV.\n",
      "2023-08-02 16:37:09,509 [INFO] Finished HE601612.2:DUP.\n",
      "2023-08-02 16:37:09,521 [INFO] Finished OU426850.1:INV.\n",
      "2023-08-02 16:37:09,522 [INFO] Finished HE601624.3:DUP.\n",
      "2023-08-02 16:37:09,536 [INFO] Finished HE601625.3:DUP.\n",
      "2023-08-02 16:37:09,538 [INFO] Finished HE601626.3:DUP.\n",
      "2023-08-02 16:37:09,547 [INFO] Finished HE601627.3:DUP.\n",
      "2023-08-02 16:37:09,550 [INFO] Finished HE601628.3:DUP.\n",
      "2023-08-02 16:37:09,558 [INFO] Finished HE601629.3:DUP.\n",
      "2023-08-02 16:37:09,560 [INFO] Finished HE601630.3:DUP.\n",
      "2023-08-02 16:37:09,567 [INFO] Finished HE601627.3:INS.\n",
      "2023-08-02 16:37:09,568 [INFO] Finished HE601631.3:DUP.\n",
      "2023-08-02 16:37:09,571 [INFO] Finished OU426848.1:DUP.\n",
      "2023-08-02 16:37:09,579 [INFO] Finished OU426849.1:DUP.\n",
      "2023-08-02 16:37:09,582 [INFO] Finished HE601625.3:INS.\n",
      "2023-08-02 16:37:09,583 [INFO] Finished OU426850.1:DUP.\n",
      "2023-08-02 16:37:09,589 [INFO] Finished HE601630.3:INS.\n",
      "2023-08-02 16:37:09,645 [INFO] Finished HE601612.2-HE601624.3:TRA/BND.\n",
      "2023-08-02 16:37:09,645 [INFO] Finished HE601629.3:INS.\n",
      "2023-08-02 16:37:09,649 [INFO] Finished HE601612.2-HE601625.3:TRA/BND.\n",
      "2023-08-02 16:37:09,649 [INFO] Finished HE601628.3:INS.\n",
      "2023-08-02 16:37:09,651 [INFO] Finished HE601612.2-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:37:09,658 [INFO] Finished HE601612.2-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:37:09,661 [INFO] Finished HE601626.3:INS.\n",
      "2023-08-02 16:37:09,669 [INFO] Finished HE601624.3:INS.\n",
      "2023-08-02 16:37:09,687 [INFO] Finished OU426848.1:INS.\n",
      "2023-08-02 16:37:09,688 [INFO] Finished HE601612.2-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:37:09,714 [INFO] Finished HE601612.2-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:09,716 [INFO] Finished HE601612.2-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:09,718 [INFO] Finished HE601612.2-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:09,718 [INFO] Finished HE601612.2-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:09,725 [INFO] Finished HE601612.2-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:09,725 [INFO] Finished HE601612.2-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:09,742 [INFO] Finished HE601624.3-HE601625.3:TRA/BND.\n",
      "2023-08-02 16:37:09,756 [INFO] Finished HE601624.3-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:37:09,759 [INFO] Finished HE601624.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:37:09,761 [INFO] Finished HE601624.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:37:09,768 [INFO] Finished OU426849.1:INS.\n",
      "2023-08-02 16:37:09,785 [INFO] Finished HE601624.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:09,786 [INFO] Finished HE601624.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:09,787 [INFO] Finished HE601624.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:09,788 [INFO] Finished HE601624.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:09,794 [INFO] Finished HE601624.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:09,797 [INFO] Finished HE601624.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:09,813 [INFO] Finished HE601625.3-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:37:09,826 [INFO] Finished HE601625.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:37:09,827 [INFO] Finished HE601625.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:37:09,830 [INFO] Finished HE601625.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:09,837 [INFO] Finished HE601625.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:09,852 [INFO] Finished HE601625.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:09,853 [INFO] Finished HE601625.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:09,855 [INFO] Finished HE601625.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:09,858 [INFO] Finished HE601625.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:09,865 [INFO] Finished HE601626.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:37:09,867 [INFO] Finished HE601626.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:37:09,883 [INFO] Finished HE601626.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:09,894 [INFO] Finished HE601626.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:09,897 [INFO] Finished HE601626.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:09,898 [INFO] Finished HE601626.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:09,906 [INFO] Finished HE601626.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:09,920 [INFO] Finished HE601627.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:37:09,923 [INFO] Finished HE601626.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:09,924 [INFO] Finished HE601627.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:09,926 [INFO] Finished HE601627.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:09,933 [INFO] Finished HE601627.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:09,934 [INFO] Finished HE601627.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:09,951 [INFO] Finished HE601627.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:09,961 [INFO] Finished HE601627.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:09,965 [INFO] Finished HE601628.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:09,965 [INFO] Finished HE601628.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:09,973 [INFO] Finished HE601628.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:09,987 [INFO] Finished HE601628.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:09,990 [INFO] Finished HE601628.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:09,993 [INFO] Finished HE601628.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:09,993 [INFO] Finished HE601629.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:10,000 [INFO] Finished HE601629.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:10,001 [INFO] Finished HE601629.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:10,018 [INFO] Finished HE601629.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:10,028 [INFO] Finished HE601629.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:10,033 [INFO] Finished HE601630.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:10,034 [INFO] Finished HE601630.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:10,042 [INFO] Finished HE601630.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:10,054 [INFO] Finished HE601630.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:10,058 [INFO] Finished HE601631.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:10,061 [INFO] Finished HE601631.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:10,061 [INFO] Finished HE601631.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:10,067 [INFO] Finished OU426848.1-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:10,068 [INFO] Finished OU426848.1-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:10,075 [INFO] Finished OU426850.1:INS.\n",
      "2023-08-02 16:37:10,085 [INFO] Finished OU426849.1-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:10,130 [INFO] Writing to your output file.\n",
      "2023-08-02 16:37:10,149 [INFO] Loading reference genome...\n",
      "2023-08-02 16:37:13,071 [INFO] Cleaning temporary files.\n",
      "2023-08-02 16:37:13,692 [INFO] Finished in 143.17 seconds.\n",
      "###################### smle_pzq_er ##################\n",
      "2023-08-02 16:37:14,452 [INFO] Running /master/nplatt/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --report_readid --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 /master/nplatt/sch_man_ont/results/minimap2/smle_pzq_er.bam /master/nplatt/sch_man_ont/data/genome/GCA_000237925.5.fa smle_pzq_er.cutesv.vcf /master/nplatt/sch_man_ont/results/cutesv/smle_pzq_er\n",
      "2023-08-02 16:37:14,643 [INFO] The total number of chromsomes: 12\n",
      "2023-08-02 16:37:16,452 [INFO] Finished OU426849.1:10000000-10680109.\n",
      "2023-08-02 16:37:23,000 [INFO] Finished HE601624.3:80000000-87984036.\n",
      "2023-08-02 16:37:24,096 [INFO] Finished HE601624.3:10000000-20000000.\n",
      "2023-08-02 16:37:24,158 [INFO] Finished HE601624.3:50000000-60000000.\n",
      "2023-08-02 16:37:24,206 [INFO] Finished OU426849.1:0-10000000.\n",
      "2023-08-02 16:37:24,371 [INFO] Finished HE601625.3:0-10000000.\n",
      "2023-08-02 16:37:24,441 [INFO] Finished HE601624.3:20000000-30000000.\n",
      "2023-08-02 16:37:24,463 [INFO] Finished HE601624.3:60000000-70000000.\n",
      "2023-08-02 16:37:24,506 [INFO] Finished HE601624.3:40000000-50000000.\n",
      "2023-08-02 16:37:24,553 [INFO] Finished HE601624.3:30000000-40000000.\n",
      "2023-08-02 16:37:24,762 [INFO] Finished HE601624.3:70000000-80000000.\n",
      "2023-08-02 16:37:25,031 [INFO] Finished HE601624.3:0-10000000.\n",
      "2023-08-02 16:37:25,964 [INFO] Finished HE601625.3:10000000-20000000.\n",
      "2023-08-02 16:37:27,639 [INFO] Finished OU426850.1:40000000-42949100.\n",
      "2023-08-02 16:37:29,306 [INFO] Finished HE601625.3:40000000-45716228.\n",
      "2023-08-02 16:37:31,096 [INFO] Finished HE601625.3:20000000-30000000.\n",
      "2023-08-02 16:37:32,142 [INFO] Finished OU426850.1:10000000-20000000.\n",
      "2023-08-02 16:37:32,176 [INFO] Finished OU426850.1:0-10000000.\n",
      "2023-08-02 16:37:32,291 [INFO] Finished OU426850.1:30000000-40000000.\n",
      "2023-08-02 16:37:32,505 [INFO] Finished HE601626.3:0-10000000.\n",
      "2023-08-02 16:37:32,599 [INFO] Finished OU426850.1:20000000-30000000.\n",
      "2023-08-02 16:37:32,902 [INFO] Finished HE601626.3:20000000-30000000.\n",
      "2023-08-02 16:37:33,609 [INFO] Finished HE601626.3:30000000-40000000.\n",
      "2023-08-02 16:37:34,003 [INFO] Finished HE601626.3:10000000-20000000.\n",
      "2023-08-02 16:37:34,099 [INFO] Finished HE601625.3:30000000-40000000.\n",
      "2023-08-02 16:37:35,319 [INFO] Finished HE601626.3:40000000-49793703.\n",
      "2023-08-02 16:37:36,654 [INFO] Finished HE601628.3:20000000-24128855.\n",
      "2023-08-02 16:37:36,859 [INFO] Finished HE601627.3:0-10000000.\n",
      "2023-08-02 16:37:37,055 [INFO] Finished HE601627.3:40000000-46471565.\n",
      "2023-08-02 16:37:37,678 [INFO] Finished HE601629.3:20000000-24696071.\n",
      "2023-08-02 16:37:38,028 [INFO] Finished HE601627.3:10000000-20000000.\n",
      "2023-08-02 16:37:39,241 [INFO] Finished HE601627.3:30000000-40000000.\n",
      "2023-08-02 16:37:39,323 [INFO] Finished HE601627.3:20000000-30000000.\n",
      "2023-08-02 16:37:39,810 [INFO] Finished HE601628.3:0-10000000.\n",
      "2023-08-02 16:37:40,128 [INFO] Finished HE601628.3:10000000-20000000.\n",
      "2023-08-02 16:37:40,287 [INFO] Finished OU426848.1:30000000-33063208.\n",
      "2023-08-02 16:37:40,549 [INFO] Finished OU426848.1:0-10000000.\n",
      "2023-08-02 16:37:40,981 [INFO] Finished HE601612.2:0-26917.\n",
      "2023-08-02 16:37:41,049 [INFO] Finished HE601629.3:10000000-20000000.\n",
      "2023-08-02 16:37:41,237 [INFO] Finished OU426848.1:10000000-20000000.\n",
      "2023-08-02 16:37:41,437 [INFO] Finished OU426848.1:20000000-30000000.\n",
      "2023-08-02 16:37:41,593 [INFO] Finished HE601629.3:0-10000000.\n",
      "2023-08-02 16:37:42,048 [INFO] Finished HE601630.3:0-10000000.\n",
      "2023-08-02 16:37:42,299 [INFO] Finished HE601631.3:0-5969971.\n",
      "2023-08-02 16:37:43,102 [INFO] Finished HE601630.3:10000000-19942475.\n",
      "2023-08-02 16:37:43,192 [INFO] Rebuilding signatures of structural variants.\n",
      "2023-08-02 16:37:50,529 [INFO] Clustering structural variants.\n",
      "2023-08-02 16:37:50,681 [INFO] Finished HE601612.2:DEL.\n",
      "2023-08-02 16:37:50,686 [INFO] Finished OU426848.1:DEL.\n",
      "2023-08-02 16:37:50,689 [INFO] Finished HE601631.3:DEL.\n",
      "2023-08-02 16:37:50,702 [INFO] Finished OU426849.1:DEL.\n",
      "2023-08-02 16:37:50,718 [INFO] Finished HE601630.3:DEL.\n",
      "2023-08-02 16:37:50,750 [INFO] Finished HE601629.3:DEL.\n",
      "2023-08-02 16:37:50,775 [INFO] Finished OU426850.1:DEL.\n",
      "2023-08-02 16:37:50,786 [INFO] Finished HE601628.3:DEL.\n",
      "2023-08-02 16:37:50,786 [INFO] Finished HE601627.3:DEL.\n",
      "2023-08-02 16:37:50,823 [INFO] Finished HE601625.3:DEL.\n",
      "2023-08-02 16:37:50,838 [INFO] Finished HE601626.3:DEL.\n",
      "2023-08-02 16:37:50,872 [INFO] Finished HE601612.2:INS.\n",
      "2023-08-02 16:37:50,889 [INFO] Finished HE601624.3:DEL.\n",
      "2023-08-02 16:37:50,910 [INFO] Finished HE601612.2:INV.\n",
      "2023-08-02 16:37:50,933 [INFO] Finished HE601624.3:INV.\n",
      "2023-08-02 16:37:50,953 [INFO] Finished HE601625.3:INV.\n",
      "2023-08-02 16:37:50,973 [INFO] Finished HE601626.3:INV.\n",
      "2023-08-02 16:37:50,985 [INFO] Finished HE601631.3:INS.\n",
      "2023-08-02 16:37:50,993 [INFO] Finished HE601627.3:INV.\n",
      "2023-08-02 16:37:51,004 [INFO] Finished HE601628.3:INV.\n",
      "2023-08-02 16:37:51,008 [INFO] Finished HE601627.3:INS.\n",
      "2023-08-02 16:37:51,009 [INFO] Finished HE601628.3:INS.\n",
      "2023-08-02 16:37:51,011 [INFO] Finished HE601629.3:INV.\n",
      "2023-08-02 16:37:51,012 [INFO] Finished HE601630.3:INS.\n",
      "2023-08-02 16:37:51,017 [INFO] Finished HE601625.3:INS.\n",
      "2023-08-02 16:37:51,021 [INFO] Finished OU426848.1:INS.\n",
      "2023-08-02 16:37:51,022 [INFO] Finished HE601630.3:INV.\n",
      "2023-08-02 16:37:51,023 [INFO] Finished HE601612.2:DUP.\n",
      "2023-08-02 16:37:51,026 [INFO] Finished HE601624.3:DUP.\n",
      "2023-08-02 16:37:51,027 [INFO] Finished HE601625.3:DUP.\n",
      "2023-08-02 16:37:51,028 [INFO] Finished HE601631.3:INV.\n",
      "2023-08-02 16:37:51,029 [INFO] Finished HE601626.3:DUP.\n",
      "2023-08-02 16:37:51,030 [INFO] Finished OU426849.1:INV.\n",
      "2023-08-02 16:37:51,030 [INFO] Finished OU426848.1:INV.\n",
      "2023-08-02 16:37:51,030 [INFO] Finished HE601626.3:INS.\n",
      "2023-08-02 16:37:51,030 [INFO] Finished HE601627.3:DUP.\n",
      "2023-08-02 16:37:51,031 [INFO] Finished HE601628.3:DUP.\n",
      "2023-08-02 16:37:51,032 [INFO] Finished HE601630.3:DUP.\n",
      "2023-08-02 16:37:51,033 [INFO] Finished HE601629.3:DUP.\n",
      "2023-08-02 16:37:51,033 [INFO] Finished OU426850.1:INV.\n",
      "2023-08-02 16:37:51,034 [INFO] Finished HE601631.3:DUP.\n",
      "2023-08-02 16:37:51,034 [INFO] Finished OU426849.1:DUP.\n",
      "2023-08-02 16:37:51,034 [INFO] Finished OU426848.1:DUP.\n",
      "2023-08-02 16:37:51,035 [INFO] Finished OU426850.1:DUP.\n",
      "2023-08-02 16:37:51,040 [INFO] Finished HE601629.3:INS.\n",
      "2023-08-02 16:37:51,047 [INFO] Finished HE601612.2-HE601624.3:TRA/BND.\n",
      "2023-08-02 16:37:51,047 [INFO] Finished HE601612.2-HE601625.3:TRA/BND.\n",
      "2023-08-02 16:37:51,047 [INFO] Finished HE601612.2-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:37:51,048 [INFO] Finished HE601612.2-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:37:51,048 [INFO] Finished HE601612.2-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:37:51,049 [INFO] Finished HE601612.2-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:51,049 [INFO] Finished HE601612.2-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:51,050 [INFO] Finished HE601612.2-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:51,056 [INFO] Finished HE601612.2-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:51,059 [INFO] Finished OU426849.1:INS.\n",
      "2023-08-02 16:37:51,061 [INFO] Finished HE601612.2-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,062 [INFO] Finished HE601612.2-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:51,064 [INFO] Finished HE601624.3-HE601625.3:TRA/BND.\n",
      "2023-08-02 16:37:51,064 [INFO] Finished HE601624.3:INS.\n",
      "2023-08-02 16:37:51,064 [INFO] Finished HE601624.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:37:51,064 [INFO] Finished HE601624.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:51,064 [INFO] Finished HE601624.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:37:51,065 [INFO] Finished HE601624.3-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:37:51,065 [INFO] Finished HE601624.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:51,071 [INFO] Finished HE601624.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:51,075 [INFO] Finished HE601624.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:51,077 [INFO] Finished HE601624.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:51,078 [INFO] Finished HE601624.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,080 [INFO] Finished HE601625.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:37:51,080 [INFO] Finished HE601625.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:37:51,080 [INFO] Finished HE601625.3-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:37:51,080 [INFO] Finished HE601625.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:51,080 [INFO] Finished HE601625.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:51,081 [INFO] Finished HE601625.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:51,082 [INFO] Finished HE601625.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:51,087 [INFO] Finished HE601625.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:51,091 [INFO] Finished HE601625.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,093 [INFO] Finished HE601626.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:37:51,093 [INFO] Finished HE601626.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:37:51,095 [INFO] Finished HE601626.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:51,096 [INFO] Finished HE601626.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:51,096 [INFO] Finished HE601626.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:51,096 [INFO] Finished HE601626.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:51,096 [INFO] Finished HE601626.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:51,097 [INFO] Finished HE601626.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,097 [INFO] Finished HE601627.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:37:51,102 [INFO] Finished HE601627.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:51,106 [INFO] Finished HE601627.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:51,108 [INFO] Finished HE601627.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:51,108 [INFO] Finished HE601627.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:51,111 [INFO] Finished HE601627.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:51,111 [INFO] Finished HE601628.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:51,111 [INFO] Finished HE601628.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:37:51,111 [INFO] Finished HE601628.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:51,111 [INFO] Finished HE601627.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,112 [INFO] Finished HE601628.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:51,112 [INFO] Finished HE601628.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:51,117 [INFO] Finished HE601628.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,121 [INFO] Finished HE601629.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:37:51,123 [INFO] Finished HE601629.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:51,123 [INFO] Finished HE601629.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:51,126 [INFO] Finished HE601629.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:51,126 [INFO] Finished HE601629.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,126 [INFO] Finished HE601630.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:37:51,126 [INFO] Finished HE601630.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:51,126 [INFO] Finished HE601630.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:51,127 [INFO] Finished HE601630.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,127 [INFO] Finished HE601631.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:37:51,130 [INFO] Finished OU426850.1:INS.\n",
      "2023-08-02 16:37:51,133 [INFO] Finished HE601631.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:51,136 [INFO] Finished HE601631.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,138 [INFO] Finished OU426848.1-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:37:51,139 [INFO] Finished OU426848.1-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,141 [INFO] Finished OU426849.1-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:37:51,154 [INFO] Writing to your output file.\n",
      "2023-08-02 16:37:51,162 [INFO] Loading reference genome...\n",
      "2023-08-02 16:37:53,622 [INFO] Cleaning temporary files.\n",
      "2023-08-02 16:37:53,921 [INFO] Finished in 39.47 seconds.\n",
      "###################### smbre ##################\n",
      "2023-08-02 16:37:54,623 [INFO] Running /master/nplatt/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --report_readid --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 /master/nplatt/sch_man_ont/results/minimap2/smbre.bam /master/nplatt/sch_man_ont/data/genome/GCA_000237925.5.fa smbre.cutesv.vcf /master/nplatt/sch_man_ont/results/cutesv/smbre\n",
      "2023-08-02 16:37:54,849 [INFO] The total number of chromsomes: 12\n",
      "2023-08-02 16:37:56,991 [INFO] Finished OU426849.1:10000000-10680109.\n",
      "2023-08-02 16:38:07,338 [INFO] Finished HE601624.3:80000000-87984036.\n",
      "2023-08-02 16:38:08,921 [INFO] Finished HE601624.3:50000000-60000000.\n",
      "2023-08-02 16:38:09,019 [INFO] Finished HE601624.3:10000000-20000000.\n",
      "2023-08-02 16:38:09,044 [INFO] Finished HE601624.3:40000000-50000000.\n",
      "2023-08-02 16:38:09,266 [INFO] Finished HE601624.3:30000000-40000000.\n",
      "2023-08-02 16:38:09,387 [INFO] Finished HE601624.3:70000000-80000000.\n",
      "2023-08-02 16:38:09,404 [INFO] Finished HE601625.3:0-10000000.\n",
      "2023-08-02 16:38:09,459 [INFO] Finished HE601624.3:20000000-30000000.\n",
      "2023-08-02 16:38:09,461 [INFO] Finished HE601624.3:60000000-70000000.\n",
      "2023-08-02 16:38:09,524 [INFO] Finished OU426849.1:0-10000000.\n",
      "2023-08-02 16:38:11,305 [INFO] Finished HE601625.3:10000000-20000000.\n",
      "2023-08-02 16:38:11,828 [INFO] Finished HE601624.3:0-10000000.\n",
      "2023-08-02 16:38:15,312 [INFO] Finished OU426850.1:40000000-42949100.\n",
      "2023-08-02 16:38:19,151 [INFO] Finished HE601625.3:40000000-45716228.\n",
      "2023-08-02 16:38:21,874 [INFO] Finished HE601625.3:20000000-30000000.\n",
      "2023-08-02 16:38:22,963 [INFO] Finished OU426850.1:0-10000000.\n",
      "2023-08-02 16:38:23,451 [INFO] Finished OU426850.1:20000000-30000000.\n",
      "2023-08-02 16:38:23,463 [INFO] Finished OU426850.1:30000000-40000000.\n",
      "2023-08-02 16:38:23,556 [INFO] Finished OU426850.1:10000000-20000000.\n",
      "2023-08-02 16:38:25,463 [INFO] Finished HE601626.3:20000000-30000000.\n",
      "2023-08-02 16:38:25,509 [INFO] Finished HE601626.3:30000000-40000000.\n",
      "2023-08-02 16:38:26,079 [INFO] Finished HE601626.3:0-10000000.\n",
      "2023-08-02 16:38:29,066 [INFO] Finished HE601626.3:40000000-49793703.\n",
      "2023-08-02 16:38:29,631 [INFO] Finished HE601625.3:30000000-40000000.\n",
      "2023-08-02 16:38:29,765 [INFO] Finished HE601626.3:10000000-20000000.\n",
      "2023-08-02 16:38:31,825 [INFO] Finished HE601628.3:20000000-24128855.\n",
      "2023-08-02 16:38:32,765 [INFO] Finished HE601627.3:40000000-46471565.\n",
      "2023-08-02 16:38:33,056 [INFO] Finished HE601627.3:0-10000000.\n",
      "2023-08-02 16:38:34,674 [INFO] Finished HE601627.3:10000000-20000000.\n",
      "2023-08-02 16:38:35,444 [INFO] Finished HE601629.3:20000000-24696071.\n",
      "2023-08-02 16:38:35,973 [INFO] Finished HE601627.3:20000000-30000000.\n",
      "2023-08-02 16:38:36,667 [INFO] Finished HE601627.3:30000000-40000000.\n",
      "2023-08-02 16:38:37,918 [INFO] Finished HE601628.3:0-10000000.\n",
      "2023-08-02 16:38:37,996 [INFO] Finished OU426848.1:30000000-33063208.\n",
      "2023-08-02 16:38:38,417 [INFO] Finished HE601628.3:10000000-20000000.\n",
      "2023-08-02 16:38:39,306 [INFO] Finished HE601629.3:0-10000000.\n",
      "2023-08-02 16:38:39,558 [INFO] Finished OU426848.1:0-10000000.\n",
      "2023-08-02 16:38:39,568 [INFO] Finished HE601612.2:0-26917.\n",
      "2023-08-02 16:38:41,075 [INFO] Finished HE601629.3:10000000-20000000.\n",
      "2023-08-02 16:38:41,099 [INFO] Finished OU426848.1:10000000-20000000.\n",
      "2023-08-02 16:38:41,486 [INFO] Finished HE601630.3:0-10000000.\n",
      "2023-08-02 16:38:41,617 [INFO] Finished OU426848.1:20000000-30000000.\n",
      "2023-08-02 16:38:44,006 [INFO] Finished HE601630.3:10000000-19942475.\n",
      "2023-08-02 16:38:45,208 [INFO] Finished HE601631.3:0-5969971.\n",
      "2023-08-02 16:38:45,305 [INFO] Rebuilding signatures of structural variants.\n",
      "2023-08-02 16:39:07,676 [INFO] Clustering structural variants.\n",
      "2023-08-02 16:39:08,016 [INFO] Finished HE601612.2:DEL.\n",
      "2023-08-02 16:39:08,038 [INFO] Finished HE601631.3:DEL.\n",
      "2023-08-02 16:39:08,044 [INFO] Finished OU426848.1:DEL.\n",
      "2023-08-02 16:39:08,069 [INFO] Finished OU426849.1:DEL.\n",
      "2023-08-02 16:39:08,139 [INFO] Finished HE601630.3:DEL.\n",
      "2023-08-02 16:39:08,220 [INFO] Finished OU426850.1:DEL.\n",
      "2023-08-02 16:39:08,239 [INFO] Finished HE601629.3:DEL.\n",
      "2023-08-02 16:39:08,279 [INFO] Finished HE601628.3:DEL.\n",
      "2023-08-02 16:39:08,286 [INFO] Finished HE601627.3:DEL.\n",
      "2023-08-02 16:39:08,380 [INFO] Finished HE601625.3:DEL.\n",
      "2023-08-02 16:39:08,475 [INFO] Finished HE601626.3:DEL.\n",
      "2023-08-02 16:39:08,506 [INFO] Finished HE601624.3:DEL.\n",
      "2023-08-02 16:39:08,605 [INFO] Finished HE601612.2:INS.\n",
      "2023-08-02 16:39:08,639 [INFO] Finished HE601612.2:INV.\n",
      "2023-08-02 16:39:08,681 [INFO] Finished HE601624.3:INV.\n",
      "2023-08-02 16:39:08,720 [INFO] Finished HE601625.3:INV.\n",
      "2023-08-02 16:39:08,757 [INFO] Finished HE601626.3:INV.\n",
      "2023-08-02 16:39:08,792 [INFO] Finished HE601627.3:INV.\n",
      "2023-08-02 16:39:08,826 [INFO] Finished HE601628.3:INV.\n",
      "2023-08-02 16:39:08,859 [INFO] Finished HE601629.3:INV.\n",
      "2023-08-02 16:39:08,874 [INFO] Finished HE601631.3:INS.\n",
      "2023-08-02 16:39:08,891 [INFO] Finished HE601630.3:INV.\n",
      "2023-08-02 16:39:08,907 [INFO] Finished HE601631.3:INV.\n",
      "2023-08-02 16:39:08,923 [INFO] Finished OU426848.1:INV.\n",
      "2023-08-02 16:39:08,939 [INFO] Finished OU426849.1:INV.\n",
      "2023-08-02 16:39:08,950 [INFO] Finished HE601612.2:DUP.\n",
      "2023-08-02 16:39:08,958 [INFO] Finished OU426850.1:INV.\n",
      "2023-08-02 16:39:08,961 [INFO] Finished HE601624.3:DUP.\n",
      "2023-08-02 16:39:08,969 [INFO] Finished HE601625.3:DUP.\n",
      "2023-08-02 16:39:08,977 [INFO] Finished HE601626.3:DUP.\n",
      "2023-08-02 16:39:08,979 [INFO] Finished HE601627.3:DUP.\n",
      "2023-08-02 16:39:08,981 [INFO] Finished HE601630.3:INS.\n",
      "2023-08-02 16:39:08,987 [INFO] Finished HE601628.3:DUP.\n",
      "2023-08-02 16:39:08,989 [INFO] Finished HE601629.3:DUP.\n",
      "2023-08-02 16:39:08,994 [INFO] Finished HE601630.3:DUP.\n",
      "2023-08-02 16:39:08,997 [INFO] Finished HE601631.3:DUP.\n",
      "2023-08-02 16:39:08,999 [INFO] Finished OU426848.1:DUP.\n",
      "2023-08-02 16:39:09,003 [INFO] Finished OU426849.1:DUP.\n",
      "2023-08-02 16:39:09,007 [INFO] Finished OU426850.1:DUP.\n",
      "2023-08-02 16:39:09,007 [INFO] Finished HE601627.3:INS.\n",
      "2023-08-02 16:39:09,015 [INFO] Finished HE601625.3:INS.\n",
      "2023-08-02 16:39:09,021 [INFO] Finished OU426848.1:INS.\n",
      "2023-08-02 16:39:09,035 [INFO] Finished HE601629.3:INS.\n",
      "2023-08-02 16:39:09,043 [INFO] Finished HE601612.2-HE601624.3:TRA/BND.\n",
      "2023-08-02 16:39:09,046 [INFO] Finished HE601612.2-HE601625.3:TRA/BND.\n",
      "2023-08-02 16:39:09,049 [INFO] Finished HE601612.2-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:39:09,052 [INFO] Finished HE601612.2-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:39:09,055 [INFO] Finished HE601628.3:INS.\n",
      "2023-08-02 16:39:09,064 [INFO] Finished HE601612.2-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:39:09,068 [INFO] Finished HE601612.2-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:39:09,083 [INFO] Finished HE601612.2-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:39:09,084 [INFO] Finished HE601612.2-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:39:09,090 [INFO] Finished HE601612.2-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:39:09,090 [INFO] Finished HE601612.2-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:39:09,091 [INFO] Finished HE601612.2-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,100 [INFO] Finished OU426849.1:INS.\n",
      "2023-08-02 16:39:09,104 [INFO] Finished HE601624.3-HE601625.3:TRA/BND.\n",
      "2023-08-02 16:39:09,112 [INFO] Finished HE601624.3-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:39:09,112 [INFO] Finished HE601624.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:39:09,128 [INFO] Finished HE601624.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:39:09,128 [INFO] Finished HE601624.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:39:09,131 [INFO] Finished HE601624.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:39:09,134 [INFO] Finished HE601624.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:39:09,134 [INFO] Finished HE601624.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:39:09,138 [INFO] Finished HE601626.3:INS.\n",
      "2023-08-02 16:39:09,145 [INFO] Finished HE601624.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:39:09,148 [INFO] Finished HE601624.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,155 [INFO] Finished HE601625.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:39:09,157 [INFO] Finished HE601625.3-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:39:09,158 [INFO] Finished HE601624.3:INS.\n",
      "2023-08-02 16:39:09,170 [INFO] Finished HE601625.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:39:09,172 [INFO] Finished HE601625.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:39:09,172 [INFO] Finished HE601625.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:39:09,177 [INFO] Finished HE601625.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:39:09,178 [INFO] Finished HE601625.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:39:09,188 [INFO] Finished HE601625.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:39:09,191 [INFO] Finished HE601625.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,194 [INFO] Finished HE601626.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:39:09,201 [INFO] Finished HE601626.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:39:09,201 [INFO] Finished HE601626.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:39:09,225 [INFO] Finished HE601626.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:39:09,227 [INFO] Finished HE601626.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:39:09,228 [INFO] Finished HE601627.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:39:09,229 [INFO] Finished HE601626.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:39:09,230 [INFO] Finished HE601627.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:39:09,231 [INFO] Finished HE601626.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:39:09,232 [INFO] Finished HE601626.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,234 [INFO] Finished HE601627.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:39:09,237 [INFO] Finished HE601627.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:39:09,244 [INFO] Finished HE601627.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:39:09,244 [INFO] Finished HE601627.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:39:09,264 [INFO] Finished HE601627.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,269 [INFO] Finished HE601628.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:39:09,270 [INFO] Finished HE601628.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:39:09,271 [INFO] Finished HE601628.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:39:09,272 [INFO] Finished HE601628.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:39:09,274 [INFO] Finished HE601628.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,274 [INFO] Finished HE601628.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:39:09,278 [INFO] Finished HE601629.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:39:09,278 [INFO] Finished HE601629.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:39:09,286 [INFO] Finished HE601629.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:39:09,287 [INFO] Finished HE601629.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:39:09,303 [INFO] Finished HE601629.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,311 [INFO] Finished HE601630.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:39:09,311 [INFO] Finished HE601630.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:39:09,314 [INFO] Finished HE601630.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:39:09,314 [INFO] Finished HE601630.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,315 [INFO] Finished HE601631.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:39:09,317 [INFO] Finished HE601631.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:39:09,320 [INFO] Finished OU426848.1-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:39:09,322 [INFO] Finished HE601631.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,328 [INFO] Finished OU426848.1-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,329 [INFO] Finished OU426849.1-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:39:09,366 [INFO] Finished OU426850.1:INS.\n",
      "2023-08-02 16:39:09,406 [INFO] Writing to your output file.\n",
      "2023-08-02 16:39:09,421 [INFO] Loading reference genome...\n",
      "2023-08-02 16:39:12,177 [INFO] Cleaning temporary files.\n",
      "2023-08-02 16:39:12,608 [INFO] Finished in 77.99 seconds.\n",
      "###################### smeg ##################\n",
      "2023-08-02 16:39:13,325 [INFO] Running /master/nplatt/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --report_readid --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 /master/nplatt/sch_man_ont/results/minimap2/smeg.bam /master/nplatt/sch_man_ont/data/genome/GCA_000237925.5.fa smeg.cutesv.vcf /master/nplatt/sch_man_ont/results/cutesv/smeg\n",
      "2023-08-02 16:39:13,513 [INFO] The total number of chromsomes: 12\n",
      "2023-08-02 16:39:16,292 [INFO] Finished OU426849.1:10000000-10680109.\n",
      "2023-08-02 16:39:30,541 [INFO] Finished HE601624.3:80000000-87984036.\n",
      "2023-08-02 16:39:32,227 [INFO] Finished HE601624.3:10000000-20000000.\n",
      "2023-08-02 16:39:32,417 [INFO] Finished HE601624.3:20000000-30000000.\n",
      "2023-08-02 16:39:32,467 [INFO] Finished OU426849.1:0-10000000.\n",
      "2023-08-02 16:39:32,495 [INFO] Finished HE601624.3:50000000-60000000.\n",
      "2023-08-02 16:39:32,610 [INFO] Finished HE601624.3:40000000-50000000.\n",
      "2023-08-02 16:39:32,854 [INFO] Finished HE601624.3:60000000-70000000.\n",
      "2023-08-02 16:39:32,931 [INFO] Finished HE601625.3:0-10000000.\n",
      "2023-08-02 16:39:32,974 [INFO] Finished HE601624.3:30000000-40000000.\n",
      "2023-08-02 16:39:33,053 [INFO] Finished HE601624.3:70000000-80000000.\n",
      "2023-08-02 16:39:33,828 [INFO] Finished HE601624.3:0-10000000.\n",
      "2023-08-02 16:39:34,069 [INFO] Finished HE601625.3:10000000-20000000.\n",
      "2023-08-02 16:39:42,936 [INFO] Finished OU426850.1:40000000-42949100.\n",
      "2023-08-02 16:39:45,268 [INFO] Finished HE601625.3:40000000-45716228.\n",
      "2023-08-02 16:39:48,711 [INFO] Finished HE601625.3:20000000-30000000.\n",
      "2023-08-02 16:39:49,983 [INFO] Finished OU426850.1:0-10000000.\n",
      "2023-08-02 16:39:50,027 [INFO] Finished OU426850.1:10000000-20000000.\n",
      "2023-08-02 16:39:50,120 [INFO] Finished OU426850.1:30000000-40000000.\n",
      "2023-08-02 16:39:50,510 [INFO] Finished OU426850.1:20000000-30000000.\n",
      "2023-08-02 16:39:52,485 [INFO] Finished HE601626.3:20000000-30000000.\n",
      "2023-08-02 16:39:53,629 [INFO] Finished HE601626.3:30000000-40000000.\n",
      "2023-08-02 16:39:56,837 [INFO] Finished HE601626.3:0-10000000.\n",
      "2023-08-02 16:39:57,415 [INFO] Finished HE601626.3:10000000-20000000.\n",
      "2023-08-02 16:40:00,037 [INFO] Finished HE601626.3:40000000-49793703.\n",
      "2023-08-02 16:40:00,817 [INFO] Finished HE601625.3:30000000-40000000.\n",
      "2023-08-02 16:40:01,734 [INFO] Finished HE601628.3:20000000-24128855.\n",
      "2023-08-02 16:40:01,770 [INFO] Finished HE601627.3:40000000-46471565.\n",
      "2023-08-02 16:40:02,207 [INFO] Finished HE601627.3:0-10000000.\n",
      "2023-08-02 16:40:05,551 [INFO] Finished HE601627.3:10000000-20000000.\n",
      "2023-08-02 16:40:06,109 [INFO] Finished HE601627.3:30000000-40000000.\n",
      "2023-08-02 16:40:06,286 [INFO] Finished HE601629.3:20000000-24696071.\n",
      "2023-08-02 16:40:06,421 [INFO] Finished HE601627.3:20000000-30000000.\n",
      "2023-08-02 16:40:07,857 [INFO] Finished HE601628.3:10000000-20000000.\n",
      "2023-08-02 16:40:08,636 [INFO] Finished HE601628.3:0-10000000.\n",
      "2023-08-02 16:40:08,949 [INFO] Finished OU426848.1:30000000-33063208.\n",
      "2023-08-02 16:40:09,878 [INFO] Finished OU426848.1:0-10000000.\n",
      "2023-08-02 16:40:10,250 [INFO] Finished HE601629.3:10000000-20000000.\n",
      "2023-08-02 16:40:10,465 [INFO] Finished HE601629.3:0-10000000.\n",
      "2023-08-02 16:40:10,785 [INFO] Finished HE601612.2:0-26917.\n",
      "2023-08-02 16:40:11,512 [INFO] Finished OU426848.1:10000000-20000000.\n",
      "2023-08-02 16:40:11,897 [INFO] Finished OU426848.1:20000000-30000000.\n",
      "2023-08-02 16:40:12,304 [INFO] Finished HE601630.3:0-10000000.\n",
      "2023-08-02 16:40:12,500 [INFO] Finished HE601631.3:0-5969971.\n",
      "2023-08-02 16:40:13,682 [INFO] Finished HE601630.3:10000000-19942475.\n",
      "2023-08-02 16:40:13,708 [INFO] Rebuilding signatures of structural variants.\n",
      "2023-08-02 16:40:28,582 [INFO] Clustering structural variants.\n",
      "2023-08-02 16:40:28,823 [INFO] Finished OU426848.1:DEL.\n",
      "2023-08-02 16:40:28,830 [INFO] Finished HE601631.3:DEL.\n",
      "2023-08-02 16:40:28,833 [INFO] Finished HE601612.2:DEL.\n",
      "2023-08-02 16:40:28,837 [INFO] Finished OU426849.1:DEL.\n",
      "2023-08-02 16:40:28,875 [INFO] Finished HE601630.3:DEL.\n",
      "2023-08-02 16:40:28,923 [INFO] Finished HE601629.3:DEL.\n",
      "2023-08-02 16:40:28,939 [INFO] Finished OU426850.1:DEL.\n",
      "2023-08-02 16:40:28,941 [INFO] Finished HE601628.3:DEL.\n",
      "2023-08-02 16:40:28,957 [INFO] Finished HE601627.3:DEL.\n",
      "2023-08-02 16:40:29,008 [INFO] Finished HE601626.3:DEL.\n",
      "2023-08-02 16:40:29,010 [INFO] Finished HE601625.3:DEL.\n",
      "2023-08-02 16:40:29,101 [INFO] Finished HE601624.3:DEL.\n",
      "2023-08-02 16:40:29,194 [INFO] Finished HE601612.2:INS.\n",
      "2023-08-02 16:40:29,215 [INFO] Finished HE601612.2:INV.\n",
      "2023-08-02 16:40:29,240 [INFO] Finished HE601624.3:INV.\n",
      "2023-08-02 16:40:29,261 [INFO] Finished HE601625.3:INV.\n",
      "2023-08-02 16:40:29,283 [INFO] Finished HE601626.3:INV.\n",
      "2023-08-02 16:40:29,304 [INFO] Finished HE601627.3:INV.\n",
      "2023-08-02 16:40:29,322 [INFO] Finished HE601631.3:INS.\n",
      "2023-08-02 16:40:29,323 [INFO] Finished HE601628.3:INV.\n",
      "2023-08-02 16:40:29,342 [INFO] Finished HE601630.3:INV.\n",
      "2023-08-02 16:40:29,343 [INFO] Finished HE601629.3:INV.\n",
      "2023-08-02 16:40:29,361 [INFO] Finished HE601631.3:INV.\n",
      "2023-08-02 16:40:29,362 [INFO] Finished OU426848.1:INV.\n",
      "2023-08-02 16:40:29,380 [INFO] Finished OU426848.1:INS.\n",
      "2023-08-02 16:40:29,380 [INFO] Finished OU426849.1:INV.\n",
      "2023-08-02 16:40:29,383 [INFO] Finished HE601630.3:INS.\n",
      "2023-08-02 16:40:29,383 [INFO] Finished OU426850.1:INV.\n",
      "2023-08-02 16:40:29,387 [INFO] Finished HE601624.3:DUP.\n",
      "2023-08-02 16:40:29,388 [INFO] Finished HE601612.2:DUP.\n",
      "2023-08-02 16:40:29,390 [INFO] Finished HE601625.3:DUP.\n",
      "2023-08-02 16:40:29,393 [INFO] Finished HE601627.3:DUP.\n",
      "2023-08-02 16:40:29,393 [INFO] Finished HE601626.3:DUP.\n",
      "2023-08-02 16:40:29,394 [INFO] Finished HE601628.3:DUP.\n",
      "2023-08-02 16:40:29,396 [INFO] Finished HE601629.3:DUP.\n",
      "2023-08-02 16:40:29,397 [INFO] Finished OU426849.1:INS.\n",
      "2023-08-02 16:40:29,399 [INFO] Finished HE601630.3:DUP.\n",
      "2023-08-02 16:40:29,400 [INFO] Finished HE601631.3:DUP.\n",
      "2023-08-02 16:40:29,400 [INFO] Finished OU426848.1:DUP.\n",
      "2023-08-02 16:40:29,402 [INFO] Finished OU426849.1:DUP.\n",
      "2023-08-02 16:40:29,404 [INFO] Finished HE601627.3:INS.\n",
      "2023-08-02 16:40:29,404 [INFO] Finished OU426850.1:DUP.\n",
      "2023-08-02 16:40:29,410 [INFO] Finished HE601626.3:INS.\n",
      "2023-08-02 16:40:29,411 [INFO] Finished HE601629.3:INS.\n",
      "2023-08-02 16:40:29,416 [INFO] Finished HE601625.3:INS.\n",
      "2023-08-02 16:40:29,419 [INFO] Finished HE601628.3:INS.\n",
      "2023-08-02 16:40:29,423 [INFO] Finished HE601612.2-HE601624.3:TRA/BND.\n",
      "2023-08-02 16:40:29,423 [INFO] Finished HE601612.2-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:40:29,424 [INFO] Finished HE601612.2-HE601625.3:TRA/BND.\n",
      "2023-08-02 16:40:29,425 [INFO] Finished HE601612.2-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:40:29,428 [INFO] Finished HE601612.2-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:40:29,430 [INFO] Finished HE601612.2-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:40:29,435 [INFO] Finished HE601612.2-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:40:29,436 [INFO] Finished HE601612.2-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:40:29,443 [INFO] Finished HE601612.2-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:40:29,444 [INFO] Finished HE601612.2-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:40:29,446 [INFO] Finished HE601612.2-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,448 [INFO] Finished HE601624.3-HE601625.3:TRA/BND.\n",
      "2023-08-02 16:40:29,450 [INFO] Finished HE601624.3-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:40:29,450 [INFO] Finished HE601624.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:40:29,452 [INFO] Finished HE601624.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:40:29,453 [INFO] Finished HE601624.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:40:29,459 [INFO] Finished HE601624.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:40:29,461 [INFO] Finished HE601624.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:40:29,467 [INFO] Finished HE601624.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:40:29,468 [INFO] Finished HE601624.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:40:29,470 [INFO] Finished HE601624.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,472 [INFO] Finished HE601625.3-HE601626.3:TRA/BND.\n",
      "2023-08-02 16:40:29,473 [INFO] Finished HE601625.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:40:29,474 [INFO] Finished HE601625.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:40:29,475 [INFO] Finished HE601625.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:40:29,476 [INFO] Finished HE601625.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:40:29,483 [INFO] Finished HE601625.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:40:29,483 [INFO] Finished HE601625.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:40:29,490 [INFO] Finished HE601625.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:40:29,491 [INFO] Finished HE601625.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,493 [INFO] Finished HE601624.3:INS.\n",
      "2023-08-02 16:40:29,496 [INFO] Finished HE601626.3-HE601627.3:TRA/BND.\n",
      "2023-08-02 16:40:29,496 [INFO] Finished HE601626.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:40:29,498 [INFO] Finished HE601626.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:40:29,498 [INFO] Finished HE601626.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:40:29,500 [INFO] Finished HE601626.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:40:29,501 [INFO] Finished HE601626.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:40:29,507 [INFO] Finished HE601626.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:40:29,509 [INFO] Finished HE601626.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,513 [INFO] Finished HE601627.3-HE601628.3:TRA/BND.\n",
      "2023-08-02 16:40:29,515 [INFO] Finished HE601627.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:40:29,520 [INFO] Finished HE601627.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:40:29,520 [INFO] Finished HE601627.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:40:29,520 [INFO] Finished HE601627.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:40:29,521 [INFO] Finished HE601627.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:40:29,521 [INFO] Finished HE601627.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,523 [INFO] Finished HE601628.3-HE601629.3:TRA/BND.\n",
      "2023-08-02 16:40:29,524 [INFO] Finished HE601628.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:40:29,531 [INFO] Finished HE601628.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:40:29,532 [INFO] Finished HE601628.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:40:29,536 [INFO] Finished HE601628.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:40:29,538 [INFO] Finished HE601628.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,543 [INFO] Finished HE601629.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:40:29,543 [INFO] Finished HE601629.3-HE601630.3:TRA/BND.\n",
      "2023-08-02 16:40:29,543 [INFO] Finished HE601629.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:40:29,543 [INFO] Finished HE601629.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:40:29,544 [INFO] Finished HE601629.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,546 [INFO] Finished HE601630.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:40:29,547 [INFO] Finished HE601630.3-HE601631.3:TRA/BND.\n",
      "2023-08-02 16:40:29,554 [INFO] Finished HE601630.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:40:29,555 [INFO] Finished HE601630.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,559 [INFO] Finished HE601631.3-OU426848.1:TRA/BND.\n",
      "2023-08-02 16:40:29,561 [INFO] Finished HE601631.3-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:40:29,565 [INFO] Finished OU426848.1-OU426849.1:TRA/BND.\n",
      "2023-08-02 16:40:29,565 [INFO] Finished OU426848.1-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,566 [INFO] Finished OU426849.1-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,566 [INFO] Finished HE601631.3-OU426850.1:TRA/BND.\n",
      "2023-08-02 16:40:29,610 [INFO] Finished OU426850.1:INS.\n",
      "2023-08-02 16:40:29,710 [INFO] Writing to your output file.\n",
      "2023-08-02 16:40:29,742 [INFO] Loading reference genome...\n",
      "2023-08-02 16:40:32,272 [INFO] Cleaning temporary files.\n",
      "2023-08-02 16:40:32,611 [INFO] Finished in 79.29 seconds.\n"
     ]
    }
   ],
   "source": [
    "#sv call\n",
    "for sample in samples:\n",
    "#for sample in [\"smle_pzq_es\"]:\n",
    "\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    bam      = \"{}/results/minimap2/{}.bam\".format(proj_dir, sample)\n",
    "    query_fq = \"{}/results/nanofilt/{}.filtered.fastq.gz\".format(proj_dir, sample)\n",
    "    out_dir  = \"{}/{}\".format(os.getcwd(), sample)\n",
    "    \n",
    "    Path(\"{}\".format(out_dir)).mkdir(parents=True, exist_ok=True)\n",
    "    !~/anaconda3/envs/cutesv/bin/cuteSV --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 --report_readid --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3 --threads 12 --min_support 10 {bam} {ref_fas} {sample}.cutesv.vcf {out_dir}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ef9b409-f682-46eb-9f93-ef3efc0d0cda",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "977c7cb6-a897-439e-80c5-f7aa1a4450f5",
   "metadata": {},
   "source": [
    "## Identify SmPoMucs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a7967d5-5c6c-4989-9426-56253302adf0",
   "metadata": {},
   "source": [
    "From this [figure/paper](https://www.sciencedirect.com/science/article/pii/S0166685107003349#fig1) it seems we can use the 5' end to possibly identify single reads with SmPoMucs.\n",
    "\n",
    "```GRMNKKIILLTIIYVALMIR```\n",
    "\n",
    "\n",
    "```GGAAGAATGAACAAGAAAATTATTCTCTTAACAATTATATATGTTGCGCTGATGATAAGAGGTGAC```\n",
    "\n",
    "Then there are 3 possible VNTRs that can be present up to 55x\n",
    " - r1: `DDYALSEPT`\n",
    " - r1': `DYYALSEPT`\n",
    " - r2: `GDLASDKPT`\n",
    "\n",
    "Then there is a C-terminal region of 234 AAs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "debd5069-29b9-4433-8014-a4f4ae64958c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/smpomuc\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/smpomuc\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f961f51-4cf4-48a5-b80b-78fd6b8bf71f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#run a blast with the aa seq and see what happens"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "2109872b-3a04-4c96-ac53-7f2f3845a393",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "echo -e \">smpomuc_5p_aa\\nGRMNKKIILLTIIYVALMIR\" >smpomuc_5p_aa.fas\n",
    "echo -e \">smpomuc_c-term\\nVPEHLKTRIKHYKDAYFHSSIQKFLSLEPYTRASSTRAPQIYHEECLRLEKLYFTKWAVHYLSKNGATDITLLQSYENEYEEAKKGDKNADPRRDWGGQLRASISKKWKEREILDDVESAYIAGPRTNVNVNKEELKKQLTNTGNNIEAQLNNVKELESKAIQAANKHMNNRDDKSLEKQHKEAYSTLGKELRSLVDLMGEAEFQRILLLTTLPKDEQIKMIIQAMDKDSTNCS\" >smpomuc_3p_aa.fas\n",
    "\n",
    "\n",
    "echo -e \">smpomuc_5p_nuc\\nGGAAGAATGAACAAGAAAATTATTCTCTTAACAATTATATATGTTGCGCTGATGATAAGAGGTGAC\" >smpomuc_5p_nuc.fas\n",
    "echo -e \">smpomuc_3p_exon_nuc\\nGAGAATGATAATGTTTTCAACTAAACCTCAGTGTTTGTTTATGATATGGACAGACATGTTTTGGATCGTTAACAGTTTTCTGTGTCA\" >smpomuc_3p_exon_nuc.fas\n",
    "\n",
    "cat smpomuc_5p_nuc.fas smpomuc_3p_exon_nuc.fas >nuc.fas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "aea9114e-4faa-4217-8ad2-8ed4c3792ab4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error while terminating subprocess (pid=33553): \n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "\n",
    "for POP in \"smle_pzq_er\" \"smle_pzq_es\" \"smbre\" \"smor\" \"smeg\"; do\n",
    "    zcat /master/nplatt/sch_man_ont/results/read_correction/${POP}/${POP}.correctedReads.fasta.gz >${POP}.correctedReads.fasta\n",
    "\n",
    "    # conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "        # makeblastdb \\\n",
    "        #     -dbtype nucl \\\n",
    "        #     -in ${POP}.correctedReads.fasta\n",
    "\n",
    "    conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "        blastn \\\n",
    "            -subject ${POP}.correctedReads.fasta \\\n",
    "            -query nuc.fas \\\n",
    "            -out nuc_vs_${POP}.blastn.tsv \\\n",
    "            -outfmt 6\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "ca906ebf-fe35-4ac9-94b4-fbf7394317e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "for pop in [\"smle_pzq_er\", \"smle_pzq_es\", \"smbre\", \"smor\", \"smeg\"]:\n",
    "    #read in data\n",
    "    df = pd.read_csv(f\"smpomuc_5p_nuc_vs_{pop}.blastn.tsv\", sep=\"\\t\", header=0)\n",
    "\n",
    "    #change column names to make them useful\n",
    "    col_names = [\"q_seqid\", \"s_seqid\", \"p_ident\", \"length\", \"mismatch\", \"gapopen\", \"q_start\", \"q_end\", \"s_start\", \"s_end\", \"evalue\", \"bitscore\"]\n",
    "    df.columns=col_names\n",
    "\n",
    "    #find possibly good hits\n",
    "    df = df.loc[( (df[\"p_ident\"] > 0.9) & (df[\"length\"] > 60) )]\n",
    "\n",
    "    #get orientation\n",
    "    df[\"orient\"] = \"Na\"\n",
    "\n",
    "    df.loc[df[\"s_start\"] < df[\"s_end\"], \"orient\"] = \"+\"\n",
    "    df.loc[df[\"s_start\"] > df[\"s_end\"], \"orient\"] = \"-\"\n",
    "\n",
    "    #prep extraction by cleaning up orientation\n",
    "    df[\"extract_start\"] = df[\"s_start\"]\n",
    "    df[\"extract_end\"] = df[\"s_end\"]\n",
    "\n",
    "    mask = df['orient'] == '-'\n",
    "    df.loc[mask, ['extract_start', 'extract_end']] = df.loc[mask, ['extract_end', 'extract_start']].values\n",
    "\n",
    "    #then adding a buffer to the 3' end of 3000 bp\n",
    "    #df[\"extract_end\"] = df[\"extract_end\"] + 3_000\n",
    "\n",
    "    # Open the SAM/BAM file\n",
    "    df[['chr', 'chr_start', 'chr_stop']] = 'na'\n",
    "\n",
    "    with pysam.AlignmentFile(f'{proj_dir}/results/minimap2/{pop}.bam', 'rb') as bamfile:\n",
    "\n",
    "        s_seqids = set(df[\"s_seqid\"])\n",
    "\n",
    "        # Iterate over each read in the file\n",
    "        for read in bamfile:\n",
    "\n",
    "            # Check if the read is mapped\n",
    "            if read.query_name in s_seqids:\n",
    "\n",
    "                # Get the reference name, start, and end position\n",
    "                ref_name = read.reference_name\n",
    "                start = read.reference_start\n",
    "                end = read.reference_end\n",
    "\n",
    "                df.loc[df[\"s_seqid\"] == read.query_name, \"chr\"]       = ref_name\n",
    "                df.loc[df[\"s_seqid\"] == read.query_name, \"chr_start\"] = start\n",
    "                df.loc[df[\"s_seqid\"] == read.query_name, \"chr_stop\"]  = end\n",
    "\n",
    "    df = df.sort_values(by=[\"chr\", \"chr_start\"])\n",
    "    df.to_csv(f\"{pop}_ind_reads_with_smpomuc_signal.csv\", sep=\",\", header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "22656e35-ae8d-443d-ab95-db093b7f2bef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>q_seqid</th>\n",
       "      <th>s_seqid</th>\n",
       "      <th>p_ident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>q_start</th>\n",
       "      <th>q_end</th>\n",
       "      <th>s_start</th>\n",
       "      <th>s_end</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>orient</th>\n",
       "      <th>extract_start</th>\n",
       "      <th>extract_end</th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_start</th>\n",
       "      <th>chr_stop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>b79e0aeb-0a87-41fc-94b9-919b498379da</td>\n",
       "      <td>95.38</td>\n",
       "      <td>65</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>18861</td>\n",
       "      <td>18925</td>\n",
       "      <td>3.000000e-27</td>\n",
       "      <td>104.0</td>\n",
       "      <td>+</td>\n",
       "      <td>18861</td>\n",
       "      <td>18925</td>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38357980</td>\n",
       "      <td>38397881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>df430d48-9250-4119-851f-6ba3fb44e4d3</td>\n",
       "      <td>95.38</td>\n",
       "      <td>65</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>6852</td>\n",
       "      <td>6788</td>\n",
       "      <td>3.000000e-27</td>\n",
       "      <td>104.0</td>\n",
       "      <td>-</td>\n",
       "      <td>6788</td>\n",
       "      <td>6852</td>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38360750</td>\n",
       "      <td>38384301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>b90cf0ab-b793-48b6-af81-bf857c2ea368</td>\n",
       "      <td>95.38</td>\n",
       "      <td>65</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>46714</td>\n",
       "      <td>46650</td>\n",
       "      <td>3.000000e-27</td>\n",
       "      <td>104.0</td>\n",
       "      <td>-</td>\n",
       "      <td>46650</td>\n",
       "      <td>46714</td>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38361373</td>\n",
       "      <td>38425380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>80fcd690-442d-42ba-96ba-ca5a998061a8</td>\n",
       "      <td>95.38</td>\n",
       "      <td>65</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>14768</td>\n",
       "      <td>14832</td>\n",
       "      <td>3.000000e-27</td>\n",
       "      <td>104.0</td>\n",
       "      <td>+</td>\n",
       "      <td>14768</td>\n",
       "      <td>14832</td>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38362406</td>\n",
       "      <td>38386007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>9011db49-5b2c-4b19-af76-8925088fdc2e</td>\n",
       "      <td>95.38</td>\n",
       "      <td>65</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>13647</td>\n",
       "      <td>13711</td>\n",
       "      <td>3.000000e-27</td>\n",
       "      <td>104.0</td>\n",
       "      <td>+</td>\n",
       "      <td>13647</td>\n",
       "      <td>13711</td>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38363532</td>\n",
       "      <td>38384631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>265756c1-f961-4fc6-aa57-92d097dff38e</td>\n",
       "      <td>98.46</td>\n",
       "      <td>65</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>15436</td>\n",
       "      <td>15372</td>\n",
       "      <td>1.000000e-30</td>\n",
       "      <td>115.0</td>\n",
       "      <td>-</td>\n",
       "      <td>15372</td>\n",
       "      <td>15436</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30440967</td>\n",
       "      <td>30443696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>da1aaceb-88de-4308-9904-2a621a2141b5</td>\n",
       "      <td>98.46</td>\n",
       "      <td>65</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>5737</td>\n",
       "      <td>5801</td>\n",
       "      <td>1.000000e-30</td>\n",
       "      <td>115.0</td>\n",
       "      <td>+</td>\n",
       "      <td>5737</td>\n",
       "      <td>5801</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30446208</td>\n",
       "      <td>30446521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>da1aaceb-88de-4308-9904-2a621a2141b5</td>\n",
       "      <td>95.38</td>\n",
       "      <td>65</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>10960</td>\n",
       "      <td>10899</td>\n",
       "      <td>3.000000e-26</td>\n",
       "      <td>100.0</td>\n",
       "      <td>-</td>\n",
       "      <td>10899</td>\n",
       "      <td>10960</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30446208</td>\n",
       "      <td>30446521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>1cc203d9-3d54-43b5-bd9e-81c7793394be</td>\n",
       "      <td>95.38</td>\n",
       "      <td>65</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>13570</td>\n",
       "      <td>13634</td>\n",
       "      <td>3.000000e-27</td>\n",
       "      <td>104.0</td>\n",
       "      <td>+</td>\n",
       "      <td>13570</td>\n",
       "      <td>13634</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>35197991</td>\n",
       "      <td>35201780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>e34e77a8-6659-449e-ab63-5f11a0a59066</td>\n",
       "      <td>96.92</td>\n",
       "      <td>65</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>12267</td>\n",
       "      <td>12331</td>\n",
       "      <td>6.000000e-29</td>\n",
       "      <td>110.0</td>\n",
       "      <td>+</td>\n",
       "      <td>12267</td>\n",
       "      <td>12331</td>\n",
       "      <td>HE601631.3</td>\n",
       "      <td>5441464</td>\n",
       "      <td>5441546</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>75 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           q_seqid                               s_seqid  p_ident  length  \\\n",
       "15  smpomuc_5p_nuc  b79e0aeb-0a87-41fc-94b9-919b498379da    95.38      65   \n",
       "4   smpomuc_5p_nuc  df430d48-9250-4119-851f-6ba3fb44e4d3    95.38      65   \n",
       "43  smpomuc_5p_nuc  b90cf0ab-b793-48b6-af81-bf857c2ea368    95.38      65   \n",
       "35  smpomuc_5p_nuc  80fcd690-442d-42ba-96ba-ca5a998061a8    95.38      65   \n",
       "3   smpomuc_5p_nuc  9011db49-5b2c-4b19-af76-8925088fdc2e    95.38      65   \n",
       "..             ...                                   ...      ...     ...   \n",
       "13  smpomuc_5p_nuc  265756c1-f961-4fc6-aa57-92d097dff38e    98.46      65   \n",
       "39  smpomuc_5p_nuc  da1aaceb-88de-4308-9904-2a621a2141b5    98.46      65   \n",
       "40  smpomuc_5p_nuc  da1aaceb-88de-4308-9904-2a621a2141b5    95.38      65   \n",
       "25  smpomuc_5p_nuc  1cc203d9-3d54-43b5-bd9e-81c7793394be    95.38      65   \n",
       "60  smpomuc_5p_nuc  e34e77a8-6659-449e-ab63-5f11a0a59066    96.92      65   \n",
       "\n",
       "    mismatch  gapopen  q_start  q_end  s_start  s_end        evalue  bitscore  \\\n",
       "15         3        0        1     65    18861  18925  3.000000e-27     104.0   \n",
       "4          3        0        1     65     6852   6788  3.000000e-27     104.0   \n",
       "43         3        0        1     65    46714  46650  3.000000e-27     104.0   \n",
       "35         3        0        1     65    14768  14832  3.000000e-27     104.0   \n",
       "3          3        0        1     65    13647  13711  3.000000e-27     104.0   \n",
       "..       ...      ...      ...    ...      ...    ...           ...       ...   \n",
       "13         1        0        1     65    15436  15372  1.000000e-30     115.0   \n",
       "39         1        0        1     65     5737   5801  1.000000e-30     115.0   \n",
       "40         0        1        1     65    10960  10899  3.000000e-26     100.0   \n",
       "25         3        0        1     65    13570  13634  3.000000e-27     104.0   \n",
       "60         2        0        1     65    12267  12331  6.000000e-29     110.0   \n",
       "\n",
       "   orient  extract_start  extract_end         chr chr_start  chr_stop  \n",
       "15      +          18861        18925  HE601625.3  38357980  38397881  \n",
       "4       -           6788         6852  HE601625.3  38360750  38384301  \n",
       "43      -          46650        46714  HE601625.3  38361373  38425380  \n",
       "35      +          14768        14832  HE601625.3  38362406  38386007  \n",
       "3       +          13647        13711  HE601625.3  38363532  38384631  \n",
       "..    ...            ...          ...         ...       ...       ...  \n",
       "13      -          15372        15436  HE601627.3  30440967  30443696  \n",
       "39      +           5737         5801  HE601627.3  30446208  30446521  \n",
       "40      -          10899        10960  HE601627.3  30446208  30446521  \n",
       "25      +          13570        13634  HE601627.3  35197991  35201780  \n",
       "60      +          12267        12331  HE601631.3   5441464   5441546  \n",
       "\n",
       "[75 rows x 18 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "6ca25a9d-602c-42a6-8f1a-9031f82dd9f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "for pop in [\"smle_pzq_er\", \"smle_pzq_es\", \"smbre\", \"smor\", \"smeg\"]:\n",
    "\n",
    "    df =  pd.read_csv(f\"{pop}_ind_reads_with_smpomuc_signal.csv\", sep=\",\")\n",
    "    # Initialize a list to hold collapsed intervals\n",
    "    collapsed = []\n",
    "\n",
    "    # Set initial values to the first row\n",
    "    last_chr = df.iloc[0]['chr']\n",
    "    last_start = df.iloc[0]['chr_start']\n",
    "    last_stop = df.iloc[0]['chr_stop']\n",
    "    count = 1\n",
    "    s_seqids = [df.iloc[0]['s_seqid']]  # Initialize with the first s_seqid\n",
    "\n",
    "    for _, row in df.iterrows():\n",
    "        # Check if the current interval overlaps with the last one\n",
    "        if row['chr'] == last_chr and row['chr_start'] <= last_stop:\n",
    "            # Update the end if needed\n",
    "            last_stop = max(last_stop, row['chr_stop'])\n",
    "            count += 1\n",
    "            s_seqids.append(row['s_seqid'])\n",
    "        else:\n",
    "            # If no overlap, append the last interval, its count, and s_seqids to collapsed\n",
    "            collapsed.append([last_chr, last_start, last_stop, count, s_seqids])\n",
    "            last_chr, last_start, last_stop = row['chr'], row['chr_start'], row['chr_stop']\n",
    "            count = 1\n",
    "            s_seqids = [row['s_seqid']]  # Reset s_seqids list for the new interval\n",
    "\n",
    "    # Append the last interval\n",
    "    collapsed.append([last_chr, last_start, last_stop, count, s_seqids])\n",
    "\n",
    "    # Convert collapsed to a DataFrame\n",
    "    collapsed_df = pd.DataFrame(collapsed, columns=['chr', 'chr_start', 'chr_stop', 'element_count', 's_seqids'])\n",
    "    collapsed_df.to_csv(f\"{pop}_collapsed_smpomuc_intervals.csv\", sep=\",\", header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "7dc6f731-65ac-4e88-a0a9-16520b8dcd4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "dfs={}\n",
    "for pop in [\"smle_pzq_er\", \"smle_pzq_es\", \"smbre\", \"smor\", \"smeg\"]:\n",
    "    df =  pd.read_csv(f\"{pop}_collapsed_smpomuc_intervals.csv\", sep=\",\")\n",
    "    dfs[pop] = df.sort_values(by=['chr', 'chr_start'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "d00b3f7a-fcbc-439c-9808-7332347ca96d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_start</th>\n",
       "      <th>chr_stop</th>\n",
       "      <th>element_count</th>\n",
       "      <th>s_seqids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38363642</td>\n",
       "      <td>38420684</td>\n",
       "      <td>12</td>\n",
       "      <td>['c0ff4693-325f-43c9-ac1c-9832f1c76b08', 'c0ff...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30194954</td>\n",
       "      <td>30399004</td>\n",
       "      <td>45</td>\n",
       "      <td>['7b16334e-76a8-452a-a519-fad791876f56', '66d3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30413836</td>\n",
       "      <td>30461216</td>\n",
       "      <td>16</td>\n",
       "      <td>['e6dd5aa6-c496-41cc-b0d1-8dba89b6846b', '6be9...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601631.3</td>\n",
       "      <td>5441460</td>\n",
       "      <td>5441546</td>\n",
       "      <td>2</td>\n",
       "      <td>['49132cb7-ae19-4ef6-b6a7-34f15a6229fb', 'bb59...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          chr  chr_start  chr_stop  element_count  \\\n",
       "0  HE601625.3   38363642  38420684             12   \n",
       "1  HE601627.3   30194954  30399004             45   \n",
       "2  HE601627.3   30413836  30461216             16   \n",
       "3  HE601631.3    5441460   5441546              2   \n",
       "\n",
       "                                            s_seqids  \n",
       "0  ['c0ff4693-325f-43c9-ac1c-9832f1c76b08', 'c0ff...  \n",
       "1  ['7b16334e-76a8-452a-a519-fad791876f56', '66d3...  \n",
       "2  ['e6dd5aa6-c496-41cc-b0d1-8dba89b6846b', '6be9...  \n",
       "3  ['49132cb7-ae19-4ef6-b6a7-34f15a6229fb', 'bb59...  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs[\"smle_pzq_er\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "964bd4ff-e9b7-4095-907d-0a66b334fcb0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_start</th>\n",
       "      <th>chr_stop</th>\n",
       "      <th>element_count</th>\n",
       "      <th>s_seqids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38283194</td>\n",
       "      <td>38445490</td>\n",
       "      <td>39</td>\n",
       "      <td>['945315fe-8a3b-4002-bf71-be7fe9ee8add', '9453...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>39684131</td>\n",
       "      <td>39707800</td>\n",
       "      <td>1</td>\n",
       "      <td>['61a28ffc-4302-4cb3-94a9-8a078f28ddc3']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601626.3</td>\n",
       "      <td>2249531</td>\n",
       "      <td>2249625</td>\n",
       "      <td>1</td>\n",
       "      <td>['fac960cf-0836-49e6-b3a6-0549b24c9a3b']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30147379</td>\n",
       "      <td>30395686</td>\n",
       "      <td>94</td>\n",
       "      <td>['d5b70a33-b833-450a-8c58-e9b2a11148d1', '1095...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30400670</td>\n",
       "      <td>30493911</td>\n",
       "      <td>28</td>\n",
       "      <td>['84c2e89e-8bdd-4a73-8839-c2f20716d5ff', '10c6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>35200854</td>\n",
       "      <td>35201780</td>\n",
       "      <td>1</td>\n",
       "      <td>['86cbf181-83cc-46b8-aa03-c2ae3a8be6b3']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HE601631.3</td>\n",
       "      <td>3769918</td>\n",
       "      <td>3774437</td>\n",
       "      <td>1</td>\n",
       "      <td>['23c8af2f-c555-4c4e-9785-f791cdd53d2f']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HE601631.3</td>\n",
       "      <td>5441462</td>\n",
       "      <td>5441547</td>\n",
       "      <td>2</td>\n",
       "      <td>['a99d9f7c-db65-4d8f-8b50-b2e3d1fccc5f', '4c58...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>OU426848.1</td>\n",
       "      <td>9813261</td>\n",
       "      <td>9815366</td>\n",
       "      <td>1</td>\n",
       "      <td>['2306227b-08a2-49ec-afae-ba9e6c97a40f']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>OU426848.1</td>\n",
       "      <td>15768465</td>\n",
       "      <td>15769389</td>\n",
       "      <td>1</td>\n",
       "      <td>['456b4175-db0e-4eae-9e7e-b402f95f68ca']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          chr  chr_start  chr_stop  element_count  \\\n",
       "0  HE601625.3   38283194  38445490             39   \n",
       "1  HE601625.3   39684131  39707800              1   \n",
       "2  HE601626.3    2249531   2249625              1   \n",
       "3  HE601627.3   30147379  30395686             94   \n",
       "4  HE601627.3   30400670  30493911             28   \n",
       "5  HE601627.3   35200854  35201780              1   \n",
       "6  HE601631.3    3769918   3774437              1   \n",
       "7  HE601631.3    5441462   5441547              2   \n",
       "8  OU426848.1    9813261   9815366              1   \n",
       "9  OU426848.1   15768465  15769389              1   \n",
       "\n",
       "                                            s_seqids  \n",
       "0  ['945315fe-8a3b-4002-bf71-be7fe9ee8add', '9453...  \n",
       "1           ['61a28ffc-4302-4cb3-94a9-8a078f28ddc3']  \n",
       "2           ['fac960cf-0836-49e6-b3a6-0549b24c9a3b']  \n",
       "3  ['d5b70a33-b833-450a-8c58-e9b2a11148d1', '1095...  \n",
       "4  ['84c2e89e-8bdd-4a73-8839-c2f20716d5ff', '10c6...  \n",
       "5           ['86cbf181-83cc-46b8-aa03-c2ae3a8be6b3']  \n",
       "6           ['23c8af2f-c555-4c4e-9785-f791cdd53d2f']  \n",
       "7  ['a99d9f7c-db65-4d8f-8b50-b2e3d1fccc5f', '4c58...  \n",
       "8           ['2306227b-08a2-49ec-afae-ba9e6c97a40f']  \n",
       "9           ['456b4175-db0e-4eae-9e7e-b402f95f68ca']  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs[\"smle_pzq_es\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "ac2f2559-4037-48fe-8ad6-42ee4c40548b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_start</th>\n",
       "      <th>chr_stop</th>\n",
       "      <th>element_count</th>\n",
       "      <th>s_seqids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38351497</td>\n",
       "      <td>38420684</td>\n",
       "      <td>20</td>\n",
       "      <td>['5664ce76-28a2-43d5-86f5-2dd0146745be', '5664...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601626.3</td>\n",
       "      <td>15820345</td>\n",
       "      <td>15821272</td>\n",
       "      <td>1</td>\n",
       "      <td>['971c58f1-77e6-48a6-b57c-d41f429e9fa7']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30187303</td>\n",
       "      <td>30364768</td>\n",
       "      <td>56</td>\n",
       "      <td>['08cba9c5-51bb-4484-8106-1d9028d4718b', '6736...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30372266</td>\n",
       "      <td>30397465</td>\n",
       "      <td>14</td>\n",
       "      <td>['256c0c7f-03c4-4239-a114-bb0aa3e907e9', 'eb38...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30404618</td>\n",
       "      <td>30460203</td>\n",
       "      <td>15</td>\n",
       "      <td>['0f94b17d-97ba-4064-a8ea-f6b42f7021b8', '4d05...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>32215702</td>\n",
       "      <td>32216512</td>\n",
       "      <td>1</td>\n",
       "      <td>['03290737-30bd-415f-8475-9cae667f5f73']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>35198194</td>\n",
       "      <td>35199654</td>\n",
       "      <td>2</td>\n",
       "      <td>['1dba2e6e-1a9e-42db-aeaf-f9b1f9d8c5cc', 'e582...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HE601628.3</td>\n",
       "      <td>23177462</td>\n",
       "      <td>23177855</td>\n",
       "      <td>1</td>\n",
       "      <td>['086d5515-8bb3-407c-a142-b447c29f7aa8']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HE601631.3</td>\n",
       "      <td>5441462</td>\n",
       "      <td>5441547</td>\n",
       "      <td>1</td>\n",
       "      <td>['7c8f3faa-5ea0-4a94-85ad-8ec5a7384a55']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          chr  chr_start  chr_stop  element_count  \\\n",
       "0  HE601625.3   38351497  38420684             20   \n",
       "1  HE601626.3   15820345  15821272              1   \n",
       "2  HE601627.3   30187303  30364768             56   \n",
       "3  HE601627.3   30372266  30397465             14   \n",
       "4  HE601627.3   30404618  30460203             15   \n",
       "5  HE601627.3   32215702  32216512              1   \n",
       "6  HE601627.3   35198194  35199654              2   \n",
       "7  HE601628.3   23177462  23177855              1   \n",
       "8  HE601631.3    5441462   5441547              1   \n",
       "\n",
       "                                            s_seqids  \n",
       "0  ['5664ce76-28a2-43d5-86f5-2dd0146745be', '5664...  \n",
       "1           ['971c58f1-77e6-48a6-b57c-d41f429e9fa7']  \n",
       "2  ['08cba9c5-51bb-4484-8106-1d9028d4718b', '6736...  \n",
       "3  ['256c0c7f-03c4-4239-a114-bb0aa3e907e9', 'eb38...  \n",
       "4  ['0f94b17d-97ba-4064-a8ea-f6b42f7021b8', '4d05...  \n",
       "5           ['03290737-30bd-415f-8475-9cae667f5f73']  \n",
       "6  ['1dba2e6e-1a9e-42db-aeaf-f9b1f9d8c5cc', 'e582...  \n",
       "7           ['086d5515-8bb3-407c-a142-b447c29f7aa8']  \n",
       "8           ['7c8f3faa-5ea0-4a94-85ad-8ec5a7384a55']  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs[\"smbre\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "2f646385-a0e3-488a-beb7-38755b792eb9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_start</th>\n",
       "      <th>chr_stop</th>\n",
       "      <th>element_count</th>\n",
       "      <th>s_seqids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38323935</td>\n",
       "      <td>38410524</td>\n",
       "      <td>24</td>\n",
       "      <td>['902c4e70-d14d-418c-8412-e4dc9865462f', '902c...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601626.3</td>\n",
       "      <td>2249529</td>\n",
       "      <td>2249653</td>\n",
       "      <td>1</td>\n",
       "      <td>['e07b74df-c837-45d1-8916-ef92a539cef5']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30171304</td>\n",
       "      <td>30393183</td>\n",
       "      <td>66</td>\n",
       "      <td>['9d89c37d-28ad-4bd9-a306-b65d6fdf07cd', '36a8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30403402</td>\n",
       "      <td>30451912</td>\n",
       "      <td>10</td>\n",
       "      <td>['0c96fcdb-ca56-4bb6-9226-9ad4c69ad9c8', 'e169...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601630.3</td>\n",
       "      <td>7058606</td>\n",
       "      <td>7059275</td>\n",
       "      <td>1</td>\n",
       "      <td>['d8602e6f-b2fc-4180-b3b6-1b250aac8a84']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HE601631.3</td>\n",
       "      <td>5441461</td>\n",
       "      <td>5441535</td>\n",
       "      <td>1</td>\n",
       "      <td>['1498fed9-e443-4181-b2b2-12a6033c1ede']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          chr  chr_start  chr_stop  element_count  \\\n",
       "0  HE601625.3   38323935  38410524             24   \n",
       "1  HE601626.3    2249529   2249653              1   \n",
       "2  HE601627.3   30171304  30393183             66   \n",
       "3  HE601627.3   30403402  30451912             10   \n",
       "4  HE601630.3    7058606   7059275              1   \n",
       "5  HE601631.3    5441461   5441535              1   \n",
       "\n",
       "                                            s_seqids  \n",
       "0  ['902c4e70-d14d-418c-8412-e4dc9865462f', '902c...  \n",
       "1           ['e07b74df-c837-45d1-8916-ef92a539cef5']  \n",
       "2  ['9d89c37d-28ad-4bd9-a306-b65d6fdf07cd', '36a8...  \n",
       "3  ['0c96fcdb-ca56-4bb6-9226-9ad4c69ad9c8', 'e169...  \n",
       "4           ['d8602e6f-b2fc-4180-b3b6-1b250aac8a84']  \n",
       "5           ['1498fed9-e443-4181-b2b2-12a6033c1ede']  "
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs[\"smor\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "f84b3555-d219-4b1b-b16e-ffbf8e8a932f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_start</th>\n",
       "      <th>chr_stop</th>\n",
       "      <th>element_count</th>\n",
       "      <th>s_seqids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38357980</td>\n",
       "      <td>38425380</td>\n",
       "      <td>14</td>\n",
       "      <td>['b79e0aeb-0a87-41fc-94b9-919b498379da', 'b79e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30184437</td>\n",
       "      <td>30229327</td>\n",
       "      <td>14</td>\n",
       "      <td>['44d7e30f-b722-4218-814a-323e549a39dc', 'b00e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30234783</td>\n",
       "      <td>30399004</td>\n",
       "      <td>31</td>\n",
       "      <td>['07b185a4-5938-410f-a2e0-117560f66754', '58a9...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30399104</td>\n",
       "      <td>30447423</td>\n",
       "      <td>15</td>\n",
       "      <td>['ebbd919c-5f80-41e7-8020-3fa71dc65e20', '8b2f...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>35197991</td>\n",
       "      <td>35201780</td>\n",
       "      <td>1</td>\n",
       "      <td>['1cc203d9-3d54-43b5-bd9e-81c7793394be']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HE601631.3</td>\n",
       "      <td>5441464</td>\n",
       "      <td>5441546</td>\n",
       "      <td>1</td>\n",
       "      <td>['e34e77a8-6659-449e-ab63-5f11a0a59066']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          chr  chr_start  chr_stop  element_count  \\\n",
       "0  HE601625.3   38357980  38425380             14   \n",
       "1  HE601627.3   30184437  30229327             14   \n",
       "2  HE601627.3   30234783  30399004             31   \n",
       "3  HE601627.3   30399104  30447423             15   \n",
       "4  HE601627.3   35197991  35201780              1   \n",
       "5  HE601631.3    5441464   5441546              1   \n",
       "\n",
       "                                            s_seqids  \n",
       "0  ['b79e0aeb-0a87-41fc-94b9-919b498379da', 'b79e...  \n",
       "1  ['44d7e30f-b722-4218-814a-323e549a39dc', 'b00e...  \n",
       "2  ['07b185a4-5938-410f-a2e0-117560f66754', '58a9...  \n",
       "3  ['ebbd919c-5f80-41e7-8020-3fa71dc65e20', '8b2f...  \n",
       "4           ['1cc203d9-3d54-43b5-bd9e-81c7793394be']  \n",
       "5           ['e34e77a8-6659-449e-ab63-5f11a0a59066']  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs[\"smeg\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b535ae3-faa9-4a9e-aa41-1db036f208fa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30541608-613c-41da-a728-53bfb71ffdfb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd75bf39-12ec-4d1a-a5c2-10482db27d78",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "162590de-d986-4897-8db8-d9b0ec0a0163",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "68d4a2ef-51fd-49b7-9b85-5a55e0503dde",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5624f982-2539-4b8b-aa90-3e3464d3e633",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "3289e483-b8fb-45eb-b8fe-250b0eeb89a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "ASSEM_DIR=\"/master/nplatt/sch_man_ont/results/assembly\"\n",
    "\n",
    "for POP in \"smle_pzq_er\" \"smle_pzq_es\" \"smbre\" \"smor\" \"smeg\"; do\n",
    "\n",
    "    # conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "    #     makeblastdb \\\n",
    "    #         -dbtype nucl \\\n",
    "    #         -in ${ASSEM_DIR}/${POP}_rnd0.fasta\n",
    "\n",
    "    conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "        blastn \\\n",
    "            -subject ${ASSEM_DIR}/${POP}_rnd0.fasta \\\n",
    "            -query nuc.fas \\\n",
    "            -out ${POP}_assem.blastn.tsv \\\n",
    "            -perc_identity 0.85 \\\n",
    "            -outfmt 6\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "id": "72d6f809-e364-4151-8aab-1c40fa8ffc60",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "    blastn \\\n",
    "        -subject /master/nplatt/sch_man_ont/data/genome/SM_V9.fa \\\n",
    "        -query nuc.fas \\\n",
    "        -out smv9_assem.blastn.tsv \\\n",
    "        -perc_identity 0.85 \\\n",
    "        -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "id": "a1f82759-7513-42a8-b203-87f9bc6acf45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>95.38</td>\n",
       "      <td>65</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>38377257</td>\n",
       "      <td>38377321</td>\n",
       "      <td>1.000000e-22</td>\n",
       "      <td>104.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>98.46</td>\n",
       "      <td>65</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30334716</td>\n",
       "      <td>30334652</td>\n",
       "      <td>6.000000e-26</td>\n",
       "      <td>115.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>98.44</td>\n",
       "      <td>64</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>65</td>\n",
       "      <td>30252316</td>\n",
       "      <td>30252253</td>\n",
       "      <td>2.000000e-25</td>\n",
       "      <td>113.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>96.92</td>\n",
       "      <td>65</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30203241</td>\n",
       "      <td>30203177</td>\n",
       "      <td>3.000000e-24</td>\n",
       "      <td>110.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>96.92</td>\n",
       "      <td>65</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30364285</td>\n",
       "      <td>30364221</td>\n",
       "      <td>3.000000e-24</td>\n",
       "      <td>110.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>93.85</td>\n",
       "      <td>65</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30426262</td>\n",
       "      <td>30426198</td>\n",
       "      <td>6.000000e-21</td>\n",
       "      <td>99.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>98.85</td>\n",
       "      <td>87</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>38364601</td>\n",
       "      <td>38364687</td>\n",
       "      <td>5.000000e-38</td>\n",
       "      <td>156.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>98.85</td>\n",
       "      <td>87</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>38413014</td>\n",
       "      <td>38413100</td>\n",
       "      <td>5.000000e-38</td>\n",
       "      <td>156.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30170448</td>\n",
       "      <td>30170362</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30232621</td>\n",
       "      <td>30232535</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30262656</td>\n",
       "      <td>30262570</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30434686</td>\n",
       "      <td>30434600</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30212841</td>\n",
       "      <td>30212755</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30274916</td>\n",
       "      <td>30274830</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30296637</td>\n",
       "      <td>30296551</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30348669</td>\n",
       "      <td>30348583</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30375260</td>\n",
       "      <td>30375174</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30396858</td>\n",
       "      <td>30396772</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     0           1       2   3   4   5   6   7         8   \\\n",
       "0        smpomuc_5p_nuc  HE601625.3   95.38  65   3   0   1  65  38377257   \n",
       "1        smpomuc_5p_nuc  HE601627.3   98.46  65   1   0   1  65  30334716   \n",
       "2        smpomuc_5p_nuc  HE601627.3   98.44  64   1   0   2  65  30252316   \n",
       "3        smpomuc_5p_nuc  HE601627.3   96.92  65   2   0   1  65  30203241   \n",
       "4        smpomuc_5p_nuc  HE601627.3   96.92  65   2   0   1  65  30364285   \n",
       "5        smpomuc_5p_nuc  HE601627.3   93.85  65   4   0   1  65  30426262   \n",
       "6   smpomuc_3p_exon_nuc  HE601625.3   98.85  87   1   0   1  87  38364601   \n",
       "7   smpomuc_3p_exon_nuc  HE601625.3   98.85  87   1   0   1  87  38413014   \n",
       "8   smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30170448   \n",
       "9   smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30232621   \n",
       "10  smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30262656   \n",
       "11  smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30434686   \n",
       "12  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30212841   \n",
       "13  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30274916   \n",
       "14  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30296637   \n",
       "15  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30348669   \n",
       "16  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30375260   \n",
       "17  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30396858   \n",
       "\n",
       "          9             10     11  \n",
       "0   38377321  1.000000e-22  104.0  \n",
       "1   30334652  6.000000e-26  115.0  \n",
       "2   30252253  2.000000e-25  113.0  \n",
       "3   30203177  3.000000e-24  110.0  \n",
       "4   30364221  3.000000e-24  110.0  \n",
       "5   30426198  6.000000e-21   99.0  \n",
       "6   38364687  5.000000e-38  156.0  \n",
       "7   38413100  5.000000e-38  156.0  \n",
       "8   30170362  1.000000e-39  161.0  \n",
       "9   30232535  1.000000e-39  161.0  \n",
       "10  30262570  1.000000e-39  161.0  \n",
       "11  30434600  1.000000e-39  161.0  \n",
       "12  30212755  5.000000e-33  139.0  \n",
       "13  30274830  5.000000e-33  139.0  \n",
       "14  30296551  5.000000e-33  139.0  \n",
       "15  30348583  5.000000e-33  139.0  \n",
       "16  30375174  5.000000e-33  139.0  \n",
       "17  30396772  5.000000e-33  139.0  "
      ]
     },
     "execution_count": 347,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "id": "d893758f-fc6c-4cab-b648-dc0d248ed9be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>98.46</td>\n",
       "      <td>65</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30334716</td>\n",
       "      <td>30334652</td>\n",
       "      <td>6.000000e-26</td>\n",
       "      <td>115.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>98.44</td>\n",
       "      <td>64</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>65</td>\n",
       "      <td>30252316</td>\n",
       "      <td>30252253</td>\n",
       "      <td>2.000000e-25</td>\n",
       "      <td>113.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>96.92</td>\n",
       "      <td>65</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30203241</td>\n",
       "      <td>30203177</td>\n",
       "      <td>3.000000e-24</td>\n",
       "      <td>110.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>96.92</td>\n",
       "      <td>65</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30364285</td>\n",
       "      <td>30364221</td>\n",
       "      <td>3.000000e-24</td>\n",
       "      <td>110.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>93.85</td>\n",
       "      <td>65</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30426262</td>\n",
       "      <td>30426198</td>\n",
       "      <td>6.000000e-21</td>\n",
       "      <td>99.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30170448</td>\n",
       "      <td>30170362</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30232621</td>\n",
       "      <td>30232535</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30262656</td>\n",
       "      <td>30262570</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30434686</td>\n",
       "      <td>30434600</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30212841</td>\n",
       "      <td>30212755</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30274916</td>\n",
       "      <td>30274830</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30296637</td>\n",
       "      <td>30296551</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30348669</td>\n",
       "      <td>30348583</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30375260</td>\n",
       "      <td>30375174</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30396858</td>\n",
       "      <td>30396772</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     0           1       2   3   4   5   6   7         8   \\\n",
       "1        smpomuc_5p_nuc  HE601627.3   98.46  65   1   0   1  65  30334716   \n",
       "2        smpomuc_5p_nuc  HE601627.3   98.44  64   1   0   2  65  30252316   \n",
       "3        smpomuc_5p_nuc  HE601627.3   96.92  65   2   0   1  65  30203241   \n",
       "4        smpomuc_5p_nuc  HE601627.3   96.92  65   2   0   1  65  30364285   \n",
       "5        smpomuc_5p_nuc  HE601627.3   93.85  65   4   0   1  65  30426262   \n",
       "8   smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30170448   \n",
       "9   smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30232621   \n",
       "10  smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30262656   \n",
       "11  smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30434686   \n",
       "12  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30212841   \n",
       "13  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30274916   \n",
       "14  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30296637   \n",
       "15  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30348669   \n",
       "16  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30375260   \n",
       "17  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30396858   \n",
       "\n",
       "          9             10     11  \n",
       "1   30334652  6.000000e-26  115.0  \n",
       "2   30252253  2.000000e-25  113.0  \n",
       "3   30203177  3.000000e-24  110.0  \n",
       "4   30364221  3.000000e-24  110.0  \n",
       "5   30426198  6.000000e-21   99.0  \n",
       "8   30170362  1.000000e-39  161.0  \n",
       "9   30232535  1.000000e-39  161.0  \n",
       "10  30262570  1.000000e-39  161.0  \n",
       "11  30434600  1.000000e-39  161.0  \n",
       "12  30212755  5.000000e-33  139.0  \n",
       "13  30274830  5.000000e-33  139.0  \n",
       "14  30296551  5.000000e-33  139.0  \n",
       "15  30348583  5.000000e-33  139.0  \n",
       "16  30375174  5.000000e-33  139.0  \n",
       "17  30396772  5.000000e-33  139.0  "
      ]
     },
     "execution_count": 350,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x.loc[x[8] > x[9]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "id": "f27465cb-2428-42f4-99b2-ed4df50c7a9f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>98.46</td>\n",
       "      <td>65</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30334716</td>\n",
       "      <td>30334652</td>\n",
       "      <td>6.000000e-26</td>\n",
       "      <td>115.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>98.44</td>\n",
       "      <td>64</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>65</td>\n",
       "      <td>30252316</td>\n",
       "      <td>30252253</td>\n",
       "      <td>2.000000e-25</td>\n",
       "      <td>113.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>96.92</td>\n",
       "      <td>65</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30203241</td>\n",
       "      <td>30203177</td>\n",
       "      <td>3.000000e-24</td>\n",
       "      <td>110.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>96.92</td>\n",
       "      <td>65</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30364285</td>\n",
       "      <td>30364221</td>\n",
       "      <td>3.000000e-24</td>\n",
       "      <td>110.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>93.85</td>\n",
       "      <td>65</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65</td>\n",
       "      <td>30426262</td>\n",
       "      <td>30426198</td>\n",
       "      <td>6.000000e-21</td>\n",
       "      <td>99.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30170448</td>\n",
       "      <td>30170362</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30232621</td>\n",
       "      <td>30232535</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30262656</td>\n",
       "      <td>30262570</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>100.00</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30434686</td>\n",
       "      <td>30434600</td>\n",
       "      <td>1.000000e-39</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30212841</td>\n",
       "      <td>30212755</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30274916</td>\n",
       "      <td>30274830</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30296637</td>\n",
       "      <td>30296551</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30348669</td>\n",
       "      <td>30348583</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30375260</td>\n",
       "      <td>30375174</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>95.40</td>\n",
       "      <td>87</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>30396858</td>\n",
       "      <td>30396772</td>\n",
       "      <td>5.000000e-33</td>\n",
       "      <td>139.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     0           1       2   3   4   5   6   7         8   \\\n",
       "1        smpomuc_5p_nuc  HE601627.3   98.46  65   1   0   1  65  30334716   \n",
       "2        smpomuc_5p_nuc  HE601627.3   98.44  64   1   0   2  65  30252316   \n",
       "3        smpomuc_5p_nuc  HE601627.3   96.92  65   2   0   1  65  30203241   \n",
       "4        smpomuc_5p_nuc  HE601627.3   96.92  65   2   0   1  65  30364285   \n",
       "5        smpomuc_5p_nuc  HE601627.3   93.85  65   4   0   1  65  30426262   \n",
       "8   smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30170448   \n",
       "9   smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30232621   \n",
       "10  smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30262656   \n",
       "11  smpomuc_3p_exon_nuc  HE601627.3  100.00  87   0   0   1  87  30434686   \n",
       "12  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30212841   \n",
       "13  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30274916   \n",
       "14  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30296637   \n",
       "15  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30348669   \n",
       "16  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30375260   \n",
       "17  smpomuc_3p_exon_nuc  HE601627.3   95.40  87   4   0   1  87  30396858   \n",
       "\n",
       "          9             10     11  \n",
       "1   30334652  6.000000e-26  115.0  \n",
       "2   30252253  2.000000e-25  113.0  \n",
       "3   30203177  3.000000e-24  110.0  \n",
       "4   30364221  3.000000e-24  110.0  \n",
       "5   30426198  6.000000e-21   99.0  \n",
       "8   30170362  1.000000e-39  161.0  \n",
       "9   30232535  1.000000e-39  161.0  \n",
       "10  30262570  1.000000e-39  161.0  \n",
       "11  30434600  1.000000e-39  161.0  \n",
       "12  30212755  5.000000e-33  139.0  \n",
       "13  30274830  5.000000e-33  139.0  \n",
       "14  30296551  5.000000e-33  139.0  \n",
       "15  30348583  5.000000e-33  139.0  \n",
       "16  30375174  5.000000e-33  139.0  \n",
       "17  30396772  5.000000e-33  139.0  "
      ]
     },
     "execution_count": 353,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "xs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "id": "12aa58e8-a2c0-4524-b5a6-de8c77069443",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1     30334716\n",
       "2     30252316\n",
       "3     30203241\n",
       "4     30364285\n",
       "5     30426262\n",
       "8     30170448\n",
       "9     30232621\n",
       "10    30262656\n",
       "11    30434686\n",
       "12    30212841\n",
       "13    30274916\n",
       "14    30296637\n",
       "15    30348669\n",
       "16    30375260\n",
       "17    30396858\n",
       "Name: 8, dtype: int64"
      ]
     },
     "execution_count": 359,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "xs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 367,
   "id": "87f60c10-5cdd-4367-82f7-72fe42c4dddf",
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "list indices must be integers or slices, not list",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [367]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mcolor_list\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;28;43mlist\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mx\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m==\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mHE601627.3\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m]\u001b[49m\n",
      "\u001b[0;31mTypeError\u001b[0m: list indices must be integers or slices, not list"
     ]
    }
   ],
   "source": [
    "color_list[]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "id": "6b966ba6-dafe-40c6-aa58-5ef9c95d94e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXoAAAEFCAYAAADt1CyEAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAAsTAAALEwEAmpwYAAAXtUlEQVR4nO3df5RdZX3v8fc3ycT5BSZegoaEJoAISSkKjoigFkU0gBCl2CbqxXrtYrmWgj+uq4vaLvXWQKuLVuVWpSDcEKCkS9QlQdRbBMrVdUEHCSEQiKm0JJILoRgg5PfM9/6xT2Ayc2bOmcwZMj68X6yzcvZ+nmc/z3P2Pp/ZZ589Q2QmkqRyTdrfA5AkjS+DXpIKZ9BLUuEMekkqnEEvSYWbsr8HUM9BBx2Uc+fO3d/DkKTfGffcc8+TmTmjXtmEDPq5c+fS29u7v4chSb8zIuI/hivz0o0kFc6gl6TCGfSSVDiDXpIKZ9BLUuEMekkqnEEvSYUz6CWpcAa9JBXOoJekwhn0klQ4g16SCmfQS1LhDHpJKpxBL0mFM+glqXAGvSQVzqCXpMIZ9JJUOINekgpn0EtS4Qx6SSqcQS9JhTPoJalwBr0kFa5h0EfE1RHxRESsHqY8IuKyiFgXEasi4vhB5ZMj4t6IuLlVg5YkNa+ZM/qlwIIRyk8Hjqw9zge+Oaj8E8CafRmcJGnsGgZ9Zt4JPDVClYXAsqzcBUyLiJkAETEbOBP4VisGK0kavVZco58FrB+wvKG2DuCrwJ8D/Y02EhHnR0RvRPRu2rSpBcOSJEFrgj7qrMuIeDfwRGbe08xGMvOKzOzJzJ4ZM2a0YFiSJGhN0G8ADh2wPBt4DDgZODsi/h1YDrw9Iq5rQX+SpFFoRdDfBJxXu/vmRODpzNyYmX+RmbMzcy6wCLgtMz/Ygv4kSaMwpVGFiLgBOAU4KCI2AJ8H2gAy83LgFuAMYB2wFfjweA1WkjR6DYM+Mxc3KE/gYw3q3AHcMZqBSZJaw9+MlaTCGfSSVDiDXpIKZ9BLUuEMekkqnEEvSYUz6CWpcAa9JBXOoJekwhn0klQ4g16SCmfQS1LhDHpJKpxBL0mFM+glqXAGvSQVzqCXpMIZ9JJUOINekgpn0EtS4Qx6SSqcQS9JhTPoJalwBr0kFc6gl6TCGfSSVDiDXpIKZ9BLUuEMekkqnEEvSYVrGPQRcXVEPBERq4cpj4i4LCLWRcSqiDi+tv7QiLg9ItZExAMR8YlWD16S1FgzZ/RLgQUjlJ8OHFl7nA98s7Z+N/DfM3MecCLwsYiYv+9DlSTti4ZBn5l3Ak+NUGUhsCwrdwHTImJmZm7MzF/WtvEssAaY1YpBS5Ka14pr9LOA9QOWNzAo0CNiLnAccHcL+pMkjUIrgj7qrMvnCyO6ge8An8zMZ4bdSMT5EdEbEb2bNm1qwbAkSdCaoN8AHDpgeTbwGEBEtFGF/PWZ+d2RNpKZV2RmT2b2zJgxowXDkiRBa4L+JuC82t03JwJPZ+bGiAjgKmBNZv59C/qRJO2DKY0qRMQNwCnAQRGxAfg80AaQmZcDtwBnAOuArcCHa01PBv4rcH9ErKyt+2xm3tLC8UuSGmgY9Jm5uEF5Ah+rs/6n1L9+L0l6EfmbsZJUOINekgpn0EtS4Qx6SSqcQS9JhTPoJalwBr0kFc6gl6TCGfSSVDiDXpIKZ9BLUuEMekkqnEEvSYUz6CWpcAa9JBXOoJekwhn0klQ4g16SCmfQS1LhDHpJKpxBL0mFM+glqXAGvSQVzqCXpMIZ9JJUOINekgpn0EtS4Qx6SSqcQS9JhTPoJalwU/b3AFrmV7+C22+H6dPh3e+Gjo79PSIyk7s23MV9j9/HEdOP4NTDT2VSjPCztbe3esyZA+98J0ye3JqB9PXBrbfCI4/AccfBCSdAREs2/fT2p1mxdgU7du9gwasXMOvAWSM3ePJJuPnm6vmZZ8KMGc119OyzsGIFbN1avTa/93tjG3g9W7ZUfWzZUvUxZ07r+2ihJ7c+yQ/W/oAkOfPIM5nR1eRrucfmzdW+2LEDTj8dDjlkn8bx222/5ea1N7Orfxenv/p0Zh4ws3Gj1avhZz+DV72q6nvq1H3qe1iZcPfdsHIlHH44nHpq695Po9Cf/dz2yG2se2odf3DwH3DSoScR9d57W7ZU++LZZ+G002Du3NYOJDNHfABXA08Aq4cpD+AyYB2wCjh+QNkC4OFa2UWN+trzeP3rX59N6+/PvPDCzPb2zM7OzAMOyHz5yzN//vPmtzEOntv5XJ581cnZdXFXdizpyAMuOSBf8z9fk49veXxo5R07Mt/5zsyursyOjmoOc+ZkPvro2Afy2GOZhx9ebbOjo+rjlFMyt20b86ZvWXtLdl7cmd2XdGfnxZ3ZvqQ9L/3ZpcM3WLas2k/d3dWjvT3zmmsad3TrrdW4Dzig2sft7Zlf/OKYx7+X22+vxjSwj899rrV9tNCylcuyfUl7dl/Snd2XdGf7kvZceu/S5jewYkU1z+7u6rVtb8/8yldGPY7vrfledizpyO5LurPr4q5sX9Kel9192fAN+voyFy+ujsU9x/rBB2euWTPqvoe1dWvmH/7h3u+nI47I3LixdX00YdNzm3LeP8zL7ku6s2NJR3Zd3JUnfuvEfHbHs3tX/Nd/rcY48Nj7y78cdX9Abw6X48MV5Ath/Vbg+BGC/gzgh7XAPxG4u7Z+MvBvwOHAVOA+YH6j/nK0Qb9iRbVDq5/hLzwOPjhz9+5Rv1it8pkffybbl7QnX+D5R9tft+VZ/3TW0MoXX1wdkAPHP3ly5lveMvaBnHZata2B2+7oyPyrvxrTZp/e/nR2Xty51/z4Atm5pDNXblw5tMH69dUBPHg/dXSM/APtueeqN8Dgdp2dmXfdNaY5PG/btswDD6zfx09/2po+Wmj90+uHHFt8gexY0pGPbm7i5OC3v63mVm9f3H9/0+P4z63/mR1LOuqO48EnHqzf6Kqrhr5fIzKPPro6aWuFiy4aeqxNmZK5YEFrtt+kc5afk21/3bbXa/OyL74sL7zlwhcqbd9enZgO3hddXZl33DGq/kYK+obX6DPzTuCpEaosBJbV+roLmBYRM4ETgHWZ+evM3Aksr9VtrSuugOeeG7p+2za4666Wd9espfctZfvu7Xut29W/ix+t+xE7du/Yu/KVV1bjHaivr/ro+dRIL30DW7bAHXdU2xpo2za4+up93y7wg7U/YHIM/Si8vW87y1YtG9rgxhvrb6i/f/gygB//uP5lpu3bYenS5gbbyK231l/fgtdpPHznwe/UXd+f/Xz7wW833sCKFTCpzlt/50647rqmx3HTwzfVPQZ29e3i+vuvr9/o8suHvl8z4dFHYd26pvse0dKl1fEx0O7d8JOfVJf+XgS7+3ezYu0KdvXv2mv9jr4dXLvq2hdW3HZbNf/Btm5t6bHXii9jZwHrByxvqK0bbn1dEXF+RPRGRO+mTZua733HjvrrI4YvexHs6ttVd32S9OWg4N1Vvy4Rw5c1Y3DAN9Nnk3b27dzziW4vmcn2XdvrNNhZfzx9fSPvp507678R+vuHvpn31c6d9ddnDv0BPAHs6NtBX//Q17Iv+4aeRNTTotd0Z99O+umvO47BJznPG25fT5rUuvfrcPsTRn5PtFB/9tOfQ18bYO/wf5GOvVYEfb1v9XKE9XVl5hWZ2ZOZPTOa/YIO4AMfgK6uehuEk05qfjsttvCohUyZtPd33UHQc0gPnW2de1c+99z6X0Ydfji88pX7PoiXvxyOOWbo+rY2OOecfd8u8K5Xv4vduXvI+s62Ts6df+7QBmedBVPqfPff1gZnnz18R6edVv+HUlcX/PEfj2LEI3j724fvY9Gi1vTRQme95qwhxxZA26Q2zj5qhNdyj9NPrx94nZ3wR3/U9DjOOPKMumHW2dbJOfOGOb7e//76N0p0d8P8+U33PaL3vrc6rgaKgGOPhQMOaE0fDUydPJWTDz2ZGBSDU2IKZ73mrBdWvO1twx97ixe3bDytCPoNwKEDlmcDj42wvrXe//4q0PeE/dSp1YG0dCm0t7e8u2Z9+bQvM7N7Jl1t1bg6p3QyrX0aV59d5+PY5z9f3eHR3V0td3RUB+QoPkYP65prqsDf8+bq7obZs2HJkjFt9lXdr+JL7/gSHVM6mByTCYKuti7eN/99nDL3lKEN5s2DT32qCpNJk6pHZyd88pMjv8Ff8Qq47LJq/Ht+UHR1VXdWLVgwpjk8b9o0+MY3hvbxrndV/Uww82bM49Nv+jSdbZ1Mqv3X2dbJhW+8kN8/+Pcbb+CQQ+CSS6r5Tp5cheCeYHnzm5sex+wDZ/PFt31xr2Ogs62TDx77Qd40+031G11wQbW/9xzrL3tZdRzccEP9y0n74m/+pprjnkzo7KzeA6261NekK8++kukd058/setq6+Lg7oO59J2XvlDpwAPhH/9x6LH3jnfAwtZd6Y56H7+HVIqYC9ycmUNODyPiTODjVF/KvhG4LDNPiIgpwFrgVOA3wC+A92fmA4366+npyd7e3uZn0d8PP/wh3HILHHQQ/OmfwmGHNd9+nGzdtZXlq5fzi9/8gqMPOprzXnse0zum16+8fXt1rfqnP4Ujj4QPfaiaSys89VQV+GvXwhvfCH/yJy27/XT1E6u5btV1bNu1jXPmncNb57y1/u1je/T2wvLl1SeuRYvgDW9orqOHHoJrr61uP1u4sDoLb9Etos97+OGqj82b4T3vqW7Ja3UfLdT7WC/LVy8nM1l0zCLeMKvJ13KP+++vTia2b6/O5N/yln2a76rHV3HdquvY2beTc+efW53JjrSdXbvg+9+vrk/Pnl0d67Ma3JY7Wtu2wT//c/U911FHwXnnVScNL7LN2zdz7X3X8sCmB+g5pIfFxyyma2qdKxBr18KyZdWxt3BhFfSj3BcRcU9m9tQtaxT0EXEDcApwEPA48HmgDSAzL49qj/4D1a2UW4EPZ2Zvre0ZwFep7sC5OjMvbmbAow56SXqJGynoG/7CVGaOeKGodlvPx4YpuwW4pZlBSpLGh38CQZIKZ9BLUuEMekkqnEEvSYUz6CWpcAa9JBXOoJekwhn0klQ4g16SCmfQS1LhDHpJKpxBL0mFM+glqXAGvSQVzqCXpMIZ9JJUOINekgpn0EtS4Qx6SSqcQS9JhTPoJalwBr0kFc6gl6TCGfSSVDiDXpIKZ9BLUuEMekkqnEEvSYUz6CWpcAa9JBWuqaCPiAUR8XBErIuIi+qUT4+I70XEqoj4eUQcM6DsUxHxQESsjogbIqK9lROQJI2sYdBHxGTg68DpwHxgcUTMH1Tts8DKzDwWOA/4Wq3tLOBCoCczjwEmA4taN3xJUiPNnNGfAKzLzF9n5k5gObBwUJ35wE8AMvMhYG5EvLJWNgXoiIgpQCfwWEtGLklqSjNBPwtYP2B5Q23dQPcB5wBExAnAHGB2Zv4GuBR4FNgIPJ2Z/3usg5YkNa+ZoI8663LQ8t8C0yNiJXABcC+wOyKmU539HwYcAnRFxAfrdhJxfkT0RkTvpk2bmh2/JKmBZoJ+A3DogOXZDLr8kpnPZOaHM/N1VNfoZwCPAO8AHsnMTZm5C/gucFK9TjLziszsycyeGTNmjH4mkqS6mgn6XwBHRsRhETGV6svUmwZWiIhptTKAPwPuzMxnqC7ZnBgRnRERwKnAmtYNX5LUyJRGFTJzd0R8HPgx1V0zV2fmAxHx0Vr55cA8YFlE9AEPAh+pld0dETcCvwR2U13SuWJcZiJJqisyB19u3/96enqyt7d3fw9Dkn5nRMQ9mdlTr8zfjJWkwhn0klQ4g16SCmfQS1LhDHpJKpxBL0mFM+glqXAGvSQVzqCXpMIZ9JJUOINekgpn0EtS4Qx6SSqcQS9JhTPoJalwBr0kFc6gl6TCGfSSVDiDXpIKZ9BLUuEMekkqnEEvSYUz6CWpcAa9JBXOoJekwhn0klQ4g16SCmfQS1LhDHpJKpxBL0mFM+glqXBNBX1ELIiIhyNiXURcVKd8ekR8LyJWRcTPI+KYAWXTIuLGiHgoItZExJtaOQFJ0sgaBn1ETAa+DpwOzAcWR8T8QdU+C6zMzGOB84CvDSj7GvCjzDwaeC2wphUDlyQ1p5kz+hOAdZn568zcCSwHFg6qMx/4CUBmPgTMjYhXRsSBwFuBq2plOzNzc6sGL0lqrJmgnwWsH7C8obZuoPuAcwAi4gRgDjAbOBzYBPyviLg3Ir4VEV31OomI8yOiNyJ6N23aNMppSJKG00zQR511OWj5b4HpEbESuAC4F9gNTAGOB76ZmccBzwFDrvEDZOYVmdmTmT0zZsxocviSpEamNFFnA3DogOXZwGMDK2TmM8CHASIigEdqj05gQ2beXat6I8MEvSRpfDRzRv8L4MiIOCwipgKLgJsGVqjdWTO1tvhnwJ2Z+Uxm/j9gfUQcVSs7FXiwRWOXJDWh4Rl9Zu6OiI8DPwYmA1dn5gMR8dFa+eXAPGBZRPRRBflHBmziAuD62g+CX1M785ckvTgic/Dl9v2vp6cne3t79/cwJOl3RkTck5k99cr8zVhJKpxBL0mFM+glqXAGvSQVzqCXpMIZ9JJUOINekgpn0EtS4Qx6SSqcQS9JhTPoJalwBr0kFc6gl6TCGfSSVDiDXpIKZ9BLUuEMekkqnEEvSYUz6CWpcAa9JBXOoJekwhn0klQ4g16SCmfQS1LhIjP39xiGiIhNwH/s73GM0UHAk/t7EOPspTBHcJ4lKXmOczJzRr2CCRn0JYiI3szs2d/jGE8vhTmC8yzJS2GO9XjpRpIKZ9BLUuEM+vFzxf4ewIvgpTBHcJ4leSnMcQiv0UtS4Tyjl6TCGfSSVDiDvoGIaI+In0fEfRHxQET8jzp1IiIui4h1EbEqIo6vrT80Im6PiDW1tp8Y0OYVEfEvEfGr2r/TX8x5DTaO8/xCRPwmIlbWHme8mPMaNP6xzHHYtoXty5HmOWH2ZaOxDqhTd54DyidHxL0RcfOAdRNqf7ZEZvoY4QEE0F173gbcDZw4qM4ZwA9rdU8E7q6tnwkcX3t+ALAWmF9b/jJwUe35RcCXCp3nF4DP7O/92II5Dtu2sH050jwnzL4c6zwHlH8a+Cfg5gHrJtT+bMXDM/oGsrKltthWewz+BnshsKxW9y5gWkTMzMyNmfnL2naeBdYAswa0uab2/BrgPeM4jYbGcZ4TxhjnOFLbkvZlM20nhLHMEyAiZgNnAt+q02bC7M9WMOibUPt4txJ4AviXzLx7UJVZwPoByxsYFHQRMRc4juqsA+CVmbkRoPbvwa0f+eiM0zwBPl772Hz1/v4YPJY5jtC2qH3ZoO2E2Zcw5mP2q8CfA/2D2ky4/TlWBn0TMrMvM18HzAZOiIhjBlWJes2eL4zoBr4DfDIznxm3gY7ROM3zm8ARwOuAjcDftXjYozKWOTbRdsIYp3lOqH0J+z7PiHg38ERm3jPeY5wIDPpRyMzNwB3AgkFFG4BDByzPBh4DiIg2qvC7PjO/O6DO4wM+Qs6kOiOZEFo5z8x8vPZm7AeuBE4Yv5E3b1/mOELbovblcG0n6r6EfZrnycDZEfHvwHLg7RFxXa3OhN2f+8qgbyAiZkTEtNrzDuAdwEODqt0EnFf7hv9E4OnM3BgRAVwFrMnMv6/T5kO15x8Cvj9ec2jGeM1zzxum5r3A6vGaQyNjnONIbUval8O2nUj7EsY2z8z8i8ycnZlzgUXAbZn5wQFtJsz+bInhvqX18fw38McC9wKrqA7sz9XWfxT4aO15AF8H/g24H+iprX8z1cfhVcDK2uOMWtl/AX4C/Kr27ysKnee1tbqrqN5AM39H51i3bYH7cqR5Tph9OdZ5DtrOKex9182E2p+tePgnECSpcF66kaTCGfSSVDiDXpIKZ9BLUuEMeknaz2q/afxERDS8ZTUivhIv/GG5tRGxuWEb77qRpP0rIt4KbKH6uzxN/8Z1RFwAHJeZ/22kep7RS9J+lpl3Ak8NXBcRR0TEjyLinoj4PxFxdJ2mi4EbGm1/SovGKUlqrSuofvHrVxHxRuAbwNv3FEbEHOAw4LZGGzLoJWmCqf2BwJOAb1d/YQSAlw2qtgi4MTP7Gm3PoJekiWcSsDmrv8w5nEXAx5rdmCRpAsnqz3w/EhHvg+f/l4iv3VMeEUcB04H/28z2DHpJ2s8i4gaq0D4qIjZExEeADwAfiYj7gAeo/s9XeywGlmeTt016e6UkFc4zekkqnEEvSYUz6CWpcAa9JBXOoJekwhn0klQ4g16SCvf/AXmKdFxF0ok9AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "colors={}\n",
    "colors[\"smpomuc_5p_nuc\"] = \"green\"\n",
    "colors[\"smpomuc_3p_exon_nuc\"] = \"red\"\n",
    "\n",
    "\n",
    "x=pd.read_csv(\"smv9_assem.blastn.tsv\", header=None, sep=\"\\t\")\n",
    "x=x.loc[x[1] == \"HE601627.3\"]\n",
    "\n",
    "color_list = [colors.get(item, \"unknown\") for item in x[0]]\n",
    "\n",
    "xs=x[8]\n",
    "\n",
    "plt.scatter(x=xs, y=[1]*len(xs), c=color_list)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69212e94-6589-401f-b393-f3abc1223d6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "colors=[]\n",
    "\n",
    "colors[\"smpomuc_5p_nuc\"] = \"green\"\n",
    "colors[\"smpomuc_3p_exon_nuc\"] = \"red\"\n",
    "\n",
    "for index, row in d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "d41065bb-c812-412c-8b67-8692567b85ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "def blast_to_bed(blast_file):\n",
    "    \"\"\"\n",
    "    Convert BLAST tabular output to BED format DataFrame including strand information.\n",
    "    \n",
    "    Parameters:\n",
    "    - blast_file (str): Path to the BLAST tabular output file\n",
    "    \n",
    "    Returns:\n",
    "    - DataFrame: BED formatted DataFrame\n",
    "    \"\"\"\n",
    "    bed_records = []\n",
    "    \n",
    "    with open(blast_file, 'r') as blastf:\n",
    "        for line in blastf:\n",
    "            parts = line.strip().split('\\t')\n",
    "            \n",
    "            # Assuming BLAST output columns are:\n",
    "            # qseqid, sseqid, pident, length, mismatch, gapopen, qstart, qend, sstart, send, evalue, bitscore\n",
    "            chrom = parts[1]\n",
    "            sstart = int(parts[8])\n",
    "            send = int(parts[9])\n",
    "            \n",
    "            # Determine orientation/strand\n",
    "            if sstart < send:\n",
    "                start, end, strand = sstart - 1, send, '+'\n",
    "            else:\n",
    "                start, end, strand = send - 1, sstart, '-'\n",
    "            \n",
    "            name = parts[0]\n",
    "            \n",
    "            score = \".\"\n",
    "            bed_records.append([chrom, start, end, name, score, strand])\n",
    "\n",
    "    df = pd.DataFrame(bed_records, columns=['chrom', 'start', 'end', 'name', 'score', 'strand'])\n",
    "    \n",
    "    return df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc3f5218-8451-4b76-be34-681d7c12df00",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bbc43fa6-1ec7-4942-b31b-dcfba07da94d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "id": "319a3cb6-ee45-4069-bb1a-f4923f2f72dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['smor', 'smle_pzq_es', 'smle_pzq_er', 'smbre', 'smeg']"
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pops"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "id": "0ab08dcc-a3f6-44e1-a032-1ef817a1083e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "smpomuc_5p_nuc\tcontig_512\t93.85\t65\t4\t0\t1\t65\t24564\t24500\t7e-25\t99.0\n",
      "smpomuc_5p_nuc\tcontig_563\t95.38\t65\t3\t0\t1\t65\t4166529\t4166593\t2e-26\t104\n",
      "smpomuc_5p_nuc\tcontig_586\t98.46\t65\t1\t0\t1\t65\t26172\t26236\t7e-30\t115\n",
      "smpomuc_5p_nuc\tcontig_586\t98.44\t64\t1\t0\t2\t65\t56069\t56132\t3e-29\t113\n",
      "smpomuc_5p_nuc\tcontig_586\t95.38\t65\t3\t0\t1\t65\t107224\t107288\t2e-26\t104\n",
      "smpomuc_3p_exon_nuc\tcontig_2732\t95.40\t87\t4\t0\t1\t87\t5809\t5895\t6e-37\t139\n",
      "smpomuc_3p_exon_nuc\tcontig_512\t100.00\t87\t0\t0\t1\t87\t32990\t32904\t1e-43\t161\n",
      "smpomuc_3p_exon_nuc\tcontig_563\t98.85\t87\t1\t0\t1\t87\t4151156\t4151242\t6e-42\t156\n",
      "smpomuc_3p_exon_nuc\tcontig_563\t98.85\t87\t1\t0\t1\t87\t4195123\t4195209\t6e-42\t156\n",
      "smpomuc_3p_exon_nuc\tcontig_586\t100.00\t87\t0\t0\t1\t87\t45743\t45829\t1e-43\t161\n",
      "smpomuc_3p_exon_nuc\tcontig_586\t100.00\t87\t0\t0\t1\t87\t77866\t77952\t1e-43\t161\n",
      "smpomuc_3p_exon_nuc\tcontig_586\t95.40\t87\t4\t0\t1\t87\t9838\t9924\t6e-37\t139\n",
      "smpomuc_3p_exon_nuc\tcontig_586\t95.40\t87\t4\t0\t1\t87\t97630\t97716\t6e-37\t139\n",
      "smpomuc_3p_exon_nuc\tcontig_589\t100.00\t87\t0\t0\t1\t87\t2064\t2150\t1e-43\t161\n"
     ]
    }
   ],
   "source": [
    "!cat smor_assem.blastn.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "d38f3b07-76e7-426a-8118-79bbdc492044",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>contig_512</td>\n",
       "      <td>24499</td>\n",
       "      <td>24564</td>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>contig_563</td>\n",
       "      <td>4166528</td>\n",
       "      <td>4166593</td>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>contig_586</td>\n",
       "      <td>26171</td>\n",
       "      <td>26236</td>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>contig_586</td>\n",
       "      <td>56068</td>\n",
       "      <td>56132</td>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>contig_586</td>\n",
       "      <td>107223</td>\n",
       "      <td>107288</td>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>contig_2732</td>\n",
       "      <td>5808</td>\n",
       "      <td>5895</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>contig_512</td>\n",
       "      <td>32903</td>\n",
       "      <td>32990</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>contig_563</td>\n",
       "      <td>4151155</td>\n",
       "      <td>4151242</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>contig_563</td>\n",
       "      <td>4195122</td>\n",
       "      <td>4195209</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>contig_586</td>\n",
       "      <td>45742</td>\n",
       "      <td>45829</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>contig_586</td>\n",
       "      <td>77865</td>\n",
       "      <td>77952</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>contig_586</td>\n",
       "      <td>9837</td>\n",
       "      <td>9924</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>contig_586</td>\n",
       "      <td>97629</td>\n",
       "      <td>97716</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>contig_589</td>\n",
       "      <td>2063</td>\n",
       "      <td>2150</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          chrom    start      end                 name score strand\n",
       "0    contig_512    24499    24564       smpomuc_5p_nuc     .      -\n",
       "1    contig_563  4166528  4166593       smpomuc_5p_nuc     .      +\n",
       "2    contig_586    26171    26236       smpomuc_5p_nuc     .      +\n",
       "3    contig_586    56068    56132       smpomuc_5p_nuc     .      +\n",
       "4    contig_586   107223   107288       smpomuc_5p_nuc     .      +\n",
       "5   contig_2732     5808     5895  smpomuc_3p_exon_nuc     .      +\n",
       "6    contig_512    32903    32990  smpomuc_3p_exon_nuc     .      -\n",
       "7    contig_563  4151155  4151242  smpomuc_3p_exon_nuc     .      +\n",
       "8    contig_563  4195122  4195209  smpomuc_3p_exon_nuc     .      +\n",
       "9    contig_586    45742    45829  smpomuc_3p_exon_nuc     .      +\n",
       "10   contig_586    77865    77952  smpomuc_3p_exon_nuc     .      +\n",
       "11   contig_586     9837     9924  smpomuc_3p_exon_nuc     .      +\n",
       "12   contig_586    97629    97716  smpomuc_3p_exon_nuc     .      +\n",
       "13   contig_589     2063     2150  smpomuc_3p_exon_nuc     .      +"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "blast_to_bed(blast_if)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "id": "00674d00-34e4-4103-93bd-0385c53ca3a0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          chrom    start      end                 name score strand\n",
      "0    contig_512    24499    24564       smpomuc_5p_nuc     .      -\n",
      "1    contig_563  4166528  4166593       smpomuc_5p_nuc     .      +\n",
      "2    contig_586    26171    26236       smpomuc_5p_nuc     .      +\n",
      "3    contig_586    56068    56132       smpomuc_5p_nuc     .      +\n",
      "4    contig_586   107223   107288       smpomuc_5p_nuc     .      +\n",
      "5   contig_2732     5808     5895  smpomuc_3p_exon_nuc     .      +\n",
      "6    contig_512    32903    32990  smpomuc_3p_exon_nuc     .      -\n",
      "7    contig_563  4151155  4151242  smpomuc_3p_exon_nuc     .      +\n",
      "8    contig_563  4195122  4195209  smpomuc_3p_exon_nuc     .      +\n",
      "9    contig_586    45742    45829  smpomuc_3p_exon_nuc     .      +\n",
      "10   contig_586    77865    77952  smpomuc_3p_exon_nuc     .      +\n",
      "11   contig_586     9837     9924  smpomuc_3p_exon_nuc     .      +\n",
      "12   contig_586    97629    97716  smpomuc_3p_exon_nuc     .      +\n",
      "13   contig_589     2063     2150  smpomuc_3p_exon_nuc     .      +\n",
      "     5p_chrom  5p_start   5p_end         5p_name 5p_score 5p_strand  \\\n",
      "0  contig_563   4166528  4166593  smpomuc_5p_nuc        .         +   \n",
      "1  contig_586     26171    26236  smpomuc_5p_nuc        .         +   \n",
      "2  contig_586     56068    56132  smpomuc_5p_nuc        .         +   \n",
      "\n",
      "     3p_chrom  3p_start   3p_end              3p_name 3p_score 3p_strand  \n",
      "0  contig_563   4195122  4195209  smpomuc_3p_exon_nuc        .         +  \n",
      "1  contig_586     97629    97716  smpomuc_3p_exon_nuc        .         +  \n",
      "2  contig_586     77865    77952  smpomuc_3p_exon_nuc        .         +  \n"
     ]
    }
   ],
   "source": [
    "for pop in [\"smor\"]:\n",
    "    \n",
    "    blast_if = f\"{pop}_assem.blastn.tsv\"\n",
    "    bed_if   = f\"{pop}_assem.blastn.bed\"\n",
    "    tsv_of  = f\"{pop}_assem.blastn.paired.tsv\"\n",
    "\n",
    "    #convert blast to bed\n",
    "    df = blast_to_bed(blast_if)\n",
    "    df.to_csv(bed_of, sep=\"\\t\", header=False, index=False)\n",
    "    \n",
    "    print(df)\n",
    "    # Split the DataFrame\n",
    "    df_5p = df[df['name'] == 'smpomuc_5p_nuc'].copy()\n",
    "    df_3p = df[df['name'] == 'smpomuc_3p_exon_nuc'].copy()\n",
    "\n",
    "    # For storing pairs\n",
    "    pairs = []\n",
    "\n",
    "    # Loop through one DataFrame and find closest match in the other DataFrame\n",
    "    for index, row_5p in df_5p.iterrows():\n",
    "        closest_distance = float('inf')\n",
    "        closest_row = None\n",
    "\n",
    "        for _, row_3p in df_3p.iterrows():\n",
    "            if row_5p['chrom'] == row_3p['chrom'] and row_5p['strand'] == row_3p['strand']:\n",
    "                distance = row_5p['start'] - row_3p['start']\n",
    "                if (row_5p['strand'] == \"+\" and distance < 0) or (row_5p['strand'] == \"-\" and distance > 0): \n",
    "                    if distance < closest_distance:\n",
    "                        closest_distance = distance\n",
    "                        closest_row = row_3p\n",
    "\n",
    "        if closest_row is not None:\n",
    "            pairs.append((row_5p, closest_row))\n",
    "            # Remove the matched row from df_3p to prevent duplicate matches\n",
    "            df_3p.drop(closest_row.name, inplace=True)\n",
    "\n",
    "\n",
    "    combined_rows = []\n",
    "\n",
    "    for pair in pairs:\n",
    "        # Extend the two rows to form a single row for the pair\n",
    "        combined_row = list(pair[0]) + list(pair[1])\n",
    "        combined_rows.append(combined_row)\n",
    "\n",
    "    # Create a DataFrame from combined rows\n",
    "    columns =  list(\"5p_\" + pair[0].index) + list(\"3p_\" + pair[0].index) \n",
    "    df_pairs = pd.DataFrame(combined_rows, columns=columns)\n",
    "\n",
    "    df_pairs.to_csv(tsv_of, sep=\"\\t\", header=True, index=False)\n",
    "    print(df_pairs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "569f06d8-4167-4c6b-991f-17627d5cafb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>5p_chrom</th>\n",
       "      <th>5p_start</th>\n",
       "      <th>5p_end</th>\n",
       "      <th>5p_name</th>\n",
       "      <th>5p_score</th>\n",
       "      <th>5p_strand</th>\n",
       "      <th>3p_chrom</th>\n",
       "      <th>3p_start</th>\n",
       "      <th>3p_end</th>\n",
       "      <th>3p_name</th>\n",
       "      <th>3p_score</th>\n",
       "      <th>3p_strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>contig_1873</td>\n",
       "      <td>78173</td>\n",
       "      <td>78238</td>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1873</td>\n",
       "      <td>56527</td>\n",
       "      <td>56614</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>contig_1873</td>\n",
       "      <td>46215</td>\n",
       "      <td>46279</td>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_1873</td>\n",
       "      <td>29572</td>\n",
       "      <td>29659</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>contig_324</td>\n",
       "      <td>725269</td>\n",
       "      <td>725334</td>\n",
       "      <td>smpomuc_5p_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>contig_324</td>\n",
       "      <td>690545</td>\n",
       "      <td>690632</td>\n",
       "      <td>smpomuc_3p_exon_nuc</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      5p_chrom  5p_start  5p_end         5p_name 5p_score 5p_strand  \\\n",
       "0  contig_1873     78173   78238  smpomuc_5p_nuc        .         -   \n",
       "1  contig_1873     46215   46279  smpomuc_5p_nuc        .         -   \n",
       "2   contig_324    725269  725334  smpomuc_5p_nuc        .         -   \n",
       "\n",
       "      3p_chrom  3p_start  3p_end              3p_name 3p_score 3p_strand  \n",
       "0  contig_1873     56527   56614  smpomuc_3p_exon_nuc        .         -  \n",
       "1  contig_1873     29572   29659  smpomuc_3p_exon_nuc        .         -  \n",
       "2   contig_324    690545  690632  smpomuc_3p_exon_nuc        .         -  "
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "921ecb7a-6a4a-43ec-aa2e-ac37f17afd85",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        chrom    start     stop                           name  . orient\n",
      "0  contig_512    24499    32990      contig_512:24499-32990(-)  .      -\n",
      "1  contig_563  4151155  4166593  contig_563:4151155-4166593(+)  .      +\n",
      "2  contig_586     9837    26236       contig_586:9837-26236(+)  .      +\n",
      "3  contig_586    45742    56132      contig_586:45742-56132(+)  .      +\n",
      "4  contig_586    97629   107288     contig_586:97629-107288(+)  .      +\n",
      "        chrom    start     stop                           name  . orient\n",
      "0  contig_292  3053118  3068566  contig_292:3053118-3068566(+)  .      +\n",
      "1   contig_86  2132083  2140572   contig_86:2132083-2140572(+)  .      +\n",
      "2   contig_88    53616    67630       contig_88:53616-67630(+)  .      +\n",
      "3   contig_88    87165    97561       contig_88:87165-97561(+)  .      +\n",
      "4   contig_88   140016   149678     contig_88:140016-149678(+)  .      +\n",
      "        chrom    start     stop                           name  . orient\n",
      "0  contig_471  2097237  2105657  contig_471:2097237-2105657(+)  .      +\n",
      "1  contig_561  1024252  1039690  contig_561:1024252-1039690(+)  .      +\n",
      "2  contig_908    17853    28259      contig_908:17853-28259(-)  .      -\n",
      "3  contig_908    47770    61785      contig_908:47770-61785(-)  .      -\n",
      "4  contig_908    72350    83383      contig_908:72350-83383(-)  .      -\n",
      "         chrom    start     stop                           name  . orient\n",
      "0  contig_1224    57518    80234     contig_1224:57518-80234(-)  .      -\n",
      "1  contig_1224    29569    47272     contig_1224:29569-47272(-)  .      -\n",
      "2  contig_1224      239     9896        contig_1224:239-9896(-)  .      -\n",
      "3   contig_150  2115700  2124184  contig_150:2115700-2124184(+)  .      +\n",
      "4   contig_161  2861984  2877421  contig_161:2861984-2877421(+)  .      +\n",
      "         chrom   start    stop                         name  . orient\n",
      "0  contig_1873   78173   92189   contig_1873:78173-92189(-)  .      -\n",
      "1  contig_1873   46215   56614   contig_1873:46215-56614(-)  .      -\n",
      "2  contig_1873     239    9897      contig_1873:239-9897(-)  .      -\n",
      "3   contig_324  725269  740728  contig_324:725269-740728(-)  .      -\n"
     ]
    }
   ],
   "source": [
    "for pop in pops:\n",
    "    \n",
    "    bed_of   = f\"{pop}_assem.paired.bed\"\n",
    "    tsv_if  = f\"{pop}_assem.blastn.paired.tsv\"\n",
    "    df_pairs = pd.read_csv(tsv_if, sep=\"\\t\")\n",
    "    \n",
    "    bed_rows=[]\n",
    "\n",
    "    for index, row in df_pairs.iterrows():\n",
    "        start = row[[\"5p_start\", \"5p_end\", \"3p_start\", \"3p_end\"]].min()\n",
    "        stop = row[[\"5p_start\", \"5p_end\", \"3p_start\", \"3p_end\"]].max()\n",
    "        chrom = row[\"5p_chrom\"]\n",
    "        orient= row[\"5p_strand\"]\n",
    "\n",
    "        name=f\"{chrom}:{start}-{stop}({orient})\"\n",
    "\n",
    "        bed_rows.append([chrom, start, stop, name, \".\", orient])\n",
    "        columns=[\"chrom\", \"start\", \"stop\", \"name\", \".\", \"orient\"]\n",
    "\n",
    "        df_bed=pd.DataFrame(bed_rows, columns=columns)\n",
    "        df_bed.to_csv(bed_of, sep=\"\\t\", header=False, index=False)\n",
    "    print(df_bed)\n",
    "#df_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "42099be5-6a7d-45a1-9307-1422fbcfd926",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash\n",
    "\n",
    "# #slop\n",
    "# L_BUFFER=3\n",
    "# R_BUFFER=3\n",
    "\n",
    "# ASSEM_DIR=\"/master/nplatt/sch_man_ont/results/assembly\"\n",
    "\n",
    "# for POP in smle_pzq_er smle_pzq_es smbre smor smeg; do\n",
    "#     FAS=\"/master/nplatt/sch_man_ont/results/assembly/${POP}_rnd0.fasta\"\n",
    "#     I_BED=${POP}\"_assem.blastn.bed\"\n",
    "#     O_BED=${POP}\"_assem.blastn.slop.bed\"\n",
    "    \n",
    "#     conda run -n sm_ont-process_reads samtools faidx ${FAS}\n",
    "#     cut -f1,2 ${FAS}.fai > ${FAS}.size\n",
    "    \n",
    "#     conda run -n sm_ont-process_reads bedtools slop -s -l ${L_BUFFER} -r ${R_BUFFER} -g ${FAS}.size -i ${I_BED} >${O_BED}\n",
    "\n",
    "# done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "fe85c525-f39a-4a97-be4c-84358fceca8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#exctract those regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "07f9067a-92f0-44c1-a840-12f0e46cc9ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "ASSEM_DIR=\"/master/nplatt/sch_man_ont/results/assembly\"\n",
    "\n",
    "for POP in \"smle_pzq_er\" \"smle_pzq_es\" \"smbre\" \"smor\" \"smeg\"; do\n",
    "\n",
    "    FI=${ASSEM_DIR}/${POP}_rnd0.fasta\n",
    "    FO=${POP}_assem.paired.fasta\n",
    "    BED=${POP}_assem.paired.bed\n",
    "    \n",
    "    conda run -n sm_ont-process_reads bedtools getfasta -s -fi ${FI} -bed ${BED} -fo ${FO}\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "b087645e-a2ad-4adc-b489-da75ef5b3b47",
   "metadata": {},
   "outputs": [],
   "source": [
    "for pop in pops:\n",
    "    input_fasta = f\"{pop}_assem.paired.fasta\"\n",
    "    output_fasta =f\"{pop}_assem.paired.mod.fasta\"\n",
    "\n",
    "    sequences = SeqIO.parse(input_fasta, \"fasta\")\n",
    "\n",
    "    # Modify the headers\n",
    "    modified_sequences = []\n",
    "    for seq in sequences:\n",
    "        seq.id = f\"{pop}_{seq.id}\"\n",
    "        seq.description = ''  # Resetting the description ensures that only the ID gets written out\n",
    "        modified_sequences.append(seq)\n",
    "\n",
    "    # Write the modified sequences to the output file\n",
    "    SeqIO.write(modified_sequences, output_fasta, \"fasta\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "48c6387e-12b2-464e-99bf-f2107ab65dc0",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat *_assem.paired.mod.fasta >test.paired.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ea4dacc-bf78-4a20-b73c-0b3aefaa52ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "r1 GATGACTATGCATTGAGCGAACCAACA\n",
    "r1p GATTACTATGCATTGAGCGAACCAACA\n",
    "r2 GGTGACCTCGCATCAGACAAACCCACA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e14c2cd-b383-412c-9373-029622b880cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "ASSEM_DIR=\"/master/nplatt/sch_man_ont/results/assembly\"\n",
    "cat smpomuc_5p_aa.fas smpomuc_3p_aa.fas >aa.fas\n",
    "\n",
    "\n",
    "for POP in \"smle_pzq_er\" \"smle_pzq_es\" \"smbre\" \"smor\" \"smeg\"; do\n",
    "\n",
    "    conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "        makeblastdb \\\n",
    "            -dbtype nucl \\\n",
    "            -in ${ASSEM_DIR}/${POP}_rnd0.fasta\n",
    "\n",
    "    conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "        tblastn \\\n",
    "            -subject ${ASSEM_DIR}/${POP}_rnd0.fasta \\\n",
    "            -query aa.fas \\\n",
    "            -out ${POP}_assem.aa.tblastn.tsv \\\n",
    "            -outfmt 6\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "ba39d4a3-5428-4995-bcdf-3122d8f2d7ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for pop in [\"smle_pzq_er\", \"smle_pzq_es\", \"smbre\", \"smor\", \"smeg\"]:\n",
    "\n",
    "pop=\"smbre\"\n",
    "\n",
    "#read in data\n",
    "df = pd.read_csv(f\"{pop}_assm.aa.tblastn.tsv\", sep=\"\\t\", header=0)\n",
    "\n",
    "#change column names to make them useful\n",
    "col_names = [\"q_seqid\", \"s_seqid\", \"p_ident\", \"length\", \"mismatch\", \"gapopen\", \"q_start\", \"q_end\", \"s_start\", \"s_end\", \"evalue\", \"bitscore\"]\n",
    "df.columns=col_names\n",
    "\n",
    "# #find possibly good hits\n",
    "# df.loc[( (df[\"p_ident\"] > 0.9) & (df[\"length\"] > 62) )]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "7f80b7a8-bd9a-4a6c-b1a2-2baa4a3e75e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>q_seqid</th>\n",
       "      <th>s_seqid</th>\n",
       "      <th>p_ident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>q_start</th>\n",
       "      <th>q_end</th>\n",
       "      <th>s_start</th>\n",
       "      <th>s_end</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>373</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_105</td>\n",
       "      <td>84.62</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>234</td>\n",
       "      <td>27564</td>\n",
       "      <td>27641</td>\n",
       "      <td>7.000000e-17</td>\n",
       "      <td>49.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>374</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_105</td>\n",
       "      <td>58.97</td>\n",
       "      <td>39</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>180</td>\n",
       "      <td>216</td>\n",
       "      <td>27436</td>\n",
       "      <td>27552</td>\n",
       "      <td>7.000000e-17</td>\n",
       "      <td>40.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>376</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_105</td>\n",
       "      <td>92.00</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>151</td>\n",
       "      <td>175</td>\n",
       "      <td>26625</td>\n",
       "      <td>26699</td>\n",
       "      <td>2.000000e-07</td>\n",
       "      <td>47.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_105</td>\n",
       "      <td>73.08</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>71</td>\n",
       "      <td>22392</td>\n",
       "      <td>22469</td>\n",
       "      <td>3.000000e-07</td>\n",
       "      <td>47.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_105</td>\n",
       "      <td>71.43</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>44</td>\n",
       "      <td>21887</td>\n",
       "      <td>21970</td>\n",
       "      <td>8.000000e-07</td>\n",
       "      <td>45.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>384</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_105</td>\n",
       "      <td>53.33</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>107</td>\n",
       "      <td>136</td>\n",
       "      <td>24927</td>\n",
       "      <td>25016</td>\n",
       "      <td>1.200000e-02</td>\n",
       "      <td>33.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_111</td>\n",
       "      <td>51.72</td>\n",
       "      <td>29</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>46</td>\n",
       "      <td>74</td>\n",
       "      <td>2150682</td>\n",
       "      <td>2150599</td>\n",
       "      <td>7.400000e+00</td>\n",
       "      <td>25.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>84.62</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>234</td>\n",
       "      <td>10624</td>\n",
       "      <td>10547</td>\n",
       "      <td>1.000000e-19</td>\n",
       "      <td>49.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>84.62</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>234</td>\n",
       "      <td>58331</td>\n",
       "      <td>58254</td>\n",
       "      <td>4.000000e-19</td>\n",
       "      <td>49.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>84.62</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>234</td>\n",
       "      <td>30384</td>\n",
       "      <td>30307</td>\n",
       "      <td>4.000000e-19</td>\n",
       "      <td>49.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>92.00</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>151</td>\n",
       "      <td>175</td>\n",
       "      <td>59271</td>\n",
       "      <td>59197</td>\n",
       "      <td>9.000000e-09</td>\n",
       "      <td>47.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>480</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>92.00</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>151</td>\n",
       "      <td>175</td>\n",
       "      <td>31324</td>\n",
       "      <td>31250</td>\n",
       "      <td>9.000000e-09</td>\n",
       "      <td>47.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>92.00</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>151</td>\n",
       "      <td>175</td>\n",
       "      <td>11565</td>\n",
       "      <td>11491</td>\n",
       "      <td>9.000000e-09</td>\n",
       "      <td>47.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>71.43</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>44</td>\n",
       "      <td>63987</td>\n",
       "      <td>63904</td>\n",
       "      <td>3.000000e-08</td>\n",
       "      <td>45.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>71.43</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>44</td>\n",
       "      <td>39253</td>\n",
       "      <td>39170</td>\n",
       "      <td>3.000000e-08</td>\n",
       "      <td>45.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>51.52</td>\n",
       "      <td>33</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>46</td>\n",
       "      <td>78</td>\n",
       "      <td>38384</td>\n",
       "      <td>38301</td>\n",
       "      <td>2.000000e-04</td>\n",
       "      <td>34.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>53.33</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>107</td>\n",
       "      <td>136</td>\n",
       "      <td>60969</td>\n",
       "      <td>60880</td>\n",
       "      <td>5.000000e-04</td>\n",
       "      <td>33.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_1224</td>\n",
       "      <td>53.33</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>107</td>\n",
       "      <td>136</td>\n",
       "      <td>36236</td>\n",
       "      <td>36147</td>\n",
       "      <td>5.000000e-04</td>\n",
       "      <td>33.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>528</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_127</td>\n",
       "      <td>52.00</td>\n",
       "      <td>25</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>71</td>\n",
       "      <td>160518</td>\n",
       "      <td>160447</td>\n",
       "      <td>4.200000e+00</td>\n",
       "      <td>24.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_150</td>\n",
       "      <td>73.08</td>\n",
       "      <td>26</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>234</td>\n",
       "      <td>2114973</td>\n",
       "      <td>2115050</td>\n",
       "      <td>7.000000e-18</td>\n",
       "      <td>43.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_150</td>\n",
       "      <td>50.94</td>\n",
       "      <td>53</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>2108921</td>\n",
       "      <td>2109079</td>\n",
       "      <td>5.000000e-09</td>\n",
       "      <td>52.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>618</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_150</td>\n",
       "      <td>92.00</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>151</td>\n",
       "      <td>175</td>\n",
       "      <td>2114032</td>\n",
       "      <td>2114106</td>\n",
       "      <td>2.000000e-07</td>\n",
       "      <td>47.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>619</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_150</td>\n",
       "      <td>67.86</td>\n",
       "      <td>28</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>44</td>\n",
       "      <td>2109478</td>\n",
       "      <td>2109561</td>\n",
       "      <td>2.000000e-05</td>\n",
       "      <td>41.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>677</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_161</td>\n",
       "      <td>80.77</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>234</td>\n",
       "      <td>2861554</td>\n",
       "      <td>2861631</td>\n",
       "      <td>3.000000e-16</td>\n",
       "      <td>47.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>681</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_161</td>\n",
       "      <td>68.00</td>\n",
       "      <td>25</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>44</td>\n",
       "      <td>2856338</td>\n",
       "      <td>2856412</td>\n",
       "      <td>3.000000e-04</td>\n",
       "      <td>38.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>799</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_211</td>\n",
       "      <td>80.77</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>234</td>\n",
       "      <td>1040879</td>\n",
       "      <td>1040802</td>\n",
       "      <td>3.000000e-16</td>\n",
       "      <td>47.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>801</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_211</td>\n",
       "      <td>55.56</td>\n",
       "      <td>36</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>140</td>\n",
       "      <td>175</td>\n",
       "      <td>1041846</td>\n",
       "      <td>1041739</td>\n",
       "      <td>1.000000e-05</td>\n",
       "      <td>41.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_211</td>\n",
       "      <td>53.85</td>\n",
       "      <td>26</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>111</td>\n",
       "      <td>136</td>\n",
       "      <td>1043557</td>\n",
       "      <td>1043480</td>\n",
       "      <td>8.400000e-01</td>\n",
       "      <td>26.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>967</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_51</td>\n",
       "      <td>60.00</td>\n",
       "      <td>25</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>71</td>\n",
       "      <td>3514520</td>\n",
       "      <td>3514591</td>\n",
       "      <td>4.200000e+00</td>\n",
       "      <td>25.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_51</td>\n",
       "      <td>53.85</td>\n",
       "      <td>26</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>72</td>\n",
       "      <td>3321440</td>\n",
       "      <td>3321514</td>\n",
       "      <td>5.900000e+00</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1116</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_763</td>\n",
       "      <td>84.62</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>234</td>\n",
       "      <td>49550</td>\n",
       "      <td>49473</td>\n",
       "      <td>3.000000e-20</td>\n",
       "      <td>50.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1117</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_763</td>\n",
       "      <td>84.62</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>234</td>\n",
       "      <td>27832</td>\n",
       "      <td>27755</td>\n",
       "      <td>3.000000e-20</td>\n",
       "      <td>50.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1118</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_763</td>\n",
       "      <td>84.62</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>234</td>\n",
       "      <td>6231</td>\n",
       "      <td>6154</td>\n",
       "      <td>3.000000e-20</td>\n",
       "      <td>50.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1120</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_763</td>\n",
       "      <td>92.00</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>151</td>\n",
       "      <td>175</td>\n",
       "      <td>50487</td>\n",
       "      <td>50413</td>\n",
       "      <td>6.000000e-09</td>\n",
       "      <td>47.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1121</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_763</td>\n",
       "      <td>92.00</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>151</td>\n",
       "      <td>175</td>\n",
       "      <td>28769</td>\n",
       "      <td>28695</td>\n",
       "      <td>6.000000e-09</td>\n",
       "      <td>47.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1122</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_763</td>\n",
       "      <td>92.00</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>151</td>\n",
       "      <td>175</td>\n",
       "      <td>7168</td>\n",
       "      <td>7094</td>\n",
       "      <td>6.000000e-09</td>\n",
       "      <td>47.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1144</th>\n",
       "      <td>smpomuc_c-term</td>\n",
       "      <td>contig_77</td>\n",
       "      <td>53.85</td>\n",
       "      <td>26</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>72</td>\n",
       "      <td>299580</td>\n",
       "      <td>299654</td>\n",
       "      <td>1.600000e+00</td>\n",
       "      <td>24.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             q_seqid      s_seqid  p_ident  length  mismatch  gapopen  \\\n",
       "373   smpomuc_c-term   contig_105    84.62      26         4        0   \n",
       "374   smpomuc_c-term   contig_105    58.97      39        14        1   \n",
       "376   smpomuc_c-term   contig_105    92.00      25         2        0   \n",
       "377   smpomuc_c-term   contig_105    73.08      26         7        0   \n",
       "378   smpomuc_c-term   contig_105    71.43      28         8        0   \n",
       "384   smpomuc_c-term   contig_105    53.33      30        14        0   \n",
       "412   smpomuc_c-term   contig_111    51.72      29        13        1   \n",
       "472   smpomuc_c-term  contig_1224    84.62      26         4        0   \n",
       "473   smpomuc_c-term  contig_1224    84.62      26         4        0   \n",
       "474   smpomuc_c-term  contig_1224    84.62      26         4        0   \n",
       "479   smpomuc_c-term  contig_1224    92.00      25         2        0   \n",
       "480   smpomuc_c-term  contig_1224    92.00      25         2        0   \n",
       "481   smpomuc_c-term  contig_1224    92.00      25         2        0   \n",
       "483   smpomuc_c-term  contig_1224    71.43      28         8        0   \n",
       "484   smpomuc_c-term  contig_1224    71.43      28         8        0   \n",
       "492   smpomuc_c-term  contig_1224    51.52      33        11        2   \n",
       "494   smpomuc_c-term  contig_1224    53.33      30        14        0   \n",
       "495   smpomuc_c-term  contig_1224    53.33      30        14        0   \n",
       "528   smpomuc_c-term   contig_127    52.00      25        11        1   \n",
       "616   smpomuc_c-term   contig_150    73.08      26         7        0   \n",
       "617   smpomuc_c-term   contig_150    50.94      53        20        1   \n",
       "618   smpomuc_c-term   contig_150    92.00      25         2        0   \n",
       "619   smpomuc_c-term   contig_150    67.86      28         9        0   \n",
       "677   smpomuc_c-term   contig_161    80.77      26         5        0   \n",
       "681   smpomuc_c-term   contig_161    68.00      25         8        0   \n",
       "799   smpomuc_c-term   contig_211    80.77      26         5        0   \n",
       "801   smpomuc_c-term   contig_211    55.56      36        16        0   \n",
       "810   smpomuc_c-term   contig_211    53.85      26        12        0   \n",
       "967   smpomuc_c-term    contig_51    60.00      25         9        1   \n",
       "968   smpomuc_c-term    contig_51    53.85      26        11        1   \n",
       "1116  smpomuc_c-term   contig_763    84.62      26         4        0   \n",
       "1117  smpomuc_c-term   contig_763    84.62      26         4        0   \n",
       "1118  smpomuc_c-term   contig_763    84.62      26         4        0   \n",
       "1120  smpomuc_c-term   contig_763    92.00      25         2        0   \n",
       "1121  smpomuc_c-term   contig_763    92.00      25         2        0   \n",
       "1122  smpomuc_c-term   contig_763    92.00      25         2        0   \n",
       "1144  smpomuc_c-term    contig_77    53.85      26        11        1   \n",
       "\n",
       "      q_start  q_end  s_start    s_end        evalue  bitscore  \n",
       "373       209    234    27564    27641  7.000000e-17      49.3  \n",
       "374       180    216    27436    27552  7.000000e-17      40.4  \n",
       "376       151    175    26625    26699  2.000000e-07      47.8  \n",
       "377        46     71    22392    22469  3.000000e-07      47.0  \n",
       "378        17     44    21887    21970  8.000000e-07      45.8  \n",
       "384       107    136    24927    25016  1.200000e-02      33.1  \n",
       "412        46     74  2150682  2150599  7.400000e+00      25.4  \n",
       "472       209    234    10624    10547  1.000000e-19      49.3  \n",
       "473       209    234    58331    58254  4.000000e-19      49.3  \n",
       "474       209    234    30384    30307  4.000000e-19      49.3  \n",
       "479       151    175    59271    59197  9.000000e-09      47.8  \n",
       "480       151    175    31324    31250  9.000000e-09      47.8  \n",
       "481       151    175    11565    11491  9.000000e-09      47.8  \n",
       "483        17     44    63987    63904  3.000000e-08      45.8  \n",
       "484        17     44    39253    39170  3.000000e-08      45.8  \n",
       "492        46     78    38384    38301  2.000000e-04      34.3  \n",
       "494       107    136    60969    60880  5.000000e-04      33.1  \n",
       "495       107    136    36236    36147  5.000000e-04      33.1  \n",
       "528        47     71   160518   160447  4.200000e+00      24.3  \n",
       "616       209    234  2114973  2115050  7.000000e-18      43.5  \n",
       "617         1     47  2108921  2109079  5.000000e-09      52.8  \n",
       "618       151    175  2114032  2114106  2.000000e-07      47.8  \n",
       "619        17     44  2109478  2109561  2.000000e-05      41.6  \n",
       "677       209    234  2861554  2861631  3.000000e-16      47.4  \n",
       "681        20     44  2856338  2856412  3.000000e-04      38.5  \n",
       "799       209    234  1040879  1040802  3.000000e-16      47.4  \n",
       "801       140    175  1041846  1041739  1.000000e-05      41.6  \n",
       "810       111    136  1043557  1043480  8.400000e-01      26.2  \n",
       "967        47     71  3514520  3514591  4.200000e+00      25.4  \n",
       "968        47     72  3321440  3321514  5.900000e+00      25.0  \n",
       "1116      209    234    49550    49473  3.000000e-20      50.1  \n",
       "1117      209    234    27832    27755  3.000000e-20      50.1  \n",
       "1118      209    234     6231     6154  3.000000e-20      50.1  \n",
       "1120      151    175    50487    50413  6.000000e-09      47.8  \n",
       "1121      151    175    28769    28695  6.000000e-09      47.8  \n",
       "1122      151    175     7168     7094  6.000000e-09      47.8  \n",
       "1144       47     72   299580   299654  1.600000e+00      24.6  "
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[( (df[\"p_ident\"] > 50) & (df[\"length\"] > 24) )]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "id": "5195af04-0bee-4163-8700-7397c272020b",
   "metadata": {},
   "outputs": [],
   "source": [
    "!echo -e \"smpomuc\\tGGAAGAATGAACAAGAAAATTATTCTCTTAACAATTATATATGTTGCGCTGATGATAAGAGGTGAC\\tGAGAATGATAATGTTTTCAACTAAACCTCAGTGTTTGTTTATGATATGGACAGACATGTTTTGGATCGTTAACAGTTTTCTGTGTCA\" >primers.txt\n",
    "!echo -e \"smpomuc\\tATGAACAAGAAAATTATTCTC\\tCGTTAACAGTTTTCTGTGTCA\" >primers.txt\n",
    "                                            #12345678901123456789012"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "159b0e5b-3252-484b-8925-c362bf770949",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "conda run -n sm_ont-process_reads isPcr -out=bed -maxSize=1000000 -flipReverse ../../data/genome/SM_V9.fa primers.txt smv9.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "9457ec20-8d36-4b8c-92e0-30363bef5342",
   "metadata": {},
   "outputs": [],
   "source": [
    "for pop in pops + [\"smv9\"]:\n",
    "    \n",
    "    # fas = f\"/master/nplatt/sch_man_ont/results/assembly/{pop}_rnd0.fasta\"\n",
    "    # cmd=f\"isPcr -out=bed -maxSize=1000000 -flipReverse {fas} primers.txt {pop}.pcr.bed\"\n",
    "    # print(cmd)\n",
    "    # !cmd\n",
    "\n",
    "    df=pd.read_csv(f\"{pop}.pcr.bed\", sep=\"\\t\", header=None)\n",
    "    df.columns=[\"chrom\", \"start\", \"stop\", \"name\", \"score\", \"strand\"]\n",
    "    df[\"len\"] = df[\"stop\"] - df[\"start\"]\n",
    "\n",
    "    df=df.sort_values(by=['start', 'len'], ascending=True).reset_index(drop=True)\n",
    "\n",
    "\n",
    "    idx = df.groupby(['chrom', 'start'])['len'].idxmin()\n",
    "    df = df.loc[idx].reset_index(drop=True)\n",
    "    idx = df.groupby(['chrom', 'stop'])['len'].idxmin()\n",
    "    df = df.loc[idx].reset_index(drop=True)\n",
    "    df.to_csv(f\"{pop}.pcr.collapsed.bed\", sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "id": "8c9e8e0e-3e72-4397-90b0-021d4ef3d8ad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38377262</td>\n",
       "      <td>38413100</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>35838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30170361</td>\n",
       "      <td>30203235</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>32874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30232534</td>\n",
       "      <td>30252311</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>19777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30296550</td>\n",
       "      <td>30334710</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>38160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30348582</td>\n",
       "      <td>30364279</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>15697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30396771</td>\n",
       "      <td>30426256</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>29485</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom     start      stop     name  score strand    len\n",
       "0  HE601625.3  38377262  38413100  smpomuc   1000      +  35838\n",
       "1  HE601627.3  30170361  30203235  smpomuc   1000      -  32874\n",
       "2  HE601627.3  30232534  30252311  smpomuc   1000      -  19777\n",
       "3  HE601627.3  30296550  30334710  smpomuc   1000      -  38160\n",
       "4  HE601627.3  30348582  30364279  smpomuc   1000      -  15697\n",
       "5  HE601627.3  30396771  30426256  smpomuc   1000      -  29485"
      ]
     },
     "execution_count": 290,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "44d8963f-db51-4bc3-980e-75bfe5ab7f6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "ASSEM_DIR=\"/master/nplatt/sch_man_ont/results/assembly\"\n",
    "\n",
    "for POP in \"smle_pzq_er\" \"smle_pzq_es\" \"smbre\" \"smor\" \"smeg\"; do\n",
    "\n",
    "    FI=${ASSEM_DIR}/${POP}_rnd0.fasta\n",
    "    FO=${POP}.pcr.collapsed.fasta\n",
    "    BED=${POP}.pcr.collapsed.bed\n",
    "    \n",
    "    conda run -n sm_ont-process_reads bedtools getfasta -s -fi ${FI} -bed ${BED} -fo ${FO}\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "id": "23ba9020-1b7e-4281-9099-25c4d555030d",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "conda run -n sm_ont-process_reads bedtools getfasta -s -fi ../../data/genome/SM_V9.fa -bed smv9.pcr.collapsed.bed -fo smv9.pcr.collapsed.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "ea027684-ffac-4495-98ff-b01ab28569e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "for pop in pops + [\"smv9\"]:\n",
    "    input_fasta = f\"{pop}.pcr.collapsed.fasta\"\n",
    "    output_fasta =f\"{pop}.pcr.collapsed.mod.fasta\"\n",
    "\n",
    "    sequences = SeqIO.parse(input_fasta, \"fasta\")\n",
    "\n",
    "    # Modify the headers\n",
    "    modified_sequences = []\n",
    "    for seq in sequences:\n",
    "        seq.id = f\"{pop}_{seq.id}\"\n",
    "        seq.description = ''  # Resetting the description ensures that only the ID gets written out\n",
    "        modified_sequences.append(seq)\n",
    "\n",
    "    # Write the modified sequences to the output file\n",
    "    SeqIO.write(modified_sequences, output_fasta, \"fasta\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "2dff5705-9255-4130-8c2b-972da1c29988",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat *.pcr.collapsed.mod.fasta >ispcr.paired.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71ea1bc3-3128-436d-8fd7-0dee86b6acae",
   "metadata": {},
   "outputs": [],
   "source": [
    "HE601627.3:30170361-30426256"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "id": "e4268d52-0723-490c-af8b-90733c564e52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38377262</td>\n",
       "      <td>38413100</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>35838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30170361</td>\n",
       "      <td>30203235</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>32874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30170361</td>\n",
       "      <td>30252311</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>81950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30170361</td>\n",
       "      <td>30334710</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>164349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30170361</td>\n",
       "      <td>30364279</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>193918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30170361</td>\n",
       "      <td>30426256</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>255895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30212754</td>\n",
       "      <td>30252311</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>39557</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30212754</td>\n",
       "      <td>30334710</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>121956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30212754</td>\n",
       "      <td>30364279</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>151525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30212754</td>\n",
       "      <td>30426256</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>213502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30232534</td>\n",
       "      <td>30252311</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>19777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30232534</td>\n",
       "      <td>30334710</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>102176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30232534</td>\n",
       "      <td>30364279</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>131745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30232534</td>\n",
       "      <td>30426256</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>193722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30262569</td>\n",
       "      <td>30334710</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>72141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30262569</td>\n",
       "      <td>30364279</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30262569</td>\n",
       "      <td>30426256</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>163687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30274829</td>\n",
       "      <td>30334710</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>59881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30274829</td>\n",
       "      <td>30364279</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>89450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30274829</td>\n",
       "      <td>30426256</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>151427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30296550</td>\n",
       "      <td>30334710</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>38160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30296550</td>\n",
       "      <td>30364279</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>67729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30296550</td>\n",
       "      <td>30426256</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>129706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30348582</td>\n",
       "      <td>30364279</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>15697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30348582</td>\n",
       "      <td>30426256</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>77674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30375173</td>\n",
       "      <td>30426256</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>51083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30396771</td>\n",
       "      <td>30426256</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>29485</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom     start      stop     name  score strand     len\n",
       "0   HE601625.3  38377262  38413100  smpomuc   1000      +   35838\n",
       "1   HE601627.3  30170361  30203235  smpomuc   1000      -   32874\n",
       "2   HE601627.3  30170361  30252311  smpomuc   1000      -   81950\n",
       "3   HE601627.3  30170361  30334710  smpomuc   1000      -  164349\n",
       "4   HE601627.3  30170361  30364279  smpomuc   1000      -  193918\n",
       "5   HE601627.3  30170361  30426256  smpomuc   1000      -  255895\n",
       "6   HE601627.3  30212754  30252311  smpomuc   1000      -   39557\n",
       "7   HE601627.3  30212754  30334710  smpomuc   1000      -  121956\n",
       "8   HE601627.3  30212754  30364279  smpomuc   1000      -  151525\n",
       "9   HE601627.3  30212754  30426256  smpomuc   1000      -  213502\n",
       "10  HE601627.3  30232534  30252311  smpomuc   1000      -   19777\n",
       "11  HE601627.3  30232534  30334710  smpomuc   1000      -  102176\n",
       "12  HE601627.3  30232534  30364279  smpomuc   1000      -  131745\n",
       "13  HE601627.3  30232534  30426256  smpomuc   1000      -  193722\n",
       "14  HE601627.3  30262569  30334710  smpomuc   1000      -   72141\n",
       "15  HE601627.3  30262569  30364279  smpomuc   1000      -  101710\n",
       "16  HE601627.3  30262569  30426256  smpomuc   1000      -  163687\n",
       "17  HE601627.3  30274829  30334710  smpomuc   1000      -   59881\n",
       "18  HE601627.3  30274829  30364279  smpomuc   1000      -   89450\n",
       "19  HE601627.3  30274829  30426256  smpomuc   1000      -  151427\n",
       "20  HE601627.3  30296550  30334710  smpomuc   1000      -   38160\n",
       "21  HE601627.3  30296550  30364279  smpomuc   1000      -   67729\n",
       "22  HE601627.3  30296550  30426256  smpomuc   1000      -  129706\n",
       "23  HE601627.3  30348582  30364279  smpomuc   1000      -   15697\n",
       "24  HE601627.3  30348582  30426256  smpomuc   1000      -   77674\n",
       "25  HE601627.3  30375173  30426256  smpomuc   1000      -   51083\n",
       "26  HE601627.3  30396771  30426256  smpomuc   1000      -   29485"
      ]
     },
     "execution_count": 260,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idx = df.groupby(['chrom', 'start', 'stop'])['len'].idxmin()\n",
    "\n",
    "# Use the indices to filter the dataframe\n",
    "collapsed_df = df.loc[idx].reset_index(drop=True)\n",
    "\n",
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78c02c89-fb21-4d2e-aa96-dba963929e5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pysam\n",
    "\n",
    "# Open the SAM/BAM file\n",
    "with pysam.AlignmentFile('path_to_your_file.bam', 'rb') as bamfile:\n",
    "    # Iterate over each read in the file\n",
    "    for read in bamfile:\n",
    "        # Check if the read is mapped\n",
    "        if not read.is_unmapped:\n",
    "            # Get the reference name, start, and end position\n",
    "            ref_name = read.reference_name\n",
    "            start = read.reference_start\n",
    "            end = read.reference_end\n",
    "\n",
    "            print(f\"Read: {read.query_name}, Mapped to: {ref_name}, Start: {start}, End: {end}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47de5a9c-eee1-4dfe-a168-671e93178518",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6547b65-0cc4-4652-87a8-50d388824653",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cat \"\">smv9_smpomuc_locus.fas\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "id": "e01564bb-e20b-45f8-b7b3-6c89f527623e",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "ASSEM_DIR=\"/master/nplatt/sch_man_ont/results/assembly\"\n",
    "\n",
    "for POP in \"smle_pzq_er\" \"smle_pzq_es\" \"smbre\" \"smor\" \"smeg\"; do\n",
    "\n",
    "    conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "        blastn \\\n",
    "            -subject ${ASSEM_DIR}/${POP}_rnd0.fasta \\\n",
    "            -query chr4_smpomuc_smv9.fas \\\n",
    "            -out ${POP}_loc.blastn.tsv \\\n",
    "            -perc_identity 0.85 \\\n",
    "            -outfmt 6\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "id": "b7548f01-0d5e-449a-aebf-2d7589938bd6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_1001</td>\n",
       "      <td>97.79</td>\n",
       "      <td>3985</td>\n",
       "      <td>82</td>\n",
       "      <td>3</td>\n",
       "      <td>207044</td>\n",
       "      <td>211025</td>\n",
       "      <td>4937</td>\n",
       "      <td>956</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6866.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_1001</td>\n",
       "      <td>97.79</td>\n",
       "      <td>3985</td>\n",
       "      <td>82</td>\n",
       "      <td>3</td>\n",
       "      <td>106700</td>\n",
       "      <td>110681</td>\n",
       "      <td>4937</td>\n",
       "      <td>956</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6866.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_1001</td>\n",
       "      <td>97.76</td>\n",
       "      <td>3747</td>\n",
       "      <td>78</td>\n",
       "      <td>3</td>\n",
       "      <td>180454</td>\n",
       "      <td>184197</td>\n",
       "      <td>4937</td>\n",
       "      <td>1194</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6449.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_1001</td>\n",
       "      <td>96.03</td>\n",
       "      <td>2417</td>\n",
       "      <td>89</td>\n",
       "      <td>6</td>\n",
       "      <td>207044</td>\n",
       "      <td>209456</td>\n",
       "      <td>12703</td>\n",
       "      <td>10290</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3925.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_1001</td>\n",
       "      <td>96.03</td>\n",
       "      <td>2417</td>\n",
       "      <td>89</td>\n",
       "      <td>6</td>\n",
       "      <td>180454</td>\n",
       "      <td>182866</td>\n",
       "      <td>12703</td>\n",
       "      <td>10290</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3925.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113591</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_998</td>\n",
       "      <td>90.91</td>\n",
       "      <td>110</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>135119</td>\n",
       "      <td>135228</td>\n",
       "      <td>111282</td>\n",
       "      <td>111391</td>\n",
       "      <td>2.000000e-35</td>\n",
       "      <td>148.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113592</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_998</td>\n",
       "      <td>87.60</td>\n",
       "      <td>121</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>51476</td>\n",
       "      <td>51596</td>\n",
       "      <td>145041</td>\n",
       "      <td>144921</td>\n",
       "      <td>3.000000e-33</td>\n",
       "      <td>141.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113593</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_998</td>\n",
       "      <td>91.01</td>\n",
       "      <td>89</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>131997</td>\n",
       "      <td>132085</td>\n",
       "      <td>141723</td>\n",
       "      <td>141635</td>\n",
       "      <td>5.000000e-27</td>\n",
       "      <td>121.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113594</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_998</td>\n",
       "      <td>91.01</td>\n",
       "      <td>89</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>131997</td>\n",
       "      <td>132085</td>\n",
       "      <td>153893</td>\n",
       "      <td>153981</td>\n",
       "      <td>5.000000e-27</td>\n",
       "      <td>121.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113595</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_998</td>\n",
       "      <td>88.16</td>\n",
       "      <td>76</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>134776</td>\n",
       "      <td>134850</td>\n",
       "      <td>180289</td>\n",
       "      <td>180215</td>\n",
       "      <td>1.000000e-17</td>\n",
       "      <td>89.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>113596 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              qseqid       sseqid  pident  length  mismatch  \\\n",
       "0       HE601627.3:30170361-30426256  contig_1001   97.79    3985        82   \n",
       "1       HE601627.3:30170361-30426256  contig_1001   97.79    3985        82   \n",
       "2       HE601627.3:30170361-30426256  contig_1001   97.76    3747        78   \n",
       "3       HE601627.3:30170361-30426256  contig_1001   96.03    2417        89   \n",
       "4       HE601627.3:30170361-30426256  contig_1001   96.03    2417        89   \n",
       "...                              ...          ...     ...     ...       ...   \n",
       "113591  HE601627.3:30170361-30426256   contig_998   90.91     110        10   \n",
       "113592  HE601627.3:30170361-30426256   contig_998   87.60     121        15   \n",
       "113593  HE601627.3:30170361-30426256   contig_998   91.01      89         8   \n",
       "113594  HE601627.3:30170361-30426256   contig_998   91.01      89         8   \n",
       "113595  HE601627.3:30170361-30426256   contig_998   88.16      76         7   \n",
       "\n",
       "        gapopen  qstart    qend  sstart    send        evalue  bitscore  \n",
       "0             3  207044  211025    4937     956  0.000000e+00    6866.0  \n",
       "1             3  106700  110681    4937     956  0.000000e+00    6866.0  \n",
       "2             3  180454  184197    4937    1194  0.000000e+00    6449.0  \n",
       "3             6  207044  209456   12703   10290  0.000000e+00    3925.0  \n",
       "4             6  180454  182866   12703   10290  0.000000e+00    3925.0  \n",
       "...         ...     ...     ...     ...     ...           ...       ...  \n",
       "113591        0  135119  135228  111282  111391  2.000000e-35     148.0  \n",
       "113592        0   51476   51596  145041  144921  3.000000e-33     141.0  \n",
       "113593        0  131997  132085  141723  141635  5.000000e-27     121.0  \n",
       "113594        0  131997  132085  153893  153981  5.000000e-27     121.0  \n",
       "113595        2  134776  134850  180289  180215  1.000000e-17      89.8  \n",
       "\n",
       "[113596 rows x 12 columns]"
      ]
     },
     "execution_count": 308,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for pop in [\"smle_pzq_er\"]:\n",
    "    \n",
    "    blast_if = f\"{pop}_loc.blastn.tsv\"\n",
    "#     bed_if   = f\"{pop}_assem.blastn.bed\"\n",
    "#     tsv_of  = f\"{pop}_assem.blastn.paired.tsv\"\n",
    "\n",
    "#     #convert blast to bed\n",
    "#     df = blast_to_bed(blast_if)\n",
    "#     df.to_csv(bed_of, sep=\"\\t\", header=False, index=False)\n",
    "\n",
    "df=pd.read_csv(blast_if, sep=\"\\t\", header=None)\n",
    "df.columns=[\"qseqid\", \"sseqid\", \"pident\", \"length\", \"mismatch\", \"gapopen\", \"qstart\", \"qend\", \"sstart\", \"send\", \"evalue\", \"bitscore\"]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "id": "382f7c13-5e5f-4d4a-a6b0-7f92ac8419fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>hit_len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_1001</td>\n",
       "      <td>97.79</td>\n",
       "      <td>3985</td>\n",
       "      <td>82</td>\n",
       "      <td>3</td>\n",
       "      <td>207044</td>\n",
       "      <td>211025</td>\n",
       "      <td>4937</td>\n",
       "      <td>956</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6866.0</td>\n",
       "      <td>3981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_1001</td>\n",
       "      <td>97.79</td>\n",
       "      <td>3985</td>\n",
       "      <td>82</td>\n",
       "      <td>3</td>\n",
       "      <td>106700</td>\n",
       "      <td>110681</td>\n",
       "      <td>4937</td>\n",
       "      <td>956</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6866.0</td>\n",
       "      <td>3981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_1001</td>\n",
       "      <td>97.76</td>\n",
       "      <td>3747</td>\n",
       "      <td>78</td>\n",
       "      <td>3</td>\n",
       "      <td>180454</td>\n",
       "      <td>184197</td>\n",
       "      <td>4937</td>\n",
       "      <td>1194</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6449.0</td>\n",
       "      <td>3743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_1001</td>\n",
       "      <td>96.03</td>\n",
       "      <td>2417</td>\n",
       "      <td>89</td>\n",
       "      <td>6</td>\n",
       "      <td>207044</td>\n",
       "      <td>209456</td>\n",
       "      <td>12703</td>\n",
       "      <td>10290</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3925.0</td>\n",
       "      <td>2413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_1001</td>\n",
       "      <td>96.03</td>\n",
       "      <td>2417</td>\n",
       "      <td>89</td>\n",
       "      <td>6</td>\n",
       "      <td>180454</td>\n",
       "      <td>182866</td>\n",
       "      <td>12703</td>\n",
       "      <td>10290</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3925.0</td>\n",
       "      <td>2413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113591</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_998</td>\n",
       "      <td>90.91</td>\n",
       "      <td>110</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>135119</td>\n",
       "      <td>135228</td>\n",
       "      <td>111282</td>\n",
       "      <td>111391</td>\n",
       "      <td>2.000000e-35</td>\n",
       "      <td>148.0</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113592</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_998</td>\n",
       "      <td>87.60</td>\n",
       "      <td>121</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>51476</td>\n",
       "      <td>51596</td>\n",
       "      <td>145041</td>\n",
       "      <td>144921</td>\n",
       "      <td>3.000000e-33</td>\n",
       "      <td>141.0</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113593</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_998</td>\n",
       "      <td>91.01</td>\n",
       "      <td>89</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>131997</td>\n",
       "      <td>132085</td>\n",
       "      <td>141723</td>\n",
       "      <td>141635</td>\n",
       "      <td>5.000000e-27</td>\n",
       "      <td>121.0</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113594</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_998</td>\n",
       "      <td>91.01</td>\n",
       "      <td>89</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>131997</td>\n",
       "      <td>132085</td>\n",
       "      <td>153893</td>\n",
       "      <td>153981</td>\n",
       "      <td>5.000000e-27</td>\n",
       "      <td>121.0</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113595</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>contig_998</td>\n",
       "      <td>88.16</td>\n",
       "      <td>76</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>134776</td>\n",
       "      <td>134850</td>\n",
       "      <td>180289</td>\n",
       "      <td>180215</td>\n",
       "      <td>1.000000e-17</td>\n",
       "      <td>89.8</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>113596 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              qseqid       sseqid  pident  length  mismatch  \\\n",
       "0       HE601627.3:30170361-30426256  contig_1001   97.79    3985        82   \n",
       "1       HE601627.3:30170361-30426256  contig_1001   97.79    3985        82   \n",
       "2       HE601627.3:30170361-30426256  contig_1001   97.76    3747        78   \n",
       "3       HE601627.3:30170361-30426256  contig_1001   96.03    2417        89   \n",
       "4       HE601627.3:30170361-30426256  contig_1001   96.03    2417        89   \n",
       "...                              ...          ...     ...     ...       ...   \n",
       "113591  HE601627.3:30170361-30426256   contig_998   90.91     110        10   \n",
       "113592  HE601627.3:30170361-30426256   contig_998   87.60     121        15   \n",
       "113593  HE601627.3:30170361-30426256   contig_998   91.01      89         8   \n",
       "113594  HE601627.3:30170361-30426256   contig_998   91.01      89         8   \n",
       "113595  HE601627.3:30170361-30426256   contig_998   88.16      76         7   \n",
       "\n",
       "        gapopen  qstart    qend  sstart    send        evalue  bitscore  \\\n",
       "0             3  207044  211025    4937     956  0.000000e+00    6866.0   \n",
       "1             3  106700  110681    4937     956  0.000000e+00    6866.0   \n",
       "2             3  180454  184197    4937    1194  0.000000e+00    6449.0   \n",
       "3             6  207044  209456   12703   10290  0.000000e+00    3925.0   \n",
       "4             6  180454  182866   12703   10290  0.000000e+00    3925.0   \n",
       "...         ...     ...     ...     ...     ...           ...       ...   \n",
       "113591        0  135119  135228  111282  111391  2.000000e-35     148.0   \n",
       "113592        0   51476   51596  145041  144921  3.000000e-33     141.0   \n",
       "113593        0  131997  132085  141723  141635  5.000000e-27     121.0   \n",
       "113594        0  131997  132085  153893  153981  5.000000e-27     121.0   \n",
       "113595        2  134776  134850  180289  180215  1.000000e-17      89.8   \n",
       "\n",
       "        hit_len  \n",
       "0          3981  \n",
       "1          3981  \n",
       "2          3743  \n",
       "3          2413  \n",
       "4          2413  \n",
       "...         ...  \n",
       "113591      109  \n",
       "113592      120  \n",
       "113593       88  \n",
       "113594       88  \n",
       "113595       74  \n",
       "\n",
       "[113596 rows x 13 columns]"
      ]
     },
     "execution_count": 309,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"hit_len\"] = abs(df[\"send\"] - df[\"sstart\"])\n",
    "df.sort_values(by='hit_len', ascending=True).reset_index(drop=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "id": "a3aae71d-6e1e-4c45-b315-4770b6715cbc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::8.442*1.65] collected minimizers\n",
      "[M::mm_idx_gen::9.858*2.78] sorted minimizers\n",
      "[M::main::9.859*2.78] loaded/built the index for 1968 target sequence(s)\n",
      "[M::mm_mapopt_update::10.615*2.65] mid_occ = 373\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1968\n",
      "[M::mm_idx_stat::11.135*2.57] distinct minimizers: 30209362 (65.19% are singletons); average occurrences: 2.377; average spacing: 5.323; total length: 382213358\n",
      "[M::worker_pipeline::32.461*1.54] mapped 1 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: /master/nplatt/anaconda3/envs/sm_ont-process_reads/bin/minimap2 -a -o test.sam -t 48 -x map-ont ../assembly/smle_pzq_er_rnd0.fasta chr4_smpomuc_smv9.fas\n",
      "[M::main] Real time: 32.710 sec; CPU: 50.182 sec; Peak RSS: 3.120 GB\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "~/anaconda3/envs/sm_ont-process_reads/bin/minimap2 -a -o test.sam -t 48 -x map-ont ../assembly/smle_pzq_er_rnd0.fasta chr4_smpomuc_smv9.fas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "id": "f0ca0888-fdcf-4ac8-94d9-e685e968381c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "      <th>22</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>0</td>\n",
       "      <td>contig_908</td>\n",
       "      <td>1385</td>\n",
       "      <td>60</td>\n",
       "      <td>68821S291M1I862M1D399M1D283M1I11M2D2M45D102M1D...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TTGACACAGAAAACTGTTAACGATCCAAAACATGTCTGTCCATATC...</td>\n",
       "      <td>...</td>\n",
       "      <td>AS:i:122090</td>\n",
       "      <td>nn:i:200</td>\n",
       "      <td>tp:A:P</td>\n",
       "      <td>cm:i:15615</td>\n",
       "      <td>s1:i:80466</td>\n",
       "      <td>s2:i:33012</td>\n",
       "      <td>de:f:0.0102</td>\n",
       "      <td>zd:i:1</td>\n",
       "      <td>SA:Z:contig_1657,92,-,188036S34560M21I33279S,6...</td>\n",
       "      <td>rl:i:10614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>2064</td>\n",
       "      <td>contig_1657</td>\n",
       "      <td>92</td>\n",
       "      <td>60</td>\n",
       "      <td>188036H1831M1D2764M1D1988M1I590M1I350M8I766M6D...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TTATTTCTTCTCGTCTAGTCCTACACGTTTGAATCCAGAACACAAT...</td>\n",
       "      <td>...</td>\n",
       "      <td>AS:i:68198</td>\n",
       "      <td>nn:i:0</td>\n",
       "      <td>tp:A:P</td>\n",
       "      <td>cm:i:5222</td>\n",
       "      <td>s1:i:29788</td>\n",
       "      <td>s2:i:0</td>\n",
       "      <td>de:f:0.0042</td>\n",
       "      <td>SA:Z:contig_908,1385,+,68821S100592M52553I3393...</td>\n",
       "      <td>rl:i:10614</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>2048</td>\n",
       "      <td>contig_191</td>\n",
       "      <td>5569193</td>\n",
       "      <td>60</td>\n",
       "      <td>201909H512M1D160M2I593M16D289M3I4M7I2M4I135M1I...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TTCGGAAAATTCAGGGCCTAGTTTTATTTCGCTTATACAAAGGTGC...</td>\n",
       "      <td>...</td>\n",
       "      <td>AS:i:38364</td>\n",
       "      <td>nn:i:100</td>\n",
       "      <td>tp:A:P</td>\n",
       "      <td>cm:i:5577</td>\n",
       "      <td>s1:i:27979</td>\n",
       "      <td>s2:i:24967</td>\n",
       "      <td>de:f:0.0249</td>\n",
       "      <td>SA:Z:contig_908,1385,+,68821S100592M52553I3393...</td>\n",
       "      <td>rl:i:10614</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>256</td>\n",
       "      <td>contig_191</td>\n",
       "      <td>5589245</td>\n",
       "      <td>0</td>\n",
       "      <td>228744S88M1I115M1D27M2I10M1D19M14I8M2I439M1I2M...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>*</td>\n",
       "      <td>...</td>\n",
       "      <td>AS:i:50787</td>\n",
       "      <td>nn:i:0</td>\n",
       "      <td>tp:A:S</td>\n",
       "      <td>cm:i:4567</td>\n",
       "      <td>s1:i:24967</td>\n",
       "      <td>de:f:0.0082</td>\n",
       "      <td>rl:i:10614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>272</td>\n",
       "      <td>contig_757</td>\n",
       "      <td>647</td>\n",
       "      <td>0</td>\n",
       "      <td>402S19M3I77M17D918M6I10M6I10M8I1064M2D6M4D9M5D...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>*</td>\n",
       "      <td>...</td>\n",
       "      <td>AS:i:48992</td>\n",
       "      <td>nn:i:0</td>\n",
       "      <td>tp:A:S</td>\n",
       "      <td>cm:i:4150</td>\n",
       "      <td>s1:i:22491</td>\n",
       "      <td>de:f:0.0183</td>\n",
       "      <td>zd:i:1</td>\n",
       "      <td>rl:i:10614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>256</td>\n",
       "      <td>contig_191</td>\n",
       "      <td>5591314</td>\n",
       "      <td>0</td>\n",
       "      <td>228744S88M1I142M2I33M13I5M2I442M1D8M1D303M1D24...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>*</td>\n",
       "      <td>...</td>\n",
       "      <td>AS:i:49127</td>\n",
       "      <td>nn:i:0</td>\n",
       "      <td>tp:A:S</td>\n",
       "      <td>cm:i:4443</td>\n",
       "      <td>s1:i:24102</td>\n",
       "      <td>de:f:0.0067</td>\n",
       "      <td>rl:i:10614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>2048</td>\n",
       "      <td>contig_908</td>\n",
       "      <td>1385</td>\n",
       "      <td>60</td>\n",
       "      <td>3433H291M1I862M1D399M1D271M1D28M7D6M1D2M4D3M10...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TAATGATTCTCATTATAAACCTATCAAATTACTAAACGAACAGTTT...</td>\n",
       "      <td>...</td>\n",
       "      <td>AS:i:3669</td>\n",
       "      <td>nn:i:0</td>\n",
       "      <td>tp:A:P</td>\n",
       "      <td>cm:i:2613</td>\n",
       "      <td>s1:i:13415</td>\n",
       "      <td>s2:i:29788</td>\n",
       "      <td>de:f:0.0313</td>\n",
       "      <td>zd:i:1</td>\n",
       "      <td>SA:Z:contig_908,1385,+,68821S100592M52553I3393...</td>\n",
       "      <td>rl:i:10614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>2064</td>\n",
       "      <td>contig_1657</td>\n",
       "      <td>23032</td>\n",
       "      <td>49</td>\n",
       "      <td>253385H469M1I1055M1D986M</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TATCAGAACGAACAGCCTAGAGTCAACTCTCAGTCCGTATTATTTC...</td>\n",
       "      <td>...</td>\n",
       "      <td>AS:i:4732</td>\n",
       "      <td>nn:i:0</td>\n",
       "      <td>tp:A:P</td>\n",
       "      <td>cm:i:385</td>\n",
       "      <td>s1:i:2182</td>\n",
       "      <td>s2:i:2549</td>\n",
       "      <td>de:f:0.0191</td>\n",
       "      <td>zd:i:2</td>\n",
       "      <td>SA:Z:contig_908,1385,+,68821S100592M52553I3393...</td>\n",
       "      <td>rl:i:10614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>272</td>\n",
       "      <td>contig_908</td>\n",
       "      <td>102218</td>\n",
       "      <td>0</td>\n",
       "      <td>30277S7M1I378M1I448M16D591M4I681M223508S</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>*</td>\n",
       "      <td>...</td>\n",
       "      <td>AS:i:3964</td>\n",
       "      <td>nn:i:0</td>\n",
       "      <td>tp:A:i</td>\n",
       "      <td>cm:i:0</td>\n",
       "      <td>s1:i:0</td>\n",
       "      <td>de:f:0.0166</td>\n",
       "      <td>rl:i:10614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>2064</td>\n",
       "      <td>contig_471</td>\n",
       "      <td>2105599</td>\n",
       "      <td>60</td>\n",
       "      <td>336M255560H</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ATGAACAAGAAAATTATTCTCTTGACAATAATATATATTGCGCTGA...</td>\n",
       "      <td>...</td>\n",
       "      <td>AS:i:666</td>\n",
       "      <td>nn:i:0</td>\n",
       "      <td>tp:A:P</td>\n",
       "      <td>cm:i:54</td>\n",
       "      <td>s1:i:320</td>\n",
       "      <td>s2:i:78</td>\n",
       "      <td>de:f:0.0030</td>\n",
       "      <td>SA:Z:contig_908,1385,+,68821S100592M52553I3393...</td>\n",
       "      <td>rl:i:10614</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             0     1            2        3   4   \\\n",
       "0  HE601627.3:30170361-30426256     0   contig_908     1385  60   \n",
       "1  HE601627.3:30170361-30426256  2064  contig_1657       92  60   \n",
       "2  HE601627.3:30170361-30426256  2048   contig_191  5569193  60   \n",
       "3  HE601627.3:30170361-30426256   256   contig_191  5589245   0   \n",
       "4  HE601627.3:30170361-30426256   272   contig_757      647   0   \n",
       "5  HE601627.3:30170361-30426256   256   contig_191  5591314   0   \n",
       "6  HE601627.3:30170361-30426256  2048   contig_908     1385  60   \n",
       "7  HE601627.3:30170361-30426256  2064  contig_1657    23032  49   \n",
       "8  HE601627.3:30170361-30426256   272   contig_908   102218   0   \n",
       "9  HE601627.3:30170361-30426256  2064   contig_471  2105599  60   \n",
       "\n",
       "                                                  5  6   7   8   \\\n",
       "0  68821S291M1I862M1D399M1D283M1I11M2D2M45D102M1D...  *   0   0   \n",
       "1  188036H1831M1D2764M1D1988M1I590M1I350M8I766M6D...  *   0   0   \n",
       "2  201909H512M1D160M2I593M16D289M3I4M7I2M4I135M1I...  *   0   0   \n",
       "3  228744S88M1I115M1D27M2I10M1D19M14I8M2I439M1I2M...  *   0   0   \n",
       "4  402S19M3I77M17D918M6I10M6I10M8I1064M2D6M4D9M5D...  *   0   0   \n",
       "5  228744S88M1I142M2I33M13I5M2I442M1D8M1D303M1D24...  *   0   0   \n",
       "6  3433H291M1I862M1D399M1D271M1D28M7D6M1D2M4D3M10...  *   0   0   \n",
       "7                           253385H469M1I1055M1D986M  *   0   0   \n",
       "8           30277S7M1I378M1I448M16D591M4I681M223508S  *   0   0   \n",
       "9                                        336M255560H  *   0   0   \n",
       "\n",
       "                                                  9   ...           13  \\\n",
       "0  TTGACACAGAAAACTGTTAACGATCCAAAACATGTCTGTCCATATC...  ...  AS:i:122090   \n",
       "1  TTATTTCTTCTCGTCTAGTCCTACACGTTTGAATCCAGAACACAAT...  ...   AS:i:68198   \n",
       "2  TTCGGAAAATTCAGGGCCTAGTTTTATTTCGCTTATACAAAGGTGC...  ...   AS:i:38364   \n",
       "3                                                  *  ...   AS:i:50787   \n",
       "4                                                  *  ...   AS:i:48992   \n",
       "5                                                  *  ...   AS:i:49127   \n",
       "6  TAATGATTCTCATTATAAACCTATCAAATTACTAAACGAACAGTTT...  ...    AS:i:3669   \n",
       "7  TATCAGAACGAACAGCCTAGAGTCAACTCTCAGTCCGTATTATTTC...  ...    AS:i:4732   \n",
       "8                                                  *  ...    AS:i:3964   \n",
       "9  ATGAACAAGAAAATTATTCTCTTGACAATAATATATATTGCGCTGA...  ...     AS:i:666   \n",
       "\n",
       "         14      15          16          17           18           19  \\\n",
       "0  nn:i:200  tp:A:P  cm:i:15615  s1:i:80466   s2:i:33012  de:f:0.0102   \n",
       "1    nn:i:0  tp:A:P   cm:i:5222  s1:i:29788       s2:i:0  de:f:0.0042   \n",
       "2  nn:i:100  tp:A:P   cm:i:5577  s1:i:27979   s2:i:24967  de:f:0.0249   \n",
       "3    nn:i:0  tp:A:S   cm:i:4567  s1:i:24967  de:f:0.0082   rl:i:10614   \n",
       "4    nn:i:0  tp:A:S   cm:i:4150  s1:i:22491  de:f:0.0183       zd:i:1   \n",
       "5    nn:i:0  tp:A:S   cm:i:4443  s1:i:24102  de:f:0.0067   rl:i:10614   \n",
       "6    nn:i:0  tp:A:P   cm:i:2613  s1:i:13415   s2:i:29788  de:f:0.0313   \n",
       "7    nn:i:0  tp:A:P    cm:i:385   s1:i:2182    s2:i:2549  de:f:0.0191   \n",
       "8    nn:i:0  tp:A:i      cm:i:0      s1:i:0  de:f:0.0166   rl:i:10614   \n",
       "9    nn:i:0  tp:A:P     cm:i:54    s1:i:320      s2:i:78  de:f:0.0030   \n",
       "\n",
       "                                                  20  \\\n",
       "0                                             zd:i:1   \n",
       "1  SA:Z:contig_908,1385,+,68821S100592M52553I3393...   \n",
       "2  SA:Z:contig_908,1385,+,68821S100592M52553I3393...   \n",
       "3                                                NaN   \n",
       "4                                         rl:i:10614   \n",
       "5                                                NaN   \n",
       "6                                             zd:i:1   \n",
       "7                                             zd:i:2   \n",
       "8                                                NaN   \n",
       "9  SA:Z:contig_908,1385,+,68821S100592M52553I3393...   \n",
       "\n",
       "                                                  21          22  \n",
       "0  SA:Z:contig_1657,92,-,188036S34560M21I33279S,6...  rl:i:10614  \n",
       "1                                         rl:i:10614         NaN  \n",
       "2                                         rl:i:10614         NaN  \n",
       "3                                                NaN         NaN  \n",
       "4                                                NaN         NaN  \n",
       "5                                                NaN         NaN  \n",
       "6  SA:Z:contig_908,1385,+,68821S100592M52553I3393...  rl:i:10614  \n",
       "7  SA:Z:contig_908,1385,+,68821S100592M52553I3393...  rl:i:10614  \n",
       "8                                                NaN         NaN  \n",
       "9                                         rl:i:10614         NaN  \n",
       "\n",
       "[10 rows x 23 columns]"
      ]
     },
     "execution_count": 314,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.read_csv(\"test.sam\", sep=\"\\t\", header=None, comment='@')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "702841c8-2209-4c69-9b6d-f4e26896d7f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "[\"QNAME\",\n",
    "\"FLAG\",\n",
    "\"RNAME\",\n",
    "\"POS\",\n",
    "\"MAPQ\",\n",
    "\"CIGAR\",\n",
    "\"RNEXT\",\n",
    "\"PNEXT\",\n",
    "\"TLEN\",\n",
    "\"SEQ\",\n",
    "\"QUA\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "id": "6908fca5-c2bf-4ba4-bfb5-ca177480d966",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2730335-ed83-40b6-8fd9-3a3ea3f7da89",
   "metadata": {},
   "outputs": [],
   "source": [
    "#split the smv9 smpomuc locus into 20kb chunks then map these to the genomes with minimap2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "id": "0407db4a-076e-4eef-9b2c-097d377bf700",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sliding_window(sequence, window_size, step_size):\n",
    "    \"\"\"Generate chunks of sequence based on a sliding window.\"\"\"\n",
    "    for i in range(0, len(sequence) - window_size + 1, step_size):\n",
    "        yield sequence[i:i+window_size]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "id": "36945656-f25b-4ec5-be93-0680b9b0957d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read the sequence from a FASTA file\n",
    "input_file = 'chr4_smpomuc_smv9.fas'\n",
    "output_file = 'chr4_smpomuc_smv9_chunked.fas'\n",
    "\n",
    "window=30_000\n",
    "step=30_000\n",
    "\n",
    "chunks = []\n",
    "\n",
    "with open(input_file, 'r') as fasta_file:\n",
    "    for record in SeqIO.parse(fasta_file, 'fasta'):\n",
    "        seq_id = record.id\n",
    "        seq = record.seq\n",
    "\n",
    "        # Split the sequence using the sliding window and store the chunks\n",
    "        for index, chunk in enumerate(sliding_window(seq, window, step), 1):\n",
    "            new_record = SeqRecord(chunk, id=f\"{seq_id}_chunk{index}\", description=\"\")\n",
    "            chunks.append(new_record)\n",
    "\n",
    "# Save the chunks to a new FASTA file\n",
    "with open(output_file, 'w') as output:\n",
    "    SeqIO.write(chunks, output, 'fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "id": "96c6fb04-8a61-432c-8aac-5814ce36bb2a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::7.514*1.70] collected minimizers\n",
      "[M::mm_idx_gen::8.702*2.76] sorted minimizers\n",
      "[M::main::8.702*2.76] loaded/built the index for 1674 target sequence(s)\n",
      "[M::mm_mapopt_update::9.411*2.62] mid_occ = 384\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1674\n",
      "[M::mm_idx_stat::9.927*2.54] distinct minimizers: 30374815 (65.19% are singletons); average occurrences: 2.391; average spacing: 5.323; total length: 386656054\n",
      "[M::worker_pipeline::38.716*1.39] mapped 1 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: /master/nplatt/anaconda3/envs/sm_ont-process_reads/bin/minimap2 -a -o test.sam -t 48 -x map-ont ../assembly/smor_rnd0.fasta chr4_smpomuc_smv9.fas\n",
      "[M::main] Real time: 39.012 sec; CPU: 54.223 sec; Peak RSS: 3.220 GB\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#chr4_smpomuc_smv9.fas\n",
    "#chr4_smpomuc_smv9_chunked.fas\n",
    "~/anaconda3/envs/sm_ont-process_reads/bin/minimap2 -a -o test.sam -t 48 -x map-ont ../assembly/smor_rnd0.fasta chr4_smpomuc_smv9.fas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "id": "d5f1c725-20a0-4099-87de-efb20ce50a31",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bam_header_read] EOF marker is absent. The input is probably truncated.\n",
      "[bam_header_read] invalid BAM binary header (this is not a BAM file).\n",
      "[main_samview] fail to read the header from \"chunks.sam\".\n"
     ]
    }
   ],
   "source": [
    "!samtools view -F0x100 chunks.sam >filtered.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "id": "e094093f-3129-4abb-96ff-39e38dbdff10",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>0</td>\n",
       "      <td>contig_908</td>\n",
       "      <td>1385</td>\n",
       "      <td>60</td>\n",
       "      <td>68821S291M1I862M1D399M1D283M1I11M2D2M45D102M1D...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TTGACACAGAAAACTGTTAACGATCCAAAACATGTCTGTCCATATC...</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>2064</td>\n",
       "      <td>contig_1657</td>\n",
       "      <td>92</td>\n",
       "      <td>60</td>\n",
       "      <td>188036H1831M1D2764M1D1988M1I590M1I350M8I766M6D...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TTATTTCTTCTCGTCTAGTCCTACACGTTTGAATCCAGAACACAAT...</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>2048</td>\n",
       "      <td>contig_191</td>\n",
       "      <td>5569193</td>\n",
       "      <td>60</td>\n",
       "      <td>201909H512M1D160M2I593M16D289M3I4M7I2M4I135M1I...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TTCGGAAAATTCAGGGCCTAGTTTTATTTCGCTTATACAAAGGTGC...</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>256</td>\n",
       "      <td>contig_191</td>\n",
       "      <td>5589245</td>\n",
       "      <td>0</td>\n",
       "      <td>228744S88M1I115M1D27M2I10M1D19M14I8M2I439M1I2M...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>272</td>\n",
       "      <td>contig_757</td>\n",
       "      <td>647</td>\n",
       "      <td>0</td>\n",
       "      <td>402S19M3I77M17D918M6I10M6I10M8I1064M2D6M4D9M5D...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>256</td>\n",
       "      <td>contig_191</td>\n",
       "      <td>5591314</td>\n",
       "      <td>0</td>\n",
       "      <td>228744S88M1I142M2I33M13I5M2I442M1D8M1D303M1D24...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>2048</td>\n",
       "      <td>contig_908</td>\n",
       "      <td>1385</td>\n",
       "      <td>60</td>\n",
       "      <td>3433H291M1I862M1D399M1D271M1D28M7D6M1D2M4D3M10...</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TAATGATTCTCATTATAAACCTATCAAATTACTAAACGAACAGTTT...</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>2064</td>\n",
       "      <td>contig_1657</td>\n",
       "      <td>23032</td>\n",
       "      <td>49</td>\n",
       "      <td>253385H469M1I1055M1D986M</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TATCAGAACGAACAGCCTAGAGTCAACTCTCAGTCCGTATTATTTC...</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>272</td>\n",
       "      <td>contig_908</td>\n",
       "      <td>102218</td>\n",
       "      <td>0</td>\n",
       "      <td>30277S7M1I378M1I448M16D591M4I681M223508S</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HE601627.3:30170361-30426256</td>\n",
       "      <td>2064</td>\n",
       "      <td>contig_471</td>\n",
       "      <td>2105599</td>\n",
       "      <td>60</td>\n",
       "      <td>336M255560H</td>\n",
       "      <td>*</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ATGAACAAGAAAATTATTCTCTTGACAATAATATATATTGCGCTGA...</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             0     1            2        3   4   \\\n",
       "0  HE601627.3:30170361-30426256     0   contig_908     1385  60   \n",
       "1  HE601627.3:30170361-30426256  2064  contig_1657       92  60   \n",
       "2  HE601627.3:30170361-30426256  2048   contig_191  5569193  60   \n",
       "3  HE601627.3:30170361-30426256   256   contig_191  5589245   0   \n",
       "4  HE601627.3:30170361-30426256   272   contig_757      647   0   \n",
       "5  HE601627.3:30170361-30426256   256   contig_191  5591314   0   \n",
       "6  HE601627.3:30170361-30426256  2048   contig_908     1385  60   \n",
       "7  HE601627.3:30170361-30426256  2064  contig_1657    23032  49   \n",
       "8  HE601627.3:30170361-30426256   272   contig_908   102218   0   \n",
       "9  HE601627.3:30170361-30426256  2064   contig_471  2105599  60   \n",
       "\n",
       "                                                  5  6   7   8   \\\n",
       "0  68821S291M1I862M1D399M1D283M1I11M2D2M45D102M1D...  *   0   0   \n",
       "1  188036H1831M1D2764M1D1988M1I590M1I350M8I766M6D...  *   0   0   \n",
       "2  201909H512M1D160M2I593M16D289M3I4M7I2M4I135M1I...  *   0   0   \n",
       "3  228744S88M1I115M1D27M2I10M1D19M14I8M2I439M1I2M...  *   0   0   \n",
       "4  402S19M3I77M17D918M6I10M6I10M8I1064M2D6M4D9M5D...  *   0   0   \n",
       "5  228744S88M1I142M2I33M13I5M2I442M1D8M1D303M1D24...  *   0   0   \n",
       "6  3433H291M1I862M1D399M1D271M1D28M7D6M1D2M4D3M10...  *   0   0   \n",
       "7                           253385H469M1I1055M1D986M  *   0   0   \n",
       "8           30277S7M1I378M1I448M16D591M4I681M223508S  *   0   0   \n",
       "9                                        336M255560H  *   0   0   \n",
       "\n",
       "                                                  9  10  \n",
       "0  TTGACACAGAAAACTGTTAACGATCCAAAACATGTCTGTCCATATC...  *  \n",
       "1  TTATTTCTTCTCGTCTAGTCCTACACGTTTGAATCCAGAACACAAT...  *  \n",
       "2  TTCGGAAAATTCAGGGCCTAGTTTTATTTCGCTTATACAAAGGTGC...  *  \n",
       "3                                                  *  *  \n",
       "4                                                  *  *  \n",
       "5                                                  *  *  \n",
       "6  TAATGATTCTCATTATAAACCTATCAAATTACTAAACGAACAGTTT...  *  \n",
       "7  TATCAGAACGAACAGCCTAGAGTCAACTCTCAGTCCGTATTATTTC...  *  \n",
       "8                                                  *  *  \n",
       "9  ATGAACAAGAAAATTATTCTCTTGACAATAATATATATTGCGCTGA...  *  "
      ]
     },
     "execution_count": 334,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.read_csv(\"test.sam\", sep=\"\\t\", header=None, comment='@', usecols=range(11))\n",
    "# df.loc[df[2] == \"contig_908\"]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d43058de-ca2f-4983-850e-c21ef37955a0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "id": "15a52718-c91b-44c5-aca5-13f7320bf6d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "echo -e \">Exon1F12\\nGGAAGAATGAACAAGAAAATTATTCTC\" >primers.fas\n",
    "echo -e \">Exon15R\\nTGACACAGAAAACTGTTAACGATCC\" >>primers.fas\n",
    "\n",
    "echo -e \"smpomuc\\tGGAAGAATGAACAAGAAAATTATTCTC\\tTGACACAGAAAACTGTTAACGATCC\" >sm_primers.txt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86d0cb32-a117-48e4-bce8-45df40c9e1e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "conda run -n sm_ont-process_reads isPcr -out=bed -maxSize=1000000 ../../data/genome/SM_V9.fa sm_primers.txt  pcr.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 376,
   "id": "e877ecf4-00a0-47e8-83a0-743ce263fb15",
   "metadata": {},
   "outputs": [],
   "source": [
    "for pop in pops + [\"smv9\"]:\n",
    "    \n",
    "    # fas = f\"/master/nplatt/sch_man_ont/results/assembly/{pop}_rnd0.fasta\"\n",
    "    # cmd=f\"isPcr -out=bed -maxSize=1000000 -flipReverse {fas} primers.txt {pop}.pcr.bed\"\n",
    "    # print(cmd)\n",
    "    # !cmd\n",
    "\n",
    "    df=pd.read_csv(f\"{pop}.pcr.bed\", sep=\"\\t\", header=None)\n",
    "    df.columns=[\"chrom\", \"start\", \"stop\", \"name\", \"score\", \"strand\"]\n",
    "    df[\"len\"] = df[\"stop\"] - df[\"start\"]\n",
    "\n",
    "    df=df.sort_values(by=['start', 'len'], ascending=True).reset_index(drop=True)\n",
    "\n",
    "\n",
    "    # idx = df.groupby(['chrom', 'start'])['len'].idxmin()\n",
    "    # df = df.loc[idx].reset_index(drop=True)\n",
    "    # idx = df.groupby(['chrom', 'stop'])['len'].idxmin()\n",
    "    # df = df.loc[idx].reset_index(drop=True)\n",
    "    #df.to_csv(f\"{pop}.pcr.collapsed.bed\", sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 375,
   "id": "ac88b4ed-213e-4c36-a9df-3179aa411b86",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HE601625.3</td>\n",
       "      <td>38377256</td>\n",
       "      <td>38413100</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>35844</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30170361</td>\n",
       "      <td>30203241</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>32880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30232534</td>\n",
       "      <td>30252317</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>981</td>\n",
       "      <td>-</td>\n",
       "      <td>19783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30296550</td>\n",
       "      <td>30334716</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>38166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30348582</td>\n",
       "      <td>30364285</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>15703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HE601627.3</td>\n",
       "      <td>30396771</td>\n",
       "      <td>30426262</td>\n",
       "      <td>smpomuc</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>29491</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom     start      stop     name  score strand    len\n",
       "0  HE601625.3  38377256  38413100  smpomuc   1000      +  35844\n",
       "1  HE601627.3  30170361  30203241  smpomuc   1000      -  32880\n",
       "2  HE601627.3  30232534  30252317  smpomuc    981      -  19783\n",
       "3  HE601627.3  30296550  30334716  smpomuc   1000      -  38166\n",
       "4  HE601627.3  30348582  30364285  smpomuc   1000      -  15703\n",
       "5  HE601627.3  30396771  30426262  smpomuc   1000      -  29491"
      ]
     },
     "execution_count": 375,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 377,
   "id": "00e136ed-78a4-401b-84b9-8fd782701562",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: cwd: command not found\n"
     ]
    }
   ],
   "source": [
    "!cwd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b993cb4c-b9ca-49f0-863b-f9382cc7ef4a",
   "metadata": {
    "tags": [],
    "toc-hr-collapsed": true
   },
   "source": [
    "## Genome assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1e83da5-f68a-40fb-aa3a-29a1cf626880",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Read Correction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "462324a1-d21b-4ce4-89ea-4da1a49eb363",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/read_correction\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/read_correction\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0badcb5-e1b7-4e00-983f-e616c46e438e",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "    Path(\"{}/results/read_correction/{}\".format(proj_dir, sample)).mkdir(parents=True, exist_ok=True)\n",
    "    \n",
    "    os.chdir(\"{}/results/read_correction\".format(proj_dir, sample))\n",
    "    \n",
    "    !conda run -n ont_assembly --cwd . --live-stream canu -correct -genomesize=385m -useGrid=False -p {sample}  -nanopore {proj_dir}/results/raw_reads/{sample}.raw_pass.fastq\n",
    "    \n",
    "    os.chdir(\"{}/results/read_correction\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39d8bb33-5379-4a91-b4bc-6c0f510761db",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash\n",
    "\n",
    "for SAMPLE in 'smor' 'smle_pzq_es' 'smle_pzq_er' 'smbre' 'smeg'; do\n",
    "\n",
    "    if [ -d ${SAMPLE} ]; then\n",
    "        rm -rf ${SAMPLE}\n",
    "    fi\n",
    "    \n",
    "    mkdir ${SAMPLE}\n",
    "    cd ${SAMPLE}\n",
    "    \n",
    "    LOG=$(pwd)/${SAMPLE}.log \n",
    "    \n",
    "    QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 192 -N ${SAMPLE} -o ${LOG}\"\n",
    "    CMD=\"conda run -n ont_assembly --cwd . --live-stream canu -correct -genomesize=385m -useGrid=False -p ${SAMPLE}  -nanopore /master/nplatt/sm_nanopore/results/raw_reads/${SAMPLE}.raw_pass.fastq\"\n",
    "    \n",
    "    echo $CMD | $QSUB\n",
    "    cd ..\n",
    "done\n",
    "\n",
    "for SAMPLE in 'smor' 'smle_pzq_es' 'smbre'; do\n",
    "    rm -rf $SAMPLE*\n",
    "    QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 192 -N ${SAMPLE} -o ${SAMPLE}.log\"\n",
    "    CMD=\"conda run -n ont_assembly --cwd . --live-stream canu -correct -genomesize=385m -useGrid=False -p ${SAMPLE}/${SAMPLE}  -nanopore /master/nplatt/sm_nanopore/results/raw_reads/${SAMPLE}.raw_pass.fastq\"\n",
    "    \n",
    "    echo $CMD | $QSUB\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "218b0bb6-ecc0-4350-9c66-cbcb0666e51f",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Genome assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "2622026b-669a-41a4-b722-2ff5ccb79193",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/assembly\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/assembly\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13c30768-4f95-4904-a110-4d27b38ff59b",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "    !flye --nano-cor {proj_dir}/results/read_correction/{sample}.correctedReads.fasta.gz -g 385m -t 128 -i 3 -o {sample}_rnd0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2a56d32-e431-4195-a5f8-ea56d1accc5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "for SAMPLE in 'smor' 'smle_pzq_es' 'smle_pzq_er' 'smbre' 'smeg'; do\n",
    "\n",
    "    if [ -d ${SAMPLE} ]; then\n",
    "        rm -rf ${SAMPLE}\n",
    "    fi\n",
    "    \n",
    "    mkdir ${SAMPLE}\n",
    "    cd ${SAMPLE}\n",
    "    \n",
    "    LOG=$(pwd)/${SAMPLE}.log \n",
    "    READS=\"/master/nplatt/sm_nanopore/results/read_correction/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz\"\n",
    "    OUT=\"/master/nplatt/sm_nanopore/results/assembly/${SAMPLE}/${SAMPLE}_rnd0\"\n",
    "    \n",
    "    QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 128 -N ${SAMPLE} -o ${LOG}\"\n",
    "    CMD=\"conda run -n sm_ont-process_reads --cwd . --live-stream flye --nano-cor ${READS} -g 385m -t 128 -i 3 -o ${OUT}\"\n",
    "    \n",
    "    echo $CMD | $QSUB\n",
    "    cd ..\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "47970cec-d20a-4f7a-b46f-a6a210585b9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>total_length</th>\n",
       "      <th>fragments</th>\n",
       "      <th>fragment_n50</th>\n",
       "      <th>largest_fragment</th>\n",
       "      <th>scaffolds</th>\n",
       "      <th>mean_coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>smor</td>\n",
       "      <td>386656054</td>\n",
       "      <td>1674</td>\n",
       "      <td>2150179</td>\n",
       "      <td>8090473</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>smle_pzq_es</td>\n",
       "      <td>403425661</td>\n",
       "      <td>1122</td>\n",
       "      <td>3481572</td>\n",
       "      <td>14813648</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>smle_pzq_er</td>\n",
       "      <td>382213358</td>\n",
       "      <td>1968</td>\n",
       "      <td>1273691</td>\n",
       "      <td>5768858</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>smbre</td>\n",
       "      <td>384951218</td>\n",
       "      <td>1020</td>\n",
       "      <td>2298594</td>\n",
       "      <td>14522024</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>smeg</td>\n",
       "      <td>370379461</td>\n",
       "      <td>1963</td>\n",
       "      <td>871323</td>\n",
       "      <td>6130221</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample total_length fragments fragment_n50 largest_fragment scaffolds  \\\n",
       "0         smor    386656054      1674      2150179          8090473         0   \n",
       "1  smle_pzq_es    403425661      1122      3481572         14813648         0   \n",
       "2  smle_pzq_er    382213358      1968      1273691          5768858         0   \n",
       "3        smbre    384951218      1020      2298594         14522024         0   \n",
       "4         smeg    370379461      1963       871323          6130221         0   \n",
       "\n",
       "  mean_coverage  \n",
       "0            23  \n",
       "1            33  \n",
       "2            15  \n",
       "3            24  \n",
       "4            14  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#get df of assembly stats\n",
    "total_lengths=[]\n",
    "fragments=[]\n",
    "fragment_n50s=[]\n",
    "largest_frags=[]\n",
    "scaffolds=[]\n",
    "mean_coverages=[]\n",
    "    \n",
    "for sample in samples:\n",
    "    info = !tail -n 8 {sample}/{sample}.log | head -n 6\n",
    "    total_lengths.append(info[0].split(\"\\t\")[-1])\n",
    "    fragments.append(info[1].split(\"\\t\")[-1])\n",
    "    fragment_n50s.append(info[2].split(\"\\t\")[-1])\n",
    "    largest_frags.append(info[3].split(\"\\t\")[-1])\n",
    "    scaffolds.append(info[4].split(\"\\t\")[-1])\n",
    "    mean_coverages.append(info[5].split(\"\\t\")[-1])\n",
    "    \n",
    "assembly_df=pd.DataFrame(data = [samples, total_lengths, fragments, fragment_n50s, largest_frags, scaffolds, mean_coverages]).T\n",
    "assembly_df.columns=[\"sample\", \"total_length\", \"fragments\", \"fragment_n50\", \"largest_fragment\", \"scaffolds\", \"mean_coverage\"]\n",
    "assembly_df.to_csv(\"assembly_info.csv\", index=False)\n",
    "assembly_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "c87716bd-431b-4e06-ab21-e8afc04f2a24",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "smor\n",
      "smle_pzq_es\n",
      "smle_pzq_er\n",
      "smbre\n",
      "smeg\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    print(sample)\n",
    "    ! cp /master/nplatt/sm_nanopore/results/assembly/{sample}/{sample}_rnd0/assembly.fasta {sample}_rnd0.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12de9d93-c60c-41f4-8556-370f4e265f84",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### polish assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "c1ea2ab8-cc08-4c9e-8819-9803b66f777c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/polish_assembly\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/polish_assembly\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "e71a2c33-555b-4c0a-aa42-7af40c7e2947",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "smor\n",
      "smle_pzq_es\n",
      "smle_pzq_er\n",
      "smbre\n",
      "smeg\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    print(sample)\n",
    "    !ln -s /master/nplatt/sm_nanopore/results/assembly/{sample}_rnd0.fasta {sample}.assembly_rnd0.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "f2841bc5-98dc-490b-b1a6-c1968fdbf9a6",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### smor ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::7.984*1.68] collected minimizers\n",
      "[M::mm_idx_gen::9.287*2.93] sorted minimizers\n",
      "[M::main::9.288*2.93] loaded/built the index for 1674 target sequence(s)\n",
      "[M::mm_mapopt_update::10.191*2.76] mid_occ = 384\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1674\n",
      "[M::mm_idx_stat::10.857*2.65] distinct minimizers: 30374815 (65.19% are singletons); average occurrences: 2.391; average spacing: 5.323; total length: 386656054\n",
      "[M::worker_pipeline::152.226*39.59] mapped 79894 sequences\n",
      "[M::worker_pipeline::307.705*40.33] mapped 77039 sequences\n",
      "[M::worker_pipeline::476.213*40.86] mapped 74651 sequences\n",
      "[M::worker_pipeline::629.258*41.74] mapped 76580 sequences\n",
      "[M::worker_pipeline::794.384*41.56] mapped 73726 sequences\n",
      "[M::worker_pipeline::965.772*41.19] mapped 76750 sequences\n",
      "[M::worker_pipeline::1144.195*40.83] mapped 76077 sequences\n",
      "[M::worker_pipeline::1289.656*41.18] mapped 77343 sequences\n",
      "[M::worker_pipeline::1453.221*41.22] mapped 75661 sequences\n",
      "[M::worker_pipeline::1617.462*40.85] mapped 78790 sequences\n",
      "[M::worker_pipeline::1764.942*41.05] mapped 78853 sequences\n",
      "[M::worker_pipeline::1933.526*41.06] mapped 76202 sequences\n",
      "[M::worker_pipeline::2095.499*41.02] mapped 73981 sequences\n",
      "[M::worker_pipeline::2273.516*41.00] mapped 70832 sequences\n",
      "[M::worker_pipeline::2436.701*41.12] mapped 70994 sequences\n",
      "[M::worker_pipeline::2600.532*41.11] mapped 72574 sequences\n",
      "[M::worker_pipeline::2774.577*40.99] mapped 71286 sequences\n",
      "[M::worker_pipeline::2945.959*40.93] mapped 72852 sequences\n",
      "[M::worker_pipeline::3124.810*40.87] mapped 73865 sequences\n",
      "[M::worker_pipeline::3281.401*40.90] mapped 75586 sequences\n",
      "[M::worker_pipeline::3432.424*40.51] mapped 60781 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smor.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smor/smor.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3432.825 sec; CPU: 139062.435 sec; Peak RSS: 22.556 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.139771 s\n",
      "[racon::Polisher::initialize] loaded sequences 101.550652 s\n",
      "[racon::Polisher::initialize] loaded overlaps 69.268296 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 19.301929 s\n",
      "[racon::Polisher::initialize] transformed data into windows 18.209538 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 195.514138 s\n",
      "[racon::Polisher::] total = 410.996559 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::7.581*1.74] collected minimizers\n",
      "[M::mm_idx_gen::8.810*2.82] sorted minimizers\n",
      "[M::main::8.811*2.82] loaded/built the index for 1399 target sequence(s)\n",
      "[M::mm_mapopt_update::9.549*2.68] mid_occ = 381\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1399\n",
      "[M::mm_idx_stat::10.053*2.59] distinct minimizers: 30397504 (65.24% are singletons); average occurrences: 2.386; average spacing: 5.322; total length: 385948710\n",
      "[M::worker_pipeline::147.574*39.59] mapped 79894 sequences\n",
      "[M::worker_pipeline::306.812*39.36] mapped 77039 sequences\n",
      "[M::worker_pipeline::475.756*40.33] mapped 74651 sequences\n",
      "[M::worker_pipeline::626.544*41.34] mapped 76580 sequences\n",
      "[M::worker_pipeline::782.457*41.45] mapped 73726 sequences\n",
      "[M::worker_pipeline::953.096*40.96] mapped 76750 sequences\n",
      "[M::worker_pipeline::1131.399*40.50] mapped 76077 sequences\n",
      "[M::worker_pipeline::1274.639*40.97] mapped 77343 sequences\n",
      "[M::worker_pipeline::1429.701*41.17] mapped 75661 sequences\n",
      "[M::worker_pipeline::1598.395*40.68] mapped 78790 sequences\n",
      "[M::worker_pipeline::1744.064*40.88] mapped 78853 sequences\n",
      "[M::worker_pipeline::1908.716*40.89] mapped 76202 sequences\n",
      "[M::worker_pipeline::2065.932*40.87] mapped 73981 sequences\n",
      "[M::worker_pipeline::2236.008*40.94] mapped 70832 sequences\n",
      "[M::worker_pipeline::2390.563*41.10] mapped 70994 sequences\n",
      "[M::worker_pipeline::2546.478*41.15] mapped 72574 sequences\n",
      "[M::worker_pipeline::2715.094*41.06] mapped 71286 sequences\n",
      "[M::worker_pipeline::2882.604*41.01] mapped 72852 sequences\n",
      "[M::worker_pipeline::3054.230*40.98] mapped 73865 sequences\n",
      "[M::worker_pipeline::3206.423*41.03] mapped 75586 sequences\n",
      "[M::worker_pipeline::3349.963*40.66] mapped 60781 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smor.assembly_rnd1.fasta /master/nplatt/sm_nanopore/results/read_correction/smor/smor.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3350.317 sec; CPU: 136217.193 sec; Peak RSS: 23.181 GB\n",
      "R2: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.144430 s\n",
      "[racon::Polisher::initialize] loaded sequences 102.767956 s\n",
      "[racon::Polisher::initialize] loaded overlaps 26.603356 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 14.940295 s\n",
      "[racon::Polisher::initialize] transformed data into windows 19.916187 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 185.418094 s\n",
      "[racon::Polisher::] total = 356.933251 s\n",
      "#\n",
      "R3: map\n",
      "[M::mm_idx_gen::7.605*1.71] collected minimizers\n",
      "[M::mm_idx_gen::8.852*2.83] sorted minimizers\n",
      "[M::main::8.852*2.83] loaded/built the index for 1369 target sequence(s)\n",
      "[M::mm_mapopt_update::9.541*2.69] mid_occ = 380\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1369\n",
      "[M::mm_idx_stat::10.048*2.61] distinct minimizers: 30396595 (65.25% are singletons); average occurrences: 2.385; average spacing: 5.322; total length: 385763855\n",
      "[M::worker_pipeline::146.197*39.51] mapped 79894 sequences\n",
      "[M::worker_pipeline::302.011*39.61] mapped 77039 sequences\n",
      "[M::worker_pipeline::468.536*40.38] mapped 74651 sequences\n",
      "[M::worker_pipeline::617.183*41.40] mapped 76580 sequences\n",
      "[M::worker_pipeline::773.340*41.48] mapped 73726 sequences\n",
      "[M::worker_pipeline::942.110*41.02] mapped 76750 sequences\n",
      "[M::worker_pipeline::1116.024*40.56] mapped 76077 sequences\n",
      "[M::worker_pipeline::1255.488*41.05] mapped 77343 sequences\n",
      "[M::worker_pipeline::1407.854*41.27] mapped 75661 sequences\n",
      "[M::worker_pipeline::1573.150*40.80] mapped 78790 sequences\n",
      "[M::worker_pipeline::1716.360*40.98] mapped 78853 sequences\n",
      "[M::worker_pipeline::1882.881*40.93] mapped 76202 sequences\n",
      "[M::worker_pipeline::2039.833*40.90] mapped 73981 sequences\n",
      "[M::worker_pipeline::2211.866*40.93] mapped 70832 sequences\n",
      "[M::worker_pipeline::2363.239*41.12] mapped 70994 sequences\n",
      "[M::worker_pipeline::2526.774*41.04] mapped 72574 sequences\n",
      "[M::worker_pipeline::2688.484*41.03] mapped 71286 sequences\n",
      "[M::worker_pipeline::2854.296*40.93] mapped 72852 sequences\n",
      "[M::worker_pipeline::3021.929*40.90] mapped 73865 sequences\n",
      "[M::worker_pipeline::3172.232*40.99] mapped 75586 sequences\n",
      "[M::worker_pipeline::3315.903*40.62] mapped 60781 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smor.assembly_rnd2.fasta /master/nplatt/sm_nanopore/results/read_correction/smor/smor.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3316.345 sec; CPU: 134699.331 sec; Peak RSS: 23.595 GB\n",
      "R3: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.149253 s\n",
      "[racon::Polisher::initialize] loaded sequences 101.847246 s\n",
      "[racon::Polisher::initialize] loaded overlaps 30.492327 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 17.914960 s\n",
      "[racon::Polisher::initialize] transformed data into windows 19.132240 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 188.459564 s\n",
      "[racon::Polisher::] total = 364.686051 s\n",
      "#\n",
      "###################### smle_pzq_es ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::7.896*1.71] collected minimizers\n",
      "[M::mm_idx_gen::9.202*2.90] sorted minimizers\n",
      "[M::main::9.202*2.90] loaded/built the index for 1122 target sequence(s)\n",
      "[M::mm_mapopt_update::9.895*2.77] mid_occ = 407\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1122\n",
      "[M::mm_idx_stat::10.381*2.69] distinct minimizers: 30553214 (64.52% are singletons); average occurrences: 2.479; average spacing: 5.325; total length: 403425661\n",
      "[M::worker_pipeline::172.686*39.54] mapped 19853 sequences\n",
      "[M::worker_pipeline::386.693*37.10] mapped 19771 sequences\n",
      "[M::worker_pipeline::603.000*37.46] mapped 20643 sequences\n",
      "[M::worker_pipeline::811.530*37.68] mapped 20836 sequences\n",
      "[M::worker_pipeline::1011.733*37.64] mapped 21265 sequences\n",
      "[M::worker_pipeline::1197.420*38.36] mapped 21689 sequences\n",
      "[M::worker_pipeline::1399.452*38.45] mapped 21616 sequences\n",
      "[M::worker_pipeline::1618.992*37.78] mapped 21757 sequences\n",
      "[M::worker_pipeline::1838.404*37.67] mapped 22496 sequences\n",
      "[M::worker_pipeline::2029.941*37.99] mapped 22806 sequences\n",
      "[M::worker_pipeline::2225.667*38.09] mapped 22165 sequences\n",
      "[M::worker_pipeline::2440.700*37.72] mapped 22957 sequences\n",
      "[M::worker_pipeline::2630.425*37.87] mapped 23544 sequences\n",
      "[M::worker_pipeline::2796.484*38.21] mapped 23371 sequences\n",
      "[M::worker_pipeline::2984.129*38.37] mapped 24068 sequences\n",
      "[M::worker_pipeline::3170.378*38.56] mapped 24289 sequences\n",
      "[M::worker_pipeline::3426.888*38.10] mapped 19452 sequences\n",
      "[M::worker_pipeline::3632.253*38.14] mapped 20038 sequences\n",
      "[M::worker_pipeline::3866.863*38.10] mapped 20058 sequences\n",
      "[M::worker_pipeline::4075.356*38.07] mapped 20416 sequences\n",
      "[M::worker_pipeline::4231.939*38.23] mapped 34327 sequences\n",
      "[M::worker_pipeline::4367.387*38.40] mapped 41954 sequences\n",
      "[M::worker_pipeline::4494.029*38.60] mapped 41664 sequences\n",
      "[M::worker_pipeline::4626.822*38.78] mapped 41677 sequences\n",
      "[M::worker_pipeline::4780.803*38.85] mapped 41139 sequences\n",
      "[M::worker_pipeline::4940.716*39.09] mapped 38763 sequences\n",
      "[M::worker_pipeline::5111.640*39.18] mapped 39457 sequences\n",
      "[M::worker_pipeline::5289.543*39.24] mapped 39956 sequences\n",
      "[M::worker_pipeline::5445.584*39.43] mapped 40366 sequences\n",
      "[M::worker_pipeline::5601.922*39.59] mapped 38901 sequences\n",
      "[M::worker_pipeline::5639.609*39.60] mapped 8242 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_es.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_es/smle_pzq_es.correctedReads.fasta.gz\n",
      "[M::main] Real time: 5640.001 sec; CPU: 223305.887 sec; Peak RSS: 23.044 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.158625 s\n",
      "[racon::Polisher::initialize] loaded sequences 146.908151 s\n",
      "[racon::Polisher::initialize] loaded overlaps 42.327313 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 15.036950 s\n",
      "[racon::Polisher::initialize] transformed data into windows 21.089318 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 254.333967 s\n",
      "[racon::Polisher::] total = 484.502313 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::8.046*1.69] collected minimizers\n",
      "[M::mm_idx_gen::9.265*2.75] sorted minimizers\n",
      "[M::main::9.265*2.75] loaded/built the index for 841 target sequence(s)\n",
      "[M::mm_mapopt_update::9.942*2.63] mid_occ = 407\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 841\n",
      "[M::mm_idx_stat::10.428*2.56] distinct minimizers: 30580531 (64.58% are singletons); average occurrences: 2.471; average spacing: 5.324; total length: 402322301\n",
      "[M::worker_pipeline::168.352*40.18] mapped 19853 sequences\n",
      "[M::worker_pipeline::373.321*37.82] mapped 19771 sequences\n",
      "[M::worker_pipeline::590.770*37.80] mapped 20643 sequences\n",
      "[M::worker_pipeline::783.351*38.57] mapped 20836 sequences\n",
      "[M::worker_pipeline::983.700*38.27] mapped 21265 sequences\n",
      "[M::worker_pipeline::1170.170*38.76] mapped 21689 sequences\n",
      "[M::worker_pipeline::1364.587*38.98] mapped 21616 sequences\n",
      "[M::worker_pipeline::1580.043*38.22] mapped 21757 sequences\n",
      "[M::worker_pipeline::1793.657*38.14] mapped 22496 sequences\n",
      "[M::worker_pipeline::1985.303*38.38] mapped 22806 sequences\n",
      "[M::worker_pipeline::2168.712*38.64] mapped 22165 sequences\n",
      "[M::worker_pipeline::2383.598*38.23] mapped 22957 sequences\n",
      "[M::worker_pipeline::2569.444*38.34] mapped 23544 sequences\n",
      "[M::worker_pipeline::2729.772*38.70] mapped 23371 sequences\n",
      "[M::worker_pipeline::2910.914*38.89] mapped 24068 sequences\n",
      "[M::worker_pipeline::3096.942*39.04] mapped 24289 sequences\n",
      "[M::worker_pipeline::3352.871*38.55] mapped 19452 sequences\n",
      "[M::worker_pipeline::3553.176*38.58] mapped 20038 sequences\n",
      "[M::worker_pipeline::3786.482*38.50] mapped 20058 sequences\n",
      "[M::worker_pipeline::3987.821*38.53] mapped 20416 sequences\n",
      "[M::worker_pipeline::4133.279*38.73] mapped 34327 sequences\n",
      "[M::worker_pipeline::4264.550*38.92] mapped 41954 sequences\n",
      "[M::worker_pipeline::4391.924*39.11] mapped 41664 sequences\n",
      "[M::worker_pipeline::4523.552*39.28] mapped 41677 sequences\n",
      "[M::worker_pipeline::4675.127*39.36] mapped 41139 sequences\n",
      "[M::worker_pipeline::4834.138*39.59] mapped 38763 sequences\n",
      "[M::worker_pipeline::5000.242*39.70] mapped 39457 sequences\n",
      "[M::worker_pipeline::5174.190*39.76] mapped 39956 sequences\n",
      "[M::worker_pipeline::5327.673*39.93] mapped 40366 sequences\n",
      "[M::worker_pipeline::5480.221*40.07] mapped 38901 sequences\n",
      "[M::worker_pipeline::5517.473*40.07] mapped 8242 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_es.assembly_rnd1.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_es/smle_pzq_es.correctedReads.fasta.gz\n",
      "[M::main] Real time: 5517.820 sec; CPU: 221092.124 sec; Peak RSS: 22.918 GB\n",
      "R2: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.256752 s\n",
      "[racon::Polisher::initialize] loaded sequences 147.591536 s\n",
      "[racon::Polisher::initialize] loaded overlaps 40.335285 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 17.719516 s\n",
      "[racon::Polisher::initialize] transformed data into windows 20.023588 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 253.454532 s\n",
      "[racon::Polisher::] total = 484.181523 s\n",
      "#\n",
      "R3: map\n",
      "[M::mm_idx_gen::9.634*1.66] collected minimizers\n",
      "[M::mm_idx_gen::10.842*2.57] sorted minimizers\n",
      "[M::main::10.842*2.57] loaded/built the index for 789 target sequence(s)\n",
      "[M::mm_mapopt_update::11.581*2.47] mid_occ = 407\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 789\n",
      "[M::mm_idx_stat::12.072*2.41] distinct minimizers: 30573180 (64.62% are singletons); average occurrences: 2.470; average spacing: 5.324; total length: 401969069\n",
      "[M::worker_pipeline::171.402*39.34] mapped 19853 sequences\n",
      "[M::worker_pipeline::380.799*36.89] mapped 19771 sequences\n",
      "[M::worker_pipeline::594.338*37.59] mapped 20643 sequences\n",
      "[M::worker_pipeline::783.218*38.59] mapped 20836 sequences\n",
      "[M::worker_pipeline::978.978*38.38] mapped 21265 sequences\n",
      "[M::worker_pipeline::1162.084*39.02] mapped 21689 sequences\n",
      "[M::worker_pipeline::1362.541*38.99] mapped 21616 sequences\n",
      "[M::worker_pipeline::1576.207*38.31] mapped 21757 sequences\n",
      "[M::worker_pipeline::1788.328*38.24] mapped 22496 sequences\n",
      "[M::worker_pipeline::1977.860*38.59] mapped 22806 sequences\n",
      "[M::worker_pipeline::2170.103*38.70] mapped 22165 sequences\n",
      "[M::worker_pipeline::2381.712*38.33] mapped 22957 sequences\n",
      "[M::worker_pipeline::2566.540*38.50] mapped 23544 sequences\n",
      "[M::worker_pipeline::2727.783*38.87] mapped 23371 sequences\n",
      "[M::worker_pipeline::2906.828*39.04] mapped 24068 sequences\n",
      "[M::worker_pipeline::3093.745*39.18] mapped 24289 sequences\n",
      "[M::worker_pipeline::3343.000*38.73] mapped 19452 sequences\n",
      "[M::worker_pipeline::3544.310*38.78] mapped 20038 sequences\n",
      "[M::worker_pipeline::3779.800*38.68] mapped 20058 sequences\n",
      "[M::worker_pipeline::3984.200*38.67] mapped 20416 sequences\n",
      "[M::worker_pipeline::4138.044*38.83] mapped 34327 sequences\n",
      "[M::worker_pipeline::4273.452*38.97] mapped 41954 sequences\n",
      "[M::worker_pipeline::4401.064*39.15] mapped 41664 sequences\n",
      "[M::worker_pipeline::4532.924*39.32] mapped 41677 sequences\n",
      "[M::worker_pipeline::4684.672*39.39] mapped 41139 sequences\n",
      "[M::worker_pipeline::4843.334*39.62] mapped 38763 sequences\n",
      "[M::worker_pipeline::5007.122*39.75] mapped 39457 sequences\n",
      "[M::worker_pipeline::5182.820*39.81] mapped 39956 sequences\n",
      "[M::worker_pipeline::5334.561*40.00] mapped 40366 sequences\n",
      "[M::worker_pipeline::5487.920*40.14] mapped 38901 sequences\n",
      "[M::worker_pipeline::5525.582*40.14] mapped 8242 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_es.assembly_rnd2.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_es/smle_pzq_es.correctedReads.fasta.gz\n",
      "[M::main] Real time: 5525.873 sec; CPU: 221794.069 sec; Peak RSS: 23.020 GB\n",
      "R3: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.346187 s\n",
      "[racon::Polisher::initialize] loaded sequences 148.885734 s\n",
      "[racon::Polisher::initialize] loaded overlaps 41.326534 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 14.228298 s\n",
      "[racon::Polisher::initialize] transformed data into windows 18.844883 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 252.791732 s\n",
      "[racon::Polisher::] total = 481.391174 s\n",
      "#\n",
      "###################### smle_pzq_er ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::8.547*1.66] collected minimizers\n",
      "[M::mm_idx_gen::9.878*2.76] sorted minimizers\n",
      "[M::main::9.878*2.76] loaded/built the index for 1968 target sequence(s)\n",
      "[M::mm_mapopt_update::10.615*2.64] mid_occ = 373\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1968\n",
      "[M::mm_idx_stat::11.119*2.57] distinct minimizers: 30209362 (65.19% are singletons); average occurrences: 2.377; average spacing: 5.323; total length: 382213358\n",
      "[M::worker_pipeline::101.212*37.43] mapped 76770 sequences\n",
      "[M::worker_pipeline::197.376*39.82] mapped 71320 sequences\n",
      "[M::worker_pipeline::292.345*40.90] mapped 73006 sequences\n",
      "[M::worker_pipeline::395.987*41.15] mapped 69078 sequences\n",
      "[M::worker_pipeline::507.789*39.95] mapped 68270 sequences\n",
      "[M::worker_pipeline::605.266*39.60] mapped 76406 sequences\n",
      "[M::worker_pipeline::690.203*40.15] mapped 75581 sequences\n",
      "[M::worker_pipeline::780.265*40.53] mapped 73411 sequences\n",
      "[M::worker_pipeline::874.280*40.80] mapped 73667 sequences\n",
      "[M::worker_pipeline::968.959*40.55] mapped 74686 sequences\n",
      "[M::worker_pipeline::1064.246*40.48] mapped 77941 sequences\n",
      "[M::worker_pipeline::1150.558*40.63] mapped 78752 sequences\n",
      "[M::worker_pipeline::1239.019*40.81] mapped 76620 sequences\n",
      "[M::worker_pipeline::1294.453*40.67] mapped 43491 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_er.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_er/smle_pzq_er.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1294.799 sec; CPU: 52650.649 sec; Peak RSS: 22.097 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.077128 s\n",
      "[racon::Polisher::initialize] loaded sequences 65.019206 s\n",
      "[racon::Polisher::initialize] loaded overlaps 24.796660 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 11.954963 s\n",
      "[racon::Polisher::initialize] transformed data into windows 7.371992 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 143.387897 s4.845812 s\n",
      "[racon::Polisher::] total = 257.503844 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::7.541*1.69] collected minimizers\n",
      "[M::mm_idx_gen::8.788*2.77] sorted minimizers\n",
      "[M::main::8.789*2.77] loaded/built the index for 1611 target sequence(s)\n",
      "[M::mm_mapopt_update::9.488*2.64] mid_occ = 364\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1611\n",
      "[M::mm_idx_stat::9.973*2.56] distinct minimizers: 30322343 (65.28% are singletons); average occurrences: 2.365; average spacing: 5.321; total length: 381614371\n",
      "[M::worker_pipeline::100.131*37.83] mapped 76770 sequences\n",
      "[M::worker_pipeline::195.433*39.64] mapped 71320 sequences\n",
      "[M::worker_pipeline::286.855*40.64] mapped 73006 sequences\n",
      "[M::worker_pipeline::389.978*40.81] mapped 69078 sequences\n",
      "[M::worker_pipeline::498.317*39.88] mapped 68270 sequences\n",
      "[M::worker_pipeline::592.994*39.57] mapped 76406 sequences\n",
      "[M::worker_pipeline::673.276*40.20] mapped 75581 sequences\n",
      "[M::worker_pipeline::761.828*40.59] mapped 73411 sequences\n",
      "[M::worker_pipeline::855.219*40.80] mapped 73667 sequences\n",
      "[M::worker_pipeline::944.630*40.70] mapped 74686 sequences\n",
      "[M::worker_pipeline::1034.611*40.65] mapped 77941 sequences\n",
      "[M::worker_pipeline::1125.942*40.59] mapped 78752 sequences\n",
      "[M::worker_pipeline::1209.989*40.83] mapped 76620 sequences\n",
      "[M::worker_pipeline::1262.212*40.77] mapped 43491 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_er.assembly_rnd1.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_er/smle_pzq_er.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1262.587 sec; CPU: 51454.778 sec; Peak RSS: 21.712 GB\n",
      "R2: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.109804 s\n",
      "[racon::Polisher::initialize] loaded sequences 65.880837 s\n",
      "[racon::Polisher::initialize] loaded overlaps 26.117030 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 12.140027 s\n",
      "[racon::Polisher::initialize] transformed data into windows 7.580020 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 140.047301 s\n",
      "[racon::Polisher::] total = 256.554459 s\n",
      "#\n",
      "R3: map\n",
      "[M::mm_idx_gen::7.511*1.70] collected minimizers\n",
      "[M::mm_idx_gen::8.833*2.85] sorted minimizers\n",
      "[M::main::8.834*2.85] loaded/built the index for 1582 target sequence(s)\n",
      "[M::mm_mapopt_update::9.492*2.72] mid_occ = 364\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1582\n",
      "[M::mm_idx_stat::9.977*2.64] distinct minimizers: 30331368 (65.29% are singletons); average occurrences: 2.365; average spacing: 5.321; total length: 381704002\n",
      "[M::worker_pipeline::100.601*38.07] mapped 76770 sequences\n",
      "[M::worker_pipeline::193.603*40.14] mapped 71320 sequences\n",
      "[M::worker_pipeline::286.347*40.77] mapped 73006 sequences\n",
      "[M::worker_pipeline::387.207*41.10] mapped 69078 sequences\n",
      "[M::worker_pipeline::491.405*40.24] mapped 68270 sequences\n",
      "[M::worker_pipeline::584.664*40.02] mapped 76406 sequences\n",
      "[M::worker_pipeline::664.283*40.59] mapped 75581 sequences\n",
      "[M::worker_pipeline::751.438*41.05] mapped 73411 sequences\n",
      "[M::worker_pipeline::844.293*41.29] mapped 73667 sequences\n",
      "[M::worker_pipeline::935.497*41.07] mapped 74686 sequences\n",
      "[M::worker_pipeline::1024.562*41.09] mapped 77941 sequences\n",
      "[M::worker_pipeline::1109.975*41.20] mapped 78752 sequences\n",
      "[M::worker_pipeline::1194.609*41.44] mapped 76620 sequences\n",
      "[M::worker_pipeline::1245.823*41.34] mapped 43491 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smle_pzq_er.assembly_rnd2.fasta /master/nplatt/sm_nanopore/results/read_correction/smle_pzq_er/smle_pzq_er.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1246.252 sec; CPU: 51501.835 sec; Peak RSS: 21.574 GB\n",
      "R3: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.172103 s\n",
      "[racon::Polisher::initialize] loaded sequences 66.222945 s\n",
      "[racon::Polisher::initialize] loaded overlaps 26.870027 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 12.611046 s\n",
      "[racon::Polisher::initialize] transformed data into windows 10.259038 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 146.096231 s\n",
      "[racon::Polisher::] total = 267.307171 s\n",
      "#\n",
      "###################### smbre ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::7.514*1.69] collected minimizers\n",
      "[M::mm_idx_gen::8.769*2.80] sorted minimizers\n",
      "[M::main::8.770*2.80] loaded/built the index for 1020 target sequence(s)\n",
      "[M::mm_mapopt_update::9.478*2.66] mid_occ = 394\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1020\n",
      "[M::mm_idx_stat::9.978*2.58] distinct minimizers: 30091754 (65.42% are singletons); average occurrences: 2.403; average spacing: 5.324; total length: 384951218\n",
      "[M::worker_pipeline::193.150*34.78] mapped 53283 sequences\n",
      "[M::worker_pipeline::345.736*38.72] mapped 52871 sequences\n",
      "[M::worker_pipeline::512.004*39.90] mapped 48725 sequences\n",
      "[M::worker_pipeline::677.763*40.43] mapped 46760 sequences\n",
      "[M::worker_pipeline::843.969*40.72] mapped 43965 sequences\n",
      "[M::worker_pipeline::1001.822*41.18] mapped 43922 sequences\n",
      "[M::worker_pipeline::1163.905*41.65] mapped 42570 sequences\n",
      "[M::worker_pipeline::1336.817*41.55] mapped 40977 sequences\n",
      "[M::worker_pipeline::1513.143*41.51] mapped 39946 sequences\n",
      "[M::worker_pipeline::1676.543*41.50] mapped 46552 sequences\n",
      "[M::worker_pipeline::1856.737*41.03] mapped 47237 sequences\n",
      "[M::worker_pipeline::2018.012*41.15] mapped 46567 sequences\n",
      "[M::worker_pipeline::2178.742*41.25] mapped 46788 sequences\n",
      "[M::worker_pipeline::2330.101*41.50] mapped 45522 sequences\n",
      "[M::worker_pipeline::2488.746*41.61] mapped 45538 sequences\n",
      "[M::worker_pipeline::2634.157*41.77] mapped 45607 sequences\n",
      "[M::worker_pipeline::2790.556*41.71] mapped 45182 sequences\n",
      "[M::worker_pipeline::2942.096*41.77] mapped 45409 sequences\n",
      "[M::worker_pipeline::3073.416*41.84] mapped 44380 sequences\n",
      "[M::worker_pipeline::3211.393*41.90] mapped 44938 sequences\n",
      "[M::worker_pipeline::3357.080*41.72] mapped 45630 sequences\n",
      "[M::worker_pipeline::3498.487*41.47] mapped 40156 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smbre.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smbre/smbre.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3498.906 sec; CPU: 145074.942 sec; Peak RSS: 24.759 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.078661 s\n",
      "[racon::Polisher::initialize] loaded sequences 107.383010 s\n",
      "[racon::Polisher::initialize] loaded overlaps 36.186241 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 16.406332 s\n",
      "[racon::Polisher::initialize] transformed data into windows 15.943155 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 195.406518 s\n",
      "[racon::Polisher::] total = 376.592673 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::7.578*1.70] collected minimizers\n",
      "[M::mm_idx_gen::8.792*2.78] sorted minimizers\n",
      "[M::main::8.793*2.78] loaded/built the index for 863 target sequence(s)\n",
      "[M::mm_mapopt_update::9.495*2.64] mid_occ = 392\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 863\n",
      "[M::mm_idx_stat::9.976*2.57] distinct minimizers: 30104826 (65.45% are singletons); average occurrences: 2.399; average spacing: 5.324; total length: 384537830\n",
      "[M::worker_pipeline::196.025*35.18] mapped 53283 sequences\n",
      "[M::worker_pipeline::348.845*38.96] mapped 52871 sequences\n",
      "[M::worker_pipeline::528.009*39.62] mapped 48725 sequences\n",
      "[M::worker_pipeline::700.069*39.96] mapped 46760 sequences\n",
      "[M::worker_pipeline::874.344*40.26] mapped 43965 sequences\n",
      "[M::worker_pipeline::1036.449*40.81] mapped 43922 sequences\n",
      "[M::worker_pipeline::1206.986*41.14] mapped 42570 sequences\n",
      "[M::worker_pipeline::1380.384*41.19] mapped 40977 sequences\n",
      "[M::worker_pipeline::1561.838*41.20] mapped 39946 sequences\n",
      "[M::worker_pipeline::1730.506*41.10] mapped 46552 sequences\n",
      "[M::worker_pipeline::1911.670*40.70] mapped 47237 sequences\n",
      "[M::worker_pipeline::2066.317*41.02] mapped 46567 sequences\n",
      "[M::worker_pipeline::2233.712*41.04] mapped 46788 sequences\n",
      "[M::worker_pipeline::2394.546*41.20] mapped 45522 sequences\n",
      "[M::worker_pipeline::2558.987*41.24] mapped 45538 sequences\n",
      "[M::worker_pipeline::2715.758*41.36] mapped 45607 sequences\n",
      "[M::worker_pipeline::2871.871*41.39] mapped 45182 sequences\n",
      "[M::worker_pipeline::3039.750*41.31] mapped 45409 sequences\n",
      "[M::worker_pipeline::3175.911*41.36] mapped 44380 sequences\n",
      "[M::worker_pipeline::3327.163*41.34] mapped 44938 sequences\n",
      "[M::worker_pipeline::3478.397*41.17] mapped 45630 sequences\n",
      "[M::worker_pipeline::3613.450*41.07] mapped 40156 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smbre.assembly_rnd1.fasta /master/nplatt/sm_nanopore/results/read_correction/smbre/smbre.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3613.843 sec; CPU: 148414.394 sec; Peak RSS: 23.428 GB\n",
      "R2: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.138405 s\n",
      "[racon::Polisher::initialize] loaded sequences 102.947766 s\n",
      "[racon::Polisher::initialize] loaded overlaps 25.953495 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 12.424720 s\n",
      "[racon::Polisher::initialize] transformed data into windows 14.520990 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 193.301941 s\n",
      "[racon::Polisher::] total = 354.562270 s\n",
      "#\n",
      "R3: map\n",
      "[M::mm_idx_gen::8.107*1.69] collected minimizers\n",
      "[M::mm_idx_gen::9.529*2.98] sorted minimizers\n",
      "[M::main::9.530*2.98] loaded/built the index for 831 target sequence(s)\n",
      "[M::mm_mapopt_update::10.381*2.81] mid_occ = 391\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 831\n",
      "[M::mm_idx_stat::10.960*2.72] distinct minimizers: 30104374 (65.45% are singletons); average occurrences: 2.399; average spacing: 5.324; total length: 384487880\n",
      "[M::worker_pipeline::191.993*35.65] mapped 53283 sequences\n",
      "[M::worker_pipeline::344.222*39.09] mapped 52871 sequences\n",
      "[M::worker_pipeline::522.196*39.74] mapped 48725 sequences\n",
      "[M::worker_pipeline::690.916*40.17] mapped 46760 sequences\n",
      "[M::worker_pipeline::864.704*40.33] mapped 43965 sequences\n",
      "[M::worker_pipeline::1021.268*40.83] mapped 43922 sequences\n",
      "[M::worker_pipeline::1192.620*41.12] mapped 42570 sequences\n",
      "[M::worker_pipeline::1362.209*41.19] mapped 40977 sequences\n",
      "[M::worker_pipeline::1537.998*41.34] mapped 39946 sequences\n",
      "[M::worker_pipeline::1704.273*41.23] mapped 46552 sequences\n",
      "[M::worker_pipeline::1889.321*40.75] mapped 47237 sequences\n",
      "[M::worker_pipeline::2048.109*40.89] mapped 46567 sequences\n",
      "[M::worker_pipeline::2213.962*40.91] mapped 46788 sequences\n",
      "[M::worker_pipeline::2369.793*41.09] mapped 45522 sequences\n",
      "[M::worker_pipeline::2533.122*41.16] mapped 45538 sequences\n",
      "[M::worker_pipeline::2685.127*41.29] mapped 45607 sequences\n",
      "[M::worker_pipeline::2839.448*41.33] mapped 45182 sequences\n",
      "[M::worker_pipeline::3000.577*41.33] mapped 45409 sequences\n",
      "[M::worker_pipeline::3134.637*41.40] mapped 44380 sequences\n",
      "[M::worker_pipeline::3279.134*41.38] mapped 44938 sequences\n",
      "[M::worker_pipeline::3422.723*41.27] mapped 45630 sequences\n",
      "[M::worker_pipeline::3563.041*41.09] mapped 40156 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smbre.assembly_rnd2.fasta /master/nplatt/sm_nanopore/results/read_correction/smbre/smbre.correctedReads.fasta.gz\n",
      "[M::main] Real time: 3563.511 sec; CPU: 146416.730 sec; Peak RSS: 23.257 GB\n",
      "R3: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.310798 s\n",
      "[racon::Polisher::initialize] loaded sequences 107.066274 s\n",
      "[racon::Polisher::initialize] loaded overlaps 26.866338 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 14.815388 s\n",
      "[racon::Polisher::initialize] transformed data into windows 16.890523 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 188.803997 s\n",
      "[racon::Polisher::] total = 360.313372 s\n",
      "#\n",
      "###################### smeg ##################\n",
      "R1: map\n",
      "[M::mm_idx_gen::7.068*1.68] collected minimizers\n",
      "[M::mm_idx_gen::8.334*2.85] sorted minimizers\n",
      "[M::main::8.334*2.85] loaded/built the index for 1963 target sequence(s)\n",
      "[M::mm_mapopt_update::9.029*2.71] mid_occ = 374\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1963\n",
      "[M::mm_idx_stat::9.508*2.62] distinct minimizers: 29693941 (65.87% are singletons); average occurrences: 2.343; average spacing: 5.325; total length: 370379461\n",
      "[M::worker_pipeline::150.602*36.83] mapped 46550 sequences\n",
      "[M::worker_pipeline::269.723*39.19] mapped 48696 sequences\n",
      "[M::worker_pipeline::410.898*39.44] mapped 48598 sequences\n",
      "[M::worker_pipeline::548.265*39.24] mapped 46410 sequences\n",
      "[M::worker_pipeline::688.176*38.50] mapped 49713 sequences\n",
      "[M::worker_pipeline::843.005*38.13] mapped 49778 sequences\n",
      "[M::worker_pipeline::979.189*38.40] mapped 48029 sequences\n",
      "[M::worker_pipeline::1119.284*38.46] mapped 47397 sequences\n",
      "[M::worker_pipeline::1236.508*38.74] mapped 63067 sequences\n",
      "[M::worker_pipeline::1343.908*39.18] mapped 69436 sequences\n",
      "[M::worker_pipeline::1466.845*39.07] mapped 67208 sequences\n",
      "[M::worker_pipeline::1591.167*39.05] mapped 65917 sequences\n",
      "[M::worker_pipeline::1608.341*38.89] mapped 7204 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smeg.assembly_rnd0.fasta /master/nplatt/sm_nanopore/results/read_correction/smeg/smeg.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1608.539 sec; CPU: 62555.827 sec; Peak RSS: 21.167 GB\n",
      "R1: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.004812 s\n",
      "[racon::Polisher::initialize] loaded sequences 57.975932 s\n",
      "[racon::Polisher::initialize] loaded overlaps 18.154140 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 7.779614 s\n",
      "[racon::Polisher::initialize] transformed data into windows 6.775523 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 106.537008 s\n",
      "[racon::Polisher::] total = 201.228917 s\n",
      "#\n",
      "R2: map\n",
      "[M::mm_idx_gen::7.289*1.65] collected minimizers\n",
      "[M::mm_idx_gen::8.646*2.85] sorted minimizers\n",
      "[M::main::8.646*2.85] loaded/built the index for 1704 target sequence(s)\n",
      "[M::mm_mapopt_update::9.344*2.71] mid_occ = 369\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1704\n",
      "[M::mm_idx_stat::9.822*2.63] distinct minimizers: 29717886 (65.87% are singletons); average occurrences: 2.337; average spacing: 5.324; total length: 369728076\n",
      "[M::worker_pipeline::148.591*36.65] mapped 46550 sequences\n",
      "[M::worker_pipeline::266.160*39.31] mapped 48696 sequences\n",
      "[M::worker_pipeline::398.206*40.43] mapped 48598 sequences\n",
      "[M::worker_pipeline::528.110*40.40] mapped 46410 sequences\n",
      "[M::worker_pipeline::666.205*39.49] mapped 49713 sequences\n",
      "[M::worker_pipeline::818.088*38.86] mapped 49778 sequences\n",
      "[M::worker_pipeline::952.898*39.00] mapped 48029 sequences\n",
      "[M::worker_pipeline::1090.429*39.05] mapped 47397 sequences\n",
      "[M::worker_pipeline::1203.600*39.29] mapped 63067 sequences\n",
      "[M::worker_pipeline::1305.587*39.72] mapped 69436 sequences\n",
      "[M::worker_pipeline::1426.536*39.66] mapped 67208 sequences\n",
      "[M::worker_pipeline::1542.291*39.80] mapped 65917 sequences\n",
      "[M::worker_pipeline::1559.713*39.65] mapped 7204 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smeg.assembly_rnd1.fasta /master/nplatt/sm_nanopore/results/read_correction/smeg/smeg.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1560.060 sec; CPU: 61840.668 sec; Peak RSS: 22.912 GB\n",
      "R2: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.715573 s\n",
      "[racon::Polisher::initialize] loaded sequences 57.372261 s\n",
      "[racon::Polisher::initialize] loaded overlaps 16.191489 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 8.778752 s\n",
      "[racon::Polisher::initialize] transformed data into windows 7.200109 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 109.529461 s\n",
      "[racon::Polisher::] total = 203.975579 s\n",
      "#\n",
      "R3: map\n",
      "[M::mm_idx_gen::7.185*1.67] collected minimizers\n",
      "[M::mm_idx_gen::8.396*2.79] sorted minimizers\n",
      "[M::main::8.397*2.79] loaded/built the index for 1669 target sequence(s)\n",
      "[M::mm_mapopt_update::9.086*2.65] mid_occ = 368\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1669\n",
      "[M::mm_idx_stat::9.561*2.57] distinct minimizers: 29717896 (65.88% are singletons); average occurrences: 2.336; average spacing: 5.324; total length: 369543290\n",
      "[M::worker_pipeline::149.730*38.03] mapped 46550 sequences\n",
      "[M::worker_pipeline::273.541*39.97] mapped 48696 sequences\n",
      "[M::worker_pipeline::413.320*40.10] mapped 48598 sequences\n",
      "[M::worker_pipeline::544.038*40.43] mapped 46410 sequences\n",
      "[M::worker_pipeline::683.224*39.58] mapped 49713 sequences\n",
      "[M::worker_pipeline::832.897*39.42] mapped 49778 sequences\n",
      "[M::worker_pipeline::967.910*39.63] mapped 48029 sequences\n",
      "[M::worker_pipeline::1108.349*39.56] mapped 47397 sequences\n",
      "[M::worker_pipeline::1226.402*39.75] mapped 63067 sequences\n",
      "[M::worker_pipeline::1329.992*40.16] mapped 69436 sequences\n",
      "[M::worker_pipeline::1454.251*40.01] mapped 67208 sequences\n",
      "[M::worker_pipeline::1574.498*40.08] mapped 65917 sequences\n",
      "[M::worker_pipeline::1591.432*39.93] mapped 7204 sequences\n",
      "[M::main] Version: 2.24-r1122\n",
      "[M::main] CMD: minimap2 -a -map-ont -t 48 smeg.assembly_rnd2.fasta /master/nplatt/sm_nanopore/results/read_correction/smeg/smeg.correctedReads.fasta.gz\n",
      "[M::main] Real time: 1591.774 sec; CPU: 63541.456 sec; Peak RSS: 21.523 GB\n",
      "R3: polish\n",
      "[racon::Polisher::initialize] loaded target sequences 2.071946 s\n",
      "[racon::Polisher::initialize] loaded sequences 58.422955 s\n",
      "[racon::Polisher::initialize] loaded overlaps 22.068410 s\n",
      "[racon::Polisher::initialize] aligning overlaps [====================] 6.597411 s\n",
      "[racon::Polisher::initialize] transformed data into windows 6.582953 s\n",
      "[racon::Polisher::polish] generating consensus [====================] 108.250975 s\n",
      "[racon::Polisher::] total = 206.464460 s\n",
      "#\n"
     ]
    }
   ],
   "source": [
    "#set the initial round\n",
    "rnd=1\n",
    "for sample in samples:\n",
    "    print(\"###################### {} ##################\".format(sample))\n",
    "\n",
    "    for rnd in [1, 2, 3]:\n",
    "        #define major i/o files\n",
    "        current_assembly = \"{}.assembly_rnd{}.fasta\".format(sample, str(rnd-1))\n",
    "        corrected_reads = \"{}/results/read_correction/{}/{}.correctedReads.fasta.gz\".format(proj_dir, sample, sample)\n",
    "        polished_assembly = \"{}.assembly_rnd{}.fasta\".format(sample, str(rnd))\n",
    "        n_threads=48\n",
    "        \n",
    "        #map corrected reads to the assembly\n",
    "        print(\"R{}: map\".format(rnd))\n",
    "        minimap_cmd = \"conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "            minimap2 \\\n",
    "                -a \\\n",
    "                -map-ont \\\n",
    "                -t {} \\\n",
    "                {} \\\n",
    "                {}  \\\n",
    "                >{}.racon_rnd{}.sam\".format(n_threads, current_assembly, corrected_reads, sample, rnd)\n",
    "        minimap_cmd=' '.join(minimap_cmd.split())\n",
    "        ! {minimap_cmd}\n",
    "\n",
    "        #polish the assembly\n",
    "        print(\"R{}: polish\".format(rnd))\n",
    "        racon_cmd = \"conda run -n sm_ont-process_reads --cwd . --live-stream \\\n",
    "            racon \\\n",
    "                -t {} \\\n",
    "                {} \\\n",
    "                {}.racon_rnd{}.sam \\\n",
    "                {} \\\n",
    "                >{}\".format(n_threads, corrected_reads, sample, rnd, current_assembly, polished_assembly)\n",
    "        racon_cmd=' '.join(racon_cmd.split())\n",
    "        ! {racon_cmd}\n",
    "        print(\"#\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a914a034-81cd-4d2d-927c-b0318826eb12",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Genome contamination "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "36f5c495-d3f7-47d8-8bb5-bbfef42a70ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(f\"{proj_dir}/results/remove_contamination\").mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(f\"{proj_dir}/results/remove_contamination\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f093c49-3b44-418e-86a9-b7dc7b250b0a",
   "metadata": {
    "tags": []
   },
   "source": [
    "https://github.com/timkahlke/BASTA/wiki/3.-BASTA-Usage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d36c2fd-7da4-4c2c-81fe-c7235e4646b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "conda run -n basta_py3 --cwd . --live-stream genome_updater.sh \\\n",
    "    -g \"\" \\\n",
    "    -d \"refseq\" \\\n",
    "    -c \"representative genome\" \\\n",
    "    -l \"complete genome,chromosome\" \\\n",
    "    -f \"genomic.fna.gz,assembly_report.txt\" \\\n",
    "    -o \"refseq_216_fas\" \\\n",
    "    -b \"v1\" \\\n",
    "    -a \\\n",
    "    -A species:1 \\\n",
    "    -m \\\n",
    "    -u \\\n",
    "    -r \\\n",
    "    -p \\\n",
    "    -t 48"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4cb02c95-be25-4c46-a008-df182698c5eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "#create blast database\n",
    "zcat refseq_216_fas/v1/files/GCF_*gz >refseq_v216.fas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "7d04f719-c4f9-4018-8f8f-9835b0d29d4c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 03/01/2023 15:11:26\n",
      "New DB name:   /master/nplatt/sch_man_ont/results/remove_contamination/refseq_v216.fas\n",
      "New DB title:  refseq_v216.fas\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing Nucleotide BLAST database named /master/nplatt/sch_man_ont/results/remove_contamination/refseq_v216.fas\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 2751379 sequences in 7269.04 seconds.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "\n",
    "conda run -n basta_py3 --cwd . --live-stream makeblastdb -dbtype nucl -in refseq_v216.fas "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73defe93-b772-4160-9b68-eaf749526c4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in samples:\n",
    "    assembly = f\"{proj_dir}/results/polish_assembly/{sample}.assembly_rnd3.fasta\"\n",
    "    !conda run -n basta_py3 --cwd . --live-stream blastn -outfmt 6 -query {assembly} -db refseq_v216.fas -out {sample}.blastn -num_threads 12 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3de6022-1acc-405b-95e8-f20e510c835e",
   "metadata": {},
   "outputs": [],
   "source": [
    "!conda run -n basta_py3 --cwd . --live-stream basta taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66a39e42-ef28-4e3c-95b3-43a947b9ce38",
   "metadata": {},
   "outputs": [],
   "source": [
    "!conda run -n basta_py3 --cwd . --live-stream basta download gb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "55039b2c-0aa4-4dc9-a570-0281d131bbcd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "#### Assigning taxonomy to each sequence ###\n",
      "\n",
      "\n",
      "# [BASTA STATUS] Assigning taxonomies ...\n",
      "\n",
      "# [BASTA STATUS] Initializing taxonomy database\n",
      "\n",
      "# [BASTA STATUS] Initializing mapping database\n",
      "\n",
      "#### Done. Output written to smor.basta\n",
      "\n",
      "#### Assigning taxonomy to each sequence ###\n",
      "\n",
      "\n",
      "# [BASTA STATUS] Assigning taxonomies ...\n",
      "\n",
      "# [BASTA STATUS] Initializing taxonomy database\n",
      "\n",
      "# [BASTA STATUS] Initializing mapping database\n",
      "\n",
      "#### Done. Output written to smle_pzq_es.basta\n",
      "\n",
      "#### Assigning taxonomy to each sequence ###\n",
      "\n",
      "\n",
      "# [BASTA STATUS] Assigning taxonomies ...\n",
      "\n",
      "# [BASTA STATUS] Initializing taxonomy database\n",
      "\n",
      "# [BASTA STATUS] Initializing mapping database\n",
      "\n",
      "#### Done. Output written to smle_pzq_er.basta\n",
      "\n",
      "#### Assigning taxonomy to each sequence ###\n",
      "\n",
      "\n",
      "# [BASTA STATUS] Assigning taxonomies ...\n",
      "\n",
      "# [BASTA STATUS] Initializing taxonomy database\n",
      "\n",
      "# [BASTA STATUS] Initializing mapping database\n",
      "\n",
      "#### Done. Output written to smbre.basta\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    !conda run -n basta_py3 --cwd . --live-stream basta sequence {sample}.blastn {sample}.basta gb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "9892835e-c3ba-4ad7-acc3-71e79593c7ea",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    418 Eukaryota;\n",
      "     23 Eukaryota;Chordata;\n",
      "     24 Eukaryota;Chordata;Mammalia;\n",
      "    110 Eukaryota;Chordata;Mammalia;Rodentia;\n",
      "     16 Eukaryota;Chordata;Mammalia;Rodentia;Cricetidae;\n",
      "      1 Eukaryota;Chordata;Mammalia;Rodentia;Muridae;\n",
      "      1 Eukaryota;Chordata;Mammalia;Rodentia;Muridae;Acomys;Acomys_russatus;\n",
      "      1 Eukaryota;Chordata;Mammalia;Rodentia;Muridae;Rattus;Rattus_norvegicus;\n",
      "     60 Unknown\n"
     ]
    }
   ],
   "source": [
    "#get an idea of non-schisto contigs/classifications and what should be removed\n",
    "! cat *.basta | cut -f2 | grep -v Platyhelminthes | sort | uniq -c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "467a630e-65fd-4f81-861d-2500b3198a46",
   "metadata": {},
   "outputs": [],
   "source": [
    "#based on these results I want to remove any contigs with chordata classifications.  Not sure about Eukaryota\n",
    "for sample in samples:\n",
    "    !grep Chordata {sample}.basta | cut -f1 >{sample}.contam_to_remove.txt\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "3f2383e6-5d97-4f79-acb7-a1bfedfd2704",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "smor\n",
      "smle_pzq_es\n",
      "smle_pzq_er\n",
      "smbre\n",
      "smeg\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    print(sample)\n",
    "    assembly = f\"{proj_dir}/results/polish_assembly/{sample}.assembly_rnd3.fasta\"\n",
    "\n",
    "    # List of contig IDs to filter out\n",
    "\n",
    "    contaminated_contigs = np.genfromtxt(f\"{sample}.contam_to_remove.txt\", delimiter=',', dtype=str)\n",
    "\n",
    "    # Input and output FASTA file names\n",
    "    input_fasta = assembly\n",
    "    output_fasta = f\"{sample}.basta.fasta\"\n",
    "\n",
    "    # Filter the contigs from the input FASTA file and write to the output FASTA file\n",
    "    with open(input_fasta, \"r\") as input_handle, open(output_fasta, \"w\") as output_handle:\n",
    "        for record in SeqIO.parse(input_handle, \"fasta\"):\n",
    "            if record.id not in contaminated_contigs:\n",
    "                SeqIO.write(record, output_handle, \"fasta\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf5d162e-af14-49b6-9d97-ed0ce9101400",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### genome quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "b64bd824-e1a6-439e-9b07-2d4addd4dd33",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/busco\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/busco\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "8434c2ca-cfee-4a25-9d97-750a6b90928c",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-03-15 12:09:38 INFO:\t***** Start a BUSCO v5.4.3 analysis, current time: 03/15/2023 12:09:38 *****\n",
      "2023-03-15 12:09:38 INFO:\tConfiguring BUSCO with local environment\n",
      "2023-03-15 12:09:38 INFO:\tMode is genome\n",
      "2023-03-15 12:09:38 INFO:\tDownloading information on latest versions of BUSCO data...\n",
      "2023-03-15 12:09:41 INFO:\tInput file is /master/nplatt/sch_man_ont/results/remove_contamination/smor.basta.fasta\n",
      "2023-03-15 12:09:41 INFO:\tNo lineage specified. Running lineage auto selector.\n",
      "\n",
      "2023-03-15 12:09:41 INFO:\t***** Starting Auto Select Lineage *****\n",
      "\tThis process runs BUSCO on the generic lineage datasets for the domains archaea, bacteria and eukaryota. Once the optimal domain is selected, BUSCO automatically attempts to find the most appropriate BUSCO dataset to use based on phylogenetic placement.\n",
      "\t--auto-lineage-euk and --auto-lineage-prok are also available if you know your input assembly is, or is not, an eukaryote. See the user guide for more information.\n",
      "\tA reminder: Busco evaluations are valid when an appropriate dataset is used, i.e., the dataset belongs to the lineage of the species to test. Because of overlapping markers/spurious matches among domains, busco matches in another domain do not necessarily mean that your genome/proteome contains sequences from this domain. However, a high busco score in multiple domains might help you identify possible contaminations.\n",
      "2023-03-15 12:09:43 INFO:\tRunning BUSCO using lineage dataset eukaryota_odb10 (eukaryota, 2020-09-10)\n",
      "2023-03-15 12:09:43 INFO:\tRunning 1 job(s) on bbtools, starting at 03/15/2023 12:09:43\n",
      "2023-03-15 12:09:50 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-03-15 12:09:50 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 12:09:50\n",
      "2023-03-15 12:17:02 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 12:17:14 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 12:17:14 INFO:\tRunning 255 job(s) on hmmsearch, starting at 03/15/2023 12:17:14\n",
      "2023-03-15 12:17:20 INFO:\t[hmmsearch]\t26 of 255 task(s) completed\n",
      "2023-03-15 12:17:20 INFO:\t[hmmsearch]\t51 of 255 task(s) completed\n",
      "2023-03-15 12:17:20 INFO:\t[hmmsearch]\t77 of 255 task(s) completed\n",
      "2023-03-15 12:17:20 INFO:\t[hmmsearch]\t102 of 255 task(s) completed\n",
      "2023-03-15 12:17:20 INFO:\t[hmmsearch]\t128 of 255 task(s) completed\n",
      "2023-03-15 12:17:20 INFO:\t[hmmsearch]\t153 of 255 task(s) completed\n",
      "2023-03-15 12:17:21 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-03-15 12:17:21 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-03-15 12:17:21 INFO:\t[hmmsearch]\t204 of 255 task(s) completed\n",
      "2023-03-15 12:17:21 INFO:\t[hmmsearch]\t230 of 255 task(s) completed\n",
      "2023-03-15 12:17:21 INFO:\t[hmmsearch]\t255 of 255 task(s) completed\n",
      "2023-03-15 12:17:22 INFO:\tResults:\tC:82.0%[S:81.2%,D:0.8%],F:10.2%,M:7.8%,n:255\t   \n",
      "\n",
      "2023-03-15 12:17:22 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-03-15 12:17:26 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 12:17:26\n",
      "2023-03-15 12:28:06 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 12:28:08 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 12:28:08 INFO:\tRunning 46 job(s) on hmmsearch, starting at 03/15/2023 12:28:08\n",
      "2023-03-15 12:28:14 INFO:\t[hmmsearch]\t5 of 46 task(s) completed\n",
      "2023-03-15 12:28:14 INFO:\t[hmmsearch]\t10 of 46 task(s) completed\n",
      "2023-03-15 12:28:14 INFO:\t[hmmsearch]\t14 of 46 task(s) completed\n",
      "2023-03-15 12:28:14 INFO:\t[hmmsearch]\t19 of 46 task(s) completed\n",
      "2023-03-15 12:28:14 INFO:\t[hmmsearch]\t23 of 46 task(s) completed\n",
      "2023-03-15 12:28:14 INFO:\t[hmmsearch]\t28 of 46 task(s) completed\n",
      "2023-03-15 12:28:14 INFO:\t[hmmsearch]\t33 of 46 task(s) completed\n",
      "2023-03-15 12:28:14 INFO:\t[hmmsearch]\t37 of 46 task(s) completed\n",
      "2023-03-15 12:28:14 INFO:\t[hmmsearch]\t42 of 46 task(s) completed\n",
      "2023-03-15 12:28:14 INFO:\t[hmmsearch]\t46 of 46 task(s) completed\n",
      "2023-03-15 12:28:17 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-03-15 12:28:18 INFO:\tResults:\tC:83.6%[S:82.4%,D:1.2%],F:9.0%,M:7.4%,n:255\t   \n",
      "\n",
      "2023-03-15 12:28:32 INFO:\teukaryota_odb10 selected\n",
      "\n",
      "2023-03-15 12:28:32 INFO:\t***** Searching tree for chosen lineage to find best taxonomic match *****\n",
      "\n",
      "2023-03-15 12:28:34 INFO:\tExtract markers...\n",
      "2023-03-15 12:28:34 INFO:\tPlace the markers on the reference tree...\n",
      "2023-03-15 12:28:34 INFO:\tRunning 1 job(s) on sepp, starting at 03/15/2023 12:28:34\n",
      "2023-03-15 12:29:37 INFO:\t[sepp]\t1 of 1 task(s) completed\n",
      "2023-03-15 12:29:40 INFO:\tLineage metazoa is selected, supported by 48 markers out of 91\n",
      "2023-03-15 12:29:40 WARNING:\tThe file or folder /master/nplatt/sch_man_ont/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "2023-03-15 12:29:41 INFO:\tRunning BUSCO using lineage dataset metazoa_odb10 (eukaryota, 2021-02-17)\n",
      "2023-03-15 12:29:41 INFO:\tRunning 1 job(s) on bbtools, starting at 03/15/2023 12:29:41\n",
      "2023-03-15 12:29:49 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-03-15 12:29:49 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 12:29:49\n",
      "2023-03-15 12:36:13 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 12:36:29 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 12:36:29 INFO:\tRunning 954 job(s) on hmmsearch, starting at 03/15/2023 12:36:29\n",
      "2023-03-15 12:36:35 INFO:\t[hmmsearch]\t96 of 954 task(s) completed\n",
      "2023-03-15 12:36:36 INFO:\t[hmmsearch]\t191 of 954 task(s) completed\n",
      "2023-03-15 12:36:36 INFO:\t[hmmsearch]\t191 of 954 task(s) completed\n",
      "2023-03-15 12:36:36 INFO:\t[hmmsearch]\t287 of 954 task(s) completed\n",
      "2023-03-15 12:36:36 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-03-15 12:36:37 INFO:\t[hmmsearch]\t478 of 954 task(s) completed\n",
      "2023-03-15 12:36:37 INFO:\t[hmmsearch]\t573 of 954 task(s) completed\n",
      "2023-03-15 12:36:37 INFO:\t[hmmsearch]\t668 of 954 task(s) completed\n",
      "2023-03-15 12:36:37 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-03-15 12:36:38 INFO:\t[hmmsearch]\t859 of 954 task(s) completed\n",
      "2023-03-15 12:36:39 INFO:\t[hmmsearch]\t954 of 954 task(s) completed\n",
      "2023-03-15 12:36:42 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-03-15 12:36:57 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 12:36:57\n",
      "2023-03-15 12:47:47 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 12:47:56 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 12:47:56 INFO:\tRunning 327 job(s) on hmmsearch, starting at 03/15/2023 12:47:56\n",
      "2023-03-15 12:48:02 INFO:\t[hmmsearch]\t33 of 327 task(s) completed\n",
      "2023-03-15 12:48:02 INFO:\t[hmmsearch]\t66 of 327 task(s) completed\n",
      "2023-03-15 12:48:03 INFO:\t[hmmsearch]\t99 of 327 task(s) completed\n",
      "2023-03-15 12:48:03 INFO:\t[hmmsearch]\t131 of 327 task(s) completed\n",
      "2023-03-15 12:48:03 INFO:\t[hmmsearch]\t164 of 327 task(s) completed\n",
      "2023-03-15 12:48:03 INFO:\t[hmmsearch]\t197 of 327 task(s) completed\n",
      "2023-03-15 12:48:04 INFO:\t[hmmsearch]\t229 of 327 task(s) completed\n",
      "2023-03-15 12:48:04 INFO:\t[hmmsearch]\t262 of 327 task(s) completed\n",
      "2023-03-15 12:48:04 INFO:\t[hmmsearch]\t295 of 327 task(s) completed\n",
      "2023-03-15 12:48:04 INFO:\t[hmmsearch]\t327 of 327 task(s) completed\n",
      "2023-03-15 12:48:14 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-03-15 12:48:19 INFO:\tResults:\tC:68.1%[S:66.9%,D:1.2%],F:7.1%,M:24.8%,n:954\t   \n",
      "\n",
      "2023-03-15 12:48:39 INFO:\t\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from generic domain eukaryota_odb10      |\n",
      "\t--------------------------------------------------\n",
      "\t|C:83.6%[S:82.4%,D:1.2%],F:9.0%,M:7.4%,n:255      |\n",
      "\t|213\tComplete BUSCOs (C)                       |\n",
      "\t|210\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|3\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|23\tFragmented BUSCOs (F)                     |\n",
      "\t|19\tMissing BUSCOs (M)                        |\n",
      "\t|255\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from dataset metazoa_odb10               |\n",
      "\t--------------------------------------------------\n",
      "\t|C:68.1%[S:66.9%,D:1.2%],F:7.1%,M:24.8%,n:954     |\n",
      "\t|649\tComplete BUSCOs (C)                       |\n",
      "\t|638\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|11\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|68\tFragmented BUSCOs (F)                     |\n",
      "\t|237\tMissing BUSCOs (M)                        |\n",
      "\t|954\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "2023-03-15 12:48:39 INFO:\tBUSCO analysis done with WARNING(s). Total running time: 2337 seconds\n",
      "\n",
      "***** Summary of warnings: *****\n",
      "2023-03-15 12:29:40 WARNING:busco.BuscoDownloadManager\tThe file or folder /master/nplatt/sch_man_ont/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "\n",
      "2023-03-15 12:48:39 INFO:\tResults written in /master/nplatt/sch_man_ont/results/busco/smor\n",
      "2023-03-15 12:48:39 INFO:\tFor assistance with interpreting the results, please consult the userguide: https://busco.ezlab.org/busco_userguide.html\n",
      "\n",
      "2023-03-15 12:48:39 INFO:\tVisit this page https://gitlab.com/ezlab/busco#how-to-cite-busco to see how to cite BUSCO\n",
      "2023-03-15 12:48:42 INFO:\t***** Start a BUSCO v5.4.3 analysis, current time: 03/15/2023 12:48:42 *****\n",
      "2023-03-15 12:48:42 INFO:\tConfiguring BUSCO with local environment\n",
      "2023-03-15 12:48:42 INFO:\tMode is genome\n",
      "2023-03-15 12:48:42 INFO:\tDownloading information on latest versions of BUSCO data...\n",
      "2023-03-15 12:48:45 INFO:\tInput file is /master/nplatt/sch_man_ont/results/remove_contamination/smle_pzq_es.basta.fasta\n",
      "2023-03-15 12:48:45 INFO:\tNo lineage specified. Running lineage auto selector.\n",
      "\n",
      "2023-03-15 12:48:45 INFO:\t***** Starting Auto Select Lineage *****\n",
      "\tThis process runs BUSCO on the generic lineage datasets for the domains archaea, bacteria and eukaryota. Once the optimal domain is selected, BUSCO automatically attempts to find the most appropriate BUSCO dataset to use based on phylogenetic placement.\n",
      "\t--auto-lineage-euk and --auto-lineage-prok are also available if you know your input assembly is, or is not, an eukaryote. See the user guide for more information.\n",
      "\tA reminder: Busco evaluations are valid when an appropriate dataset is used, i.e., the dataset belongs to the lineage of the species to test. Because of overlapping markers/spurious matches among domains, busco matches in another domain do not necessarily mean that your genome/proteome contains sequences from this domain. However, a high busco score in multiple domains might help you identify possible contaminations.\n",
      "2023-03-15 12:48:47 INFO:\tRunning BUSCO using lineage dataset eukaryota_odb10 (eukaryota, 2020-09-10)\n",
      "2023-03-15 12:48:47 INFO:\tRunning 1 job(s) on bbtools, starting at 03/15/2023 12:48:47\n",
      "2023-03-15 12:48:51 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-03-15 12:48:51 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 12:48:51\n",
      "2023-03-15 12:55:48 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 12:55:58 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 12:55:58 INFO:\tRunning 255 job(s) on hmmsearch, starting at 03/15/2023 12:55:58\n",
      "2023-03-15 12:56:03 INFO:\t[hmmsearch]\t26 of 255 task(s) completed\n",
      "2023-03-15 12:56:03 INFO:\t[hmmsearch]\t51 of 255 task(s) completed\n",
      "2023-03-15 12:56:03 INFO:\t[hmmsearch]\t77 of 255 task(s) completed\n",
      "2023-03-15 12:56:04 INFO:\t[hmmsearch]\t128 of 255 task(s) completed\n",
      "2023-03-15 12:56:04 INFO:\t[hmmsearch]\t153 of 255 task(s) completed\n",
      "2023-03-15 12:56:04 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-03-15 12:56:04 INFO:\t[hmmsearch]\t204 of 255 task(s) completed\n",
      "2023-03-15 12:56:04 INFO:\t[hmmsearch]\t230 of 255 task(s) completed\n",
      "2023-03-15 12:56:04 INFO:\t[hmmsearch]\t255 of 255 task(s) completed\n",
      "2023-03-15 12:56:05 INFO:\tResults:\tC:87.5%[S:85.9%,D:1.6%],F:6.7%,M:5.8%,n:255\t   \n",
      "\n",
      "2023-03-15 12:56:05 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-03-15 12:56:09 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 12:56:09\n",
      "2023-03-15 13:05:33 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:05:35 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 13:05:35 INFO:\tRunning 32 job(s) on hmmsearch, starting at 03/15/2023 13:05:35\n",
      "2023-03-15 13:05:40 INFO:\t[hmmsearch]\t4 of 32 task(s) completed\n",
      "2023-03-15 13:05:40 INFO:\t[hmmsearch]\t7 of 32 task(s) completed\n",
      "2023-03-15 13:05:40 INFO:\t[hmmsearch]\t10 of 32 task(s) completed\n",
      "2023-03-15 13:05:40 INFO:\t[hmmsearch]\t13 of 32 task(s) completed\n",
      "2023-03-15 13:05:41 INFO:\t[hmmsearch]\t17 of 32 task(s) completed\n",
      "2023-03-15 13:05:41 INFO:\t[hmmsearch]\t20 of 32 task(s) completed\n",
      "2023-03-15 13:05:41 INFO:\t[hmmsearch]\t20 of 32 task(s) completed\n",
      "2023-03-15 13:05:41 INFO:\t[hmmsearch]\t23 of 32 task(s) completed\n",
      "2023-03-15 13:05:41 INFO:\t[hmmsearch]\t26 of 32 task(s) completed\n",
      "2023-03-15 13:05:41 INFO:\t[hmmsearch]\t29 of 32 task(s) completed\n",
      "2023-03-15 13:05:41 INFO:\t[hmmsearch]\t32 of 32 task(s) completed\n",
      "2023-03-15 13:05:43 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-03-15 13:05:45 INFO:\tResults:\tC:88.3%[S:86.7%,D:1.6%],F:6.3%,M:5.4%,n:255\t   \n",
      "\n",
      "2023-03-15 13:05:58 INFO:\teukaryota_odb10 selected\n",
      "\n",
      "2023-03-15 13:05:58 INFO:\t***** Searching tree for chosen lineage to find best taxonomic match *****\n",
      "\n",
      "2023-03-15 13:05:58 INFO:\tExtract markers...\n",
      "2023-03-15 13:05:58 INFO:\tPlace the markers on the reference tree...\n",
      "2023-03-15 13:05:58 INFO:\tRunning 1 job(s) on sepp, starting at 03/15/2023 13:05:58\n",
      "2023-03-15 13:06:59 INFO:\t[sepp]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:07:02 INFO:\tLineage metazoa is selected, supported by 52 markers out of 92\n",
      "2023-03-15 13:07:02 WARNING:\tThe file or folder /master/nplatt/sch_man_ont/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "2023-03-15 13:07:03 INFO:\tRunning BUSCO using lineage dataset metazoa_odb10 (eukaryota, 2021-02-17)\n",
      "2023-03-15 13:07:03 INFO:\tRunning 1 job(s) on bbtools, starting at 03/15/2023 13:07:03\n",
      "2023-03-15 13:07:10 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:07:10 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 13:07:10\n",
      "2023-03-15 13:13:45 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:14:00 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 13:14:00 INFO:\tRunning 954 job(s) on hmmsearch, starting at 03/15/2023 13:14:00\n",
      "2023-03-15 13:14:06 INFO:\t[hmmsearch]\t96 of 954 task(s) completed\n",
      "2023-03-15 13:14:06 INFO:\t[hmmsearch]\t191 of 954 task(s) completed\n",
      "2023-03-15 13:14:07 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-03-15 13:14:07 INFO:\t[hmmsearch]\t478 of 954 task(s) completed\n",
      "2023-03-15 13:14:07 INFO:\t[hmmsearch]\t573 of 954 task(s) completed\n",
      "2023-03-15 13:14:07 INFO:\t[hmmsearch]\t668 of 954 task(s) completed\n",
      "2023-03-15 13:14:08 INFO:\t[hmmsearch]\t859 of 954 task(s) completed\n",
      "2023-03-15 13:14:09 INFO:\t[hmmsearch]\t954 of 954 task(s) completed\n",
      "2023-03-15 13:14:12 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-03-15 13:14:27 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 13:14:27\n",
      "2023-03-15 13:25:23 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:25:29 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 13:25:29 INFO:\tRunning 302 job(s) on hmmsearch, starting at 03/15/2023 13:25:29\n",
      "2023-03-15 13:25:34 INFO:\t[hmmsearch]\t31 of 302 task(s) completed\n",
      "2023-03-15 13:25:35 INFO:\t[hmmsearch]\t61 of 302 task(s) completed\n",
      "2023-03-15 13:25:35 INFO:\t[hmmsearch]\t91 of 302 task(s) completed\n",
      "2023-03-15 13:25:35 INFO:\t[hmmsearch]\t121 of 302 task(s) completed\n",
      "2023-03-15 13:25:36 INFO:\t[hmmsearch]\t151 of 302 task(s) completed\n",
      "2023-03-15 13:25:36 INFO:\t[hmmsearch]\t182 of 302 task(s) completed\n",
      "2023-03-15 13:25:36 INFO:\t[hmmsearch]\t212 of 302 task(s) completed\n",
      "2023-03-15 13:25:36 INFO:\t[hmmsearch]\t242 of 302 task(s) completed\n",
      "2023-03-15 13:25:36 INFO:\t[hmmsearch]\t272 of 302 task(s) completed\n",
      "2023-03-15 13:25:36 INFO:\t[hmmsearch]\t302 of 302 task(s) completed\n",
      "2023-03-15 13:25:46 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-03-15 13:25:52 INFO:\tResults:\tC:69.6%[S:68.0%,D:1.6%],F:6.6%,M:23.8%,n:954\t   \n",
      "\n",
      "2023-03-15 13:26:10 INFO:\t\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from generic domain eukaryota_odb10      |\n",
      "\t--------------------------------------------------\n",
      "\t|C:88.3%[S:86.7%,D:1.6%],F:6.3%,M:5.4%,n:255      |\n",
      "\t|225\tComplete BUSCOs (C)                       |\n",
      "\t|221\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|4\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|16\tFragmented BUSCOs (F)                     |\n",
      "\t|14\tMissing BUSCOs (M)                        |\n",
      "\t|255\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from dataset metazoa_odb10               |\n",
      "\t--------------------------------------------------\n",
      "\t|C:69.6%[S:68.0%,D:1.6%],F:6.6%,M:23.8%,n:954     |\n",
      "\t|664\tComplete BUSCOs (C)                       |\n",
      "\t|649\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|15\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|63\tFragmented BUSCOs (F)                     |\n",
      "\t|227\tMissing BUSCOs (M)                        |\n",
      "\t|954\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "2023-03-15 13:26:10 INFO:\tBUSCO analysis done with WARNING(s). Total running time: 2246 seconds\n",
      "\n",
      "***** Summary of warnings: *****\n",
      "2023-03-15 13:07:02 WARNING:busco.BuscoDownloadManager\tThe file or folder /master/nplatt/sch_man_ont/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "\n",
      "2023-03-15 13:26:10 INFO:\tResults written in /master/nplatt/sch_man_ont/results/busco/smle_pzq_es\n",
      "2023-03-15 13:26:10 INFO:\tFor assistance with interpreting the results, please consult the userguide: https://busco.ezlab.org/busco_userguide.html\n",
      "\n",
      "2023-03-15 13:26:10 INFO:\tVisit this page https://gitlab.com/ezlab/busco#how-to-cite-busco to see how to cite BUSCO\n",
      "2023-03-15 13:26:14 INFO:\t***** Start a BUSCO v5.4.3 analysis, current time: 03/15/2023 13:26:14 *****\n",
      "2023-03-15 13:26:14 INFO:\tConfiguring BUSCO with local environment\n",
      "2023-03-15 13:26:14 INFO:\tMode is genome\n",
      "2023-03-15 13:26:14 INFO:\tDownloading information on latest versions of BUSCO data...\n",
      "2023-03-15 13:26:17 INFO:\tInput file is /master/nplatt/sch_man_ont/results/remove_contamination/smle_pzq_er.basta.fasta\n",
      "2023-03-15 13:26:17 INFO:\tNo lineage specified. Running lineage auto selector.\n",
      "\n",
      "2023-03-15 13:26:17 INFO:\t***** Starting Auto Select Lineage *****\n",
      "\tThis process runs BUSCO on the generic lineage datasets for the domains archaea, bacteria and eukaryota. Once the optimal domain is selected, BUSCO automatically attempts to find the most appropriate BUSCO dataset to use based on phylogenetic placement.\n",
      "\t--auto-lineage-euk and --auto-lineage-prok are also available if you know your input assembly is, or is not, an eukaryote. See the user guide for more information.\n",
      "\tA reminder: Busco evaluations are valid when an appropriate dataset is used, i.e., the dataset belongs to the lineage of the species to test. Because of overlapping markers/spurious matches among domains, busco matches in another domain do not necessarily mean that your genome/proteome contains sequences from this domain. However, a high busco score in multiple domains might help you identify possible contaminations.\n",
      "2023-03-15 13:26:19 INFO:\tRunning BUSCO using lineage dataset eukaryota_odb10 (eukaryota, 2020-09-10)\n",
      "2023-03-15 13:26:19 INFO:\tRunning 1 job(s) on bbtools, starting at 03/15/2023 13:26:19\n",
      "2023-03-15 13:26:23 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:26:23 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 13:26:23\n",
      "2023-03-15 13:32:45 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:32:59 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 13:32:59 INFO:\tRunning 255 job(s) on hmmsearch, starting at 03/15/2023 13:32:59\n",
      "2023-03-15 13:33:05 INFO:\t[hmmsearch]\t26 of 255 task(s) completed\n",
      "2023-03-15 13:33:05 INFO:\t[hmmsearch]\t51 of 255 task(s) completed\n",
      "2023-03-15 13:33:05 INFO:\t[hmmsearch]\t77 of 255 task(s) completed\n",
      "2023-03-15 13:33:05 INFO:\t[hmmsearch]\t102 of 255 task(s) completed\n",
      "2023-03-15 13:33:05 INFO:\t[hmmsearch]\t128 of 255 task(s) completed\n",
      "2023-03-15 13:33:05 INFO:\t[hmmsearch]\t153 of 255 task(s) completed\n",
      "2023-03-15 13:33:05 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-03-15 13:33:06 INFO:\t[hmmsearch]\t204 of 255 task(s) completed\n",
      "2023-03-15 13:33:06 INFO:\t[hmmsearch]\t230 of 255 task(s) completed\n",
      "2023-03-15 13:33:06 INFO:\t[hmmsearch]\t255 of 255 task(s) completed\n",
      "2023-03-15 13:33:07 INFO:\tResults:\tC:79.3%[S:77.3%,D:2.0%],F:11.0%,M:9.7%,n:255\t   \n",
      "\n",
      "2023-03-15 13:33:07 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-03-15 13:33:13 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 13:33:13\n",
      "2023-03-15 13:43:55 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:44:02 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 13:44:02 INFO:\tRunning 53 job(s) on hmmsearch, starting at 03/15/2023 13:44:02\n",
      "2023-03-15 13:44:08 INFO:\t[hmmsearch]\t6 of 53 task(s) completed\n",
      "2023-03-15 13:44:08 INFO:\t[hmmsearch]\t11 of 53 task(s) completed\n",
      "2023-03-15 13:44:08 INFO:\t[hmmsearch]\t16 of 53 task(s) completed\n",
      "2023-03-15 13:44:08 INFO:\t[hmmsearch]\t22 of 53 task(s) completed\n",
      "2023-03-15 13:44:08 INFO:\t[hmmsearch]\t27 of 53 task(s) completed\n",
      "2023-03-15 13:44:08 INFO:\t[hmmsearch]\t32 of 53 task(s) completed\n",
      "2023-03-15 13:44:08 INFO:\t[hmmsearch]\t38 of 53 task(s) completed\n",
      "2023-03-15 13:44:08 INFO:\t[hmmsearch]\t43 of 53 task(s) completed\n",
      "2023-03-15 13:44:08 INFO:\t[hmmsearch]\t48 of 53 task(s) completed\n",
      "2023-03-15 13:44:08 INFO:\t[hmmsearch]\t53 of 53 task(s) completed\n",
      "2023-03-15 13:44:11 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-03-15 13:44:13 INFO:\tResults:\tC:81.6%[S:79.6%,D:2.0%],F:9.0%,M:9.4%,n:255\t   \n",
      "\n",
      "2023-03-15 13:44:28 INFO:\teukaryota_odb10 selected\n",
      "\n",
      "2023-03-15 13:44:28 INFO:\t***** Searching tree for chosen lineage to find best taxonomic match *****\n",
      "\n",
      "2023-03-15 13:44:28 INFO:\tExtract markers...\n",
      "2023-03-15 13:44:28 INFO:\tPlace the markers on the reference tree...\n",
      "2023-03-15 13:44:28 INFO:\tRunning 1 job(s) on sepp, starting at 03/15/2023 13:44:28\n",
      "2023-03-15 13:45:28 INFO:\t[sepp]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:45:31 INFO:\tLineage metazoa is selected, supported by 45 markers out of 88\n",
      "2023-03-15 13:45:31 WARNING:\tThe file or folder /master/nplatt/sch_man_ont/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "2023-03-15 13:45:32 INFO:\tRunning BUSCO using lineage dataset metazoa_odb10 (eukaryota, 2021-02-17)\n",
      "2023-03-15 13:45:32 INFO:\tRunning 1 job(s) on bbtools, starting at 03/15/2023 13:45:32\n",
      "2023-03-15 13:45:38 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:45:38 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 13:45:38\n",
      "2023-03-15 13:52:01 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 13:52:19 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 13:52:19 INFO:\tRunning 954 job(s) on hmmsearch, starting at 03/15/2023 13:52:19\n",
      "2023-03-15 13:52:25 INFO:\t[hmmsearch]\t96 of 954 task(s) completed\n",
      "2023-03-15 13:52:25 INFO:\t[hmmsearch]\t191 of 954 task(s) completed\n",
      "2023-03-15 13:52:26 INFO:\t[hmmsearch]\t287 of 954 task(s) completed\n",
      "2023-03-15 13:52:26 INFO:\t[hmmsearch]\t478 of 954 task(s) completed\n",
      "2023-03-15 13:52:26 INFO:\t[hmmsearch]\t478 of 954 task(s) completed\n",
      "2023-03-15 13:52:27 INFO:\t[hmmsearch]\t668 of 954 task(s) completed\n",
      "2023-03-15 13:52:27 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-03-15 13:52:27 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-03-15 13:52:27 INFO:\t[hmmsearch]\t859 of 954 task(s) completed\n",
      "2023-03-15 13:52:28 INFO:\t[hmmsearch]\t954 of 954 task(s) completed\n",
      "2023-03-15 13:52:31 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-03-15 13:52:50 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 13:52:50\n",
      "2023-03-15 14:04:26 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 14:04:36 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 14:04:36 INFO:\tRunning 375 job(s) on hmmsearch, starting at 03/15/2023 14:04:36\n",
      "2023-03-15 14:04:42 INFO:\t[hmmsearch]\t38 of 375 task(s) completed\n",
      "2023-03-15 14:04:42 INFO:\t[hmmsearch]\t75 of 375 task(s) completed\n",
      "2023-03-15 14:04:43 INFO:\t[hmmsearch]\t113 of 375 task(s) completed\n",
      "2023-03-15 14:04:43 INFO:\t[hmmsearch]\t150 of 375 task(s) completed\n",
      "2023-03-15 14:04:44 INFO:\t[hmmsearch]\t188 of 375 task(s) completed\n",
      "2023-03-15 14:04:44 INFO:\t[hmmsearch]\t225 of 375 task(s) completed\n",
      "2023-03-15 14:04:44 INFO:\t[hmmsearch]\t263 of 375 task(s) completed\n",
      "2023-03-15 14:04:45 INFO:\t[hmmsearch]\t300 of 375 task(s) completed\n",
      "2023-03-15 14:04:45 INFO:\t[hmmsearch]\t338 of 375 task(s) completed\n",
      "2023-03-15 14:04:45 INFO:\t[hmmsearch]\t375 of 375 task(s) completed\n",
      "2023-03-15 14:04:56 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-03-15 14:05:01 INFO:\tResults:\tC:63.8%[S:62.5%,D:1.3%],F:9.0%,M:27.2%,n:954\t   \n",
      "\n",
      "2023-03-15 14:05:22 INFO:\t\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from generic domain eukaryota_odb10      |\n",
      "\t--------------------------------------------------\n",
      "\t|C:81.6%[S:79.6%,D:2.0%],F:9.0%,M:9.4%,n:255      |\n",
      "\t|208\tComplete BUSCOs (C)                       |\n",
      "\t|203\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|5\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|23\tFragmented BUSCOs (F)                     |\n",
      "\t|24\tMissing BUSCOs (M)                        |\n",
      "\t|255\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from dataset metazoa_odb10               |\n",
      "\t--------------------------------------------------\n",
      "\t|C:63.8%[S:62.5%,D:1.3%],F:9.0%,M:27.2%,n:954     |\n",
      "\t|608\tComplete BUSCOs (C)                       |\n",
      "\t|596\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|12\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|86\tFragmented BUSCOs (F)                     |\n",
      "\t|260\tMissing BUSCOs (M)                        |\n",
      "\t|954\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "2023-03-15 14:05:22 INFO:\tBUSCO analysis done with WARNING(s). Total running time: 2345 seconds\n",
      "\n",
      "***** Summary of warnings: *****\n",
      "2023-03-15 13:45:31 WARNING:busco.BuscoDownloadManager\tThe file or folder /master/nplatt/sch_man_ont/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "\n",
      "2023-03-15 14:05:22 INFO:\tResults written in /master/nplatt/sch_man_ont/results/busco/smle_pzq_er\n",
      "2023-03-15 14:05:22 INFO:\tFor assistance with interpreting the results, please consult the userguide: https://busco.ezlab.org/busco_userguide.html\n",
      "\n",
      "2023-03-15 14:05:22 INFO:\tVisit this page https://gitlab.com/ezlab/busco#how-to-cite-busco to see how to cite BUSCO\n",
      "2023-03-15 14:05:25 INFO:\t***** Start a BUSCO v5.4.3 analysis, current time: 03/15/2023 14:05:25 *****\n",
      "2023-03-15 14:05:25 INFO:\tConfiguring BUSCO with local environment\n",
      "2023-03-15 14:05:25 INFO:\tMode is genome\n",
      "2023-03-15 14:05:25 INFO:\tDownloading information on latest versions of BUSCO data...\n",
      "2023-03-15 14:05:28 INFO:\tInput file is /master/nplatt/sch_man_ont/results/remove_contamination/smbre.basta.fasta\n",
      "2023-03-15 14:05:28 INFO:\tNo lineage specified. Running lineage auto selector.\n",
      "\n",
      "2023-03-15 14:05:28 INFO:\t***** Starting Auto Select Lineage *****\n",
      "\tThis process runs BUSCO on the generic lineage datasets for the domains archaea, bacteria and eukaryota. Once the optimal domain is selected, BUSCO automatically attempts to find the most appropriate BUSCO dataset to use based on phylogenetic placement.\n",
      "\t--auto-lineage-euk and --auto-lineage-prok are also available if you know your input assembly is, or is not, an eukaryote. See the user guide for more information.\n",
      "\tA reminder: Busco evaluations are valid when an appropriate dataset is used, i.e., the dataset belongs to the lineage of the species to test. Because of overlapping markers/spurious matches among domains, busco matches in another domain do not necessarily mean that your genome/proteome contains sequences from this domain. However, a high busco score in multiple domains might help you identify possible contaminations.\n",
      "2023-03-15 14:05:31 INFO:\tRunning BUSCO using lineage dataset eukaryota_odb10 (eukaryota, 2020-09-10)\n",
      "2023-03-15 14:05:31 INFO:\tRunning 1 job(s) on bbtools, starting at 03/15/2023 14:05:31\n",
      "2023-03-15 14:05:35 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-03-15 14:05:35 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 14:05:35\n",
      "2023-03-15 14:12:18 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 14:12:29 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 14:12:29 INFO:\tRunning 255 job(s) on hmmsearch, starting at 03/15/2023 14:12:29\n",
      "2023-03-15 14:12:36 INFO:\t[hmmsearch]\t26 of 255 task(s) completed\n",
      "2023-03-15 14:12:36 INFO:\t[hmmsearch]\t51 of 255 task(s) completed\n",
      "2023-03-15 14:12:36 INFO:\t[hmmsearch]\t77 of 255 task(s) completed\n",
      "2023-03-15 14:12:36 INFO:\t[hmmsearch]\t102 of 255 task(s) completed\n",
      "2023-03-15 14:12:36 INFO:\t[hmmsearch]\t153 of 255 task(s) completed\n",
      "2023-03-15 14:12:36 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-03-15 14:12:36 INFO:\t[hmmsearch]\t204 of 255 task(s) completed\n",
      "2023-03-15 14:12:36 INFO:\t[hmmsearch]\t230 of 255 task(s) completed\n",
      "2023-03-15 14:12:36 INFO:\t[hmmsearch]\t255 of 255 task(s) completed\n",
      "2023-03-15 14:12:37 INFO:\tResults:\tC:82.8%[S:81.6%,D:1.2%],F:9.8%,M:7.4%,n:255\t   \n",
      "\n",
      "2023-03-15 14:12:37 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-03-15 14:12:42 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 14:12:42\n",
      "2023-03-15 14:22:47 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 14:22:50 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 14:22:50 INFO:\tRunning 44 job(s) on hmmsearch, starting at 03/15/2023 14:22:50\n",
      "2023-03-15 14:22:56 INFO:\t[hmmsearch]\t5 of 44 task(s) completed\n",
      "2023-03-15 14:22:56 INFO:\t[hmmsearch]\t9 of 44 task(s) completed\n",
      "2023-03-15 14:22:56 INFO:\t[hmmsearch]\t14 of 44 task(s) completed\n",
      "2023-03-15 14:22:56 INFO:\t[hmmsearch]\t18 of 44 task(s) completed\n",
      "2023-03-15 14:22:56 INFO:\t[hmmsearch]\t23 of 44 task(s) completed\n",
      "2023-03-15 14:22:56 INFO:\t[hmmsearch]\t27 of 44 task(s) completed\n",
      "2023-03-15 14:22:56 INFO:\t[hmmsearch]\t31 of 44 task(s) completed\n",
      "2023-03-15 14:22:56 INFO:\t[hmmsearch]\t36 of 44 task(s) completed\n",
      "2023-03-15 14:22:56 INFO:\t[hmmsearch]\t40 of 44 task(s) completed\n",
      "2023-03-15 14:22:56 INFO:\t[hmmsearch]\t44 of 44 task(s) completed\n",
      "2023-03-15 14:22:59 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-03-15 14:23:00 INFO:\tResults:\tC:83.2%[S:82.0%,D:1.2%],F:9.8%,M:7.0%,n:255\t   \n",
      "\n",
      "2023-03-15 14:23:15 INFO:\teukaryota_odb10 selected\n",
      "\n",
      "2023-03-15 14:23:15 INFO:\t***** Searching tree for chosen lineage to find best taxonomic match *****\n",
      "\n",
      "2023-03-15 14:23:16 INFO:\tExtract markers...\n",
      "2023-03-15 14:23:16 INFO:\tPlace the markers on the reference tree...\n",
      "2023-03-15 14:23:16 INFO:\tRunning 1 job(s) on sepp, starting at 03/15/2023 14:23:16\n",
      "2023-03-15 14:24:18 INFO:\t[sepp]\t1 of 1 task(s) completed\n",
      "2023-03-15 14:24:20 INFO:\tLineage metazoa is selected, supported by 48 markers out of 89\n",
      "2023-03-15 14:24:20 WARNING:\tThe file or folder /master/nplatt/sch_man_ont/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "2023-03-15 14:24:21 INFO:\tRunning BUSCO using lineage dataset metazoa_odb10 (eukaryota, 2021-02-17)\n",
      "2023-03-15 14:24:21 INFO:\tRunning 1 job(s) on bbtools, starting at 03/15/2023 14:24:21\n",
      "2023-03-15 14:24:28 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-03-15 14:24:28 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 14:24:28\n",
      "2023-03-15 14:30:57 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 14:31:12 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 14:31:12 INFO:\tRunning 954 job(s) on hmmsearch, starting at 03/15/2023 14:31:12\n",
      "2023-03-15 14:31:19 INFO:\t[hmmsearch]\t96 of 954 task(s) completed\n",
      "2023-03-15 14:31:20 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-03-15 14:31:21 INFO:\t[hmmsearch]\t573 of 954 task(s) completed\n",
      "2023-03-15 14:31:21 INFO:\t[hmmsearch]\t668 of 954 task(s) completed\n",
      "2023-03-15 14:31:21 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-03-15 14:31:23 INFO:\t[hmmsearch]\t954 of 954 task(s) completed\n",
      "2023-03-15 14:31:26 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-03-15 14:31:42 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 14:31:42\n",
      "2023-03-15 14:42:58 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 14:43:03 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 14:43:03 INFO:\tRunning 335 job(s) on hmmsearch, starting at 03/15/2023 14:43:03\n",
      "2023-03-15 14:43:09 INFO:\t[hmmsearch]\t34 of 335 task(s) completed\n",
      "2023-03-15 14:43:10 INFO:\t[hmmsearch]\t101 of 335 task(s) completed\n",
      "2023-03-15 14:43:10 INFO:\t[hmmsearch]\t134 of 335 task(s) completed\n",
      "2023-03-15 14:43:11 INFO:\t[hmmsearch]\t168 of 335 task(s) completed\n",
      "2023-03-15 14:43:11 INFO:\t[hmmsearch]\t201 of 335 task(s) completed\n",
      "2023-03-15 14:43:11 INFO:\t[hmmsearch]\t235 of 335 task(s) completed\n",
      "2023-03-15 14:43:11 INFO:\t[hmmsearch]\t268 of 335 task(s) completed\n",
      "2023-03-15 14:43:12 INFO:\t[hmmsearch]\t302 of 335 task(s) completed\n",
      "2023-03-15 14:43:12 INFO:\t[hmmsearch]\t335 of 335 task(s) completed\n",
      "2023-03-15 14:43:18 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-03-15 14:43:23 INFO:\tResults:\tC:66.6%[S:65.3%,D:1.3%],F:7.4%,M:26.0%,n:954\t   \n",
      "\n",
      "2023-03-15 14:43:43 INFO:\t\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from generic domain eukaryota_odb10      |\n",
      "\t--------------------------------------------------\n",
      "\t|C:83.2%[S:82.0%,D:1.2%],F:9.8%,M:7.0%,n:255      |\n",
      "\t|212\tComplete BUSCOs (C)                       |\n",
      "\t|209\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|3\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|25\tFragmented BUSCOs (F)                     |\n",
      "\t|18\tMissing BUSCOs (M)                        |\n",
      "\t|255\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from dataset metazoa_odb10               |\n",
      "\t--------------------------------------------------\n",
      "\t|C:66.6%[S:65.3%,D:1.3%],F:7.4%,M:26.0%,n:954     |\n",
      "\t|635\tComplete BUSCOs (C)                       |\n",
      "\t|623\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|12\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|71\tFragmented BUSCOs (F)                     |\n",
      "\t|248\tMissing BUSCOs (M)                        |\n",
      "\t|954\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "2023-03-15 14:43:43 INFO:\tBUSCO analysis done with WARNING(s). Total running time: 2294 seconds\n",
      "\n",
      "***** Summary of warnings: *****\n",
      "2023-03-15 14:24:20 WARNING:busco.BuscoDownloadManager\tThe file or folder /master/nplatt/sch_man_ont/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "\n",
      "2023-03-15 14:43:43 INFO:\tResults written in /master/nplatt/sch_man_ont/results/busco/smbre\n",
      "2023-03-15 14:43:43 INFO:\tFor assistance with interpreting the results, please consult the userguide: https://busco.ezlab.org/busco_userguide.html\n",
      "\n",
      "2023-03-15 14:43:43 INFO:\tVisit this page https://gitlab.com/ezlab/busco#how-to-cite-busco to see how to cite BUSCO\n",
      "2023-03-15 14:43:46 INFO:\t***** Start a BUSCO v5.4.3 analysis, current time: 03/15/2023 14:43:46 *****\n",
      "2023-03-15 14:43:46 INFO:\tConfiguring BUSCO with local environment\n",
      "2023-03-15 14:43:46 INFO:\tMode is genome\n",
      "2023-03-15 14:43:46 INFO:\tDownloading information on latest versions of BUSCO data...\n",
      "2023-03-15 14:43:49 INFO:\tInput file is /master/nplatt/sch_man_ont/results/remove_contamination/smeg.basta.fasta\n",
      "2023-03-15 14:43:49 INFO:\tNo lineage specified. Running lineage auto selector.\n",
      "\n",
      "2023-03-15 14:43:49 INFO:\t***** Starting Auto Select Lineage *****\n",
      "\tThis process runs BUSCO on the generic lineage datasets for the domains archaea, bacteria and eukaryota. Once the optimal domain is selected, BUSCO automatically attempts to find the most appropriate BUSCO dataset to use based on phylogenetic placement.\n",
      "\t--auto-lineage-euk and --auto-lineage-prok are also available if you know your input assembly is, or is not, an eukaryote. See the user guide for more information.\n",
      "\tA reminder: Busco evaluations are valid when an appropriate dataset is used, i.e., the dataset belongs to the lineage of the species to test. Because of overlapping markers/spurious matches among domains, busco matches in another domain do not necessarily mean that your genome/proteome contains sequences from this domain. However, a high busco score in multiple domains might help you identify possible contaminations.\n",
      "2023-03-15 14:43:51 INFO:\tRunning BUSCO using lineage dataset eukaryota_odb10 (eukaryota, 2020-09-10)\n",
      "2023-03-15 14:43:51 INFO:\tRunning 1 job(s) on bbtools, starting at 03/15/2023 14:43:51\n",
      "2023-03-15 14:43:55 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-03-15 14:43:56 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 14:43:56\n",
      "2023-03-15 14:50:16 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 14:50:32 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 14:50:32 INFO:\tRunning 255 job(s) on hmmsearch, starting at 03/15/2023 14:50:32\n",
      "2023-03-15 14:50:39 INFO:\t[hmmsearch]\t26 of 255 task(s) completed\n",
      "2023-03-15 14:50:39 INFO:\t[hmmsearch]\t51 of 255 task(s) completed\n",
      "2023-03-15 14:50:39 INFO:\t[hmmsearch]\t77 of 255 task(s) completed\n",
      "2023-03-15 14:50:39 INFO:\t[hmmsearch]\t128 of 255 task(s) completed\n",
      "2023-03-15 14:50:39 INFO:\t[hmmsearch]\t153 of 255 task(s) completed\n",
      "2023-03-15 14:50:40 INFO:\t[hmmsearch]\t179 of 255 task(s) completed\n",
      "2023-03-15 14:50:40 INFO:\t[hmmsearch]\t204 of 255 task(s) completed\n",
      "2023-03-15 14:50:40 INFO:\t[hmmsearch]\t230 of 255 task(s) completed\n",
      "2023-03-15 14:50:40 INFO:\t[hmmsearch]\t255 of 255 task(s) completed\n",
      "2023-03-15 14:50:41 INFO:\tResults:\tC:80.8%[S:79.2%,D:1.6%],F:10.2%,M:9.0%,n:255\t   \n",
      "\n",
      "2023-03-15 14:50:41 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-03-15 14:50:46 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 14:50:46\n",
      "2023-03-15 15:01:21 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 15:01:25 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 15:01:25 INFO:\tRunning 49 job(s) on hmmsearch, starting at 03/15/2023 15:01:25\n",
      "2023-03-15 15:01:30 INFO:\t[hmmsearch]\t5 of 49 task(s) completed\n",
      "2023-03-15 15:01:30 INFO:\t[hmmsearch]\t10 of 49 task(s) completed\n",
      "2023-03-15 15:01:30 INFO:\t[hmmsearch]\t15 of 49 task(s) completed\n",
      "2023-03-15 15:01:30 INFO:\t[hmmsearch]\t20 of 49 task(s) completed\n",
      "2023-03-15 15:01:30 INFO:\t[hmmsearch]\t25 of 49 task(s) completed\n",
      "2023-03-15 15:01:30 INFO:\t[hmmsearch]\t30 of 49 task(s) completed\n",
      "2023-03-15 15:01:31 INFO:\t[hmmsearch]\t35 of 49 task(s) completed\n",
      "2023-03-15 15:01:31 INFO:\t[hmmsearch]\t40 of 49 task(s) completed\n",
      "2023-03-15 15:01:31 INFO:\t[hmmsearch]\t45 of 49 task(s) completed\n",
      "2023-03-15 15:01:31 INFO:\t[hmmsearch]\t49 of 49 task(s) completed\n",
      "2023-03-15 15:01:33 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-03-15 15:01:35 INFO:\tResults:\tC:82.0%[S:80.4%,D:1.6%],F:10.2%,M:7.8%,n:255\t   \n",
      "\n",
      "2023-03-15 15:01:50 INFO:\teukaryota_odb10 selected\n",
      "\n",
      "2023-03-15 15:01:50 INFO:\t***** Searching tree for chosen lineage to find best taxonomic match *****\n",
      "\n",
      "2023-03-15 15:01:51 INFO:\tExtract markers...\n",
      "2023-03-15 15:01:51 INFO:\tPlace the markers on the reference tree...\n",
      "2023-03-15 15:01:51 INFO:\tRunning 1 job(s) on sepp, starting at 03/15/2023 15:01:51\n",
      "2023-03-15 15:02:53 INFO:\t[sepp]\t1 of 1 task(s) completed\n",
      "2023-03-15 15:02:56 INFO:\tLineage metazoa is selected, supported by 43 markers out of 85\n",
      "2023-03-15 15:02:56 WARNING:\tThe file or folder /master/nplatt/sch_man_ont/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "2023-03-15 15:02:57 INFO:\tRunning BUSCO using lineage dataset metazoa_odb10 (eukaryota, 2021-02-17)\n",
      "2023-03-15 15:02:57 INFO:\tRunning 1 job(s) on bbtools, starting at 03/15/2023 15:02:57\n",
      "2023-03-15 15:03:04 INFO:\t[bbtools]\t1 of 1 task(s) completed\n",
      "2023-03-15 15:03:04 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 15:03:04\n",
      "2023-03-15 15:09:10 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 15:09:31 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 15:09:31 INFO:\tRunning 954 job(s) on hmmsearch, starting at 03/15/2023 15:09:31\n",
      "2023-03-15 15:09:39 INFO:\t[hmmsearch]\t96 of 954 task(s) completed\n",
      "2023-03-15 15:09:39 INFO:\t[hmmsearch]\t191 of 954 task(s) completed\n",
      "2023-03-15 15:09:39 INFO:\t[hmmsearch]\t287 of 954 task(s) completed\n",
      "2023-03-15 15:09:40 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-03-15 15:09:40 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-03-15 15:09:40 INFO:\t[hmmsearch]\t382 of 954 task(s) completed\n",
      "2023-03-15 15:09:40 INFO:\t[hmmsearch]\t478 of 954 task(s) completed\n",
      "2023-03-15 15:09:40 INFO:\t[hmmsearch]\t573 of 954 task(s) completed\n",
      "2023-03-15 15:09:40 INFO:\t[hmmsearch]\t668 of 954 task(s) completed\n",
      "2023-03-15 15:09:41 INFO:\t[hmmsearch]\t764 of 954 task(s) completed\n",
      "2023-03-15 15:09:41 INFO:\t[hmmsearch]\t859 of 954 task(s) completed\n",
      "2023-03-15 15:09:42 INFO:\t[hmmsearch]\t954 of 954 task(s) completed\n",
      "2023-03-15 15:09:45 INFO:\tExtracting missing and fragmented buscos from the file refseq_db.faa...\n",
      "2023-03-15 15:10:00 INFO:\tRunning 1 job(s) on metaeuk, starting at 03/15/2023 15:10:00\n",
      "2023-03-15 15:20:37 INFO:\t[metaeuk]\t1 of 1 task(s) completed\n",
      "2023-03-15 15:20:48 INFO:\t***** Run HMMER on gene sequences *****\n",
      "2023-03-15 15:20:48 INFO:\tRunning 331 job(s) on hmmsearch, starting at 03/15/2023 15:20:48\n",
      "2023-03-15 15:20:55 INFO:\t[hmmsearch]\t34 of 331 task(s) completed\n",
      "2023-03-15 15:20:55 INFO:\t[hmmsearch]\t67 of 331 task(s) completed\n",
      "2023-03-15 15:20:55 INFO:\t[hmmsearch]\t100 of 331 task(s) completed\n",
      "2023-03-15 15:20:56 INFO:\t[hmmsearch]\t133 of 331 task(s) completed\n",
      "2023-03-15 15:20:56 INFO:\t[hmmsearch]\t166 of 331 task(s) completed\n",
      "2023-03-15 15:20:56 INFO:\t[hmmsearch]\t199 of 331 task(s) completed\n",
      "2023-03-15 15:20:57 INFO:\t[hmmsearch]\t232 of 331 task(s) completed\n",
      "2023-03-15 15:20:57 INFO:\t[hmmsearch]\t265 of 331 task(s) completed\n",
      "2023-03-15 15:20:57 INFO:\t[hmmsearch]\t298 of 331 task(s) completed\n",
      "2023-03-15 15:20:58 INFO:\t[hmmsearch]\t331 of 331 task(s) completed\n",
      "2023-03-15 15:21:09 INFO:\tValidating exons and removing overlapping matches\n",
      "2023-03-15 15:21:14 INFO:\tResults:\tC:67.4%[S:66.4%,D:1.0%],F:7.4%,M:25.2%,n:954\t   \n",
      "\n",
      "2023-03-15 15:21:35 INFO:\t\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from generic domain eukaryota_odb10      |\n",
      "\t--------------------------------------------------\n",
      "\t|C:82.0%[S:80.4%,D:1.6%],F:10.2%,M:7.8%,n:255     |\n",
      "\t|209\tComplete BUSCOs (C)                       |\n",
      "\t|205\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|4\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|26\tFragmented BUSCOs (F)                     |\n",
      "\t|20\tMissing BUSCOs (M)                        |\n",
      "\t|255\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "\n",
      "\t--------------------------------------------------\n",
      "\t|Results from dataset metazoa_odb10               |\n",
      "\t--------------------------------------------------\n",
      "\t|C:67.4%[S:66.4%,D:1.0%],F:7.4%,M:25.2%,n:954     |\n",
      "\t|643\tComplete BUSCOs (C)                       |\n",
      "\t|633\tComplete and single-copy BUSCOs (S)       |\n",
      "\t|10\tComplete and duplicated BUSCOs (D)        |\n",
      "\t|71\tFragmented BUSCOs (F)                     |\n",
      "\t|240\tMissing BUSCOs (M)                        |\n",
      "\t|954\tTotal BUSCO groups searched               |\n",
      "\t--------------------------------------------------\n",
      "2023-03-15 15:21:35 INFO:\tBUSCO analysis done with WARNING(s). Total running time: 2266 seconds\n",
      "\n",
      "***** Summary of warnings: *****\n",
      "2023-03-15 15:02:56 WARNING:busco.BuscoDownloadManager\tThe file or folder /master/nplatt/sch_man_ont/results/busco/busco_downloads/lineages/metazoa_odb10 is not the last available version. To update all data files to the last version, add the parameter --update-data in your next run.\n",
      "\n",
      "2023-03-15 15:21:35 INFO:\tResults written in /master/nplatt/sch_man_ont/results/busco/smeg\n",
      "2023-03-15 15:21:35 INFO:\tFor assistance with interpreting the results, please consult the userguide: https://busco.ezlab.org/busco_userguide.html\n",
      "\n",
      "2023-03-15 15:21:35 INFO:\tVisit this page https://gitlab.com/ezlab/busco#how-to-cite-busco to see how to cite BUSCO\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    assembly = \"{}/results/remove_contamination/{}.basta.fasta\".format(proj_dir, sample)\n",
    "    !conda run -n busco --cwd . --live-stream busco -i {assembly} --auto-lineage-euk -c 48 -o {sample} -m genome "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9ffd5276-7d80-4904-8fcb-2e8d90c0ca33",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Synteny"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e5bd150b-d84a-49f9-ae65-3dfdd8424f27",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/synteny/\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/synteny\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "9d44fdf5-3046-4977-a5d0-7cb8a9580055",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "#repeatmask the genomes\n",
    "ln -s ../../data/genome/GCA_000237925.5.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b54261cb-eb07-4c49-8b35-62ce5d79323e",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir repeatmodeler\n",
    "cd repeatmodeler\n",
    "\n",
    "BuildDatabase -name GCA_000237925.5 ../GCA_000237925.5.fa\n",
    "\n",
    "RepeatModeler -pa 48 -database GCA_000237925.5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d9f3981-7603-45b0-b849-9ddae0bfa8c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "SAMPLES=(\"smor\" \"smle_pzq_es\" \"smle_pzq_er\" \"smbre\" \"smeg\")\n",
    "PROJ_DIR = \"/master/nplatt/sch_man_ont\"\n",
    "\n",
    "RMOD_LIB = \"${PROJ_DIR}/results/synteny/repeatmodeler/GCA_000237925.5-families.fa\"\n",
    "\n",
    "mkdir ${PROJ_DIR}/results/synteny/repeatmasker\n",
    "cd ${PROJ_DIR}/results/synteny/repeatmasker\n",
    "\n",
    "for SAMPLE in \"${SAMPLES[@]}\"; do\n",
    "\n",
    "    mkdir ${PROJ_DIR}/results/synteny/repeatmasker/${SAMPLE}\n",
    "    cd ${PROJ_DIR}/results/synteny/repeatmasker/${SAMPLE}\n",
    "    \n",
    "    \n",
    "    ASSEMBLY = \"${PROJ_DIR}/results/remove_contamination/${SAMPLE}.basta.fasta\"\n",
    "    conda run -n repeatmasker RepeatMasker -pa 48 -s -gff -dir . -lib ${RMOD_LIB} ${ASSEMBLY}\n",
    "    \n",
    "    cd ${PROJ_DIR}/results/synteny/repeatmasker"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "faae6500-e42f-45ab-b73c-127fa4a4aecf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "make: *** No rule to make target `/master/nplatt/sch_man_ont/synteny/smor.basta.fasta.masked', needed by `smor_scaffolds.fa'.  Stop.\n",
      "make: *** No rule to make target `/master/nplatt/sch_man_ont/synteny/smle_pzq_es.basta.fasta.masked', needed by `smle_pzq_es_scaffolds.fa'.  Stop.\n",
      "make: *** No rule to make target `/master/nplatt/sch_man_ont/synteny/smle_pzq_er.basta.fasta.masked', needed by `smle_pzq_er_scaffolds.fa'.  Stop.\n",
      "make: *** No rule to make target `/master/nplatt/sch_man_ont/synteny/smbre.basta.fasta.masked', needed by `smbre_scaffolds.fa'.  Stop.\n",
      "make: *** No rule to make target `/master/nplatt/sch_man_ont/synteny/smeg.basta.fasta.masked', needed by `smeg_scaffolds.fa'.  Stop.\n"
     ]
    }
   ],
   "source": [
    "for sample in samples:\n",
    "    assembly = \"{}/synteny/{}.basta.fasta.masked\".format(proj_dir, sample)\n",
    "    \n",
    "    Path(f\"{proj_dir}/results/synteny/{sample}\").mkdir(parents=True, exist_ok=True)\n",
    "    os.chdir(f\"{proj_dir}/results/synteny/{sample}\")\n",
    "    \n",
    "    !{proj_dir}/bin/JupiterPlot/jupiter t=48 name={sample} ng=85 ref=../GCA_000237925.5.fa fa={assembly}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d0b43a9-4a3d-4319-9832-62a17e381f1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "SAMPLES=(\"smor\" \"smle_pzq_es\" \"smle_pzq_er\" \"smbre\" \"smeg\")\n",
    "\n",
    "for SAMPLE in \"${SAMPLES[@]}\"; do\n",
    "    WORK_DIR=\"$(pwd)/${SAMPLE}/\"\n",
    "    mkdir -p ${WORK_DIR} && cd ${WORK_DIR}\n",
    "    ~/sch_man_ont/bin/JupiterPlot/jupiter t=48 name=${SAMPLE} n=85 ref=../GCA_000237925.5.fa fa=${SAMPLE}.basta.fasta.masked\n",
    "    cd ..\n",
    "done\n",
    "    \n",
    "# ~/sch_man_ont/bin/JupiterPlot/jupiter t=48 name=smle_pzq_es n=85 ref=GCA_000237925.5.fa fa=smle_pzq_es.basta.fasta.masked\n",
    "# ~/sch_man_ont/bin/JupiterPlot/jupiter t=48 name=smle_pzq_er n=85 ref=GCA_000237925.5.fa fa=smle_pzq_er.basta.fasta.masked\n",
    "# ~/sch_man_ont/bin/JupiterPlot/jupiter t=48 name=smbre n=85 ref=GCA_000237925.5.fa fa=smbre.basta.fasta.masked\n",
    "# ~/sch_man_ont/bin/JupiterPlot/jupiter t=48 name=smeg n=85 ref=GCA_000237925.5.fa fa=smeg.basta.fasta.masked"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93197ef3-c9a1-4f1d-a8ec-12330f64d926",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## TE annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "dc5c6957-e7c5-4552-8f74-9d01f51afb12",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(\"{}/results/edta\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/edta\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "757ef2c1-8889-4df0-a72f-1c1fc6a2524c",
   "metadata": {},
   "source": [
    "This step gets a CDS file for EDTA from the v10 version of the genome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "96fad7e4-dc9b-4e15-a130-7c5d0f54b830",
   "metadata": {},
   "outputs": [],
   "source": [
    "%bash\n",
    "\n",
    "wget https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS18/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS18.CDS_transcripts.fa.gz\n",
    "gunzip schistosoma_mansoni.PRJEA36577.WBPS18.CDS_transcripts.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "418eed82-ab63-4c8e-bd88-684a4542b6dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda run -n EDTA --cwd . --live-stream \\\n",
    "    perl ~/sch_man_ont/bin/EDTA/EDTA.pl \\\n",
    "        --genome ~/sch_man_ont/data/genome/SM_V9.fa \\\n",
    "        --species others \\\n",
    "        --step all \\\n",
    "        --overwrite 1 \\\n",
    "        --cds schistosoma_mansoni.PRJEA36577.WBPS18.CDS_transcripts.fa \\\n",
    "        --sensitive 1 \\\n",
    "        --anno 1 \\\n",
    "        --force 0 \\\n",
    "        --threads 48"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "818182ca-701c-4c0c-86be-8094904e6032",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Grave Yard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "316d74b7-320a-4a5c-9201-bba9709eb023",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate sm_ont-process_reads\n",
    "\n",
    "cd /master/nplatt/sm_nanopore\n",
    "\n",
    "cd sm_nanopore/results/nanofilt\n",
    "3\n",
    " ~/anaconda3/envs/sm_ont-process_reads/bin/flye --nano-cor ../read_correction_canu/${SAMPLE}/${SAMPLE}.correctedReads.fasta.gz -g 385m -t 96  -i 3 -o ${SAMPLE} >${SAMPLE}.canu.log 2>&1 &\n",
    "\n",
    "conda activate repeatmasker\n",
    "RepeatMasker \\\n",
    "    -pa 12 \\\n",
    "    -lib ../repeatmodeler/Smansoni_v7-families.fa \\\n",
    "    -gff \\\n",
    "    -s \\\n",
    "    -dir $(pwd) \\\n",
    "    -small \\\n",
    "    -x \\\n",
    "    $REF\n",
    "\n",
    "\n",
    "\n",
    "SmOR_Female_6_02_22_vSM_V9_sorted.vcf\n",
    "schisto_pairs_04132022_vSM_V9_sorted.vcf\n",
    "SmEG_Female_5_25_22_vSM_V9_sorted.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500) && AF>0.75' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pzq-es_filtered.vcf\n",
    "\n",
    "bedtools intersect \\\n",
    "    -c \\\n",
    "    -b smle-pzq-es_filtered.vcf \\\n",
    "    -a ../lift/v9.bed \\\n",
    "    >smle-pzq-es_intersect.bed\n",
    "\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-200 || SVLEN>200)' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout\n",
    "\n",
    "#############################################################################################\n",
    "\n",
    "#genrate reads\n",
    "art_illumina \\\n",
    "    --len 150 \\\n",
    "    --fcov 100 \\\n",
    "    --errfree \\\n",
    "    -na \\\n",
    "    --in SM_V9.fa \\\n",
    "    --out sm_v9_illumina \\\n",
    "    >sm_v9_illumina.log 2>&1 &\n",
    "\n",
    "nanosim-h \\\n",
    "    --perfect \\\n",
    "    -o sm_v9_ont \\\n",
    "    -n 5000000 \\\n",
    "    SM_V9.fa \\\n",
    "    >sm_v9_ont.log 2>&1 &\n",
    "\n",
    "#convert to fasta\n",
    "sed -n '1~4s/^@/>/p;2~4p' sm_v9_illumina.fq > sm_v9_illumina.fa\n",
    "\n",
    "#make sure amount of data is similar\n",
    "echo sm_v9_ont.fa $(cat sm_v9_ont.fa | paste  - - | cut -f2 | wc -c) >>counts &\n",
    "echo sm_v9_illumina.fa $(cat sm_v9_illumina.fa | paste  - - | cut -f2 | wc -c) >>counts &\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#map reads (perfectly and unambiguosly) back to the reference\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "    bbmap.sh \\\n",
    "        ref=SM_V9.fa \\\n",
    "        perfectmode=t \\\n",
    "        threads=48 \\\n",
    "        ambiguous=toss \\\n",
    "        in=${SAMPLE}.fa \\\n",
    "        overwrite=t \\\n",
    "        out=${SAMPLE}.sam \\\n",
    "        mappedonly=t \\\n",
    "        >${SAMPLE}.bbmap.log 2>&1 &\n",
    "done\n",
    "\n",
    "\n",
    "#calculate coverage\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "\n",
    "    echo $SAMPLE\n",
    "    echo \"...sort\"\n",
    "    samtools view -b -F 4 ${SAMPLE}.sam | samtools sort >${SAMPLE}.bam\n",
    "    echo \"...idx\"\n",
    "    samtools index ${SAMPLE}.bam\n",
    "    echo \"...cov\"\n",
    "    mosdepth \\\n",
    "        --threads 4 \\\n",
    "        ${SAMPLE} \\\n",
    "        ${SAMPLE}.bam\n",
    "done\n",
    "\n",
    "#how much is inaccessible per read type\n",
    "for SAMPLE in sm_v9_illumina sm_v9_ont; do\n",
    "\n",
    "    MB=$(zcat ${SAMPLE}.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}')\n",
    "    echo -e $SAMPLE\"\\t\"$MB\n",
    "\n",
    "done\n",
    "\n",
    "art_illumina \\\n",
    "--paired \\\n",
    "--len 150 \\\n",
    "--fcov 10 \\\n",
    "--mflen 300 \\\n",
    "--sdev 80 \\\n",
    "-na \\\n",
    "--in SM_V9.fa \\\n",
    "--out sm_v9_wes\n",
    "\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_wes1.fq \\\n",
    "    in2=sm_v9_wes2.fq \\\n",
    "    samestrandpairs=t \\\n",
    "    killbadpairs=t \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_wes.sam \\\n",
    "    mappedonly=t \\\n",
    "    ignorebadquality=t\n",
    "\n",
    "samtools view -b -F 4 sm_v9_wes.sam | samtools sort >sm_v9_wes.bam\n",
    "samtools index sm_v9_wes.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_wes \\\n",
    "    sm_v9_wes.bam\n",
    "\n",
    "\n",
    "\n",
    "zcat sm_v9_wes.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_wes.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "cd ~/sch_man_ont/data/\n",
    "wget https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS17/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS17.canonical_geneset.gtf.gz\n",
    "gunzip schistosoma_mansoni.PRJEA36577.WBPS17.canonical_geneset.gtf.gz\n",
    "\n",
    "cp SM_V9.gtf SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_1/HE601624.3/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_2/HE601625.3/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_3/HE601626.3/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_4/HE601627.3/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_5/HE601628.3/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_6/HE601629.3/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_7/HE601630.3/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_WSR/HE601631.3/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_PAR1/OU426849.1/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_PAR2/OU426850.1/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_ZSR/OU426848.1/' SM_V9.mod.gtf\n",
    "sed -i 's/SM_V9_MITO/HE601612.2/' SM_V9.mod.gtf\n",
    "\n",
    "bedtools intersect \\\n",
    "    -wb \\\n",
    "    -a smle-pzq-es_filtered.vcf \\\n",
    "    -b sm_v9.gff3\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500) && AF>0.75' \\\n",
    "    schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pzq-es_filtered.vcf\n",
    "\n",
    "bedtools intersect \\\n",
    "    -c \\\n",
    "    -b smle-pzq-es_filtered.vcf \\\n",
    "    -a ../lift/v9.bed\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "cp ../cutesv/sm_v9.gff3 .\n",
    "awk '{if ($3==\"exon\") print $0}' sm_v9.gff3 >sm_v9_exons.gff3\n",
    "cut -f1 -d\";\" sm_v9_exons.gff3 | cut -f1,4,5,9 | sed 's/ID=exon://' >sm_v9_exons.bed\n",
    "bedtools intersect -a sm_v9_exons.bed  -b qtls_v9.bed >qtl_exons.bed\n",
    "\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smeg_filtered.vcf  >smeg_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smeg_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smeg_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smeg_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smeg_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smor_filtered.vcf  >smor_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smor_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smor_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smor_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smor_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "\n",
    "bedtools intersect -wb -a qtls_v9.bed -b smle-pzq-es_filtered.vcf  >smle-pzq-es_qtl_snvs.vcf\n",
    "bedtools intersect -wa -b smle-pzq-es_qtl_snvs.vcf -a qtl_exons.bed   | sort -u -k4,4 > smle-pzq-es_snvs_in_qtl_exons.bed\n",
    "bedtools intersect -c -b smle-pzq-es_snvs_in_qtl_exons.bed -a qtls_v9.bed\n",
    "bedtools intersect -wb -b smle-pzq-es_snvs_in_qtl_exons.bed -a qtls_v9.bed | cut -f4,8 | cut -f1 -d\".\" | sort -u -k2,2 | cut -f1 | sort | uniq -c\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "nanosim-h --perfect -o sm_v9_ont_reads -n 500000 SM_V9.fa\n",
    "\n",
    "\n",
    "    minimap2 \\\n",
    "        -a \\\n",
    "        -o sm_v9_ont.sam \\\n",
    "        -t 48 \\\n",
    "        -x map-ont \\\n",
    "        SM_V9.fa \\\n",
    "        sm_v9_ont_reads.fa\n",
    "\n",
    "\n",
    "\n",
    "samtools view -b -F 4 sm_v9_ont.sam | samtools sort >sm_v9_ont.bam\n",
    "\n",
    "samtools index sm_v9_ont.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_ont \\\n",
    "    sm_v9_ont.bam\n",
    "\n",
    "zcat sm_v9_ont.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_ont.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_ont_reads.fa \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_ont_bbmap.sam \\\n",
    "    mappedonly=t \\\n",
    "    ignorebadquality=t\n",
    "\n",
    "bbmap.sh \\\n",
    "    ref=SM_V9.fa \\\n",
    "    perfectmode=t \\\n",
    "    threads=48 \\\n",
    "    ambiguous=toss \\\n",
    "    in=sm_v9_ont_reads.fa \\\n",
    "    overwrite=t \\\n",
    "    out=sm_v9_ont_bbmap.sam \\\n",
    "    mappedonly=t\n",
    "\n",
    "\n",
    "samtools view -b -F 4 sm_v9_ont_bbmap.sam | samtools sort >sm_v9_ont_bbmap.bam\n",
    "samtools index sm_v9_ont_bbmap.bam\n",
    "\n",
    "mosdepth \\\n",
    "    --threads 4 \\\n",
    "    sm_v9_ont_bbmap \\\n",
    "    sm_v9_ont_bbmap.bam\n",
    "\n",
    "zcat sm_v9_ont_bbmap.per-base.bed.gz | awk '{if ($4>=12) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "zcat sm_v9_ont_bbmap.per-base.bed.gz | awk '{if ($4<=2) SUM+=$3-$2} END {print SUM/1000000}'\n",
    "\n",
    "RepeatMasker -qq -pa 48 -nolow -norna -dir . -gff -lib sm_v7_tes.fas SM_V9.fa\n",
    "\n",
    "SmEG_Female_5_25_22\n",
    "schisto_pairs_04132022  ../cutesv/SmOR_Female_6_02_22_vSM_V9_sorted.vcf\n",
    "\n",
    "for VCF in $(ls ../cutesv/*_vSM_V9_sorted.vcf); do \n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/schisto_pairs_04132022_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smle-pze-es_bigsvs.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/SmEG_Female_5_25_22_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smeg_bigsvs.vcf\n",
    "\n",
    "bcftools filter \\\n",
    "    -sLOWFREQ_SMALL -i' (SVLEN<-500 || SVLEN>500)' \\\n",
    "    ../cutesv/SmOR_Female_6_02_22_vSM_V9_sorted.vcf \\\n",
    "    | vcftools \\\n",
    "        --remove-filtered-all \\\n",
    "        --vcf - \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout \\\n",
    "        >smor_bigsvs.vcf\n",
    "\n",
    "   396,000,000\n",
    "39,777,694,771\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "sm_ont-process_reads",
   "language": "python",
   "name": "sm_ont-process_reads"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
