{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5fa1b64e-a799-4ff8-896c-36534c559df6",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# Nanopore long-read sequencing uncovers hundreds of structural variants in the Schistosome genome\n",
    "*Shalini Nair, Elisha Enabulele, Xue Li, Tim Anderson, Neal Platt*"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af9f12f9-924c-4ced-9a24-70fe777660e7",
   "metadata": {
    "tags": []
   },
   "source": [
    "__ABSTRACT__: Nanopore sequencing generates extremely long sequence reads allowing detailed characterization of genomic features (genome rearrangements/structural variants) that are difficult to accurately assay using Illumina short-read sequencing approaches. Technological improvements in Nanopore long-read sequence have resulted in reduced prices, as well as increased accuracy and output. This method was declared “Method of the year” in Nature in 2022. Our laboratory is exploring the use of nanopore for characterizing pathogen genomes. Here, we detail application of nanopore sequencing of schistosome parasites. These parasites have a 380 Mb genome that is riddled with repetitive elements and structural variants. We have developed protocols that allow generation of reads up to 200kb, with 50% of reads > 24kb, and generation of 13.5 Gb of high-quality data from a single R10 flow cell, providing 27x coverage of the parasite genome.  Using this new data, we have identified multiple structural variants segregating in a lab strain of S. mansoni that would not have been visible using short read sequencing alone.  For example, we have found 468 large structural variants (>500bp) that were completely fixed in our laboratory population compared to the reference genome/strain. We anticipate that characterization of large structural variants may help to identify causative features underlying important schistosome phenotypes. Nanopore also has many possible uses for other researchers at Texas biomed, including sequencing entire viral genomes or host MHC, and rapid monitoring of infectious diseases in global populations."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77d21088-87ad-4785-97c9-9e142e6208c5",
   "metadata": {
    "tags": []
   },
   "source": [
    "**[08/04/2023] Nanopore analysis ideas**\n",
    "\n",
    "- Are the genomes of lab populations syntenic or have large changes in gene structure/order been introduced?\n",
    "  - __This would be extremely interesting__\n",
    "  - Aside – one possible explanation for the division between Sh populations might be rearrangements/inversions\n",
    "- What impact do large SVs have on the coding regions of the genome?\n",
    "  - Comment – define impact here. It would certainly be interesting to know whether SVs in coding sequences seem likely to reduce/eliminate gene expression\n",
    "- Is there evidence of population specific SVs (including CNVs/gene family expansions/contractions)?\n",
    "- Can we recover and examine variability in complex regions like SmPoMucs and how is this variability distributed among and within lab populations?\n",
    "  - __Yes – v important.__\n",
    "  - Simply identifying the genome location of these genes will be very important – we would like to be able to definitively state that smPoMucs do/do not colocalize with the host specificity QTLs that Fred has mapped.\n",
    "- Are there large SVs in important regions associated with parasite phenotypes?\n",
    "  - Important question but difficult to do. I think we can certainly examine how many SVs are segregating in QTL regions, to highlight possible causative SVs\n",
    "- (Some interesting question about the tandem repeat region in the mitochondrial genome...not sure what yet)\n",
    "  - There is some old literature about these regions, suggesting v rapid change and extensive polymorphism within individuals. This might be difficult with the population data, but would be great to do when we have data from individual genotypes. Perhaps worth waiting until we have that data.\n",
    "- __Is it possible to identify SVs that are being selected (for/against)?__\n",
    "  - I think this is particularly interesting. We expect most SVs to be selected against.\n",
    "  - Comparing allele frequencies within populations would be informative – we expect most to be at low frequency. High frequency SVs are good candidates for positive selection\n",
    "  - __Comparing SV burden in coding vs non-coding regions would be informative to understand strength of selection against SVs__\n",
    "  - __Likewise, comparing SVs in low and high recombination rate regions. We predict higher frequency/larger SVs in high recombination regions__"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f44f39c-ec40-4ed0-aee4-7b899c9a589b",
   "metadata": {},
   "source": [
    "# Analyzing structural variants"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db002030-e5c3-4e4b-bf17-4ab6628a7b44",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Prep analyses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 448,
   "id": "9cd4c83d-03e3-44f3-8e5a-5e64db850897",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from pathlib import Path\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "# from Bio import SeqIO\n",
    "# from Bio.SeqRecord import SeqRecord\n",
    "# import vcf\n",
    "# from itertools import combinations\n",
    "# import venn\n",
    "# import io\n",
    "# import gffutils\n",
    "# from sklearn.decomposition import PCA\n",
    "# from sklearn.preprocessing import StandardScaler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 449,
   "id": "3bb0e454-1134-4194-8c67-e685847b6669",
   "metadata": {},
   "outputs": [],
   "source": [
    "proj_dir = \"/master/nplatt/sch_man_ont\"\n",
    "ref_fas = \"{}/data/genome/SM_V10.fa\".format(proj_dir)\n",
    "Path(\"{}/results\".format(proj_dir)).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "625a0a9e-c38c-4c9a-a768-c3a22131bfb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "samples = [ \"smor\", \n",
    "            \"smle_pzq_es\",\n",
    "            \"smle_pzq_er\",\n",
    "            \"smbre\", \n",
    "            \"smeg\"        ]\n",
    "\n",
    "pops=samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d62d2d02-6e9f-4c61-8028-73b97326de49",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"{}/results\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81f6d702-84ea-41e3-946b-5e568af5d41b",
   "metadata": {},
   "source": [
    "# Fst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 958,
   "id": "fab23e9a-7d1a-42ac-abdb-790c67db63c2",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "Path(\"{}/results/fst\".format(proj_dir)).mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(\"{}/results/fst\".format(proj_dir))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 959,
   "id": "c3317d54-796b-47c0-b9f1-1d63b17b62c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SVTYPE</th>\n",
       "      <th>SVLEN</th>\n",
       "      <th>smor.MAF</th>\n",
       "      <th>smle_pzq_es.MAF</th>\n",
       "      <th>smle_pzq_er.MAF</th>\n",
       "      <th>smbre.MAF</th>\n",
       "      <th>smeg.MAF</th>\n",
       "      <th>smor.COV</th>\n",
       "      <th>smle_pzq_es.COV</th>\n",
       "      <th>smle_pzq_er.COV</th>\n",
       "      <th>smbre.COV</th>\n",
       "      <th>smeg.COV</th>\n",
       "      <th>gene_ids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SM_V10_1</td>\n",
       "      <td>108036</td>\n",
       "      <td>cuteSV.INS.4</td>\n",
       "      <td>C</td>\n",
       "      <td>['CATTATTATTATTATTATTACTACTATTATTATTACTATT']</td>\n",
       "      <td>INS</td>\n",
       "      <td>39.0</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9429</td>\n",
       "      <td>0.8889</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9565</td>\n",
       "      <td>25</td>\n",
       "      <td>70</td>\n",
       "      <td>18</td>\n",
       "      <td>25</td>\n",
       "      <td>23</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SM_V10_1</td>\n",
       "      <td>156118</td>\n",
       "      <td>cuteSV.DEL.7</td>\n",
       "      <td>ATGTTATCTTTGGCAGCCTATTTAAACATCTGGGCTACCTGTTCCT...</td>\n",
       "      <td>['A']</td>\n",
       "      <td>DEL</td>\n",
       "      <td>-401.0</td>\n",
       "      <td>0.3500</td>\n",
       "      <td>0.1587</td>\n",
       "      <td>0.3846</td>\n",
       "      <td>0.8750</td>\n",
       "      <td>0.2800</td>\n",
       "      <td>20</td>\n",
       "      <td>63</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SM_V10_1</td>\n",
       "      <td>214273</td>\n",
       "      <td>cuteSV.INS.8</td>\n",
       "      <td>T</td>\n",
       "      <td>['TGATGATGATGATAGATATTGATGATGATGATGATGATGATGAT...</td>\n",
       "      <td>INS</td>\n",
       "      <td>51.0</td>\n",
       "      <td>0.1304</td>\n",
       "      <td>0.6667</td>\n",
       "      <td>0.9333</td>\n",
       "      <td>0.8696</td>\n",
       "      <td>0.3125</td>\n",
       "      <td>23</td>\n",
       "      <td>36</td>\n",
       "      <td>15</td>\n",
       "      <td>23</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SM_V10_1</td>\n",
       "      <td>258698</td>\n",
       "      <td>cuteSV.INS.9</td>\n",
       "      <td>A</td>\n",
       "      <td>['ACTCGGGAATAACATTAGGATCACTTCAATTTTTTTAATAATTT...</td>\n",
       "      <td>INS</td>\n",
       "      <td>4470.0</td>\n",
       "      <td>0.2778</td>\n",
       "      <td>0.1897</td>\n",
       "      <td>0.1429</td>\n",
       "      <td>0.0930</td>\n",
       "      <td>0.3125</td>\n",
       "      <td>18</td>\n",
       "      <td>58</td>\n",
       "      <td>14</td>\n",
       "      <td>43</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SM_V10_1</td>\n",
       "      <td>264300</td>\n",
       "      <td>cuteSV.INS.10</td>\n",
       "      <td>C</td>\n",
       "      <td>['CCTGGAAGCACTGGACGGCCGTTTCGTCCTATTGGCGGACCCCT...</td>\n",
       "      <td>INS</td>\n",
       "      <td>324.0</td>\n",
       "      <td>0.6579</td>\n",
       "      <td>0.2637</td>\n",
       "      <td>0.5455</td>\n",
       "      <td>0.6269</td>\n",
       "      <td>0.1522</td>\n",
       "      <td>38</td>\n",
       "      <td>91</td>\n",
       "      <td>22</td>\n",
       "      <td>67</td>\n",
       "      <td>46</td>\n",
       "      <td>Smp_318880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8096</th>\n",
       "      <td>SM_V10_Z</td>\n",
       "      <td>86357919</td>\n",
       "      <td>cuteSV.INS.4444</td>\n",
       "      <td>T</td>\n",
       "      <td>['TTGTTATGTCCCCTGGTGAATTATAAATGGTAACTCTGAGTCTA...</td>\n",
       "      <td>INS</td>\n",
       "      <td>604.0</td>\n",
       "      <td>0.3333</td>\n",
       "      <td>0.2222</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>0.9545</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>18</td>\n",
       "      <td>36</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8097</th>\n",
       "      <td>SM_V10_Z</td>\n",
       "      <td>86361688</td>\n",
       "      <td>cuteSV.INS.4445</td>\n",
       "      <td>A</td>\n",
       "      <td>['AATCAGTAAGCGAGTAGTGATGGAAGTTTGGTTATTTTGGTACA...</td>\n",
       "      <td>INS</td>\n",
       "      <td>325.0</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>17</td>\n",
       "      <td>47</td>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8098</th>\n",
       "      <td>SM_V10_Z</td>\n",
       "      <td>86465601</td>\n",
       "      <td>cuteSV.DEL.5146</td>\n",
       "      <td>GTGCGCGTTCAAAAAGCGGATCACAAACTTTGTAGAAAAAT</td>\n",
       "      <td>['G']</td>\n",
       "      <td>DEL</td>\n",
       "      <td>-40.0</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.2022</td>\n",
       "      <td>0.2292</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>38</td>\n",
       "      <td>89</td>\n",
       "      <td>48</td>\n",
       "      <td>29</td>\n",
       "      <td>54</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8099</th>\n",
       "      <td>SM_V10_Z</td>\n",
       "      <td>86467302</td>\n",
       "      <td>cuteSV.INS.4446</td>\n",
       "      <td>G</td>\n",
       "      <td>['GAATTCTATATTTTTCAAGGGAGTTGAAATCATAAGTCAATCGA...</td>\n",
       "      <td>INS</td>\n",
       "      <td>229.0</td>\n",
       "      <td>0.5238</td>\n",
       "      <td>0.1860</td>\n",
       "      <td>0.0263</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.1500</td>\n",
       "      <td>21</td>\n",
       "      <td>86</td>\n",
       "      <td>38</td>\n",
       "      <td>35</td>\n",
       "      <td>40</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8100</th>\n",
       "      <td>SM_V10_Z</td>\n",
       "      <td>86470505</td>\n",
       "      <td>cuteSV.INV.281</td>\n",
       "      <td>C</td>\n",
       "      <td>['&lt;INV&gt;']</td>\n",
       "      <td>INV</td>\n",
       "      <td>3663.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.6667</td>\n",
       "      <td>0.6471</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.6250</td>\n",
       "      <td>14</td>\n",
       "      <td>30</td>\n",
       "      <td>17</td>\n",
       "      <td>20</td>\n",
       "      <td>24</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8101 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM       POS               ID  \\\n",
       "0     SM_V10_1    108036     cuteSV.INS.4   \n",
       "1     SM_V10_1    156118     cuteSV.DEL.7   \n",
       "2     SM_V10_1    214273     cuteSV.INS.8   \n",
       "3     SM_V10_1    258698     cuteSV.INS.9   \n",
       "4     SM_V10_1    264300    cuteSV.INS.10   \n",
       "...        ...       ...              ...   \n",
       "8096  SM_V10_Z  86357919  cuteSV.INS.4444   \n",
       "8097  SM_V10_Z  86361688  cuteSV.INS.4445   \n",
       "8098  SM_V10_Z  86465601  cuteSV.DEL.5146   \n",
       "8099  SM_V10_Z  86467302  cuteSV.INS.4446   \n",
       "8100  SM_V10_Z  86470505   cuteSV.INV.281   \n",
       "\n",
       "                                                    REF  \\\n",
       "0                                                     C   \n",
       "1     ATGTTATCTTTGGCAGCCTATTTAAACATCTGGGCTACCTGTTCCT...   \n",
       "2                                                     T   \n",
       "3                                                     A   \n",
       "4                                                     C   \n",
       "...                                                 ...   \n",
       "8096                                                  T   \n",
       "8097                                                  A   \n",
       "8098          GTGCGCGTTCAAAAAGCGGATCACAAACTTTGTAGAAAAAT   \n",
       "8099                                                  G   \n",
       "8100                                                  C   \n",
       "\n",
       "                                                    ALT SVTYPE   SVLEN  \\\n",
       "0          ['CATTATTATTATTATTATTACTACTATTATTATTACTATT']    INS    39.0   \n",
       "1                                                 ['A']    DEL  -401.0   \n",
       "2     ['TGATGATGATGATAGATATTGATGATGATGATGATGATGATGAT...    INS    51.0   \n",
       "3     ['ACTCGGGAATAACATTAGGATCACTTCAATTTTTTTAATAATTT...    INS  4470.0   \n",
       "4     ['CCTGGAAGCACTGGACGGCCGTTTCGTCCTATTGGCGGACCCCT...    INS   324.0   \n",
       "...                                                 ...    ...     ...   \n",
       "8096  ['TTGTTATGTCCCCTGGTGAATTATAAATGGTAACTCTGAGTCTA...    INS   604.0   \n",
       "8097  ['AATCAGTAAGCGAGTAGTGATGGAAGTTTGGTTATTTTGGTACA...    INS   325.0   \n",
       "8098                                              ['G']    DEL   -40.0   \n",
       "8099  ['GAATTCTATATTTTTCAAGGGAGTTGAAATCATAAGTCAATCGA...    INS   229.0   \n",
       "8100                                          ['<INV>']    INV  3663.0   \n",
       "\n",
       "      smor.MAF  smle_pzq_es.MAF  smle_pzq_er.MAF  smbre.MAF  smeg.MAF  \\\n",
       "0       1.0000           0.9429           0.8889     1.0000    0.9565   \n",
       "1       0.3500           0.1587           0.3846     0.8750    0.2800   \n",
       "2       0.1304           0.6667           0.9333     0.8696    0.3125   \n",
       "3       0.2778           0.1897           0.1429     0.0930    0.3125   \n",
       "4       0.6579           0.2637           0.5455     0.6269    0.1522   \n",
       "...        ...              ...              ...        ...       ...   \n",
       "8096    0.3333           0.2222           0.1250     0.9545    0.3750   \n",
       "8097    1.0000           1.0000           1.0000     1.0000    1.0000   \n",
       "8098    0.5000           0.2022           0.2292     0.0000    0.1667   \n",
       "8099    0.5238           0.1860           0.0263     1.0000    0.1500   \n",
       "8100    0.0000           0.6667           0.6471     0.0000    0.6250   \n",
       "\n",
       "      smor.COV  smle_pzq_es.COV  smle_pzq_er.COV  smbre.COV  smeg.COV  \\\n",
       "0           25               70               18         25        23   \n",
       "1           20               63               13          8        25   \n",
       "2           23               36               15         23        16   \n",
       "3           18               58               14         43        16   \n",
       "4           38               91               22         67        46   \n",
       "...        ...              ...              ...        ...       ...   \n",
       "8096        18               36               16         22        16   \n",
       "8097        17               47               26         24        20   \n",
       "8098        38               89               48         29        54   \n",
       "8099        21               86               38         35        40   \n",
       "8100        14               30               17         20        24   \n",
       "\n",
       "        gene_ids  \n",
       "0            NaN  \n",
       "1            NaN  \n",
       "2            NaN  \n",
       "3            NaN  \n",
       "4     Smp_318880  \n",
       "...          ...  \n",
       "8096         NaN  \n",
       "8097         NaN  \n",
       "8098         NaN  \n",
       "8099         NaN  \n",
       "8100         NaN  \n",
       "\n",
       "[8101 rows x 18 columns]"
      ]
     },
     "execution_count": 959,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cov_thresh=8\n",
    "\n",
    "filt_df=pd.read_csv(f\"{proj_dir}/results/svs_and_genes/sv_cat_afs_dp_ge_{cov_thresh}_w_genes.csv\", sep=\",\", header=0)\n",
    "filt_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 764,
   "id": "bcb345dc-9621-4a25-8b04-bd2830f3d574",
   "metadata": {},
   "outputs": [],
   "source": [
    "#use grenedalf to calculate per site fst from freq table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 962,
   "id": "c6c0feab-b587-4769-a679-ea265f48e76b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SVTYPE</th>\n",
       "      <th>SVLEN</th>\n",
       "      <th>smor.MAF</th>\n",
       "      <th>smle_pzq_es.MAF</th>\n",
       "      <th>smle_pzq_er.MAF</th>\n",
       "      <th>smbre.MAF</th>\n",
       "      <th>smeg.MAF</th>\n",
       "      <th>smor.COV</th>\n",
       "      <th>smle_pzq_es.COV</th>\n",
       "      <th>smle_pzq_er.COV</th>\n",
       "      <th>smbre.COV</th>\n",
       "      <th>smeg.COV</th>\n",
       "      <th>gene_ids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SM_V10_1</td>\n",
       "      <td>108036</td>\n",
       "      <td>cuteSV.INS.4</td>\n",
       "      <td>C</td>\n",
       "      <td>CATTATTATTATTATTATTACTACTATTATTATTACTATT</td>\n",
       "      <td>INS</td>\n",
       "      <td>39.0</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9429</td>\n",
       "      <td>0.8889</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.9565</td>\n",
       "      <td>25</td>\n",
       "      <td>70</td>\n",
       "      <td>18</td>\n",
       "      <td>25</td>\n",
       "      <td>23</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SM_V10_1</td>\n",
       "      <td>156118</td>\n",
       "      <td>cuteSV.DEL.7</td>\n",
       "      <td>ATGTTATCTTTGGCAGCCTATTTAAACATCTGGGCTACCTGTTCCT...</td>\n",
       "      <td>A</td>\n",
       "      <td>DEL</td>\n",
       "      <td>-401.0</td>\n",
       "      <td>0.3500</td>\n",
       "      <td>0.1587</td>\n",
       "      <td>0.3846</td>\n",
       "      <td>0.8750</td>\n",
       "      <td>0.2800</td>\n",
       "      <td>20</td>\n",
       "      <td>63</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SM_V10_1</td>\n",
       "      <td>214273</td>\n",
       "      <td>cuteSV.INS.8</td>\n",
       "      <td>T</td>\n",
       "      <td>TGATGATGATGATAGATATTGATGATGATGATGATGATGATGATGA...</td>\n",
       "      <td>INS</td>\n",
       "      <td>51.0</td>\n",
       "      <td>0.1304</td>\n",
       "      <td>0.6667</td>\n",
       "      <td>0.9333</td>\n",
       "      <td>0.8696</td>\n",
       "      <td>0.3125</td>\n",
       "      <td>23</td>\n",
       "      <td>36</td>\n",
       "      <td>15</td>\n",
       "      <td>23</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SM_V10_1</td>\n",
       "      <td>258698</td>\n",
       "      <td>cuteSV.INS.9</td>\n",
       "      <td>A</td>\n",
       "      <td>ACTCGGGAATAACATTAGGATCACTTCAATTTTTTTAATAATTTTT...</td>\n",
       "      <td>INS</td>\n",
       "      <td>4470.0</td>\n",
       "      <td>0.2778</td>\n",
       "      <td>0.1897</td>\n",
       "      <td>0.1429</td>\n",
       "      <td>0.0930</td>\n",
       "      <td>0.3125</td>\n",
       "      <td>18</td>\n",
       "      <td>58</td>\n",
       "      <td>14</td>\n",
       "      <td>43</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SM_V10_1</td>\n",
       "      <td>264300</td>\n",
       "      <td>cuteSV.INS.10</td>\n",
       "      <td>C</td>\n",
       "      <td>CCTGGAAGCACTGGACGGCCGTTTCGTCCTATTGGCGGACCCCTCA...</td>\n",
       "      <td>INS</td>\n",
       "      <td>324.0</td>\n",
       "      <td>0.6579</td>\n",
       "      <td>0.2637</td>\n",
       "      <td>0.5455</td>\n",
       "      <td>0.6269</td>\n",
       "      <td>0.1522</td>\n",
       "      <td>38</td>\n",
       "      <td>91</td>\n",
       "      <td>22</td>\n",
       "      <td>67</td>\n",
       "      <td>46</td>\n",
       "      <td>Smp_318880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8096</th>\n",
       "      <td>SM_V10_Z</td>\n",
       "      <td>86357919</td>\n",
       "      <td>cuteSV.INS.4444</td>\n",
       "      <td>T</td>\n",
       "      <td>TTGTTATGTCCCCTGGTGAATTATAAATGGTAACTCTGAGTCTATT...</td>\n",
       "      <td>INS</td>\n",
       "      <td>604.0</td>\n",
       "      <td>0.3333</td>\n",
       "      <td>0.2222</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>0.9545</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>18</td>\n",
       "      <td>36</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8097</th>\n",
       "      <td>SM_V10_Z</td>\n",
       "      <td>86361688</td>\n",
       "      <td>cuteSV.INS.4445</td>\n",
       "      <td>A</td>\n",
       "      <td>AATCAGTAAGCGAGTAGTGATGGAAGTTTGGTTATTTTGGTACAGA...</td>\n",
       "      <td>INS</td>\n",
       "      <td>325.0</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>17</td>\n",
       "      <td>47</td>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8098</th>\n",
       "      <td>SM_V10_Z</td>\n",
       "      <td>86465601</td>\n",
       "      <td>cuteSV.DEL.5146</td>\n",
       "      <td>GTGCGCGTTCAAAAAGCGGATCACAAACTTTGTAGAAAAAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>-40.0</td>\n",
       "      <td>0.5000</td>\n",
       "      <td>0.2022</td>\n",
       "      <td>0.2292</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.1667</td>\n",
       "      <td>38</td>\n",
       "      <td>89</td>\n",
       "      <td>48</td>\n",
       "      <td>29</td>\n",
       "      <td>54</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8099</th>\n",
       "      <td>SM_V10_Z</td>\n",
       "      <td>86467302</td>\n",
       "      <td>cuteSV.INS.4446</td>\n",
       "      <td>G</td>\n",
       "      <td>GAATTCTATATTTTTCAAGGGAGTTGAAATCATAAGTCAATCGAAG...</td>\n",
       "      <td>INS</td>\n",
       "      <td>229.0</td>\n",
       "      <td>0.5238</td>\n",
       "      <td>0.1860</td>\n",
       "      <td>0.0263</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.1500</td>\n",
       "      <td>21</td>\n",
       "      <td>86</td>\n",
       "      <td>38</td>\n",
       "      <td>35</td>\n",
       "      <td>40</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8100</th>\n",
       "      <td>SM_V10_Z</td>\n",
       "      <td>86470505</td>\n",
       "      <td>cuteSV.INV.281</td>\n",
       "      <td>C</td>\n",
       "      <td>&lt;INV&gt;</td>\n",
       "      <td>INV</td>\n",
       "      <td>3663.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.6667</td>\n",
       "      <td>0.6471</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.6250</td>\n",
       "      <td>14</td>\n",
       "      <td>30</td>\n",
       "      <td>17</td>\n",
       "      <td>20</td>\n",
       "      <td>24</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8101 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM       POS               ID  \\\n",
       "0     SM_V10_1    108036     cuteSV.INS.4   \n",
       "1     SM_V10_1    156118     cuteSV.DEL.7   \n",
       "2     SM_V10_1    214273     cuteSV.INS.8   \n",
       "3     SM_V10_1    258698     cuteSV.INS.9   \n",
       "4     SM_V10_1    264300    cuteSV.INS.10   \n",
       "...        ...       ...              ...   \n",
       "8096  SM_V10_Z  86357919  cuteSV.INS.4444   \n",
       "8097  SM_V10_Z  86361688  cuteSV.INS.4445   \n",
       "8098  SM_V10_Z  86465601  cuteSV.DEL.5146   \n",
       "8099  SM_V10_Z  86467302  cuteSV.INS.4446   \n",
       "8100  SM_V10_Z  86470505   cuteSV.INV.281   \n",
       "\n",
       "                                                    REF  \\\n",
       "0                                                     C   \n",
       "1     ATGTTATCTTTGGCAGCCTATTTAAACATCTGGGCTACCTGTTCCT...   \n",
       "2                                                     T   \n",
       "3                                                     A   \n",
       "4                                                     C   \n",
       "...                                                 ...   \n",
       "8096                                                  T   \n",
       "8097                                                  A   \n",
       "8098          GTGCGCGTTCAAAAAGCGGATCACAAACTTTGTAGAAAAAT   \n",
       "8099                                                  G   \n",
       "8100                                                  C   \n",
       "\n",
       "                                                    ALT SVTYPE   SVLEN  \\\n",
       "0              CATTATTATTATTATTATTACTACTATTATTATTACTATT    INS    39.0   \n",
       "1                                                     A    DEL  -401.0   \n",
       "2     TGATGATGATGATAGATATTGATGATGATGATGATGATGATGATGA...    INS    51.0   \n",
       "3     ACTCGGGAATAACATTAGGATCACTTCAATTTTTTTAATAATTTTT...    INS  4470.0   \n",
       "4     CCTGGAAGCACTGGACGGCCGTTTCGTCCTATTGGCGGACCCCTCA...    INS   324.0   \n",
       "...                                                 ...    ...     ...   \n",
       "8096  TTGTTATGTCCCCTGGTGAATTATAAATGGTAACTCTGAGTCTATT...    INS   604.0   \n",
       "8097  AATCAGTAAGCGAGTAGTGATGGAAGTTTGGTTATTTTGGTACAGA...    INS   325.0   \n",
       "8098                                                  G    DEL   -40.0   \n",
       "8099  GAATTCTATATTTTTCAAGGGAGTTGAAATCATAAGTCAATCGAAG...    INS   229.0   \n",
       "8100                                              <INV>    INV  3663.0   \n",
       "\n",
       "      smor.MAF  smle_pzq_es.MAF  smle_pzq_er.MAF  smbre.MAF  smeg.MAF  \\\n",
       "0       1.0000           0.9429           0.8889     1.0000    0.9565   \n",
       "1       0.3500           0.1587           0.3846     0.8750    0.2800   \n",
       "2       0.1304           0.6667           0.9333     0.8696    0.3125   \n",
       "3       0.2778           0.1897           0.1429     0.0930    0.3125   \n",
       "4       0.6579           0.2637           0.5455     0.6269    0.1522   \n",
       "...        ...              ...              ...        ...       ...   \n",
       "8096    0.3333           0.2222           0.1250     0.9545    0.3750   \n",
       "8097    1.0000           1.0000           1.0000     1.0000    1.0000   \n",
       "8098    0.5000           0.2022           0.2292     0.0000    0.1667   \n",
       "8099    0.5238           0.1860           0.0263     1.0000    0.1500   \n",
       "8100    0.0000           0.6667           0.6471     0.0000    0.6250   \n",
       "\n",
       "      smor.COV  smle_pzq_es.COV  smle_pzq_er.COV  smbre.COV  smeg.COV  \\\n",
       "0           25               70               18         25        23   \n",
       "1           20               63               13          8        25   \n",
       "2           23               36               15         23        16   \n",
       "3           18               58               14         43        16   \n",
       "4           38               91               22         67        46   \n",
       "...        ...              ...              ...        ...       ...   \n",
       "8096        18               36               16         22        16   \n",
       "8097        17               47               26         24        20   \n",
       "8098        38               89               48         29        54   \n",
       "8099        21               86               38         35        40   \n",
       "8100        14               30               17         20        24   \n",
       "\n",
       "        gene_ids  \n",
       "0            NaN  \n",
       "1            NaN  \n",
       "2            NaN  \n",
       "3            NaN  \n",
       "4     Smp_318880  \n",
       "...          ...  \n",
       "8096         NaN  \n",
       "8097         NaN  \n",
       "8098         NaN  \n",
       "8099         NaN  \n",
       "8100         NaN  \n",
       "\n",
       "[8101 rows x 18 columns]"
      ]
     },
     "execution_count": 962,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#prep for grenedalf with coverages\n",
    "g_df=filt_df.copy()\n",
    "\n",
    "g_df[\"ALT\"] = g_df[\"ALT\"].str.replace(\"^\\['\", \"\", regex=True)\n",
    "g_df[\"ALT\"] = g_df[\"ALT\"].str.replace(\"\\']$\", \"\", regex=True)\n",
    "g_df.to_csv(\"to_grenedalf.csv\", sep=\",\", header=True, index=False)\n",
    "g_df\n",
    " \n",
    "    "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "sm_ont-svs",
   "language": "python",
   "name": "sm_ont-svs"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
